// stimulsoft-viewer-angular/esm5/lib/stimulsoft-viewer-angular.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild, Output, EventEmitter } from '@angular/core';
import { ControllerService } from './services/controller.service';
import { MenuService } from './menu/menu.service';
import { ModelService } from './services/model.service';
import { ExportService } from './services/export.service';
import { FormService } from './forms/form.service';
import { ApiService } from './services/api.service';
import { PrintService } from './services/print.service';
import { DashboardService } from './services/dashboard.service';
import { StiHttpClientService } from './services/http-client.service';
import { StylesService } from './services/styles.service';
import { HelperService } from './services/helper.service';
import { MouseService } from './services/mouse.service';
import { PageService } from './services/page.service';
import { TooltipService } from './services/tooltip.service';
import { FindService } from './services/find.service';
import { AnimationService } from './services/animation.service';
import { InteractionsService } from './services/interactions.service';
import { RadioButtonService } from './services/radio-button.service';
import { CollectionsService } from './services/collections.service';
import { MailService } from './services/mail.service';
import { EditableFieldsService } from './services/editable-fields.service';
import { DrillDownService } from './services/drill-down.service';
import { Md5Service } from './services/md5.service';
import { FullScreenService } from './services/full-screen.service';
import { ToolbarService } from './services/toolbar.service';
//1bfb351c1add27b1537ffce9c723e3cb1f5b27bb 02.03.2021 17:29:20
var StimulsoftViewerComponent = /** @class */ (function () {
    function StimulsoftViewerComponent(controller, menuService, model, exportService, formService, api, printService, dashboardService) {
        var _this = this;
        this.controller = controller;
        this.menuService = menuService;
        this.model = model;
        this.exportService = exportService;
        this.formService = formService;
        this.api = api;
        this.printService = printService;
        this.dashboardService = dashboardService;
        /**
         * Occurs when report/dashboard loaded
         */
        this.loaded = new EventEmitter();
        /**
         * Occurs on error, $event is ErrorMessage object contains error: string & type: any  (if present)
         */
        this.error = new EventEmitter();
        /**
         * Occurs on export, $event object contains exportFormat: string & exportSettings: {}
         */
        this.export = new EventEmitter();
        /**
         * Occurs on export & email, $event object contains exportFormat: string & exportSettings: {}
         */
        this.email = new EventEmitter();
        /**
         * Occurs on export & email, $event object contains format: string : 'PrintPdf' or 'PrintWithoutPreview' or 'PrintWithPreview'
         */
        this.print = new EventEmitter();
        /**
         * Occurs on pressing 'Design' button
         */
        this.design = new EventEmitter();
        /**
         * Viewer background color
         */
        this.backgroundColor = 'White';
        this.initialized = false;
        this.viewInit = false;
        window.onresize = function () { return _this.resize(); };
        this.initEvents();
    }
    StimulsoftViewerComponent.prototype.ngOnChanges = function (changes) {
        this.loadViewer();
    };
    StimulsoftViewerComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.controller.getMessage().subscribe(function (message) {
            switch (message.action) {
                case 'viewer_loaded':
                    _this.initialized = true;
                    break;
            }
        });
    };
    StimulsoftViewerComponent.prototype.ngAfterViewInit = function () {
        this.viewInit = true;
        this.loadViewer();
    };
    StimulsoftViewerComponent.prototype.initEvents = function () {
        var _this = this;
        this.controller.getMessage().subscribe(function (message) {
            switch (message.action) {
                case 'GetReport':
                case 'OpenReport':
                    setTimeout(function () {
                        _this.loaded.next();
                    });
                    break;
            }
        });
        this.controller.getActionMessage().subscribe(function (message) {
            switch (message.action) {
                case 'Error':
                    setTimeout(function () {
                        _this.error.next(_this.model.errorMessage);
                    });
                    break;
                case 'ExportReport':
                    setTimeout(function () {
                        _this.export.next(message.data);
                    });
                    break;
                case 'Email':
                    setTimeout(function () {
                        _this.email.next(message.data);
                    });
                    break;
                case 'Print':
                    setTimeout(function () {
                        _this.print.next(message.data);
                    });
                    break;
                case 'Design':
                    setTimeout(function () {
                        _this.design.next();
                    });
                    break;
            }
        });
    };
    StimulsoftViewerComponent.prototype.loadViewer = function () {
        if (this.viewInit) {
            this.model.requestUrl = this.requestUrl;
            this.model.action = this.action;
            this.model.properties = this.properties;
            this.controller.loadViewer();
            this.resize();
            this.model.controls.viewer.el = this.viewerElement;
        }
    };
    StimulsoftViewerComponent.prototype.resize = function () {
        var _a;
        if ((_a = this.viewerElement) === null || _a === void 0 ? void 0 : _a.nativeElement) {
            this.model.viewerSize = { width: this.viewerElement.nativeElement.offsetWidth, height: this.viewerElement.nativeElement.offsetHeight };
            this.controller.viewerResized();
        }
    };
    StimulsoftViewerComponent.ctorParameters = function () { return [
        { type: ControllerService },
        { type: MenuService },
        { type: ModelService },
        { type: ExportService },
        { type: FormService },
        { type: ApiService },
        { type: PrintService },
        { type: DashboardService }
    ]; };
    __decorate([
        ViewChild('viewer')
    ], StimulsoftViewerComponent.prototype, "viewerElement", void 0);
    __decorate([
        Output()
    ], StimulsoftViewerComponent.prototype, "loaded", void 0);
    __decorate([
        Output()
    ], StimulsoftViewerComponent.prototype, "error", void 0);
    __decorate([
        Output()
    ], StimulsoftViewerComponent.prototype, "export", void 0);
    __decorate([
        Output()
    ], StimulsoftViewerComponent.prototype, "email", void 0);
    __decorate([
        Output()
    ], StimulsoftViewerComponent.prototype, "print", void 0);
    __decorate([
        Output()
    ], StimulsoftViewerComponent.prototype, "design", void 0);
    __decorate([
        Input()
    ], StimulsoftViewerComponent.prototype, "requestUrl", void 0);
    __decorate([
        Input()
    ], StimulsoftViewerComponent.prototype, "action", void 0);
    __decorate([
        Input()
    ], StimulsoftViewerComponent.prototype, "properties", void 0);
    __decorate([
        Input()
    ], StimulsoftViewerComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], StimulsoftViewerComponent.prototype, "height", void 0);
    __decorate([
        Input()
    ], StimulsoftViewerComponent.prototype, "backgroundColor", void 0);
    __decorate([
        Input()
    ], StimulsoftViewerComponent.prototype, "style", void 0);
    StimulsoftViewerComponent = __decorate([
        Component({
            selector: 'stimulsoft-viewer-angular',
            template: "\n    <span #viewer [style]=\"style\"\n      [style.top.px]=\"0\"\n      [style.right.px]=\"0\"\n      [style.bottom.px]=\"0\"\n      [style.left.px]=\"0\"\n      [style.display]=\"'inline-block'\"\n      [style.backgroundColor]=\"backgroundColor\"\n      [style.width]=\"this.width != null ? this.width : '100%'\"\n      [style.height]=\"this.height != null ? this.height : (model.options?.appearance.scrollbarsMode ? '650px' : '100%')\">\n      <div class=\"stiJsViewerMainPanel\">\n        <sti-center-text *ngIf=\"initialized\"></sti-center-text>\n        <sti-toolbar *ngIf=\"initialized\"></sti-toolbar>\n        <div *ngFor=\"let i of [1,2,3,4,5]\" [style.zIndex]=\"i * 10\" [style.display]=\"formService.form?.level == i ? '' : 'none'\" class=\"stiJsViewerDisabledPanel\"></div>\n        <sti-about-panel *ngIf=\"initialized\"></sti-about-panel>\n\n        <sti-dashboards-panel *ngIf=\"initialized\"></sti-dashboards-panel>\n        <sti-report-panel *ngIf=\"initialized\"></sti-report-panel>\n        <sti-progress *ngIf=\"initialized\"></sti-progress>\n        <sti-find-panel *ngIf=\"initialized\"></sti-find-panel>\n        <sti-drill-down-panel *ngIf=\"initialized\"></sti-drill-down-panel>\n        <sti-resources-panel *ngIf=\"initialized\"></sti-resources-panel>\n        <sti-navigate-panel *ngIf=\"initialized\"></sti-navigate-panel>\n\n        <sti-export-form *ngIf=\"initialized\" [exportFormSettings]=\"exportService.exportFormSettings\"></sti-export-form>\n        <sti-send-email-form *ngIf=\"initialized\"></sti-send-email-form>\n        <sti-password-form *ngIf=\"initialized\"></sti-password-form>\n        <sti-error-message-form *ngIf=\"initialized\"></sti-error-message-form>\n        <sti-notification-form *ngIf=\"initialized\"></sti-notification-form>\n\n        <sti-tooltip *ngIf=\"initialized\"></sti-tooltip>\n        <sti-menu *ngFor=\"let menu of menuService.menus\" [menu]=\"menu\"></sti-menu>\n        <sti-bookmarks-panel *ngIf=\"initialized\"></sti-bookmarks-panel>\n        <sti-interactions-panel *ngIf=\"initialized\"></sti-interactions-panel>\n\n        <sti-open-dialog *ngIf=\"initialized\" [fileMask]=\"model.openDialogFileMask\"></sti-open-dialog>\n      </div>\n    </span>\n  ",
            providers: [ModelService, StiHttpClientService, ControllerService, StylesService, HelperService, MenuService,
                MouseService, PageService, TooltipService, FindService, AnimationService, Md5Service, InteractionsService, FormService, RadioButtonService, ExportService,
                CollectionsService, MailService, EditableFieldsService, DrillDownService, PrintService, FullScreenService, ToolbarService, ApiService, DashboardService]
        })
    ], StimulsoftViewerComponent);
    return StimulsoftViewerComponent;
}());
export { StimulsoftViewerComponent };
