// stimulsoft-viewer-angular/esm5/lib/menu/date-picker-menu.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { ModelService } from '../services/model.service';
import { HelperService } from '../services/helper.service';
var DatePickerMenuComponent = /** @class */ (function () {
    function DatePickerMenuComponent(model, helper) {
        var _this = this;
        this.model = model;
        this.helper = helper;
        this.keyTo = false;
        this.closeOnAction = true;
        this.monthesForDatePickerItems = [];
        this.cols = [0, 1, 2, 3, 4, 5, 6];
        this.rows = [0, 1, 2, 3, 4, 5];
        this.model.months.forEach(function (m, i) { return _this.monthesForDatePickerItems.push({ name: 'Month' + i, caption: _this.model.loc('Month' + m), key: i }); });
        var firstLetters = {};
        this.model.dayOfWeek.forEach(function (d) {
            var dayName = _this.model.loc('Day' + d);
            firstLetters[dayName.toString().substring(0, 1).toUpperCase()] = true;
        });
        this.countLetters = Object.keys(firstLetters).length < 5 ? 2 : 1;
    }
    DatePickerMenuComponent.prototype.ngOnInit = function () { };
    DatePickerMenuComponent.prototype.getButtonSelected = function (col, row) {
        var firstDay = this.getFirstDay();
        var curDay = row * 7 + col;
        var day = curDay - firstDay + 1;
        return day === this.key.day;
    };
    DatePickerMenuComponent.prototype.getButtonCaption = function (col, row) {
        var firstDay = this.getFirstDay();
        var daysInMonth = this.helper.getCountDaysOfMonth(this.key.year, this.key.month - 1);
        var curDay = row * 7 + col;
        var day = curDay - firstDay + 1;
        if (curDay >= firstDay && day <= daysInMonth) {
            return day.toString();
        }
        return '';
    };
    DatePickerMenuComponent.prototype.getTime = function () {
        return this.helper.formatDate(this.helper.getDate(this.key), 'H:mm:ss');
    };
    DatePickerMenuComponent.prototype.getFirstDay = function () {
        var firstDay = new Date(this.key.year, this.key.month - 1, 1).getDay();
        if (firstDay === 0) {
            firstDay = 7;
        }
        if (this.model.options.appearance.datePickerFirstDayOfWeek !== 'Sunday') {
            firstDay--;
        }
        return firstDay;
    };
    Object.defineProperty(DatePickerMenuComponent.prototype, "showTime", {
        get: function () {
            var _a, _b;
            return ((_b = (_a = this.menu) === null || _a === void 0 ? void 0 : _a.params) === null || _b === void 0 ? void 0 : _b.dateTimeType) !== 'Date';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DatePickerMenuComponent.prototype, "key", {
        get: function () {
            return this.keyTo ? this.menu.params.keyTo : this.menu.params.key;
        },
        enumerable: true,
        configurable: true
    });
    DatePickerMenuComponent.prototype.yearAction = function (input) {
        try {
            this.key.year = parseInt(input.value, 10);
        }
        catch (_a) { }
    };
    DatePickerMenuComponent.prototype.timeAction = function (input) {
        var time = this.helper.stringToTime(input.value);
        this.key.seconds = time.seconds;
        this.key.minutes = time.minutes;
        this.key.hours = time.hours;
    };
    DatePickerMenuComponent.prototype.getShortDayName = function (index) {
        var dayName = this.model.loc('Day' + this.model.dayOfWeek[index]);
        if (dayName) {
            return dayName.toString().substring(0, this.countLetters <= dayName.length ? this.countLetters : 1).toUpperCase();
        }
        return '';
    };
    DatePickerMenuComponent.prototype.getShortDayColor = function (index) {
        if (index === (this.model.options.appearance.datePickerFirstDayOfWeek === 'Sunday' ? 6 : 5)) {
            return '#0000ff';
        }
        if (index === (this.model.options.appearance.datePickerFirstDayOfWeek === 'Sunday' ? 0 : 6)) {
            return '#ff0000';
        }
        return '';
    };
    DatePickerMenuComponent.prototype.prevMonthButtonAction = function () {
        var month = this.key.month;
        var year = this.key.year;
        month--;
        if (month === 0) {
            month = 12;
            year--;
        }
        var countDaysInMonth = this.helper.getCountDaysOfMonth(year, month - 1);
        if (countDaysInMonth < this.key.day) {
            this.key.day = countDaysInMonth;
        }
        this.key.month = month;
        this.key.year = year;
    };
    DatePickerMenuComponent.prototype.nextMonthButtonAction = function () {
        var month = this.key.month;
        var year = this.key.year;
        month++;
        if (month === 13) {
            month = 1;
            year++;
        }
        var countDaysInMonth = this.helper.getCountDaysOfMonth(year, month - 1);
        if (countDaysInMonth < this.key.day) {
            this.key.day = countDaysInMonth;
        }
        this.key.month = month;
        this.key.year = year;
    };
    DatePickerMenuComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService }
    ]; };
    __decorate([
        Input()
    ], DatePickerMenuComponent.prototype, "menu", void 0);
    __decorate([
        Input()
    ], DatePickerMenuComponent.prototype, "keyTo", void 0);
    __decorate([
        Input()
    ], DatePickerMenuComponent.prototype, "closeOnAction", void 0);
    DatePickerMenuComponent = __decorate([
        Component({
            selector: 'sti-date-picker-menu',
            template: "\n        <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\">\n                <sti-button [imageName]=\"'Arrows.BigArrowLeft.png'\" [margin]=\"'1px 2px 0 1px'\" (action)=\"prevMonthButtonAction()\">\n                </sti-button>\n              </td>\n\n              <td class=\"stiJsViewerClearAllStyles\">\n                <sti-drop-down-list\n                  [width]=\"model.options.isTouchDevice ? 79 : 81\"\n                  [readOnly]=\"true\"\n                  [margin]=\"'1px 2px 0 0'\"\n                  [items]=\"monthesForDatePickerItems\"\n                  [key]=\"key.month - 1\"\n                  (action)=\"key.month = $event.key + 1\">\n                </sti-drop-down-list>\n              </td>\n\n              <td class=\"stiJsViewerClearAllStyles\">\n                <sti-text-box [width]=\"40\" [margin]=\"'1px 2px 0 0'\" [value]=\"key.year\" (action)=\"yearAction($event)\">\n                </sti-text-box>\n              </td>\n\n              <td class=\"stiJsViewerClearAllStyles\">\n                <sti-button [imageName]=\"'Arrows.BigArrowRight.png'\" [margin]=\"'1px 1px 0 0'\" (action)=\"nextMonthButtonAction()\">\n                </sti-button>\n              </td>\n            </tr>\n          </tbody>\n        </table>\n        <div class=\"stiJsViewerDatePickerSeparator\" style=\"margin: 2px 0px;\"></div>\n\n        <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td *ngFor=\"let item of model.dayOfWeek; let i = index\" class=\"stiJsViewerDatePickerDayOfWeekCell\"\n                  [style.fontSize]=\"countLetters == 2 ? '11px' : ''\"\n                  [style.color]=\"getShortDayColor(i)\">\n                  {{getShortDayName(i)}}\n              </td>\n            </tr>\n            <tr *ngFor=\"let row of rows\" class=\"stiJsViewerClearAllStyles\">\n                <td *ngFor=\"let col of cols\" class=\"stiJsViewerClearAllStyles\">\n                  <sti-date-picker-day-button\n                      [col]=\"col\"\n                      [row]=\"row\"\n                      [date]=\"key\"\n                      [selected]=\"getButtonSelected(col, row)\"\n                      [caption]=\"getButtonCaption(col, row)\"\n                      [enabled]=\"getButtonCaption(col, row) != ''\">\n                      [closeOnAction]=\"closeOnAction\"\n                  </sti-date-picker-day-button>\n                </td>\n            </tr>\n          </tbody>\n        </table>\n\n        <div *ngIf=\"showTime\" class=\"stiJsViewerDatePickerSeparator\" style=\"margin: 2px 0px;\"></div>\n        <table *ngIf=\"showTime\" class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"width:100%\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\" style=\"padding: 0 4px 0 4px; white-space: nowrap;\">\n                {{this.model.loc('Time') + ':'}}\n              </td>\n\n              <td class=\"stiJsViewerClearAllStyles\" style=\"text-align: right;\">\n                <sti-text-box [width]=\"90\"\n                    [margin]=\"'1px 2px 2px 2px'\"\n                    [value]=\"getTime()\"\n                    (action)=\"timeAction($event)\">\n                </sti-text-box>\n              </td>\n            </tr>\n          </tbody>\n        </table>\n  "
        })
    ], DatePickerMenuComponent);
    return DatePickerMenuComponent;
}());
export { DatePickerMenuComponent };
