// stimulsoft-viewer-angular/esm5/lib/menu/digital-signature-menu.component copy.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ExportService } from '../services/export.service';
var DigitalSignatureMenuComponent = /** @class */ (function () {
    function DigitalSignatureMenuComponent(model, exportService) {
        this.model = model;
        this.exportService = exportService;
    }
    DigitalSignatureMenuComponent.prototype.ngOnInit = function () { };
    DigitalSignatureMenuComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ExportService }
    ]; };
    DigitalSignatureMenuComponent = __decorate([
        Component({
            selector: 'sti-digital-signature-menu',
            template: "\n    <div *ngFor=\"let item of model.pdfSecurityCertificates\" class=\"stiJsViewerMenuStandartItem stiJsViewerMenuStandartItemSelected\" style=\"height: auto; line-height: 1.3;\">\n      <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"height: 100%; width: 100%;\">\n        <tbody>\n          <tr class=\"stiJsViewerClearAllStyles\">\n            <td class=\"stiJsViewerClearAllStyles\" style=\"padding: 8px 20px 8px 8px; text-align: left; white-space: nowrap;\">\n            </td>\n          </tr>\n        </tbody>\n      </table>\n    </div>\n"
        })
    ], DigitalSignatureMenuComponent);
    return DigitalSignatureMenuComponent;
}());
export { DigitalSignatureMenuComponent };
