// stimulsoft-viewer-angular/esm5/lib/menu/meni-item.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { ModelService } from '../services/model.service';
var MenuItem = /** @class */ (function () {
    function MenuItem(name, caption, img, type, selected, imageSize, value) {
        this.name = name;
        this.caption = caption;
        this.img = img;
        this.type = type;
        this.selected = selected;
        this.imageSize = imageSize;
        this.value = value;
    }
    return MenuItem;
}());
export { MenuItem };
var MenuItemComponent = /** @class */ (function () {
    function MenuItemComponent(model) {
        this.model = model;
        this.styleName = 'stiJsViewerMenuStandartItem';
        this.selected = false;
        this.over = false;
        this._enabled = true;
    }
    MenuItemComponent.prototype.ngOnInit = function () { };
    Object.defineProperty(MenuItemComponent.prototype, "enabled", {
        get: function () {
            return this._enabled;
        },
        set: function (value) {
            if (!value) {
                this.over = false;
            }
            this._enabled = value;
        },
        enumerable: true,
        configurable: true
    });
    MenuItemComponent.prototype.mouseover = function () {
        if (!this.enabled || !this.enabled) {
            return;
        }
        this.over = true;
    };
    MenuItemComponent.prototype.mouseout = function () {
        if (!this.enabled || !this.enabled) {
            return;
        }
        this.over = false;
    };
    Object.defineProperty(MenuItemComponent.prototype, "height", {
        get: function () {
            if (this.item.imageSize === 'Big') {
                return '38px';
            }
            else {
                return this.model.options.isMobileDevice ? '0.4in' : (this.model.options.isTouchDevice ? '30px' : '24px');
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(MenuItemComponent.prototype, "className", {
        get: function () {
            return this.styleName ? this.styleName + ' ' + (this.enabled ? (this.styleName + ((this.selected || this.item.selected) ? 'Selected' : (this.over ? 'Over' : ''))) : this.styleName + 'Disabled') : '';
        },
        enumerable: true,
        configurable: true
    });
    MenuItemComponent.ctorParameters = function () { return [
        { type: ModelService }
    ]; };
    __decorate([
        Input()
    ], MenuItemComponent.prototype, "item", void 0);
    __decorate([
        Input()
    ], MenuItemComponent.prototype, "styleName", void 0);
    __decorate([
        Input()
    ], MenuItemComponent.prototype, "enabled", null);
    MenuItemComponent = __decorate([
        Component({
            selector: 'sti-menu-item',
            template: "\n    <div *ngIf=\"item.type!='separator'\" [class]=\"className\" (mouseover)=\"mouseover()\"\n         (mouseout)=\"mouseout()\" [style.height]=\"height\">\n         <table style=\"height: 100%; width: 100%\" class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n            <tr class=\"stiJsViewerClearAllStyles\">\n                <td *ngIf=\"item.img != null && !model.options.appearance.rightToLeft\" class=\"stiJsViewerClearAllStyles\" style=\"width:22px; min-width:22px; padding: 0; text-align:center; line-height: 0;\">\n                  <img *ngIf=\"item.img != ''\" [style.height.px]=\"item.imageSize === 'Big' ? 32 : 16\" [style.width.px]=\"item.imageSize === 'Big' ? 32 : 16\" src=\"{{model.img(item.img)}}\" />\n                </td>\n\n                <td *ngIf=\"item.caption != null\" class=\"stiJsViewerClearAllStyles\" style=\"white-space:nowrap\"\n                  [style.textAlign]=\"model.options.appearance.rightToLeft ? 'right' : 'left'\"\n                  [style.padding]=\"item.imageSize == 'None' ? '0 20px 0 30px' : '0 20px 0 7px'\">\n                  {{item.caption}}\n                </td>\n\n                <td *ngIf=\"item.img != null && model.options.appearance.rightToLeft\" class=\"stiJsViewerClearAllStyles\" style=\"width:22px; min-width:22px; padding: 0; text-align:center; line-height: 0;\">\n                  <img *ngIf=\"item.img != ''\" [style.height.px]=\"item.imageSize === 'Big' ? 32 : 16\" [style.width.px]=\"item.imageSize === 'Big' ? 32 : 16\" src=\"{{model.img(item.img)}}\" />\n                </td>\n            </tr>\n         </table>\n    </div>\n    <div *ngIf=\"item.type=='separator'\" class=\"stiJsViewerVerticalMenuSeparator\" [style.margin]=\"item.imageSize == 'Big' ? '1px 2px 1px 2px' : ''\">\n    </div>\n  "
        })
    ], MenuItemComponent);
    return MenuItemComponent;
}());
export { MenuItemComponent };
