// stimulsoft-viewer-angular/esm5/lib/menu/menu.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { Menu, MenuService } from './menu.service';
import { trigger, state, style, transition, animate } from '@angular/animations';
import { ControllerService } from '../services/controller.service';
import { HelperService } from '../services/helper.service';
var MenuComponent = /** @class */ (function () {
    function MenuComponent(model, controller, helper, menuService) {
        this.model = model;
        this.controller = controller;
        this.helper = helper;
        this.menuService = menuService;
        this.style = 'display: none';
        this.overflow = 'hidden';
    }
    MenuComponent.prototype.ngAfterViewInit = function () {
        this.menu.menuEl = this.menuEl;
        this.menu.innerContent = this.innerContent;
    };
    MenuComponent.prototype.ngOnInit = function () {
    };
    MenuComponent.prototype.startAnimation = function (event) {
        if (event.toState === 'expanded') {
            this.style = 'width: 350px; overflow: hidden';
            this.overflow = 'hidden';
        }
    };
    MenuComponent.prototype.doneAnimation = function (event) {
        if (event.toState === 'expanded') {
            this.style = 'overflow: visible;';
            this.overflow = this.menu.type.indexOf('parameterMenu') === 0 || this.model.options.isMobileDevice ? 'auto' : 'hidden';
        }
        else if (event.toState === 'initialUp' || event.toState === 'initialDown') {
            this.style = 'display: none';
            this.overflow = 'hidden';
        }
    };
    MenuComponent.prototype.menuClick = function (item) {
        this.menuService.closeAllMenus();
        this.controller.action({ name: item.name, value: item.value });
    };
    MenuComponent.prototype.mouseup = function () {
        this.menuService.menuMouseUp = this.menu.type;
    };
    Object.defineProperty(MenuComponent.prototype, "zIndex", {
        get: function () {
            return this.menu.type === 'datePickerMenu' ? 36 : (this.menu.type === 'verticalMenu' ? 37 : 35);
        },
        enumerable: true,
        configurable: true
    });
    MenuComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: HelperService },
        { type: MenuService }
    ]; };
    __decorate([
        Input()
    ], MenuComponent.prototype, "menu", void 0);
    __decorate([
        ViewChild('menuEl')
    ], MenuComponent.prototype, "menuEl", void 0);
    __decorate([
        ViewChild('innerContent')
    ], MenuComponent.prototype, "innerContent", void 0);
    MenuComponent = __decorate([
        Component({
            selector: 'sti-menu',
            template: "\n      <div #menuEl class=\"stiJsViewerParentMenu\"\n      [style]=\"menu.sizeStyle || style\"\n      [style.top.px]=\"menu.top\"\n      [style.left.px]=\"menu.left\"\n      [style.zIndex]=\"zIndex\"\n      [style.height]=\"menu.type=='buttonMenu' && model.options.isMobileDevice ? '100%' : menu.height + 'px'\"\n      [@.disabled]=\"!model.options.toolbar.menuAnimation\"\n      (mouseup)=\"mouseup()\">\n        <div #innerContent [style]=\"{overflowX: 'hidden', overflowY: overflow}\"\n        [style.color]=\"model.options.toolbar.fontColor\"\n        [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n        [style.maxHeight]=\"menu.type=='buttonMenu' && model.options.isMobileDevice ? '100%' : '450px'\"\n        [style.height]=\"menu.type=='buttonMenu' && model.options.isMobileDevice ? '100%' : null\"\n        [@expand]=\"menu.state || 'initialDown'\"\n        (@expand.start)=\"startAnimation($event)\"\n        (@expand.done)=\"doneAnimation($event)\"\n        [class]=\"helper.val(menu.menuStyleName, 'stiJsViewerMenu')\">\n            <ng-container *ngIf=\"menu.type=='buttonMenu'\">\n              <sti-menu-item *ngFor=\"let item of menu.items\" [item]=\"item\" (click)=\"menuClick(item)\"></sti-menu-item>\n            </ng-container>\n\n            <sti-parameter-menu-edit-list *ngIf=\"menu.type=='parameterMenuForEditList'\" [menu]=\"menu\">\n            </sti-parameter-menu-edit-list>\n\n            <sti-parameter-menu-not-edit-list *ngIf=\"menu.type=='parameterMenuForNotEditList'\" [menu]=\"menu\">\n            </sti-parameter-menu-not-edit-list>\n\n            <sti-parameter-menu-for-value *ngIf=\"menu.type=='parameterMenuForValue'\" [menu]=\"menu\">\n            </sti-parameter-menu-for-value>\n\n            <sti-parameter-menu-for-range *ngIf=\"menu.type=='parameterMenuForRange'\" [menu]=\"menu\">\n            </sti-parameter-menu-for-range>\n\n            <sti-date-picker-menu *ngIf=\"menu.type=='datePickerMenu'\" [menu]=\"menu\">\n            </sti-date-picker-menu>\n\n            <sti-double-date-picker-menu *ngIf=\"menu.type=='doubleDatePickerMenu'\" [menu]=\"menu\">\n            </sti-double-date-picker-menu>\n\n            <sti-document-security-menu *ngIf=\"menu.type=='documentSecurityMenu'\">\n            </sti-document-security-menu>\n\n            <sti-digital-signature-menu *ngIf=\"menu.type=='useDigitalSignatureMenu'\">\n            </sti-digital-signature-menu>\n\n            <sti-vertical-menu *ngIf=\"menu.type=='verticalMenu'\" [menu]=\"menu\">\n            </sti-vertical-menu>\n        </div>\n      </div>\n  ",
            animations: [
                trigger('expand', [
                    state('initialDown', style({ transform: 'translateY(-100%)' })),
                    state('initialUp', style({ transform: 'translateY(100%)' })),
                    state('initialLeft', style({ transform: 'translateX(-100%)' })),
                    state('expanded', style({ transform: 'translateY(0) translateX(0)' })),
                    transition('initialUp => expanded', [
                        animate('150ms ease-in-out')
                    ]),
                    transition('initialDown => expanded', [
                        animate('150ms ease-in-out')
                    ]),
                    transition('initialLeft => expanded', [
                        animate('150ms ease-in-out')
                    ]),
                    transition('expanded => initialLeft', [
                        animate('150ms ease-in-out')
                    ])
                ])
            ]
        })
    ], MenuComponent);
    return MenuComponent;
}());
export { MenuComponent };
