// stimulsoft-viewer-angular/esm5/lib/menu/menu.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { HelperService } from '../services/helper.service';
import { ModelService } from '../services/model.service';
import { MouseService } from '../services/mouse.service';
import { Subject } from 'rxjs';
var Menu = /** @class */ (function () {
    function Menu(name, items, parent, type, state, menuEl, top, left, innerContent, width, height, menuStyleName, params, verticalItems, action, itemStyleName, value, sizeStyle) {
        this.name = name;
        this.items = items;
        this.parent = parent;
        this.type = type;
        this.state = state;
        this.menuEl = menuEl;
        this.top = top;
        this.left = left;
        this.innerContent = innerContent;
        this.width = width;
        this.height = height;
        this.menuStyleName = menuStyleName;
        this.params = params;
        this.verticalItems = verticalItems;
        this.action = action;
        this.itemStyleName = itemStyleName;
        this.value = value;
        this.sizeStyle = sizeStyle;
    }
    return Menu;
}());
export { Menu };
var MenuService = /** @class */ (function () {
    function MenuService(helper, model, mouseService) {
        var _this = this;
        this.helper = helper;
        this.model = model;
        this.mouseService = mouseService;
        this.VERTICAL_MENU_NAME = 'verticalMenu';
        this.menuMouseUp = '';
        this._menus = {};
        this.subject = new Subject();
        this.mouseService.getDocumentMouseUp().subscribe(function (event) {
            if (_this.menuMouseUp === '') {
                _this.closeAllMenus();
            }
            else if (_this.menuMouseUp !== 'datePickerMenu' && _this.menuMouseUp !== _this.VERTICAL_MENU_NAME) {
                var datePicker = Object.values(_this._menus).find(function (m) { return m.type === 'datePickerMenu'; });
                if (datePicker != null) {
                    datePicker.state = 'initialDown';
                }
            }
            _this.menuMouseUp = '';
        });
    }
    MenuService.prototype.getVisibility = function () {
        return this.subject.asObservable();
    };
    MenuService.prototype.addMenu = function (menu) {
        this._menus[menu.name] = menu;
    };
    MenuService.prototype.closeAllMenus = function () {
        var _this = this;
        Object.values(this._menus).forEach(function (m) { return m.state = m.type === 'buttonMenu' && _this.model.options.isMobileDevice ? 'initialLeft' : 'initialDown'; });
    };
    MenuService.prototype.isMenuVisible = function () {
        return Object.values(this._menus).find(function (m) { return m.state === 'expanded'; }) != null;
    };
    MenuService.prototype.showMenu = function (menuName) {
        var _this = this;
        var _a, _b;
        var menu = this._menus[menuName];
        if (((_a = menu) === null || _a === void 0 ? void 0 : _a.type) !== 'datePickerMenu' && ((_b = menu) === null || _b === void 0 ? void 0 : _b.type) !== this.VERTICAL_MENU_NAME) {
            this.closeAllMenus();
        }
        if (menu) {
            menu.sizeStyle = 'opacity: 0;';
            if (this.model.options.isMobileDevice && menu.type === 'buttonMenu') {
                setTimeout(function () {
                    var innerContent = menu.innerContent.nativeElement;
                    menu.left = 0;
                    menu.top = 0;
                    menu.width = innerContent.offsetWidth;
                    menu.height = innerContent.offsetHeight;
                    menu.state = 'initialLeft';
                    menu.sizeStyle = null;
                    setTimeout(function () {
                        menu.state = 'expanded';
                    });
                });
            }
            else {
                setTimeout(function () {
                    menu.sizeStyle = null;
                    _this.showMenuInternal(menu);
                });
            }
            this.subject.next(menu);
        }
    };
    MenuService.prototype.showMenuInternal = function (menu) {
        var isVertMenu = true;
        var parentButton = menu.parent.nativeElement;
        var innerContent = menu.innerContent.nativeElement;
        var offsetHeight = menu.menuEl.nativeElement.offsetHeight;
        var style = menu.menuEl.nativeElement.style;
        var coords = this.getMenuCoordinates(parentButton, innerContent, offsetHeight, style, isVertMenu);
        menu.left = coords.left;
        menu.top = coords.top;
        menu.width = menu.width || coords.width;
        menu.height = coords.height;
        menu.state = coords.state;
        setTimeout(function () {
            menu.state = 'expanded';
        });
    };
    MenuService.prototype.getMenuCoordinates = function (parentButton, innerContent, offsetHeight, style, isVertMenu) {
        var menu = {};
        var browserWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
        var browserHeight = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
        var rightAlign = false;
        var mainClassName = 'stiJsViewerMainPanel';
        var animDirect = 'Down';
        var leftOffset = 0;
        var left = (isVertMenu)
            ? (this.model.options.appearance.rightToLeft || rightAlign
                ? (this.helper.findPosX(parentButton, mainClassName) - innerContent.offsetWidth + parentButton.offsetWidth) - (leftOffset || 0)
                : this.helper.findPosX(parentButton, mainClassName) - (leftOffset || 0))
            : ((animDirect === 'Right')
                ? (this.helper.findPosX(parentButton, mainClassName) + parentButton.offsetWidth + 2)
                : (this.helper.findPosX(parentButton, mainClassName) - parentButton.offsetWidth - 2));
        if (left + innerContent.offsetWidth > browserWidth) {
            left = browserWidth - innerContent.offsetWidth - 15;
        }
        if (left < 0) {
            left = 10;
        }
        menu.left = left;
        if (animDirect === 'Down' &&
            this.helper.findPosY(parentButton) + parentButton.offsetHeight + innerContent.offsetHeight > browserHeight &&
            this.helper.findPosY(parentButton) - innerContent.offsetHeight > 0) {
            animDirect = 'Up';
        }
        menu.top = (isVertMenu)
            ? ((animDirect === 'Down')
                ? (this.helper.findPosY(parentButton, mainClassName) + parentButton.offsetHeight + 2)
                : (this.helper.findPosY(parentButton, mainClassName) - offsetHeight))
            : (this.helper.findPosY(parentButton, mainClassName) + parentButton.offsetHeight + innerContent.offsetHeight > browserHeight &&
                (browserHeight - innerContent.offsetHeight - 10) > 0)
                ? (browserHeight - innerContent.offsetHeight - 10)
                : this.helper.findPosY(parentButton, mainClassName);
        menu.width = innerContent.offsetWidth;
        menu.height = innerContent.offsetHeight;
        if (menu.top + innerContent.offsetHeight > browserHeight) {
            menu.top = (browserHeight - innerContent.offsetHeight - 10);
        }
        if (menu.top < 0) {
            menu.top = 10;
        }
        menu.state = animDirect === 'Down' ? 'initialDown' : 'initialUp';
        return menu;
    };
    Object.defineProperty(MenuService.prototype, "menus", {
        get: function () {
            return Object.values(this._menus);
        },
        enumerable: true,
        configurable: true
    });
    MenuService.prototype.getVerticalMenu = function () {
        return this._menus[this.VERTICAL_MENU_NAME];
    };
    MenuService.prototype.getMenu = function (name) {
        return this._menus[name];
    };
    MenuService.ctorParameters = function () { return [
        { type: HelperService },
        { type: ModelService },
        { type: MouseService }
    ]; };
    MenuService = __decorate([
        Injectable()
    ], MenuService);
    return MenuService;
}());
export { MenuService };
