// stimulsoft-viewer-angular/esm5/lib/menu/parameter-menu-for-value.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { ModelService } from '../services/model.service';
import { HelperService } from '../services/helper.service';
import { InteractionsService } from '../services/interactions.service';
var ParameterMenuForValueComponent = /** @class */ (function () {
    function ParameterMenuForValueComponent(model, helper, intearctionService) {
        this.model = model;
        this.helper = helper;
        this.intearctionService = intearctionService;
    }
    ParameterMenuForValueComponent.prototype.ngOnInit = function () { };
    ParameterMenuForValueComponent.prototype.action = function (item) {
        this.menu.params.value = this.menu.params.key = item.key;
        this.menu.state = 'initialDown';
        if (this.menu.params.binding) {
            if (!this.model.options.paramsVariablesStartValues) {
                this.model.options.paramsVariablesStartValues = this.helper.copyObject(this.model.options.paramsVariables);
            }
            this.intearctionService.postInteraction({ action: 'InitVars', variables: this.intearctionService.getParametersValues() });
        }
    };
    ParameterMenuForValueComponent.prototype.getValue = function (item) {
        return (item.value !== '' && this.menu.params.type !== 'DateTime' && this.menu.params.type !== 'TimeSpan')
            ? item.value
            : this.helper.getStringKey(item.key, this.menu.params);
    };
    ParameterMenuForValueComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService },
        { type: InteractionsService }
    ]; };
    __decorate([
        Input()
    ], ParameterMenuForValueComponent.prototype, "menu", void 0);
    ParameterMenuForValueComponent = __decorate([
        Component({
            selector: 'sti-parameter-menu-for-value',
            template: "\n      <sti-parameter-find *ngIf=\"menu.params.items.length > 10\" [variable]=\"menu.params\"></sti-parameter-find>\n\n      <sti-parameter-menu-separator *ngIf=\"menu.params.items.length > 10\"></sti-parameter-menu-separator>\n\n      <div style=\"max-height: 400px\" [style.overflow]=\"menu.params.items.length > 10 ? 'hidden auto' : 'hidden'\">\n        <table class=\"stiJsViewerClearAllStyles stiJsViewerParametersMenuInnerTable\" cellpadding=\"0\" cellspacing=\"0\"\n          [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n          [style.color]=\"model.options.toolbar.fontColor\"\n          [style.fontSize]=\"'12px'\"\n          [style.width]=\"(menu.parent.nativeElement.offsetWidth - 5) + 'px'\">\n            <tbody>\n              <tr *ngFor=\"let item of menu.params.items\" class=\"stiJsViewerClearAllStyles\" [style.display]=\"item.visible ? '' : 'none'\">\n                <td class=\"stiJsViewerClearAllStyles\">\n                  <sti-parameter-menu-item>\n                      <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"width: 100%\">\n                          <tbody>\n                            <tr class=\"stiJsViewerClearAllStyles\">\n                              <td class=\"stiJsViewerClearAllStyles\">\n                                  <sti-parameter-menu-item (action)=\"action(item)\">\n                                    {{getValue(item)}}\n                                  </sti-parameter-menu-item>\n                              </td>\n                            </tr>\n                          </tbody>\n                      </table>\n                  </sti-parameter-menu-item>\n                </td>\n              </tr>\n            </tbody>\n          </table>\n      </div>\n  "
        })
    ], ParameterMenuForValueComponent);
    return ParameterMenuForValueComponent;
}());
export { ParameterMenuForValueComponent };
