// stimulsoft-viewer-angular/esm5/lib/menu/parameter-menu-edit-list.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { Menu, MenuService } from './menu.service';
import { HelperService } from '../services/helper.service';
import { Variable } from '../services/objects';
var ParameterMenuEditListComponent = /** @class */ (function () {
    function ParameterMenuEditListComponent(model, helper, menuService) {
        this.model = model;
        this.helper = helper;
        this.menuService = menuService;
    }
    ParameterMenuEditListComponent.prototype.ngOnInit = function () { };
    ParameterMenuEditListComponent.prototype.newItem = function (event) {
        var item = new Variable();
        if (this.menu.params.type === 'DateTime') {
            item.key = this.helper.getDateTimeObject();
            item.value = this.helper.dateTimeObjectToString(item.key);
        }
        else if (this.menu.params.type === 'TimeSpan') {
            item.key = '00:00:00';
            item.value = '00:00:00';
        }
        else if (this.menu.params.type === 'Bool') {
            item.key = 'False';
            item.value = 'False';
        }
        else {
            item.key = '';
            item.value = '';
        }
        item.focusOnCreate = true;
        if (!this.menu.params.items) {
            this.menu.params.items = [];
        }
        this.menu.params.items.push(item);
        setTimeout(function () {
            item.focusOnCreate = false;
        });
    };
    ParameterMenuEditListComponent.prototype.dateTimeButtonAction = function (item) {
        var _this = this;
        this.menuService.addMenu({
            type: 'datePickerMenu', name: 'datePickerMenu', items: [], parent: this.dateTimeButton,
            params: item
        });
        setTimeout(function () {
            _this.menuService.showMenu('datePickerMenu');
        });
    };
    ParameterMenuEditListComponent.prototype.guidButtonAction = function (item) {
        item.key = this.helper.newGuid();
    };
    ParameterMenuEditListComponent.prototype.removeButtonAction = function (item) {
        this.menu.params.items.splice(this.menu.params.items.indexOf(item), 1);
    };
    ParameterMenuEditListComponent.prototype.removeAllAction = function () {
        this.menu.params.items.splice(0, this.menu.params.items.length);
    };
    ParameterMenuEditListComponent.prototype.closeAction = function () {
        this.menu.state = 'initialDown';
    };
    ParameterMenuEditListComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService },
        { type: MenuService }
    ]; };
    __decorate([
        ViewChild('dateTimeButton')
    ], ParameterMenuEditListComponent.prototype, "dateTimeButton", void 0);
    __decorate([
        Input()
    ], ParameterMenuEditListComponent.prototype, "menu", void 0);
    ParameterMenuEditListComponent = __decorate([
        Component({
            selector: 'sti-parameter-menu-edit-list',
            template: "\n        <table class=\"stiJsViewerClearAllStyles stiJsViewerParametersMenuInnerTable\" cellpadding=\"0\" cellspacing=\"0\"\n        [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n        [style.color]=\"model.options.toolbar.fontColor\"\n        [style.fontSize]=\"'12px'\"\n        [style.width]=\"(menu.parent.nativeElement.offsetWidth - 5) + 'px'\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\">\n                <!--New Item Button -->\n                <sti-parameter-menu-item (action)=\"newItem($event)\">\n                  {{model.loc('NewItem')}}\n                </sti-parameter-menu-item>\n              </td>\n            </tr>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\">\n                <sti-parameter-menu-item *ngFor=\"let item of menu.params.items\" [active]=\"false\">\n                  <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n                    <tbody>\n                      <tr class=\"stiJsViewerClearAllStyles\">\n                         <td class=\"stiJsViewerClearAllStyles\" style=\"padding: 0px 1px 0px 0px;\">\n                            <sti-parameter-text-box [item]=\"item\" [isMenu]=\"true\"\n                              [variable]=\"menu.params\" [focusOnCreate]=\"item.focusOnCreate\">\n                            </sti-parameter-text-box>\n                         </td>\n\n                         <td *ngIf=\"menu.params.type == 'DateTime'\" class=\"stiJsViewerClearAllStyles\" style=\"padding: 0 1px 0 1px;\" #dateTimeButton>\n                           <sti-parameter-button [buttonType]=\"'DateTimeButton'\" (action)=\"dateTimeButtonAction(item)\">\n                           </sti-parameter-button>\n                         </td>\n\n                         <td *ngIf=\"menu.params.type == 'Guid'\" class=\"stiJsViewerClearAllStyles\" style=\"padding: 0 1px 0 1px;\">\n                           <sti-parameter-button [buttonType]=\"'GuidButton'\" (action)=\"guidButtonAction(item)\">\n                           </sti-parameter-button>\n                         </td>\n\n                         <td class=\"stiJsViewerClearAllStyles\" style=\"padding: 0 1px 0 1px;\">\n                           <sti-parameter-button [buttonType]=\"'RemoveItemButton'\" (action)=\"removeButtonAction(item)\" [params]=\"item\">\n                           </sti-parameter-button>\n                         </td>\n                       </tr>\n                     </tbody>\n                  </table>\n                </sti-parameter-menu-item>\n              </td>\n            </tr>\n\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\">\n                  <sti-parameter-menu-item (action)=\"removeAllAction()\">\n                      {{model.loc('RemoveAll')}}\n                  </sti-parameter-menu-item>\n\n                  <sti-parameter-menu-separator></sti-parameter-menu-separator>\n\n                  <sti-parameter-menu-item (action)=\"closeAction()\">\n                      {{model.loc('Close')}}\n                  </sti-parameter-menu-item>\n              </td>\n            </tr>\n          </tbody>\n        </table>\n  "
        })
    ], ParameterMenuEditListComponent);
    return ParameterMenuEditListComponent;
}());
export { ParameterMenuEditListComponent };
