// stimulsoft-viewer-angular/esm5/lib/menu/parameter-menu-item.component.js
import { __decorate } from "tslib";
import { Component, Output, EventEmitter, Input } from '@angular/core';
import { ModelService } from '../services/model.service';
var ParameterMenuItemComponent = /** @class */ (function () {
    function ParameterMenuItemComponent(model) {
        this.model = model;
        this.action = new EventEmitter();
        this.active = true;
        this.padding = '0px 5px';
        this.isOver = false;
        this.mouseDown = false;
    }
    ParameterMenuItemComponent.prototype.ngOnInit = function () { };
    ParameterMenuItemComponent.prototype.mouseover = function () {
        this.isOver = true;
    };
    ParameterMenuItemComponent.prototype.mouseout = function () {
        this.isOver = false;
        this.mouseDown = false;
    };
    ParameterMenuItemComponent.prototype.mousedown = function () {
        this.mouseDown = true;
    };
    ParameterMenuItemComponent.prototype.mouseup = function () {
        this.mouseDown = false;
        this.action.emit();
    };
    Object.defineProperty(ParameterMenuItemComponent.prototype, "className", {
        get: function () {
            return this.active ? this.mouseDown ? 'stiJsViewerParametersMenuItemPressed' : (this.isOver ? 'stiJsViewerParametersMenuItemOver' : 'stiJsViewerParametersMenuItem') : 'stiJsViewerParametersMenuItem';
        },
        enumerable: true,
        configurable: true
    });
    ParameterMenuItemComponent.ctorParameters = function () { return [
        { type: ModelService }
    ]; };
    __decorate([
        Output()
    ], ParameterMenuItemComponent.prototype, "action", void 0);
    __decorate([
        Input()
    ], ParameterMenuItemComponent.prototype, "active", void 0);
    __decorate([
        Input()
    ], ParameterMenuItemComponent.prototype, "padding", void 0);
    ParameterMenuItemComponent = __decorate([
        Component({
            selector: 'sti-parameter-menu-item',
            template: "\n      <div [class]=\"className\"\n        [style.height.px]=\"model.options.isTouchDevice ? 30 : 24\"\n        (mouseover)=\"mouseover()\"\n        (mouseout)=\"mouseout()\"\n        (mousedown)=\"mousedown()\"\n        (mouseup)=\"mouseup()\">\n        <table class=\"stiJsViewerClearAllStyles stiJsViewerParametersMenuItemInnerTable\" cellpadding=\"0\" cellspacing=\"0\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\" [style.padding]=\"padding\">\n                <ng-content>\n                </ng-content>\n              </td>\n            </tr>\n          </tbody>\n        </table>\n\n      </div>\n  "
        })
    ], ParameterMenuItemComponent);
    return ParameterMenuItemComponent;
}());
export { ParameterMenuItemComponent };
