// stimulsoft-viewer-angular/esm5/lib/menu/vertical-menu-item.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
import { MenuService } from './menu.service';
var VerticalMenuItemComponent = /** @class */ (function () {
    function VerticalMenuItemComponent(model, menuService) {
        this.model = model;
        this.menuService = menuService;
        this.styleName = 'stiJsViewerMenuStandartItem';
        this.action = new EventEmitter();
        this.over = false;
        this._enabled = true;
    }
    VerticalMenuItemComponent.prototype.ngOnInit = function () { };
    Object.defineProperty(VerticalMenuItemComponent.prototype, "enabled", {
        get: function () {
            return this._enabled;
        },
        set: function (value) {
            if (!value) {
                this.over = false;
            }
            this._enabled = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(VerticalMenuItemComponent.prototype, "className", {
        get: function () {
            var _a;
            return this.styleName ? this.styleName + ' ' + (this.enabled ? (this.styleName + (this.over ? 'Over' : (((_a = this.item) === null || _a === void 0 ? void 0 : _a.selected) ? 'Selected' : 'Default'))) : this.styleName + 'Disabled') : '';
        },
        enumerable: true,
        configurable: true
    });
    VerticalMenuItemComponent.prototype.mouseover = function () {
        if (!this.enabled) {
            return;
        }
        this.over = true;
    };
    VerticalMenuItemComponent.prototype.mouseout = function () {
        if (!this.enabled) {
            return;
        }
        this.over = false;
    };
    VerticalMenuItemComponent.prototype.click = function () {
        if (this._enabled) {
            this.action.emit(this.item);
        }
    };
    Object.defineProperty(VerticalMenuItemComponent.prototype, "isSmallItem", {
        get: function () {
            return this.styleName && this.styleName.indexOf('MenuStandartItem') >= 0;
        },
        enumerable: true,
        configurable: true
    });
    VerticalMenuItemComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: MenuService }
    ]; };
    __decorate([
        Input()
    ], VerticalMenuItemComponent.prototype, "styleName", void 0);
    __decorate([
        Input()
    ], VerticalMenuItemComponent.prototype, "item", void 0);
    __decorate([
        Input()
    ], VerticalMenuItemComponent.prototype, "imageSizesWidth", void 0);
    __decorate([
        Input()
    ], VerticalMenuItemComponent.prototype, "imageSizesHeight", void 0);
    __decorate([
        Output()
    ], VerticalMenuItemComponent.prototype, "action", void 0);
    __decorate([
        Input()
    ], VerticalMenuItemComponent.prototype, "enabled", null);
    VerticalMenuItemComponent = __decorate([
        Component({
            selector: 'sti-vertical-menu-item',
            template: "\n    <div [style.height]=\"item?.type == 'DigitalSignature' ? 'auto' :(model.options.isMobileDevice ? '0.4in' : (model.options.isTouchDevice ? '30px' : '24px'))\"\n         [class]=\"className\"\n         [style.width]=\"item?.type == 'DigitalSignature' ? '100%' :  menuService.getVerticalMenu().width + 'px'\"\n         [style.lineHeight]=\"item?.type == 'DigitalSignature' ? 1.3 : null\"\n         (mouseover)=\"mouseover()\"\n         (mouseout)=\"mouseout()\"\n         (click)=\"click()\">\n         <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"height: 100%; width: 100%\">\n            <tbody>\n              <tr class=\"stiJsViewerClearAllStyles\">\n                 <td *ngIf=\"item?.imageName != null && !model.options.appearance.rightToLeft\"\n                     style=\"width:22px; min-width = 22px; padding: 0; textAlign: center; lineHeight: 0\"\n                     class=\"stiJsViewerClearAllStyles\">\n                     <img [style.width.px]=\"(imageSizesWidth ? imageSizesWidth : (isSmallItem ? 16 : 32))\"\n                          [style.height.px]=\"(imageSizesHeight ? imageSizesHeight : (isSmallItem ? 16 : 32))\"\n                          [src]=\"model.img(item?.imageName)\"\n                          [style.visible]=\"model.img(item?.imageName) === '' ? 'none' : ''\" />\n                </td>\n\n                <td *ngIf=\"item?.caption != null && !model.options.appearance.rightToLeft && item?.type != 'DigitalSignature'\"\n                     style=\"padding: 0px 20px 0px 7px; text-align: left; white-space: nowrap;\"\n                     [style.fontSize]=\"model.options.isMobileDevice ? '0.16in' : ''\"\n                     class=\"stiJsViewerClearAllStyles\">\n                     {{item?.caption}}\n                </td>\n\n                <td *ngIf=\"item?.caption != null && !model.options.appearance.rightToLeft && item?.type == 'DigitalSignature'\"\n                     style=\"padding: 8px 20px 8px 8px ; text-align: left; white-space: nowrap;\"\n                     [style.fontSize]=\"model.options.isMobileDevice ? '0.16in' : ''\"\n                     class=\"stiJsViewerClearAllStyles\" [innerHtml]=\"item?.caption\">\n                </td>\n\n                <!-- right-to-left -->\n                <td *ngIf=\"item?.caption != null && model.options.appearance.rightToLeft && item?.type == 'DigitalSignature'\"\n                     style=\"padding: 8px 20px 8px 8px; text-align: right; white-space: nowrap;\"\n                     [style.fontSize]=\"model.options.isMobileDevice ? '0.16in' : ''\"\n                     class=\"stiJsViewerClearAllStyles\" [innerHtml]=\"item?.caption\">\n                </td>\n\n                <td *ngIf=\"item?.caption != null && model.options.appearance.rightToLeft && item?.type != 'DigitalSignature'\"\n                     style=\"padding: 0 7px 0 20px; text-align: right; white-space: nowrap;\"\n                     [style.fontSize]=\"model.options.isMobileDevice ? '0.16in' : ''\"\n                     class=\"stiJsViewerClearAllStyles\">\n                     {{item?.caption}}\n                </td>\n\n                <td *ngIf=\"item?.imageName != null && model.options.appearance.rightToLeft\"\n                     style=\"width:22px; min-width = 22px; padding: 0; textAlign: center; lineHeight: 0\"\n                     class=\"stiJsViewerClearAllStyles\">\n                     <img [style.width.px]=\"(imageSizesWidth ? imageSizesWidth : (isSmallItem ? 16 : 32))\"\n                          [style.height.px]=\"(imageSizesHeight ? imageSizesHeight : (isSmallItem ? 16 : 32))\"\n                          [src]=\"model.img(item?.imageName)\"\n                          [style.visible]=\"model.img(item?.imageName) === '' ? 'none' : ''\" />\n                </td>\n              </tr>\n            </tbody>\n          </table>\n    </div>\n  "
        })
    ], VerticalMenuItemComponent);
    return VerticalMenuItemComponent;
}());
export { VerticalMenuItemComponent };
