// stimulsoft-viewer-angular/esm5/lib/services/api.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { ModelService } from './model.service';
import { ControllerService } from './controller.service';
import { ExportService } from './export.service';
import { HelperService } from './helper.service';
import { MailService } from './mail.service';
var ApiService = /** @class */ (function () {
    function ApiService(model, controller, exportService, helper, mailService) {
        this.model = model;
        this.controller = controller;
        this.exportService = exportService;
        this.helper = helper;
        this.mailService = mailService;
        this.EXPORT_FORMATS = ['Document', 'Pdf', 'Xps', 'Ppt2007', 'Html', 'Html5', 'Mht', 'Text', 'Rtf', 'Word2007', 'Odt', 'Excel',
            'ExcelBinary', 'ExcelXml', 'Excel2007', 'Ods', 'Csv', 'Dbf', 'Dif', 'Sylk', 'Json', 'Xml', 'ImageBmp',
            'ImageGif', 'ImageJpeg', 'ImagePcx', 'ImagePng', 'ImageTiff', 'ImageEmf', 'ImageSvg', 'ImageSvgz'];
    }
    Object.defineProperty(ApiService.prototype, "currentPage", {
        /**
         * The current page number
         */
        get: function () {
            return this.model.reportParams.pageNumber;
        },
        set: function (value) {
            if (value >= 0 && value < this.model.reportParams.pagesCount) {
                this.model.reportParams.pageNumber = value;
                this.controller.getPages();
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ApiService.prototype, "pageCount", {
        /**
         * The total pages count
         */
        get: function () {
            return this.model.reportParams.pageNumber;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ApiService.prototype, "viewMode", {
        /**
         * The view mode, can be 'SinglePage', 'Continuous' & 'MultiplePages'
         */
        get: function () {
            return this.model.reportParams.viewMode;
        },
        set: function (value) {
            if (value === 'SinglePage' || value === 'Continuous' || value === 'MultiplePages') {
                this.model.reportParams.viewMode = value;
                this.controller.getPages();
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ApiService.prototype, "zoom", {
        /**
         * The page zoom in percent
         * From 1 to 1000
         */
        get: function () {
            return this.model.reportParams.zoom;
        },
        set: function (value) {
            if (value >= 1 && value <= 1000) {
                this.model.reportParams.zoom = value;
                this.controller.getPages();
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * Zoom page in height
     */
    ApiService.prototype.zoomPageHeight = function () {
        this.controller.action({ name: 'ZoomOnePage' });
    };
    /**
     * Zoom page in width
     */
    ApiService.prototype.zoomPageWidth = function () {
        this.controller.action({ name: 'ZoomPageWidth' });
    };
    /**
     * Print to PDF
     */
    ApiService.prototype.printPdf = function () {
        this.controller.action({ name: 'PrintPdf' });
    };
    /**
     * Print with preview
     */
    ApiService.prototype.printWithPreview = function () {
        this.controller.action({ name: 'PrintWithPreview' });
    };
    /**
     * Print without preview
     */
    ApiService.prototype.printWithoutPreview = function () {
        this.controller.action({ name: 'PrintWithoutPreview' });
    };
    /**
     * Show export form
     *
     * @param format The format to export, can be 'Document', 'Pdf', 'Xps', 'Ppt2007', 'Html', 'Html5', 'Mht', 'Text', 'Rtf', 'Word2007', 'Odt', 'Excel',
     * 'ExcelBinary', 'ExcelXml', 'Excel2007', 'Ods', 'Csv', 'Dbf', 'Dif', 'Sylk', 'Json', 'Xml', 'ImageBmp',
     * 'ImageGif', 'ImageJpeg', 'ImagePcx', 'ImagePng', 'ImageTiff', 'ImageEmf', 'ImageSvg', 'ImageSvgz'
     */
    ApiService.prototype.showExportForm = function (format) {
        if (this.EXPORT_FORMATS.some(function (f) { return f === format; })) {
            this.controller.actionSubject.next({ action: 'Export', data: { format: format } });
        }
    };
    /**
     * Show export form & email
     *
     * @param format The format to export, can be 'Document', 'Pdf', 'Xps', 'Ppt2007', 'Html', 'Html5', 'Mht', 'Text', 'Rtf', 'Word2007', 'Odt', 'Excel',
     * 'ExcelBinary', 'ExcelXml', 'Excel2007', 'Ods', 'Csv', 'Dbf', 'Dif', 'Sylk', 'Json', 'Xml', 'ImageBmp',
     * 'ImageGif', 'ImageJpeg', 'ImagePcx', 'ImagePng', 'ImageTiff', 'ImageEmf', 'ImageSvg', 'ImageSvgz'
     */
    ApiService.prototype.showExportEmailForm = function (format) {
        if (this.EXPORT_FORMATS.some(function (f) { return f === format; })) {
            this.controller.actionSubject.next({ action: 'SendEmail', data: { format: format } });
        }
    };
    /**
     * Export report/dashboard to selected format
     * Use default settings if not specified
     *
     * @param format The format to export, can be 'Document', 'Pdf', 'Xps', 'Ppt2007', 'Html', 'Html5', 'Mht', 'Text', 'Rtf', 'Word2007', 'Odt', 'Excel',
     * 'ExcelBinary', 'ExcelXml', 'Excel2007', 'Ods', 'Csv', 'Dbf', 'Dif', 'Sylk', 'Json', 'Xml', 'ImageBmp',
     * 'ImageGif', 'ImageJpeg', 'ImagePcx', 'ImagePng', 'ImageTiff', 'ImageEmf', 'ImageSvg', 'ImageSvgz'
     *
     * @param settings The export settings
     */
    ApiService.prototype.export = function (format, settings) {
        if (this.EXPORT_FORMATS.some(function (f) { return f === format; })) {
            var exportSettings = this.getExportSettings(format, settings);
            this.controller.postExport(format, exportSettings);
        }
    };
    /**
     * Export report/dashboard to seleted format & email
     * Use default settings if not specified
     * Use default email settings if not specified
     *
     * @param format The format to export, can be 'Document', 'Pdf', 'Xps', 'Ppt2007', 'Html', 'Html5', 'Mht', 'Text', 'Rtf', 'Word2007', 'Odt', 'Excel',
     * 'ExcelBinary', 'ExcelXml', 'Excel2007', 'Ods', 'Csv', 'Dbf', 'Dif', 'Sylk', 'Json', 'Xml', 'ImageBmp',
     * 'ImageGif', 'ImageJpeg', 'ImagePcx', 'ImagePng', 'ImageTiff', 'ImageEmf', 'ImageSvg', 'ImageSvgz'
     *
     * @param settings The export settings
     *
     * @param email The email
     *
     * @param message The email message
     *
     * @param subject The email subject
     */
    ApiService.prototype.exportEmail = function (format, settings, email, subject, message) {
        if (this.EXPORT_FORMATS.some(function (f) { return f === format; })) {
            this.exportService.exportSettings = this.getExportSettings(format, settings);
            this.exportService.format = format;
            this.mailService.fillDefaults();
            if (email) {
                this.exportService.exportSettings.Email = email;
            }
            if (message) {
                this.exportService.exportSettings.Message = message;
            }
            if (subject) {
                this.exportService.exportSettings.Subject = subject;
            }
            this.mailService.sendMail();
        }
    };
    ApiService.prototype.getExportSettings = function (format, settings) {
        this.exportService.sendMail = false;
        this.exportService.format = format;
        var exportSettings = this.helper.copyObject(this.exportService.getDefaultExportSettings(format));
        if (settings) {
            Object.keys(settings).forEach(function (key) { return exportSettings[key] = settings[key]; });
        }
        return exportSettings;
    };
    ApiService.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: ExportService },
        { type: HelperService },
        { type: MailService }
    ]; };
    ApiService = __decorate([
        Injectable()
    ], ApiService);
    return ApiService;
}());
export { ApiService };
