// stimulsoft-viewer-angular/esm5/lib/services/controller.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { StiHttpClientService } from './http-client.service';
import { Subject } from 'rxjs';
import { ModelService } from './model.service';
import { StylesService } from './styles.service';
import { PageService } from './page.service';
import { HelperService } from './helper.service';
import { FormService } from '../forms/form.service';
import { FullScreenService } from './full-screen.service';
import { MenuService, Menu } from '../menu/menu.service';
var ControllerService = /** @class */ (function () {
    function ControllerService(httpClient, model, stylesService, pageService, helper, formService, fullScreenService, menuService, styleService) {
        var _this = this;
        this.httpClient = httpClient;
        this.model = model;
        this.stylesService = stylesService;
        this.pageService = pageService;
        this.helper = helper;
        this.formService = formService;
        this.fullScreenService = fullScreenService;
        this.menuService = menuService;
        this.styleService = styleService;
        this.subject = new Subject();
        this.actionSubject = new Subject();
        setInterval(function () {
            var _a, _b;
            if (_this.model.httpError) {
                if ((_a = _this.model.httpError.error) === null || _a === void 0 ? void 0 : _a.text) {
                    _this.showError((_b = _this.model.httpError.error) === null || _b === void 0 ? void 0 : _b.text);
                }
                else if (_this.model.httpError.message) {
                    _this.showError(null, _this.model.httpError.message);
                }
                else if (_this.model.httpError.error instanceof ErrorEvent) {
                    _this.showError(_this.model.httpError.error.message);
                }
                _this.model.httpError = null;
            }
            if (_this.model.navigateNode) {
                _this.postBookmarkNodeAction(_this.model.navigateNode);
                _this.model.navigateNode = null;
            }
        }, 250);
        this.setupHandlers();
    }
    ControllerService.prototype.setupHandlers = function () {
        var _this = this;
        this.getMessage().subscribe(function (message) {
            switch (message.action) {
                case 'GetReport':
                case 'OpenReport':
                    setTimeout(function () {
                        var _a, _b;
                        _this.styleService.addCustomFontStyles((_a = message.data) === null || _a === void 0 ? void 0 : _a.customFonts);
                        if (_this.model.options.appearance.fullScreenMode) {
                            _this.fullScreenService.changeFullScreenMode(true);
                        }
                        if (((_b = message.data) === null || _b === void 0 ? void 0 : _b.stimulsoftFontContent) && !_this.model.options.stimulsoftFontContent) {
                            _this.model.options.stimulsoftFontContent = message.data.stimulsoftFontContent;
                            _this.styleService.addCustomFontStyles([{
                                    contentForCss: message.data.stimulsoftFontContent,
                                    originalFontFamily: 'Stimulsoft'
                                }]);
                        }
                    });
                    break;
            }
        });
        this.model.controls.bookmarksPanel.getVisibility().subscribe(function (value) {
            if (value && _this.model.options.isMobileDevice) {
                _this.model.controls.parametersPanel.visible = false;
            }
            if (value) {
                _this.hideToolbar();
            }
            else {
                _this.showToolbar();
            }
        });
        this.model.controls.parametersPanel.getVisibility().subscribe(function (value) {
            if (value && _this.model.options.isMobileDevice) {
                _this.model.controls.bookmarksPanel.visible = false;
            }
            if (value) {
                _this.hideToolbar();
            }
            else {
                _this.showToolbar();
            }
        });
        this.menuService.getVisibility().subscribe(function (menu) {
            _this.hideToolbar();
        });
    };
    ControllerService.prototype.hideToolbar = function () {
        if (this.model.options.isMobileDevice && this.model.options.toolbar.autoHide) {
            this.model.controls.toolbar.visible = false;
            this.model.controls.navigatePanel.visible = false;
        }
    };
    ControllerService.prototype.showToolbar = function () {
        var _this = this;
        if (this.model.options.isMobileDevice && this.model.options.toolbar.autoHide) {
            this.model.controls.toolbar.visible = true;
            this.model.controls.navigatePanel.visible = true;
            setTimeout(function () {
                _this.keepToolbar();
            }, 300);
        }
    };
    ControllerService.prototype.keepToolbar = function () {
        var _this = this;
        if (this.model.options.isMobileDevice && this.model.options.toolbar.autoHide) {
            clearTimeout(this.model.toolbarHideTimer);
            this.model.toolbarHideTimer = setTimeout(function () {
                _this.hideToolbar();
            }, 4000);
        }
    };
    ControllerService.prototype.getMessage = function () {
        return this.subject.asObservable();
    };
    ControllerService.prototype.getActionMessage = function () {
        return this.actionSubject.asObservable();
    };
    ControllerService.prototype.post = function (action, cAction, postData, responseType, subAction) {
        var _this = this;
        if (responseType === void 0) { responseType = 'json'; }
        this.model.controls.navigatePanel.enabled = false;
        this.model.controls.toolbar.enabled = false;
        var url;
        if (!cAction) {
            url = this.model.requestUrl.replace('{action}', action === 'GetReport' || this.model.options.server.cacheMode === 'None' ? this.model.options.actions.getReport : this.model.options.actions.viewerEvent);
        }
        else {
            url = this.model.requestUrl.replace('{action}', cAction);
        }
        if (postData) {
            postData.action = action;
        }
        else {
            postData = { action: action };
        }
        this.httpClient.post(url, this.model.createPostParameters(postData), responseType).subscribe(function (data) {
            var _a;
            _this.model.showProgress = false;
            _this.model.controls.navigatePanel.enabled = true;
            _this.model.controls.toolbar.enabled = true;
            if (data === 'ServerError:The report is not specified.' && ((_a = _this.model.options) === null || _a === void 0 ? void 0 : _a.isReportRecieved)) {
                _this.model.options.isReportRecieved = false;
                _this.model.reportParams.prevPageNumber = _this.model.reportParams.pageNumber;
                _this.post('GetReport');
                return;
            }
            if (data) {
                _this.model.reportParams = data;
                _this.startRefreshReportTimer(data.refreshTime);
            }
            _this.subject.next({ action: action, data: data, subAction: subAction });
        });
    };
    ControllerService.prototype.loadViewer = function () {
        var _this = this;
        var url = this.model.requestUrl.replace('{action}', this.model.action);
        this.httpClient.post(url, this.model.createPostParameters({ action: 'AngularViewerData' }, true, false), 'json').subscribe(function (data) {
            _this.model.clear();
            _this.stylesService.setupStyle(atob(data['styles']), 'viewer');
            _this.model.options = data;
            _this.checkTrExp();
            _this.initAutoUpdateCache();
            _this.subject.next({ action: 'viewer_loaded' });
            _this.getReport();
        });
    };
    ControllerService.prototype.getReport = function () {
        this.model.options.paramsVariablesStartValues = null;
        this.post('GetReport');
    };
    ControllerService.prototype.getPages = function () {
        this.post('GetPages');
    };
    ControllerService.prototype.postExport = function (format, settings, elementName, isDashboardExport) {
        if (isDashboardExport === void 0) { isDashboardExport = false; }
        var data = {
            action: isDashboardExport ? 'ExportDashboard' : 'ExportReport',
            exportFormat: format,
            exportSettings: settings,
            elementName: elementName
        };
        var doc = settings && settings.OpenAfterExport && this.model.options.appearance.openExportedReportWindow === '_blank' ? this.helper.openNewWindow('about:blank', '_blank').document : null;
        var url = this.model.requestUrl;
        this.httpClient.postForm(url.replace('{action}', this.model.options.actions.exportReport), data, doc);
        this.actionSubject.next({ action: 'ExportReport', data: data });
    };
    ControllerService.prototype.loadFile = function (fileName, content) {
        if (typeof content !== 'string' || content === '') {
            return;
        }
        if (content.indexOf('<?xml') === 0 || content.indexOf('{') === 0) {
            content = btoa(content);
        }
        var data = {
            action: 'OpenReport',
            openingFileName: fileName || 'Report.mdc',
            base64Data: content.indexOf('base64,') > 0 ? content.substr(content.indexOf('base64,') + 7) : content
        };
        if (fileName && (fileName.toLowerCase().indexOf('.mdx') >= 0 || fileName.toLowerCase().indexOf('.mrx') >= 0)) {
            this.formService.showForm('passwordForm', data);
        }
        else {
            this.postOpen(data);
        }
    };
    ControllerService.prototype.postOpen = function (data) {
        this.model.clearViewerState();
        this.model.reportParams.reportFileName = data.openingFileName;
        this.post('OpenReport', this.model.options.actions.openReport, data);
    };
    ControllerService.prototype.action = function (event) {
        var _this = this;
        switch (event.name) {
            case 'Find':
                this.model.controls.findPanel.visible = !this.model.controls.findPanel.visible;
                return;
            case 'Bookmarks':
                this.model.controls.bookmarksPanel.visible = !this.model.controls.bookmarksPanel.visible;
                return;
            case 'Parameters':
                this.model.controls.parametersPanel.visible = !this.model.controls.parametersPanel.visible;
                return;
            case 'BookmarkAction':
                if (this.model.reportParams.pageNumber === event.bookmarkPage || this.model.reportParams.viewMode !== 'SinglePage') {
                    this.helper.scrollToAnchor(event.bookmarkAnchor, event.componentGuid);
                    return;
                }
                else {
                    this.model.reportParams.pageNumber = event.bookmarkPage;
                    this.model.options.bookmarkAnchor = event.bookmarkAnchor;
                    this.model.options.componentGuid = event.componentGuid;
                }
                break;
            case 'GoToPage':
                this.model.reportParams.pageNumber = event.value;
                break;
            case 'FirstPage':
                this.model.reportParams.pageNumber = 0;
                break;
            case 'PrevPage':
                this.model.reportParams.pageNumber = Math.max(0, this.model.reportParams.pageNumber - 1);
                break;
            case 'NextPage':
                this.model.reportParams.pageNumber = Math.min(this.model.reportParams.pagesCount - 1, this.model.reportParams.pageNumber + 1);
                break;
            case 'LastPage':
                this.model.reportParams.pageNumber = this.model.reportParams.pagesCount - 1;
                break;
            case 'ViewModeSinglePage':
                this.model.reportParams.viewMode = 'SinglePage';
                break;
            case 'ViewModeContinuous':
                this.model.reportParams.viewMode = 'Continuous';
                break;
            case 'ViewModeMultiplePages':
                this.model.reportParams.viewMode = 'MultiplePages';
                break;
            case 'Zoom25':
                this.model.reportParams.zoom = 25;
                break;
            case 'Zoom50':
                this.model.reportParams.zoom = 50;
                break;
            case 'Zoom75':
                this.model.reportParams.zoom = 75;
                break;
            case 'Zoom100':
                this.model.reportParams.zoom = 100;
                break;
            case 'Zoom150':
                this.model.reportParams.zoom = 150;
                break;
            case 'Zoom200':
                this.model.reportParams.zoom = 200;
                break;
            case 'ZoomOnePage':
            case 'ZoomPageWidth':
                this.model.reportParams.zoom = event.name === 'ZoomPageWidth' ? this.pageService.getZoomByPageWidth() : this.pageService.getZoomByPageHeight();
                break;
            case 'Submit':
                this.model.reportParams.editableParameters = null;
                if (this.model.reportParams.type === 'Report') {
                    this.model.reportParams.pageNumber = 0;
                }
                if (this.model.options.isMobileDevice) {
                    this.model.controls.parametersPanel.visible = false;
                }
                this.actionSubject.next({ action: 'Variables' });
                return;
            case 'Reset':
                if (this.model.options.paramsVariablesStartValues) {
                    this.model.options.paramsVariables = this.model.options.paramsVariablesStartValues;
                }
                this.actionSubject.next({ action: 'Reset' });
                return;
            case 'Editor':
                this.actionSubject.next({ action: 'Editor', data: {} });
                return;
            case 'Resources':
                this.model.controls.resourcesPanel.visible = !this.model.controls.resourcesPanel.visible;
                return;
            case 'PrintPdf':
            case 'PrintWithPreview':
            case 'PrintWithoutPreview':
                this.actionSubject.next({ action: 'Print', data: { format: event.name } });
                return;
            case 'Print':
                var format = 'PrintPdf';
                switch (this.model.options.toolbar.printDestination) {
                    case 'Pdf':
                        format = 'PrintPdf';
                        break;
                    case 'Direct':
                        format = 'PrintWithoutPreview';
                        break;
                    case 'WithPreview':
                        format = 'PrintWithPreview';
                        break;
                }
                this.actionSubject.next({ action: 'Print', data: { format: format } });
                return;
            case 'Open':
                this.model.openDialogFileMask = null;
                setTimeout(function () {
                    _this.model.openDialogFileMask = '.mdc,.mdz,.mdx,.mrt,.mrz,.mrx';
                });
                return;
            case 'ResourceView':
                this.postReportResource(event.value.name, 'View');
                return;
            case 'ResourceSaveFile':
                this.postReportResource(event.value.name, 'SaveFile');
                return;
            case 'FullScreen':
                this.fullScreenService.changeFullScreenMode(!this.model.options.appearance.fullScreenMode);
                return;
            case 'About':
                this.actionSubject.next({ action: 'About' });
                return;
            case 'Pin':
                this.model.options.toolbar.autoHide = !this.model.options.toolbar.autoHide;
                this.pageService.calculateLayout();
                if (this.model.options.toolbar.autoHide) {
                    setTimeout(function () {
                        _this.hideToolbar();
                    }, 200);
                }
                return;
            case 'Design':
                this.actionSubject.next({ action: 'Design' });
                break;
            default:
                if (event.name.indexOf('saveMenu') === 0) {
                    this.actionSubject.next({ action: 'Export', data: { format: event.name.substr(8) } });
                    return;
                }
                else if (event.name.indexOf('sendEmailMenu') === 0) {
                    this.actionSubject.next({ action: 'SendEmail', data: { format: event.name.substr(13) } });
                    return;
                }
        }
        this.getPages();
    };
    ControllerService.prototype.showError = function (message, messageText) {
        var type = 'Error';
        // Check for error in 'ServerError:' string format
        if (message != null && typeof (message) === 'string' && message.substr(0, 12) === 'ServerError:') {
            if (message.length <= 13) {
                messageText = 'An unknown error occurred (the server returned an empty value).';
            }
            else {
                messageText = message.substr(12);
            }
        }
        // Check for error in JSON format
        if (message != null && message.success === false && message.type && message.text) {
            type = message.type;
            messageText = message.text;
        }
        this.model.errorMessage = { error: 'Unknown error', type: type };
        if (messageText != null) {
            if (messageText === 'The report is not specified.' && !this.model.options.appearance.showReportIsNotSpecifiedMessage) {
                return true;
            }
            this.model.errorMessage = { error: messageText.replace('\n', '<br>'), type: type };
            if (this.model.images) {
                this.formService.showForm('errorMessageForm');
            }
            else {
                alert(messageText);
            }
            this.actionSubject.next({ action: 'Error' });
            return true;
        }
        return false;
    };
    ControllerService.prototype.postReportResource = function (resourceName, viewType) {
        var data = {
            action: 'ReportResource',
            reportResourceParams: {
                resourceName: resourceName,
                viewType: viewType
            }
        };
        var doc = viewType === 'View' ? this.helper.openNewWindow('about:blank', '_blank').document : null;
        var url = this.model.requestUrl;
        this.httpClient.postForm(url.replace('{action}', this.model.options.actions.viewerEvent), data, doc);
    };
    ControllerService.prototype.postBookmarkNodeAction = function (node) {
        node.selected = true;
        this.action({ name: 'BookmarkAction', bookmarkPage: node.page, bookmarkAnchor: node.url.replace(/\\\'/g, '\'').substr(1), componentGuid: node.compunentGuid });
    };
    ControllerService.prototype.viewerResized = function () {
        this.menuService.closeAllMenus();
    };
    ControllerService.prototype.initAutoUpdateCache = function () {
        var _this = this;
        if (this.model.options.server.allowAutoUpdateCache) {
            if (this.model.timerAutoUpdateCache) {
                clearInterval(this.model.timerAutoUpdateCache);
            }
            this.model.timerAutoUpdateCache = setInterval(function () { return _this.post('UpdateCache'); }, this.model.options.server.timeoutAutoUpdateCache);
        }
    };
    ControllerService.prototype.startRefreshReportTimer = function (timeout) {
        var _this = this;
        if (this.model.refreshReportTimer != null) {
            clearInterval(this.model.refreshReportTimer);
            this.model.refreshReportTimer = null;
        }
        if (timeout && timeout > 0) {
            this.model.refreshReportTimer = setInterval(function () {
                if (!_this.model.showProgress) {
                    _this.post('Refresh');
                }
            }, timeout * 1000);
        }
    };
    ControllerService.prototype.checkTrExp = function () {
        var _this = this;
        if (!this.model.options.cloudMode && !this.model.options.serverMode && !this.model.options.standaloneJsMode && !this.model.options.reportDesignerMode && !this.model.options.alternateValid) {
            var buildDate = new Date();
            try {
                /*if (this.model.options.jsMode && typeof Stimulsoft != 'undefined') {
                  let innerDate = Stimulsoft.StiVersion.created.innerDate;
                  if (innerDate['getFullYear'] && innerDate.getFullYear() > 2017)
                    buildDate = Stimulsoft.StiVersion.created.innerDate;
                }
                else*/ if (this.model.options.buildDate) {
                    buildDate = new Date(this.model.options.buildDate);
                }
            }
            catch (e) {
                buildDate = new Date();
            }
            var trDays_1 = Math.floor(((new Date()).getTime() - buildDate.getTime()) / 1000 / 60 / 60 / 24);
            if (trDays_1 > 60) {
                setTimeout(function () {
                    var message = trDays_1 > 120 ? _this.model.loc('NoticesYourTrialHasExpired') : _this.model.loc('NoticesYouUsingTrialVersion');
                    var image = 'Notifications.Warning.png';
                    var buttonCaption = _this.model.loc('ButtonOk');
                    var cancelAction;
                    var action;
                    if (trDays_1 > 120) {
                        action = cancelAction = function () { window.location.href = 'https://www.stimulsoft.com/en/online-store'; };
                    }
                    _this.model.notificationFormOptions = { message: message, image: image, buttonCaption: buttonCaption, cancelAction: cancelAction, action: action };
                    _this.formService.showForm('notificationForm');
                }, 3000);
            }
        }
    };
    ControllerService.ctorParameters = function () { return [
        { type: StiHttpClientService },
        { type: ModelService },
        { type: StylesService },
        { type: PageService },
        { type: HelperService },
        { type: FormService },
        { type: FullScreenService },
        { type: MenuService },
        { type: StylesService }
    ]; };
    ControllerService = __decorate([
        Injectable()
    ], ControllerService);
    return ControllerService;
}());
export { ControllerService };
