// stimulsoft-viewer-angular/esm5/lib/services/drill-down.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { ControllerService } from './controller.service';
import { ModelService } from './model.service';
import { HelperService } from './helper.service';
var DrillDownService = /** @class */ (function () {
    function DrillDownService(controller, model, helper) {
        var _this = this;
        this.controller = controller;
        this.model = model;
        this.helper = helper;
        controller.getMessage().subscribe(function (message) {
            if (message.action !== 'InitVars' && message.action !== 'viewer_loaded') {
                _this.model.options.drillDownInProgress = false;
                var parameters = message.data;
                if (_this.model.drillDownButtons.length === 0) {
                    _this.addButton(parameters.reportFileName, {});
                }
                if (message.action === 'DrillDown') {
                    var useDbsDrillDownPanel = false; // jsObject.controls.dashboardsPanel && jsObject.controls.dashboardsPanel.visible && jsObject.controls.dashboardsPanel.selectedButton;
                    if (useDbsDrillDownPanel) {
                        // jsObject.controls.dashboardsPanel.addDrillDownButton(null, parameters.drillDownGuid, parameters.drillDownParameters, parameters.previewSettings, parameters.reportFileName);
                    }
                    _this.showDrillDownPage(parameters.reportFileName, parameters.drillDownGuid, parameters.drillDownParameters, useDbsDrillDownPanel);
                }
            }
        });
    }
    DrillDownService.prototype.showDrillDownPage = function (reportFileName, drillDownGuid, drillDownParameters, useDbsDrillDownPanel) {
        var _this = this;
        if (useDbsDrillDownPanel) {
            return;
        }
        var buttonExist = false;
        this.model.drillDownButtons.forEach(function (b) {
            if (b.reportParams.drillDownGuid === drillDownGuid) {
                _this.model.drillDownButtons.forEach(function (bt) { return bt.selected = false; });
                buttonExist = b.selected = b.visible = true;
            }
        });
        if (!buttonExist) {
            this.addButton(reportFileName);
            this.model.reportParams.drillDownParameters = drillDownParameters;
            this.model.reportParams.pageNumber = 0;
            this.model.reportParams.pagesWidth = 0;
            this.model.reportParams.pagesHeight = 0;
        }
    };
    DrillDownService.prototype.addButton = function (caption, reportParams) {
        if (reportParams === void 0) { reportParams = {}; }
        this.model.drillDownButtons.forEach(function (b) { return b.selected = false; });
        this.model.drillDownButtons.push({ caption: caption, selected: true, reportParams: reportParams, visible: true });
        this.updateVisibility();
    };
    DrillDownService.prototype.updateVisibility = function () {
        this.model.controls.drillDownPanel.visible = this.model.drillDownButtons.length > 1;
    };
    DrillDownService.prototype.saveState = function () {
        var sButton = this.model.drillDownButtons.find(function (b) { return b.selected; });
        if (sButton) {
            sButton.reportParams = this.model.getReportParams();
        }
    };
    DrillDownService.ctorParameters = function () { return [
        { type: ControllerService },
        { type: ModelService },
        { type: HelperService }
    ]; };
    DrillDownService = __decorate([
        Injectable()
    ], DrillDownService);
    return DrillDownService;
}());
export { DrillDownService };
