// stimulsoft-viewer-angular/esm5/lib/services/find.service.js
import { __decorate, __values } from "tslib";
import { Injectable } from '@angular/core';
import { ModelService } from './model.service';
import { HelperService } from './helper.service';
import { AnimationService } from './animation.service';
var FindService = /** @class */ (function () {
    function FindService(model, helper, animationService) {
        var _this = this;
        this.model = model;
        this.helper = helper;
        this.animationService = animationService;
        this.symbols = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890';
        this.findLabels = [];
        this.text = '';
        this.lastFindText = '';
        this.matchCase = false;
        this.matchWholeWord = false;
        this.findMode = false;
        model.controls.findPanel.getVisibility().subscribe(function (value) {
            if (!value) {
                _this.hideFindLabels();
                _this.lastFindText = '';
            }
        });
    }
    FindService.prototype.hideFindLabels = function () {
        this.findLabels.forEach(function (findLabel) {
            var parentElement = findLabel.parentElement;
            parentElement.removeChild(findLabel);
            if (parentElement.oldPositionStyle) {
                parentElement.style.position = parentElement.oldPositionStyle;
            }
        });
        this.findLabels = [];
        this.findMode = false;
        this.lastFindText = '';
    };
    FindService.prototype.selectFindLabel = function (direction) {
        if (this.findLabels.length === 0) {
            return;
        }
        var selectedIndex = 0;
        var labels = this.findLabels;
        for (var i = 0; i < labels.length; i++) {
            if (labels[i].isSelected) {
                labels[i].setSelected(false);
                selectedIndex = i;
                break;
            }
        }
        if (direction === 'Next') {
            selectedIndex++;
            if (selectedIndex > labels.length - 1) {
                selectedIndex = 0;
            }
        }
        else {
            selectedIndex--;
            if (selectedIndex < 0) {
                selectedIndex = labels.length - 1;
            }
        }
        labels[selectedIndex].setSelected(true);
        this.goToFindedElement(labels[selectedIndex]);
    };
    FindService.prototype.showFindLabels = function () {
        var _this = this;
        this.hideFindLabels();
        this.findMode = true;
        this.lastFindText = this.text;
        var text = this.matchCase ? this.text : this.text.toLowerCase();
        this.model.pages.forEach(function (page) {
            var e_1, _a;
            var pageElements = page.page.getElementsByTagName('*');
            try {
                for (var pageElements_1 = __values(pageElements), pageElements_1_1 = pageElements_1.next(); !pageElements_1_1.done; pageElements_1_1 = pageElements_1.next()) {
                    var pageElement = pageElements_1_1.value;
                    var innerText = pageElement.innerHTML;
                    if (innerText && pageElement.childNodes.length === 1 && pageElement.childNodes[0].nodeName === '#text') {
                        if (!_this.matchCase) {
                            innerText = innerText.toLowerCase();
                        }
                        if (innerText.indexOf(text) >= 0) {
                            if (_this.matchWholeWord && !_this.isWholeWord(innerText, text)) {
                                continue;
                            }
                            var label = document.createElement('div');
                            label.ownerElement = pageElement;
                            label.className = 'stiJsViewerFindLabel';
                            label.style.width = (pageElement.offsetWidth - 4) + 'px';
                            var labelHeight = pageElement.offsetHeight - 4;
                            label.style.height = labelHeight + 'px';
                            label.style.top = '0px';
                            label.style.left = '0px';
                            label.ownerElement.oldPositionStyle = label.ownerElement.style.position;
                            if (label.ownerElement.style.position !== 'absolute' && label.ownerElement.style.position !== 'fixed') {
                                label.ownerElement.style.position = 'relative';
                            }
                            pageElement.insertBefore(label, pageElement.childNodes[0]);
                            label.setSelected = function (state) {
                                this.isSelected = state;
                                this.style.border = '2px solid ' + (state ? 'red' : '#8a8a8a');
                            };
                            if (_this.findLabels.length === 0) {
                                label.setSelected(true);
                            }
                            _this.findLabels.push(label);
                        }
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (pageElements_1_1 && !pageElements_1_1.done && (_a = pageElements_1.return)) _a.call(pageElements_1);
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
        if (this.findLabels.length > 0) {
            this.goToFindedElement(this.findLabels[0]);
        }
    };
    FindService.prototype.goToFindedElement = function (findLabel) {
        if (findLabel && findLabel.ownerElement) {
            var targetTop = this.helper.findPosY(findLabel.ownerElement, this.model.options.appearance.scrollbarsMode ? 'stiJsViewerReportPanel' : null, true) - findLabel.ownerElement.offsetHeight - 50;
            var d = new Date();
            var endTime = d.getTime() + this.model.options.scrollDuration;
            this.animationService.showAnimationForScroll(this.model.controls.reportPanel.el.nativeElement, targetTop, endTime, function () { });
        }
    };
    FindService.prototype.isWholeWord = function (str, word) {
        var index = str.indexOf(word);
        var preSymbol = str.substring(index - 1, index);
        var nextSymbol = str.substring(index + word.length, index + word.length + 1);
        return ((preSymbol === '' || this.symbols.indexOf(preSymbol) === -1) && (nextSymbol === '' || this.symbols.indexOf(nextSymbol) === -1));
    };
    FindService.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService },
        { type: AnimationService }
    ]; };
    FindService = __decorate([
        Injectable()
    ], FindService);
    return FindService;
}());
export { FindService };
