// stimulsoft-viewer-angular/esm5/lib/services/helper.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { ModelService } from './model.service';
import { AnimationService } from './animation.service';
import { DateTimeObject } from './objects';
var HelperService = /** @class */ (function () {
    function HelperService(model, animationService) {
        this.model = model;
        this.animationService = animationService;
    }
    HelperService.prototype.val = function (value, def) {
        return value != null && value.trim() !== '' ? value : (def || '');
    };
    HelperService.prototype.findPosY = function (obj, mainClassName, noScroll) {
        var curtop = noScroll ? 0 : this.getScrollYOffset(obj, mainClassName);
        if (obj.offsetParent) {
            while (obj.className !== mainClassName) {
                curtop += obj.offsetTop;
                if (!obj.offsetParent) {
                    break;
                }
                obj = obj.offsetParent;
            }
        }
        else if (obj.y) {
            curtop += obj.y;
        }
        return curtop;
    };
    HelperService.prototype.getScrollYOffset = function (obj, mainClassName) {
        var scrolltop = 0;
        if (obj.parentElement) {
            while (obj.className !== mainClassName) {
                if ('scrollTop' in obj) {
                    scrolltop -= obj.scrollTop;
                }
                if (!obj.parentElement) {
                    break;
                }
                obj = obj.parentElement;
            }
        }
        return scrolltop;
    };
    HelperService.prototype.findPosX = function (obj, mainClassName, noScroll) {
        var curleft = noScroll ? 0 : this.getScrollXOffset(obj, mainClassName);
        if (obj.offsetParent) {
            while (obj.className !== mainClassName) {
                curleft += obj.offsetLeft;
                if (!obj.offsetParent) {
                    break;
                }
                obj = obj.offsetParent;
            }
        }
        else if (obj.x) {
            curleft += obj.x;
        }
        return curleft;
    };
    HelperService.prototype.getScrollXOffset = function (obj, mainClassName) {
        var scrollleft = 0;
        if (obj.parentElement) {
            while (obj.className !== mainClassName) {
                if ('scrollLeft' in obj) {
                    scrollleft -= obj.scrollLeft;
                }
                if (!obj.parentElement) {
                    break;
                }
                obj = obj.parentElement;
            }
        }
        return scrollleft;
    };
    HelperService.prototype.showHelpWindow = function (url) {
        var helpLanguage = 'en';
        switch (this.model.options.cultureName) {
            case 'ru': helpLanguage = 'ru';
            // case 'de': helpLanguage = 'de';
        }
        this.openNewWindow('https://www.stimulsoft.com/' + helpLanguage + '/documentation/online/' + url);
    };
    HelperService.prototype.openNewWindow = function (url, name, specs) {
        return window.open(url, name, specs);
    };
    HelperService.prototype.scrollToAnchor = function (anchorName, componentGuid) {
        var _this = this;
        var aHyperlinks = this.model.controls.reportPanel.el.nativeElement.getElementsByTagName("a");
        var identicalAnchors = [];
        if (anchorName) {
            anchorName = anchorName.replace(/!!#92/g, '\\');
        }
        if (componentGuid) {
            for (var i_1 = 0; i_1 < aHyperlinks.length; i_1++) {
                if (aHyperlinks[i_1].getAttribute('guid') === componentGuid) {
                    identicalAnchors.push(aHyperlinks[i_1]);
                }
            }
        }
        if (identicalAnchors.length === 0) {
            var guidIndex = anchorName.indexOf("#GUID#");
            if (identicalAnchors.length == 0) {
                var aHyperlinks_1 = this.model.controls.reportPanel.el.nativeElement.getElementsByTagName("a");
                for (var i = 0; i < aHyperlinks_1.length; i++) {
                    if (aHyperlinks_1[i].name && ((guidIndex >= 0 && (aHyperlinks_1[i].name.indexOf(anchorName.substring(guidIndex + 6)) >= 0 || anchorName.substring(0, guidIndex) == aHyperlinks_1[i].name)) || aHyperlinks_1[i].name == anchorName))
                        identicalAnchors.push(aHyperlinks_1[i]);
                }
            }
        }
        if (identicalAnchors.length > 0) {
            var anchor_1 = identicalAnchors[0];
            var anchorParent_1 = anchor_1.parentElement || anchor_1;
            var anchorHeight_1 = anchorParent_1.offsetHeight;
            var anchorOffsetTop_1 = anchorParent_1.offsetTop;
            identicalAnchors.forEach(function (identicalAnchor) {
                var nextAnchorParent = identicalAnchor.parentElement || identicalAnchor;
                if (nextAnchorParent.offsetTop > anchorOffsetTop_1) {
                    anchorHeight_1 = Math.max(anchorHeight_1, nextAnchorParent.offsetTop - anchorOffsetTop_1 + nextAnchorParent.offsetHeight);
                }
            });
            var date = new Date();
            var endTime = date.getTime() + this.model.options.scrollDuration;
            var targetTop = this.findPosY(anchor_1, this.model.options.appearance.scrollbarsMode ? 'stiJsViewerReportPanel' : null, true) - anchorParent_1.offsetHeight * 2;
            this.animationService.showAnimationForScroll(this.model.controls.reportPanel.el.nativeElement, targetTop, endTime, function () {
                var _a;
                var page = _this.getPageFromAnchorElement(anchor_1);
                var anchorParentTopPos = _this.findPosY(anchorParent_1, 'stiJsViewerReportPanel', true);
                var pageTopPos = page ? _this.findPosY(page, 'stiJsViewerReportPanel', true) : anchorParentTopPos;
                _this.removeBookmarksLabel();
                var label = document.createElement('div');
                _this.model.controls.bookmarksLabel = label;
                label.className = 'stiJsViewerBookmarksLabel';
                var labelMargin = 20 * (_this.model.reportParams.zoom / 100);
                var labelWidth = page ? page.offsetWidth - labelMargin - 6 : anchorParent_1.offsetWidth;
                var labelHeight = anchorHeight_1 - 2;
                label.style.width = labelWidth + 'px';
                label.style.height = labelHeight + 'px';
                var margins = (_a = page.pageAttributes.margins) === null || _a === void 0 ? void 0 : _a.replace('px', '').split(' ');
                var pageLeftMargin = page.pageAttributes.margins ? _this.strToInt(margins[3]) : 0;
                var pageTopMargin = page.pageAttributes.margins ? _this.strToInt(margins[0]) : 0;
                label.style.marginLeft = (labelMargin / 2 - pageLeftMargin) + 'px';
                label.style.marginTop = (anchorParentTopPos - pageTopPos - pageTopMargin - (_this.model.reportParams.zoom / 100) - 1) + 'px';
                page.insertBefore(label, page.childNodes[0]);
            });
        }
    };
    HelperService.prototype.removeBookmarksLabel = function () {
        if (this.model.controls.bookmarksLabel) {
            this.model.controls.bookmarksLabel.parentElement.removeChild(this.model.controls.bookmarksLabel);
            this.model.controls.bookmarksLabel = null;
        }
    };
    HelperService.prototype.getPageFromAnchorElement = function (anchorElement) {
        var obj = anchorElement;
        while (obj.parentElement) {
            if (obj.className && obj.className.indexOf('stiJsViewerPage') === 0) {
                return obj;
            }
            obj = obj.parentElement;
        }
        return obj;
    };
    HelperService.prototype.strToInt = function (value) {
        var result = parseInt(value, 10);
        return result || 0;
    };
    HelperService.prototype.strToDouble = function (value) {
        if (value === null) {
            return null;
        }
        var result = parseFloat(value.toString().replace(',', '.').trim());
        return result || 0;
    };
    HelperService.prototype.copyObject = function (o) {
        if (!o || 'object' !== typeof o) {
            return o;
        }
        var c = 'function' === typeof o.pop ? [] : {};
        var p;
        var v;
        for (p in o) {
            if (o.hasOwnProperty(p)) {
                v = o[p];
                if (v && 'object' === typeof v) {
                    c[p] = this.copyObject(v);
                }
                else {
                    c[p] = v;
                }
            }
        }
        return c;
    };
    HelperService.prototype.getCountObjects = function (objectArray) {
        var count = 0;
        if (objectArray) {
            for (var singleObject in objectArray) {
                count++;
            }
        }
        return count;
    };
    HelperService.prototype.replaceMonths = function (value) {
        for (var i = 1; i <= 12; i++) {
            var enName = '';
            var locName = '';
            switch (i) {
                case 1:
                    enName = 'January';
                    locName = this.model.loc('MonthJanuary');
                    break;
                case 2:
                    enName = 'February';
                    locName = this.model.loc('MonthFebruary');
                    break;
                case 3:
                    enName = 'March';
                    locName = this.model.loc('MonthMarch');
                    break;
                case 4:
                    enName = 'April';
                    locName = this.model.loc('MonthApril');
                    break;
                case 5:
                    enName = 'May';
                    locName = this.model.loc('MonthMay');
                    break;
                case 6:
                    enName = 'June';
                    locName = this.model.loc('MonthJune');
                    break;
                case 7:
                    enName = 'July';
                    locName = this.model.loc('MonthJuly');
                    break;
                case 8:
                    enName = 'August';
                    locName = this.model.loc('MonthAugust');
                    break;
                case 9:
                    enName = 'September';
                    locName = this.model.loc('MonthSeptember');
                    break;
                case 10:
                    enName = 'October';
                    locName = this.model.loc('MonthOctober');
                    break;
                case 11:
                    enName = 'November';
                    locName = this.model.loc('MonthNovember');
                    break;
                case 12:
                    enName = 'December';
                    locName = this.model.loc('MonthDecember');
                    break;
            }
            var enShortName = enName.substring(0, 3);
            var locShortName = locName.substring(0, 3);
            value = value.replace(enName, i).replace(enName.toLowerCase(), i).replace(enShortName, i).replace(enShortName.toLowerCase(), i);
            value = value.replace(locName, i).replace(locName.toLowerCase(), i).replace(locShortName, i).replace(locShortName.toLowerCase(), i);
        }
        return value;
    };
    HelperService.prototype.getDate = function (dateTimeObj) {
        return new Date(dateTimeObj.year, dateTimeObj.month - 1, dateTimeObj.day, dateTimeObj.hours, dateTimeObj.minutes, dateTimeObj.seconds);
    };
    HelperService.prototype.getDateTimeObject = function (date) {
        if (!date) {
            date = new Date();
        }
        var dateTimeObject = new DateTimeObject();
        dateTimeObject.year = date.getFullYear();
        dateTimeObject.month = date.getMonth() + 1;
        dateTimeObject.day = date.getDate();
        dateTimeObject.hours = date.getHours();
        dateTimeObject.minutes = date.getMinutes();
        dateTimeObject.seconds = date.getSeconds();
        return dateTimeObject;
    };
    HelperService.prototype.getDateTimeFromString = function (value, format) {
        var charIsDigit = function (ch) {
            return ('0123456789'.indexOf(ch) >= 0);
        };
        if (!value) {
            return new Date();
        }
        value = this.replaceMonths(value);
        var dateTime = new Date();
        // If the date format is not specified, then deserializator for getting date and time is applied
        if (format === null) {
            format = 'dd.MM.yyyy hh:mm:ss';
        }
        // Otherwise the format is parsed. Now only numeric date and time formats are supported
        var year = 1970;
        var month = 1;
        var day = 1;
        var hour = 0;
        var minute = 0;
        var second = 0;
        var millisecond = 0;
        var char = '';
        var pos = 0;
        var values = [];
        // Parse date and time into separate numeric values
        while (pos < value.length) {
            char = value.charAt(pos);
            if (charIsDigit(char)) {
                values.push(char);
                pos++;
                while (pos < value.length && charIsDigit(value.charAt(pos))) {
                    values[values.length - 1] += value.charAt(pos);
                    pos++;
                }
                values[values.length - 1] = this.strToInt(values[values.length - 1]);
            }
            pos++;
        }
        pos = 0;
        var charCount = 0;
        var index = -1;
        var is12hour = false;
        // Parsing format and replacement of appropriate values of date and time
        while (pos < format.length) {
            char = format.charAt(pos);
            charCount = 0;
            if (char === 'Y' || char === 'y' || char === 'M' || char === 'd' || char === 'h' || char === 'H' ||
                char === 'm' || char === 's' || char === 'f' || char === 'F' || char === 't' || char === 'z') {
                index++;
                while (pos < format.length && format.charAt(pos) === char) {
                    pos++;
                    charCount++;
                }
            }
            switch (char) {
                case 'Y': // full year
                    year = values[index];
                    break;
                case 'y': // year
                    if (values[index] < 1000) {
                        year = 2000 + values[index];
                    }
                    else {
                        year = values[index];
                    }
                    break;
                case 'M': // month
                    month = values[index];
                    break;
                case 'd': // day
                    day = values[index];
                    break;
                case 'h': // (hour 12)
                    is12hour = true;
                    hour = values[index];
                    break;
                case 'H': // (hour 24)
                    hour = values[index];
                    break;
                case 'm': // minute
                    minute = values[index];
                    break;
                case 's': // second
                    second = values[index];
                    break;
                case 'f': // second fraction
                case 'F': // second fraction, trailing zeroes are trimmed
                    millisecond = values[index];
                    break;
                case 't': // PM or AM
                    if (value.toLowerCase().indexOf('am') >= 0 && hour === 12) {
                        hour = 0;
                    }
                    if (value.toLowerCase().indexOf('pm') >= 0 && hour < 12) {
                        hour += 12;
                    }
                    break;
                default:
                    pos++;
                    break;
            }
        }
        dateTime = new Date(year || new Date().getFullYear(), month - 1 || 0, day || 1, hour || 0, minute || 0, second, millisecond || 0);
        if (!dateTime || isNaN(dateTime)) {
            return new Date();
        }
        return dateTime;
    };
    HelperService.prototype.getStringDateTime = function (object, dateTimeType) {
        if (dateTimeType === 'Date') {
            object.hours = 0;
            object.minutes = 0;
            object.seconds = 0;
        }
        return object.month + '/' + object.day + '/' + object.year + ' ' +
            (object.hours > 12 ? object.hours - 12 : object.hours) + ':' + object.minutes + ':' + object.seconds + ' ' +
            (object.hours < 12 ? 'AM' : 'PM');
    };
    HelperService.prototype.dateTimeObjectToString = function (dateTimeObject, typeDateTimeObject) {
        var date = new Date(dateTimeObject.year, dateTimeObject.month - 1, dateTimeObject.day, dateTimeObject.hours, dateTimeObject.minutes, dateTimeObject.seconds);
        if (this.model.options.appearance.parametersPanelDateFormat !== '') {
            return this.formatDate(date, this.model.options.appearance.parametersPanelDateFormat);
        }
        return this.dateToLocaleString(date, typeDateTimeObject);
    };
    HelperService.prototype.dateToLocaleString = function (date, dateTimeType) {
        var timeString = date.toLocaleTimeString();
        var isAmericanFormat = timeString.toLowerCase().indexOf('am') >= 0 || timeString.toLowerCase().indexOf('pm') >= 0;
        var formatDate = isAmericanFormat ? 'MM/dd/yyyy' : 'dd.MM.yyyy';
        var yyyy = date.getFullYear();
        var yy = yyyy.toString().substring(2);
        var M = date.getMonth() + 1;
        var MM = M < 10 ? '0' + M : M;
        var d = date.getDate();
        var dd = d < 10 ? '0' + d : d;
        formatDate = formatDate.replace(/yyyy/i, yyyy);
        formatDate = formatDate.replace(/yy/i, yy);
        formatDate = formatDate.replace(/MM/i, MM);
        formatDate = formatDate.replace(/M/i, M);
        formatDate = formatDate.replace(/dd/i, dd);
        formatDate = formatDate.replace(/d/i, d);
        var h = date.getHours();
        var tt = '';
        if (isAmericanFormat) {
            tt = h < 12 ? ' AM' : ' PM';
            h = h > 12 ? h - 12 : h;
            if (h === 0) {
                h = 12;
            }
        }
        else {
            h = h < 10 ? '0' + h : h;
        }
        var m = date.getMinutes();
        m = m < 10 ? '0' + m : m;
        var s = date.getSeconds();
        s = s < 10 ? '0' + s : s;
        var formatTime = h + ':' + m + ':' + s + tt;
        if (dateTimeType === 'Time') {
            return formatTime;
        }
        if (dateTimeType === 'Date') {
            return formatDate;
        }
        return formatDate + ' ' + formatTime;
    };
    HelperService.prototype.formatDate = function (formatDate, formatString) {
        var yyyy = formatDate.getFullYear();
        var yy = yyyy.toString().substring(2);
        var m = formatDate.getMonth() + 1;
        var mm = m < 10 ? '0' + m : m;
        var d = formatDate.getDate();
        var dd = d < 10 ? '0' + d : d;
        var h = formatDate.getHours();
        var hh = h < 10 ? '0' + h : h;
        var h12 = h > 12 ? h - 12 : (h > 0 ? h : 12);
        var hh12 = h12 < 10 ? '0' + h12 : h12;
        var n = formatDate.getMinutes();
        var nn = n < 10 ? '0' + n : n;
        var s = formatDate.getSeconds();
        var ss = s < 10 ? '0' + s : s;
        var tt = h < 12 ? 'AM' : 'PM';
        formatString = formatString.replace(/yyyy/gi, yyyy);
        formatString = formatString.replace(/yy/gi, yy);
        formatString = formatString.replace(/Y/, yyyy);
        formatString = formatString.replace(/MM/g, mm);
        formatString = formatString.replace(/M/g, m);
        formatString = formatString.replace(/dd/g, dd);
        formatString = formatString.replace(/d/g, d);
        formatString = formatString.replace(/hh/g, hh12);
        formatString = formatString.replace(/h/g, h12);
        formatString = formatString.replace(/HH/g, hh);
        formatString = formatString.replace(/H/g, h);
        formatString = formatString.replace(/mm/g, nn);
        formatString = formatString.replace(/m/g, n);
        formatString = formatString.replace(/ss/g, ss);
        formatString = formatString.replace(/s/g, s);
        formatString = formatString.replace(/tt/gi, tt);
        formatString = formatString.replace(/t/gi, tt.substr(0, 1));
        return formatString;
    };
    HelperService.prototype.getStringKey = function (key, params) {
        if (key === null) {
            return '';
        }
        return ((params.type === 'DateTime') ? this.dateTimeObjectToString(key, params.dateTimeType) : key);
    };
    HelperService.prototype.stringToTime = function (timeStr) {
        var timeArray = timeStr.split(':');
        var time = { hours: 0, minutes: 0, seconds: 0 };
        time.hours = this.strToInt(timeArray[0]);
        if (timeArray.length > 1) {
            time.minutes = this.strToInt(timeArray[1]);
        }
        if (timeArray.length > 2) {
            time.seconds = this.strToInt(timeArray[2]);
        }
        if (time.hours < 0) {
            time.hours = 0;
        }
        if (time.minutes < 0) {
            time.minutes = 0;
        }
        if (time.seconds < 0) {
            time.seconds = 0;
        }
        if (time.hours > 23) {
            time.hours = 23;
        }
        if (time.minutes > 59) {
            time.minutes = 59;
        }
        if (time.seconds > 59) {
            time.seconds = 59;
        }
        return time;
    };
    HelperService.prototype.newGuid = function () {
        var chars = '0123456789abcdefghijklmnopqrstuvwxyz'.split('');
        var uuid = [];
        var rnd = Math.random;
        var r;
        uuid[8] = uuid[13] = uuid[18] = uuid[23] = '-';
        uuid[14] = '4';
        for (var i = 0; i < 36; i++) {
            if (!uuid[i]) {
                r = 0 | rnd() * 16;
                uuid[i] = chars[(i === 19) ? (r & 0x3) | 0x8 : r & 0xf];
            }
        }
        return uuid.join('');
    };
    HelperService.prototype.getCountDaysOfMonth = function (year, month) {
        var countDaysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
        var count = countDaysInMonth[month];
        if (month === 1) {
            if (year % 4 === 0 && (year % 100 !== 0 || year % 400 === 0)) {
                count = 29;
            }
            else {
                count = 28;
            }
        }
        return count;
    };
    HelperService.prototype.setObjectToCenter = function (object, defaultTop) {
        var leftPos = (this.model.viewerSize.width / 2 - object.offsetWidth / 2);
        var topPos = this.model.options.appearance.fullScreenMode ? (this.model.viewerSize.height / 2 - object.offsetHeight / 2) : (defaultTop ? defaultTop : 250);
        return { left: Math.max(leftPos, 0), top: Math.max(topPos, 0) };
    };
    HelperService.prototype.checkCloudAuthorization = function (action) {
        return true;
    };
    HelperService.prototype.getCookie_ = function (name) {
        var cookie = ' ' + document.cookie;
        var search = ' ' + name + '=';
        var setStr = null;
        var offset = 0;
        var end = 0;
        if (cookie.length > 0) {
            offset = cookie.indexOf(search);
            if (offset !== -1) {
                offset += search.length;
                end = cookie.indexOf(';', offset);
                if (end === -1) {
                    end = cookie.length;
                }
                setStr = unescape(cookie.substring(offset, end));
            }
        }
        return setStr;
    };
    HelperService.prototype.getCookie = function (name) {
        if (this.model.options.standaloneJsMode || typeof localStorage === 'undefined' || name.indexOf('sti_') == 0 || name.indexOf('login') == 0) {
            return this.getCookie_(name);
        }
        var value = localStorage.getItem(name);
        if (value != null) {
            return value;
        }
        else {
            value = this.getCookie_(name);
            if (value != null) {
                this.removeCookie(name);
                localStorage.setItem(name, value);
            }
            return value;
        }
    };
    HelperService.prototype.removeCookie = function (name) {
        document.cookie = name + '=;expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/';
    };
    HelperService.prototype.setCookie = function (name, value, path, domain, secure, expires) {
        if (this.model.options.standaloneJsMode || typeof localStorage === 'undefined' || name.indexOf('sti_') === 0 || name.indexOf('login') === 0) {
            // save to cookies
            if (value && typeof (value) === 'string' && value.length >= 4096) {
                return;
            }
            var pathName = location.pathname;
            var expDate = new Date();
            expDate.setTime(expDate.getTime() + (365 * 24 * 3600 * 1000));
            document.cookie = name + '=' + escape(value) +
                '; expires=' + (expires || expDate['toGMTString']()) +
                ((path) ? '; path=' + path : '; path=/') +
                ((domain) ? '; domain=' + domain : '') +
                ((secure) ? '; secure' : '');
        }
        else {
            // save to localstorage
            localStorage.setItem(name, value);
        }
    };
    HelperService.prototype.getNavigatorName = function () {
        if (!navigator) {
            return 'Unknown';
        }
        var userAgent = navigator.userAgent;
        if (userAgent.indexOf('Edge') >= 0) {
            return 'Edge';
        }
        if (userAgent.indexOf('MSIE') >= 0 || userAgent.indexOf('Trident') >= 0) {
            return 'MSIE';
        }
        if (userAgent.indexOf('Gecko') >= 0) {
            if (userAgent.indexOf('Chrome') >= 0) {
                return 'Chrome';
            }
            if (userAgent.indexOf('Safari') >= 0) {
                return 'Safari';
            }
            return 'Mozilla';
        }
        if (userAgent.indexOf('Opera') >= 0) {
            return 'Opera';
        }
    };
    HelperService.prototype.isTouchDevice = function () {
        return ('ontouchstart' in document.documentElement);
    };
    HelperService.prototype.isMobileDevice = function () {
        return /iPhone|iPad|iPod|Macintosh|Android|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
    };
    HelperService.prototype.getHumanFileSize = function (value, decimals) {
        var i = Math.floor(Math.log(value) / Math.log(1024));
        return ((value / Math.pow(1024, i)).toFixed(decimals)) + ' ' + ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'][i];
    };
    HelperService.prototype.getZoomMenuItems = function () {
        var items = [{ name: 'Zoom25', caption: '25%', img: '' },
            { name: 'Zoom50', caption: '50%', img: '' },
            { name: 'Zoom75', caption: '75%', img: '' },
            { name: 'Zoom100', caption: '100%', img: '' },
            { name: 'Zoom150', caption: '150%', img: '' },
            { name: 'Zoom200', caption: '200%', img: '' }];
        if (this.model.options.toolbar.displayMode !== 'Separated') {
            items.push({ type: 'separator' });
            items.push({ name: 'ZoomOnePage', caption: this.model.loc('ZoomOnePage'), img: 'ZoomOnePage.png' });
            items.push({ name: 'ZoomPageWidth', caption: this.model.loc('ZoomPageWidth'), img: 'ZoomPageWidth.png' });
        }
        return items;
    };
    Object.defineProperty(HelperService.prototype, "helpLinks", {
        get: function () {
            return {
                Print: 'user-manual/index.html?viewer_reports.htm#toolbar',
                Save: 'user-manual/index.html?viewer_reports.htm#toolbar',
                SendEmail: 'user-manual/index.html?viewer_reports.htm#toolbar',
                Bookmarks: 'user-manual/index.html?viewer_reports.htm#toolbar',
                Parameters: 'user-manual/index.html?viewer_reports.htm#toolbar',
                FirstPage: 'user-manual/index.html?viewer_reports.htm#statusbar',
                PrevPage: 'user-manual/index.html?viewer_reports.htm#statusbar',
                NextPage: 'user-manual/index.html?viewer_reports.htm#statusbar',
                LastPage: 'user-manual/index.html?viewer_reports.htm#statusbar',
                FullScreen: 'user-manual/index.html?viewer_reports.htm#toolbar',
                Zoom: 'user-manual/index.html?viewer_reports.htm#statusbar',
                ViewMode: 'user-manual/index.html?viewer_reports.htm#displayingmode',
                Editor: 'user-manual/index.html?viewer_reports.htm#toolbar',
                Find: 'user-manual/index.html?viewer_reports.htm#searchpanel',
                DashboardToolbar: 'user-manual/index.html?viewer_dashboards.htm#controlbuttonsofthedashboard',
                DashboardElementToolbar: 'user-manual/index.html?viewer_dashboards.htm#elementcontrols',
                DashboardExport: 'user-manual/index.html?exports_dashboards.htm',
                DashboardPdfExport: 'user-manual/index.html?exports_dashboards.htm#pdfexportsettings',
                DashboardExcelExport: 'user-manual/index.html?exports_dashboards.htm#excelexportsettings',
                DashboardImageExport: 'user-manual/index.html?exports_dashboards.htm#imageexportsettings',
                DashboardDataExport: 'user-manual/index.html?exports_dashboards.htm#exportsettingsofdata'
            };
        },
        enumerable: true,
        configurable: true
    });
    HelperService.prototype.getBackText = function (withoutBrackets) {
        var backText = String.fromCharCode(84) + 'r' + String.fromCharCode(105) + 'a';
        if (withoutBrackets) {
            return backText + String.fromCharCode(108);
        }
        return String.fromCharCode(91) + backText + String.fromCharCode(108) + String.fromCharCode(93);
    };
    HelperService.ctorParameters = function () { return [
        { type: ModelService },
        { type: AnimationService }
    ]; };
    HelperService = __decorate([
        Injectable()
    ], HelperService);
    return HelperService;
}());
export { HelperService };
