// stimulsoft-viewer-angular/esm5/lib/services/http-client.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { throwError } from 'rxjs';
import { catchError, timeout } from 'rxjs/operators';
import { ModelService } from './model.service';
import { HelperService } from './helper.service';
var StiHttpClientService = /** @class */ (function () {
    function StiHttpClientService(httpClient, model, helper) {
        this.httpClient = httpClient;
        this.model = model;
        this.helper = helper;
    }
    StiHttpClientService.prototype.post = function (url, data, responseType) {
        var _this = this;
        if (responseType === void 0) { responseType = 'json'; }
        var _a, _b, _c, _d;
        var model = this.model;
        model.showProgress = true;
        var reqTimeout = ((_b = (_a = this.model.options) === null || _a === void 0 ? void 0 : _a.server) === null || _b === void 0 ? void 0 : _b.requestTimeout) > 0 ? ((_d = (_c = this.model.options) === null || _c === void 0 ? void 0 : _c.server) === null || _d === void 0 ? void 0 : _d.requestTimeout) * 1000 : 1000 * 1000;
        if (responseType === 'json') {
            return this.httpClient.post(url, this.getFormData(data)).pipe(
            // retry(3),
            timeout(reqTimeout), catchError(function (error) {
                model.httpError = error;
                model.showProgress = false;
                _this.model.controls.navigatePanel.enabled = true;
                _this.model.controls.toolbar.enabled = true;
                return throwError('Something bad happened; please try again later.');
            }));
        }
        else {
            return this.httpClient.post(url, this.getFormData(data), { responseType: 'text' }).pipe(
            // retry(3),
            timeout(reqTimeout), catchError(function (error) {
                model.httpError = error;
                model.showProgress = false;
                _this.model.controls.navigatePanel.enabled = true;
                _this.model.controls.toolbar.enabled = true;
                return throwError('Something bad happened; please try again later.');
            }));
        }
    };
    StiHttpClientService.prototype.getFormData = function (data) {
        var formData = new FormData();
        Object.keys(data).forEach(function (key) { return formData.append(key, data[key]); });
        return formData;
    };
    StiHttpClientService.prototype.postForm = function (url, data, doc, postOnlyData) {
        var _this = this;
        if (postOnlyData === void 0) { postOnlyData = false; }
        if (!doc) {
            doc = document;
        }
        var form = doc.createElement('FORM');
        form.setAttribute('method', 'POST');
        form.setAttribute('action', url);
        var params = postOnlyData ? data : this.model.createPostParameters(data, true);
        Object.keys(params).forEach(function (key) {
            var paramsField = doc.createElement('INPUT');
            paramsField.setAttribute('type', 'hidden');
            paramsField.setAttribute('name', key);
            paramsField.setAttribute('value', params[key]);
            form.appendChild(paramsField);
        });
        if (this.model.options.jsDesigner) {
            this.model.options.jsDesigner.options.ignoreBeforeUnload = true;
        }
        doc.body.appendChild(form);
        form.submit();
        doc.body.removeChild(form);
        setTimeout(function () {
            if (_this.model.options.jsDesigner) {
                _this.model.options.jsDesigner.options.ignoreBeforeUnload = false;
            }
        }, 500);
    };
    StiHttpClientService.ctorParameters = function () { return [
        { type: HttpClient },
        { type: ModelService },
        { type: HelperService }
    ]; };
    StiHttpClientService = __decorate([
        Injectable()
    ], StiHttpClientService);
    return StiHttpClientService;
}());
export { StiHttpClientService };
