// stimulsoft-viewer-angular/esm5/lib/services/interactions.service.js
import { __decorate, __values } from "tslib";
import { Injectable } from '@angular/core';
import { ModelService } from './model.service';
import { Md5Service } from './md5.service';
import { HelperService } from './helper.service';
import { ControllerService } from './controller.service';
import { DrillDownService } from './drill-down.service';
var InteractionsService = /** @class */ (function () {
    function InteractionsService(model, md5, controller, helper, drillDownService) {
        var _this = this;
        this.model = model;
        this.md5 = md5;
        this.controller = controller;
        this.helper = helper;
        this.drillDownService = drillDownService;
        controller.getMessage().subscribe(function (message) {
            if (message.action === 'InitVars') {
                _this.showParametersPanel(message.data);
            }
        });
        controller.getActionMessage().subscribe(function (message) {
            switch (message.action) {
                case 'Variables':
                    _this.postInteraction({ action: 'Variables', variables: _this.getParametersValues() });
                    break;
                case 'Reset':
                    _this.showParametersPanel(_this.model.options.paramsVariables);
                    break;
                default:
                    break;
            }
        });
    }
    InteractionsService.prototype.postInteraction = function (params) {
        this.drillDownService.saveState();
        // Add new drill-down parameters to drill-down queue and calc guid
        if (params.action !== 'InitVars' && (params.action === 'DrillDown' || params.action === 'DashboardDrillDown')) {
            if (this.model.options.drillDownInProgress) {
                return;
            }
            if (params.action === 'DashboardDrillDown' && params.drillDownParameters) {
                params.drillDownParameters.isDashboardDrillDown = true; // add dbs flag
            }
            var drillDownParameters = this.model.reportParams.drillDownParameters || [];
            params.drillDownParameters = params.drillDownParameters ? drillDownParameters.concat(params.drillDownParameters) : drillDownParameters;
            if (params.action === 'DrillDown') {
                params.drillDownGuid = this.md5.hex_md5(JSON.stringify(this.sortPropsInDrillDownParameters(params.drillDownParameters)));
            }
            else {
                params.dashboardDrillDownGuid = this.md5.hex_md5(JSON.stringify(this.sortPropsInDrillDownParameters(params.drillDownParameters)));
            }
            this.model.options.drillDownInProgress = true;
        }
        this.controller.post(params.action, this.model.options.actions.interaction, params);
    };
    InteractionsService.prototype.showParametersPanel = function (data) {
        if (!data) {
            return;
        }
        if (this.controller.showError(data)) {
            data = null;
        }
        this.model.options.isParametersReceived = true;
        this.model.options.paramsVariables = typeof data === 'string' ? JSON.parse(data) : data;
        this.model.controls.parametersPanel.visible = !this.model.options.isMobileDevice;
        // if (this.model.reportParams.type === 'Dashboard') jsObject.postAction('GetPages');
        var paramsVariables = this.helper.copyObject(this.model.options.paramsVariables);
        var countParameters = this.helper.getCountObjects(paramsVariables);
        var countColumns = (countParameters <= this.model.options.minParametersCountForMultiColumns)
            ? 1 : this.model.options.appearance.parametersPanelColumnsCount;
        var countInColumn = Math.trunc(countParameters / countColumns);
        if (countInColumn * countColumns < countParameters) {
            countInColumn++;
        }
        // setup lists
        for (var i = 0; i < countParameters; i++) {
            var variable = paramsVariables[i];
            if (variable.items) {
                variable.items.forEach(function (v) {
                    v.isChecked = true;
                    v.visible = true;
                });
            }
            variable.isNull = (variable.type === 'DateTime' && variable.value === null && variable.basicType === 'NullableValue');
        }
        this.model.interactions = { paramsVariables: paramsVariables, countColumns: new Array(countColumns), countInColumn: new Array(countInColumn) };
    };
    InteractionsService.prototype.sortPropsInDrillDownParameters = function (inArray) {
        if (!inArray || !inArray.length) {
            return inArray;
        }
        var outArray = [];
        var _loop_1 = function (i) {
            var propNames = [];
            var el = inArray[i];
            var copyEl = {};
            Object.keys(el).forEach(function (p) {
                if (el.hasOwnProperty(p)) {
                    propNames.push(p);
                }
            });
            propNames.sort();
            propNames.forEach(function (propName) { return copyEl[propName] = el[propName]; });
            outArray[i] = copyEl;
        };
        for (var i = 0; i < inArray.length; i++) {
            _loop_1(i);
        }
        return outArray;
    };
    InteractionsService.prototype.getParametersValues = function () {
        var _this = this;
        var parametersValues = {};
        Object.keys(this.model.interactions.paramsVariables).forEach(function (i) {
            var parameter = _this.model.interactions.paramsVariables[i];
            parametersValues[parameter.name] = _this.getVariableValue(parameter);
        });
        return parametersValues;
    };
    InteractionsService.prototype.getVariableValue = function (parameter) {
        var _this = this;
        if (parameter.isNull) {
            return null;
        }
        switch (parameter.basicType) {
            case 'Value':
            case 'NullableValue':
                return this.val(parameter.key, parameter, parameter.allowUserValues ? parameter.value : parameter.key);
            case 'Range':
                return { from: this.val(parameter.key, parameter), to: this.val(parameter.keyTo, parameter) };
            case 'List':
                var value_1 = [];
                if (parameter.items) {
                    parameter.items.filter(function (i) { return parameter.allowUserValues || i.isChecked; }).forEach(function (item) { return value_1.push(_this.val(item.key, parameter)); });
                }
                return value_1;
        }
    };
    InteractionsService.prototype.val = function (key, parameter, value) {
        return parameter.type === 'DateTime' ? this.helper.getStringDateTime(key, parameter.dateTimeType) : (value === undefined ? key : value);
    };
    InteractionsService.prototype.initializeInteractions = function (page) {
        var e_1, _a;
        var _this = this;
        if (!page) {
            return;
        }
        var elems = page.querySelectorAll ? page.querySelectorAll('td,div,span,rect,path,ellipse') : page.getElementsByTagName('td');
        var collapsedHash = [];
        var _loop_2 = function (elem) {
            if (elem.getAttribute('interaction') && (elem.getAttribute('pageguid') ||
                elem.getAttribute('reportfile') ||
                elem.getAttribute('collapsed') ||
                elem.getAttribute('databandsort'))) {
                elem.style.cursor = 'pointer';
                var sort = elem.getAttribute('sort');
                if (sort) {
                    this_1.paintSortingArrow(elem, sort);
                }
                var collapsed = elem.getAttribute('collapsed');
                if (collapsed) {
                    var compId = elem.getAttribute('compindex') + '|' + elem.getAttribute('interaction');
                    if (collapsedHash.indexOf(compId) < 0) {
                        this_1.paintCollapsingIcon(elem, collapsed === 'true');
                        collapsedHash.push(compId);
                    }
                }
                elem.onclick = function (e) {
                    if (elem.getAttribute('pageguid') || elem.getAttribute('reportfile')) {
                        _this.postInteractionDrillDown(elem);
                    }
                    else if (elem.getAttribute('collapsed')) {
                        _this.postInteractionCollapsing(elem);
                    }
                    else {
                        _this.postInteractionSorting(elem, e.ctrlKey);
                    }
                };
                if (elem.getAttribute('pageguid') || elem.getAttribute('reportfile')) {
                    elem.onmouseover = function () { return elem.style.opacity = 0.75; };
                    elem.onmouseout = function () { return elem.style.opacity = 1; };
                }
            }
        };
        var this_1 = this;
        try {
            for (var elems_1 = __values(elems), elems_1_1 = elems_1.next(); !elems_1_1.done; elems_1_1 = elems_1.next()) {
                var elem = elems_1_1.value;
                _loop_2(elem);
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (elems_1_1 && !elems_1_1.done && (_a = elems_1.return)) _a.call(elems_1);
            }
            finally { if (e_1) throw e_1.error; }
        }
    };
    InteractionsService.prototype.getComponentOffset = function (component) {
        var offsetX = 0;
        var offsetY = 0;
        var startComponent = component;
        while (component && !isNaN(component.offsetLeft) && !isNaN(component.offsetTop)
            && (component === startComponent || component.style.position === '' || component.style.position === 'static')) {
            offsetX += component.offsetLeft - component.scrollLeft;
            offsetY += component.offsetTop - component.scrollTop;
            component = component.offsetParent;
        }
        return { top: offsetY, left: offsetX };
    };
    InteractionsService.prototype.paintSortingArrow = function (component, sort) {
        if (component.arrowImg) {
            return;
        }
        var arrowImg = document.createElement("div");
        var sortUpSrc = "<svg xmlns='http://www.w3.org/2000/svg' width='12' height='12'><path d='M1 9l5-4 5 4z' fill='#eeeeee' stroke='#666'/></svg>";
        var sortDownSrc = "<svg xmlns='http://www.w3.org/2000/svg' width='12' height='12'><path d='M1 5l5 4 5-4z' fill='#eeeeee' stroke='#666'/></svg>";
        arrowImg.innerHTML = sort == "asc" ? sortDownSrc : sortUpSrc;
        var arrowWidth = (this.model.reportParams.zoom / 100) * 16;
        var arrowHeight = (this.model.reportParams.zoom / 100) * 16;
        arrowImg.style.position = 'absolute';
        arrowImg.style.width = arrowWidth + 'px';
        arrowImg.style.height = arrowHeight + 'px';
        component.appendChild(arrowImg);
        component.arrowImg = arrowImg;
        var oldPosition = component.style.position;
        var oldClassName = component.className;
        component.style.position = 'relative';
        if (!oldClassName) {
            component.className = 'stiSortingParentElement';
        }
        var arrowLeftPos = this.helper.findPosX(arrowImg, component.className);
        var arrowTopPos = this.helper.findPosY(arrowImg, component.className);
        arrowImg.style.marginLeft = (component.offsetWidth - arrowLeftPos - arrowWidth - ((this.model.reportParams.zoom / 100) * 3)) + 'px';
        arrowImg.style.marginTop = (component.offsetHeight / 2 - arrowHeight / 2 - arrowTopPos) + 'px';
        if (oldPosition)
            component.style.position = oldPosition;
        component.className = oldClassName;
    };
    InteractionsService.prototype.paintCollapsingIcon = function (component, collapsed) {
        if (component.collapsImg) {
            return;
        }
        var collapsImg = document.createElement('img');
        collapsImg.src = collapsed ? this.model.img('CollapsingPlus.png') : this.model.img('CollapsingMinus.png');
        collapsImg.style.position = 'absolute';
        var collapsWidth = (this.model.reportParams.zoom / 100) * 10;
        var collapsHeight = (this.model.reportParams.zoom / 100) * 10;
        collapsImg.style.width = collapsWidth + 'px';
        collapsImg.style.height = collapsHeight + 'px';
        component.appendChild(collapsImg);
        component.collapsImg = collapsImg;
        var componentOffset = this.getComponentOffset(component);
        var collapsOffset = this.getComponentOffset(collapsImg);
        collapsImg.style.marginLeft = (componentOffset.left - collapsOffset.left + collapsWidth / 3) + 'px';
        collapsImg.style.marginTop = (componentOffset.top - collapsOffset.top + collapsWidth / 3) + 'px';
    };
    InteractionsService.prototype.postInteractionSorting = function (component, isCtrl) {
        var params = {
            action: 'Sorting',
            sortingParameters: {
                ComponentName: component.getAttribute('interaction') + ';' + isCtrl.toString(),
                DataBand: component.getAttribute('databandsort')
            }
        };
        if (this.model.options.isParametersReceived) {
            params.variables = this.getParametersValues();
        }
        this.postInteraction(params);
    };
    InteractionsService.prototype.postInteractionDrillDown = function (component) {
        var params = {
            action: 'DrillDown',
            drillDownParameters: {
                ComponentIndex: component.getAttribute('compindex'),
                ElementIndex: component.getAttribute('elementindex'),
                PageIndex: component.getAttribute('pageindex'),
                PageGuid: component.getAttribute('pageguid'),
                ReportFile: component.getAttribute('reportfile')
            }
        };
        this.postInteraction(params);
    };
    InteractionsService.prototype.postInteractionCollapsing = function (component) {
        var componentName = component.getAttribute('interaction');
        var collapsingIndex = component.getAttribute('compindex');
        var collapsed = component.getAttribute('collapsed') === 'true' ? false : true;
        if (!this.model.reportParams.collapsingStates) {
            this.model.reportParams.collapsingStates = {};
        }
        if (!this.model.reportParams.collapsingStates[componentName]) {
            this.model.reportParams.collapsingStates[componentName] = {};
        }
        this.model.reportParams.collapsingStates[componentName][collapsingIndex] = collapsed;
        var params = {
            action: 'Collapsing',
            collapsingParameters: {
                ComponentName: componentName,
                CollapsingStates: this.model.reportParams.collapsingStates
            }
        };
        if (this.model.options.isParametersReceived) {
            params.variables = this.getParametersValues();
        }
        this.postInteraction(params);
    };
    InteractionsService.prototype.updateAllHyperLinks = function () {
        var _this = this;
        var pointers = this.model.reportParams.tableOfContentsPointers;
        //var bookmarksPanel = this.controls.bookmarksPanel;
        if (this.model.reportParams.bookmarksContent != null || (pointers && pointers.length > 0)) {
            this.model.pages.forEach(function (p) {
                var e_2, _a;
                var page = p.page;
                var aHyperlinks = page.getElementsByTagName('a');
                var _loop_3 = function (aHyperlink) {
                    aHyperlink.hrefContent = aHyperlink.getAttribute('href');
                    if (aHyperlink.hrefContent) {
                        if (aHyperlink.hrefContent.indexOf("#") == 0) {
                            var anchorParams = aHyperlink.hrefContent.substring(1).split("#GUID#");
                            aHyperlink.anchorName = anchorParams[0];
                            aHyperlink.componentGuid = anchorParams.length > 1 ? anchorParams[1] : "";
                            aHyperlink.onclick = function () {
                                var currAnchorName = aHyperlink.anchorName;
                                var cuurCompGuid = aHyperlink.componentGuid;
                                try {
                                    currAnchorName = decodeURI(aHyperlink.anchorName);
                                }
                                catch (e) {
                                    currAnchorName = aHyperlink.anchorName;
                                }
                                if (pointers.length > 0) {
                                    var pageIndex = 1;
                                    for (var i = 0; i < pointers.length; i++) {
                                        if (cuurCompGuid) {
                                            if (pointers[i].componentGuid == cuurCompGuid) {
                                                pageIndex = pointers[i].pageIndex;
                                                break;
                                            }
                                        }
                                        else if (currAnchorName) {
                                            var pointerAnchor = pointers[i].anchor.indexOf("#") == 0 ? pointers[i].anchor.substring(1) : pointers[i].anchor;
                                            if (pointerAnchor == currAnchorName) {
                                                pageIndex = pointers[i].pageIndex;
                                                break;
                                            }
                                        }
                                    }
                                    var anchorName = "";
                                    if (currAnchorName)
                                        anchorName += currAnchorName;
                                    if (cuurCompGuid)
                                        anchorName += ("#GUID#" + cuurCompGuid);
                                    if (anchorName) {
                                        _this.controller.action({ name: 'BookmarkAction', bookmarkPage: Math.max(pageIndex - 1, 0), bookmarkAnchor: anchorName, componentGuid: aHyperlink.compunentGuid });
                                        return false;
                                    }
                                }
                                if (_this.model.reportParams.bookmarksContent != null) {
                                    var node = _this.model.nodes.find(function (n) { return n.name === currAnchorName; });
                                    if (node) {
                                        _this.model.navigateNode = node;
                                    }
                                    else {
                                        for (var k = 0; k < document.anchors.length; k++) {
                                            if (document.anchors[k].name == currAnchorName) {
                                                //jsObject.scrollToAnchor(currAnchorName);
                                                _this.helper.scrollToAnchor(currAnchorName);
                                                return;
                                            }
                                        }
                                    }
                                    _this.controller.post('BookmarkAction', null, 0);
                                    /*let aBookmarks = bookmarksPanel.getElementsByTagName("a");
                                    for (let k = 0; k < aBookmarks.length; k++) {
                                      let clickFunc = aBookmarks[k].getAttribute("onclick");
                                      if (clickFunc && clickFunc.indexOf("'" + currAnchorName + "'") >= 0) {
                                        try {
                                          eval(clickFunc);
                                          return false;
                                        }
                                        catch (e) { }
                                      }
                                    }
                  
                                    this.controller.post('BookmarkAction', null, 0);*/
                                    return false;
                                }
                                //=================================
                                /*let pageIndex = 1;
                                for (let i = 0; i < pointers.length; i++) {
                                  if (aHyperlink.componentGuid) {
                                    if (pointers[i].componentGuid == aHyperlink.componentGuid) {
                                      pageIndex = pointers[i].pageIndex;
                                      break;
                                    }
                                  }
                                  else if (aHyperlink.bookmarkAnchor) {
                                    if (pointers[i].anchor == decodeURI(aHyperlink.bookmarkAnchor)) {
                                      pageIndex = pointers[i].pageIndex;
                                      break;
                                    }
                                  }
                                }
                                this.controller.action({ name: 'BookmarkAction', bookmarkPage: pageIndex - 1, bookmarkAnchor: decodeURI(aHyperlink.bookmarkAnchor), componentGuid: aHyperlink.compunentGuid });
                                return false;
                                };
                                } else if (this.model.reportParams.bookmarksContent != null) {
                                aHyperlink.anchorName = aHyperlink.hrefContent.replace('#', '');
                
                                aHyperlink.onclick = () => {
                                const node = this.model.nodes.find(n => n.url.replace(/\\\'/g, '\'').substr(1) === aHyperlink.anchorName);
                                if (node) {
                                  this.model.navigateNode = node;
                                } else if (aHyperlink.hrefContent.indexOf('#') === 0) {
                                  for (let i = 0; i < document.anchors.length; i++) {
                                    const anchor = document.anchors[i];
                                    if (anchor.name === aHyperlink.anchorName) {
                                      this.helper.scrollToAnchor(aHyperlink.anchorName);
                                      return;
                                    }
                                  }
                                  this.controller.post('BookmarkAction', null, 0);
                                }
                                return false;
                                };*/
                            };
                        }
                    }
                };
                try {
                    for (var aHyperlinks_1 = __values(aHyperlinks), aHyperlinks_1_1 = aHyperlinks_1.next(); !aHyperlinks_1_1.done; aHyperlinks_1_1 = aHyperlinks_1.next()) {
                        var aHyperlink = aHyperlinks_1_1.value;
                        _loop_3(aHyperlink);
                    }
                }
                catch (e_2_1) { e_2 = { error: e_2_1 }; }
                finally {
                    try {
                        if (aHyperlinks_1_1 && !aHyperlinks_1_1.done && (_a = aHyperlinks_1.return)) _a.call(aHyperlinks_1);
                    }
                    finally { if (e_2) throw e_2.error; }
                }
            });
        }
    };
    InteractionsService.ctorParameters = function () { return [
        { type: ModelService },
        { type: Md5Service },
        { type: ControllerService },
        { type: HelperService },
        { type: DrillDownService }
    ]; };
    InteractionsService = __decorate([
        Injectable()
    ], InteractionsService);
    return InteractionsService;
}());
export { InteractionsService };
