// stimulsoft-viewer-angular/esm5/lib/services/mail.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { ModelService } from './model.service';
import { ExportService } from './export.service';
import { FormService } from '../forms/form.service';
import { ControllerService } from './controller.service';
var MailService = /** @class */ (function () {
    function MailService(model, exportService, formService, controller) {
        var _this = this;
        this.model = model;
        this.exportService = exportService;
        this.formService = formService;
        this.controller = controller;
        exportService.getMessage().subscribe(function (event) {
            if (event === 'postMail') {
                _this.postMail();
            }
        });
    }
    MailService.prototype.postMail = function () {
        this.fillDefaults();
        if (this.model.options.email.showEmailDialog) {
            this.formService.showForm('sendEmailForm');
        }
        else {
            this.sendMail();
        }
    };
    MailService.prototype.sendMail = function () {
        var data = {
            exportFormat: this.exportService.format,
            exportSettings: this.exportService.exportSettings
        };
        this.formService.closeForm('sendEmailForm');
        this.controller.post('EmailReport', this.model.options.actions.emailReport, data);
        this.controller.actionSubject.next({ action: 'Email', data: data });
    };
    MailService.prototype.fillDefaults = function () {
        this.exportService.exportSettings.Email = this.model.options.email.defaultEmailAddress;
        this.exportService.exportSettings.Message = this.model.options.email.defaultEmailMessage;
        this.exportService.exportSettings.Subject = this.model.options.email.defaultEmailSubject;
        var ext = this.exportService.format.toLowerCase().replace('image', '');
        switch (ext) {
            case 'excel':
                ext = 'xls';
                break;
            case 'excel2007':
                ext = 'xlsx';
                break;
            case 'excelxml':
                ext = 'xls';
                break;
            case 'html5':
                ext = 'html';
                break;
            case 'jpeg':
                ext = 'jpg';
                break;
            case 'ppt2007':
                ext = 'ppt';
                break;
            case 'text':
                ext = 'txt';
                break;
            case 'word2007':
                ext = 'docx';
                break;
        }
        this.exportService.exportSettings.Attachment = this.model.reportParams.reportFileName + '.' + ext;
    };
    MailService.ctorParameters = function () { return [
        { type: ModelService },
        { type: ExportService },
        { type: FormService },
        { type: ControllerService }
    ]; };
    MailService = __decorate([
        Injectable()
    ], MailService);
    return MailService;
}());
export { MailService };
