// stimulsoft-viewer-angular/esm5/lib/services/model.service.js
import { __decorate, __values } from "tslib";
import { Injectable } from '@angular/core';
import { ControlClass } from './objects';
var ModelService = /** @class */ (function () {
    function ModelService() {
        this._reportParams = {};
        this.controls = new ControlClass();
        this.pages = [];
        this.months = [];
        this.dayOfWeek = [];
        this.dateRanges = [];
        this.encodingData = [];
        this.pdfSecurityCertificates = [];
        this.paperSizes = [];
        this.drillDownButtons = [];
        this.showProgress = false;
    }
    Object.defineProperty(ModelService.prototype, "reportParams", {
        get: function () {
            return this._reportParams;
        },
        set: function (parameters) {
            this._reportParams.pagesArray = parameters.pagesArray;
            // Apply new report parameters, if not update current page
            if (parameters.action && parameters.action !== 'GetPages') {
                this.reportParams.type = parameters.reportType;
                this.reportParams.drillDownGuid = parameters.drillDownGuid;
                this.reportParams.dashboardDrillDownGuid = parameters.dashboardDrillDownGuid;
                this.reportParams.pagesCount = parameters.pagesCount;
                if (parameters.pageNumber != null) {
                    this.reportParams.pageNumber = parameters.pageNumber;
                }
                this.reportParams.zoom = parameters.zoom;
                this.reportParams.viewMode = parameters.viewMode;
                this.reportParams.reportFileName = parameters.reportFileName;
                this.reportParams.collapsingStates = parameters.collapsingStates;
                if (parameters.bookmarksContent) {
                    this.reportParams.bookmarksContent = parameters.bookmarksContent;
                }
                if (parameters.resources) {
                    this.reportParams.resources = parameters.resources;
                }
                this.reportParams.isCompilationMode = parameters.isCompilationMode;
                if (parameters.variablesValues) {
                    this.reportParams.variablesValues = parameters.variablesValues;
                }
                if (parameters.parametersDateFormat) {
                    this.options.appearance.parametersPanelDateFormat = parameters.parametersDateFormat;
                }
                if (parameters.tableOfContentsPointers)
                    this.reportParams.tableOfContentsPointers = parameters.tableOfContentsPointers;
                this.reportParams.isEditableReport = parameters.isEditableReport;
                if (parameters.userValues) {
                    this.options.userValues = parameters.userValues;
                }
                this.reportParams.dashboards = parameters.dashboards;
                this.reportParams.previewSettings = parameters.previewSettings;
            }
        },
        enumerable: true,
        configurable: true
    });
    ModelService.prototype.setReportParams = function (parameters) {
        this._reportParams = parameters;
    };
    ModelService.prototype.getReportParams = function () {
        return this.copyObject(this._reportParams);
    };
    ModelService.prototype.copyObject = function (o) {
        if (!o || 'object' !== typeof o) {
            return o;
        }
        var c = 'function' === typeof o.pop ? [] : {};
        var p;
        var v;
        for (p in o) {
            if (o.hasOwnProperty(p) && p !== 'pagesArray') {
                v = o[p];
                if (v && 'object' === typeof v) {
                    c[p] = this.copyObject(v);
                }
                else {
                    c[p] = v;
                }
            }
        }
        return c;
    };
    Object.defineProperty(ModelService.prototype, "options", {
        get: function () {
            return this._options;
        },
        set: function (data) {
            this._options = this.toLower(data.options);
            if (data.jsOptions) {
                for (var key in data.jsOptions) {
                    if (data.jsOptions.hasOwnProperty(key)) {
                        this._options[key] = data.jsOptions[key];
                    }
                }
            }
            this._options.exports.defaultSettings = data.defaultSettings;
            this.localization = data.loc;
            this.months = data.months;
            this.dayOfWeek = data.dayOfWeek;
            this.images = data.images;
            this.dateRanges = data.dateRanges;
            this.paperSizes = data.paperSizes;
            this.encodingData = data.encodingData;
            this.pdfSecurityCertificates = data.pdfSecurityCertificates;
            this.clearViewerState();
            this.setupOptions();
        },
        enumerable: true,
        configurable: true
    });
    ModelService.prototype.loc = function (attr) {
        var _a;
        return _a = this.localization[attr], (_a !== null && _a !== void 0 ? _a : attr);
    };
    ModelService.prototype.img = function (name) {
        var _a;
        if (name != null && name.length > 0 && !this.images[name]) {
            //console.warn(`Image: ${name} not found`);
        }
        return _a = this.images[name], (_a !== null && _a !== void 0 ? _a : '');
    };
    ModelService.prototype.toLower = function (obj) {
        var _this = this;
        var result = {};
        Object.keys(obj).forEach(function (key) {
            result[key.substr(0, 1).toLowerCase() + key.substr(1)] = (typeof (obj[key]) === 'object' && obj[key] !== null) ? _this.toLower(obj[key]) : obj[key];
        });
        return result;
    };
    ModelService.prototype.clear = function () {
        this._options = undefined;
        this._reportParams = {};
        this._imagesForBookmark = undefined;
        this.pages = [];
        this.form = null;
    };
    ModelService.prototype.clearViewerState = function () {
        this.reportParams = {};
        this.reportParams.type = 'Auto';
        this.reportParams.pageNumber = 0;
        this.reportParams.originalPageNumber = 0;
        this.reportParams.drillDownGuid = null;
        this.reportParams.dashboardDrillDownGuid = null;
        this.reportParams.collapsingStates = null;
        this.reportParams.bookmarksContent = null;
        this.reportParams.editableParameters = null;
        this.reportParams.resources = null;
        this.reportParams.drillDownParameters = [];
        this.reportParams.elementName = null;
        this.reportParams.variablesValues = null;
        this.reportParams.tableOfContentsPointers = [];
        this.reportParams.isEditableReport = false;
        this.options.viewerId = this.newToken();
        this.options.clientGuid = this.newToken();
        this.options.paramsVariables = null;
        this.options.multiFilterStates = null;
        this.options.isParametersReceived = false;
        this.options.drillDownInProgress = false;
        // this.controls.mainPanel.style.background = '';
        // this.tableElementGridStates = {};
        // Restore current page number, if reload current report
        if (this.reportParams.prevPageNumber) {
            this.reportParams.pageNumber = this.reportParams.prevPageNumber;
            delete this.reportParams.prevPageNumber;
        }
        this.fullScreenOptions = null;
        this.drillDownButtons = [];
        this.controls.parametersPanel.visible = false;
        this.controls.bookmarksPanel.visible = false;
        this.interactions = null;
        /*
        this.controls.drillDownPanel.visible = false;
        this.controls.findPanel.visible = false;
        this.controls.parametersPanel.visible = false;
        this.controls.resourcesPanel.visible = false;*/
    };
    ModelService.prototype.style = function (value) {
        return value !== '' && value != null ? value : '';
    };
    ModelService.prototype.createPostParameters = function (data, asObject, useOptions) {
        if (asObject === void 0) { asObject = true; }
        if (useOptions === void 0) { useOptions = true; }
        var _a, _b, _c, _d;
        var params;
        // Object params
        var postParams = {
            stiweb_component: 'Viewer',
            stiweb_imagesScalingFactor: this.getImagesScalingFactor()
        };
        if (this.properties) {
            postParams['properties'] = btoa(JSON.stringify(this.properties));
        }
        if (this.options && useOptions) {
            params = {
                viewerId: this.options.viewerId,
                routes: this.options.routes,
                formValues: this.options.formValues,
                clientGuid: this.options.clientGuid,
                drillDownGuid: this.reportParams.drillDownGuid,
                dashboardDrillDownGuid: this.reportParams.dashboardDrillDownGuid,
                cacheMode: this.options.server.cacheMode,
                cacheTimeout: this.options.server.cacheTimeout,
                cacheItemPriority: this.options.server.cacheItemPriority,
                pageNumber: this.reportParams.pageNumber,
                originalPageNumber: this.reportParams.originalPageNumber,
                reportType: this.reportParams.type,
                zoom: (this.reportParams.zoom && this.reportParams.zoom > 0) ? this.reportParams.zoom : (((_b = (_a = this.options) === null || _a === void 0 ? void 0 : _a.toolbar) === null || _b === void 0 ? void 0 : _b.zoom) > 0 ? (_d = (_c = this.options) === null || _c === void 0 ? void 0 : _c.toolbar) === null || _d === void 0 ? void 0 : _d.zoom : 100),
                viewMode: this.reportParams.viewMode || this.options.toolbar.viewMode,
                multiPageWidthCount: this.reportParams.multiPageWidthCount,
                multiPageHeightCount: this.reportParams.multiPageHeightCount,
                multiPageContainerWidth: this.reportParams.multiPageContainerWidth,
                multiPageContainerHeight: this.reportParams.multiPageContainerHeight,
                multiPageMargins: this.reportParams.multiPageMargins,
                showBookmarks: this.options.toolbar.showBookmarksButton,
                openLinksWindow: this.options.appearance.openLinksWindow,
                chartRenderType: this.options.appearance.chartRenderType,
                reportDisplayMode: this.options.appearance.reportDisplayMode,
                drillDownParameters: this.reportParams.drillDownParameters,
                editableParameters: this.reportParams.editableParameters,
                useRelativeUrls: this.options.server.useRelativeUrls,
                passQueryParametersForResources: this.options.server.passQueryParametersForResources,
                passQueryParametersToReport: this.options.server.passQueryParametersToReport,
                version: this.options.shortProductVersion,
                reportDesignerMode: this.options.reportDesignerMode,
                imagesQuality: this.options.appearance.imagesQuality,
                parametersPanelSortDataItems: this.options.appearance.parametersPanelSortDataItems,
                combineReportPages: this.options.appearance.combineReportPages,
                isAngular: true,
                allowAutoUpdateCookies: this.options.server.allowAutoUpdateCookies
            };
            if (this.options.server.useLocalizedCache && this.options.localization) {
                params['useLocalizedCache'] = true;
                params['localization'] = this.options.localization;
            }
            if (this.options.userValues) {
                params['userValues'] = this.options.userValues;
            }
        }
        else {
            params = {};
        }
        if (data) {
            Object.keys(data).forEach(function (key) { return params[key] = data[key]; });
        }
        // Object params
        if (params.action) {
            postParams['stiweb_action'] = params.action;
            delete params.action;
        }
        if (params.base64Data) {
            postParams['stiweb_data'] = params.base64Data;
            delete params.base64Data;
        }
        if (this.options && useOptions) {
            // Params
            var jsonParams = JSON.stringify(params);
            if (this.options.server.useCompression) {
                // postParams['stiweb_packed_parameters'] = StiGZipHelper.pack(jsonParams);
                postParams['stiweb_parameters'] = btoa(jsonParams);
            }
            else {
                postParams['stiweb_parameters'] = btoa(jsonParams);
            }
        }
        if (asObject) {
            return postParams;
        }
    };
    ModelService.prototype.newToken = function () {
        var a = '1234567890abcdefghijklmnopqrstuvwxyz'.split('');
        var b = [];
        var length = 32;
        for (var i = 0; i < length; i++) {
            var j = (Math.random() * (a.length - 1)).toFixed(0);
            b[i] = a[j];
        }
        return b.join('');
    };
    ModelService.prototype.setupOptions = function () {
        var _a, _b, _c, _d, _e, _f;
        if (!this.options.exports.showExportToPowerPoint && !this.options.exports.showExportToPdf && !this.options.exports.showExportToXps &&
            !this.options.exports.showExportToOpenDocumentWriter && !this.options.exports.showExportToOpenDocumentCalc && !this.options.exports.showExportToText &&
            !this.options.exports.showExportToRtf && !this.options.exports.showExportToWord2007 && !this.options.exports.showExportToCsv && !this.options.exports.showExportToJson &&
            !this.options.exports.showExportToDbf && !this.options.exports.showExportToXml && !this.options.exports.showExportToDif && !this.options.exports.showExportToSylk &&
            !this.options.exports.showExportToExcel && !this.options.exports.showExportToExcel2007 && !this.options.exports.showExportToExcelXml && !this.options.exports.showExportToHtml &&
            !this.options.exports.showExportToHtml5 && !this.options.exports.showExportToMht && !this.options.exports.showExportToImageBmp && !this.options.exports.showExportToImageGif &&
            !this.options.exports.showExportToImageJpeg && !this.options.exports.showExportToImageMetafile && !this.options.exports.showExportToImagePcx &&
            !this.options.exports.showExportToImagePng && !this.options.exports.showExportToImageTiff && !this.options.exports.showExportToImageSvg && !this.options.exports.showExportToImageSvgz) {
            if (!this.options.exports.showExportToDocument) {
                this.options.toolbar.showSaveButton = false;
            }
            this.options.toolbar.showSendEmailButton = false;
        }
        // Options
        this.options.isTouchDevice = this.options.appearance.interfaceType === 'Auto'
            ? this.isTouchDevice() && this.isMobileDevice()
            : this.options.appearance.interfaceType === 'Touch';
        this.options.isMobileDevice = this.options.appearance.interfaceType === 'Auto' && !this.options.reportDesignerMode
            ? this.isTouchDevice() && this.isMobileDevice()
            : this.options.appearance.interfaceType === 'Mobile';
        if (this.options.isMobileDevice) {
            this.initializeMobile();
        }
        else {
            this.options.toolbar.showPinToolbarButton = false;
        }
        this.controls.toolbar.visible = this.options.toolbar.visible;
        this.options.menuAnimDuration = 150;
        this.options.formAnimDuration = 200;
        this.options.scrollDuration = 350;
        this.options.menuHideDelay = 250;
        this.options.server.timeoutAutoUpdateCache = 180000;
        this.options.toolbar.backgroundColor = this.getHTMLColor(this.options.toolbar.backgroundColor);
        this.options.toolbar.borderColor = this.getHTMLColor(this.options.toolbar.borderColor);
        this.options.toolbar.fontColor = this.getHTMLColor(this.options.toolbar.fontColor);
        this.options.appearance.pageBorderColor = this.getHTMLColor(this.options.appearance.pageBorderColor);
        this.options.parametersValues = {};
        this.options.parameterRowHeight = this.options.isTouchDevice ? 35 : 30;
        this.options.minParametersCountForMultiColumns = 5;
        // First Day Of Week
        if (this.options.appearance.datePickerFirstDayOfWeek === 'Sunday') {
            this.dayOfWeek.splice(6, 1);
            this.dayOfWeek.splice(0, 0, 'Sunday');
        }
        if (((_b = (_a = this.options) === null || _a === void 0 ? void 0 : _a.toolbar) === null || _b === void 0 ? void 0 : _b.zoom) === -1 || ((_d = (_c = this.options) === null || _c === void 0 ? void 0 : _c.toolbar) === null || _d === void 0 ? void 0 : _d.zoom) === -2) {
            this.reportParams.autoZoom = (_f = (_e = this.options) === null || _e === void 0 ? void 0 : _e.toolbar) === null || _f === void 0 ? void 0 : _f.zoom;
        }
        // Actions
        if (!this.options.actions.getReport) {
            this.options.actions.getReport = this.options.actions.viewerEvent;
        }
        if (!this.options.actions.printReport) {
            this.options.actions.printReport = this.options.actions.viewerEvent;
        }
        if (!this.options.actions.openReport) {
            this.options.actions.openReport = this.options.actions.viewerEvent;
        }
        if (!this.options.actions.exportReport) {
            this.options.actions.exportReport = this.options.actions.viewerEvent;
        }
        if (!this.options.actions.interaction) {
            this.options.actions.interaction = this.options.actions.viewerEvent;
        }
        if (!(window.File && window.FileReader && window.FileList && window.Blob)) {
            this.options.toolbar.showOpenButton = false;
        }
        // Render JsViewer styles into HEAD
        if (this.options.requestResourcesUrl || this.options.appearance.customStylesUrl) {
            var viewerStyles = document.createElement('link');
            viewerStyles.setAttribute('type', 'text/css');
            viewerStyles.setAttribute('rel', 'stylesheet');
            viewerStyles.setAttribute('href', this.options.appearance.customStylesUrl || this.getResourceUrl('styles'));
            this.controls.head.appendChild(viewerStyles);
        }
        this.options.viewerId = (Math.random() * 1000000).toString();
    };
    ModelService.prototype.pagesNavigationIsActive = function () {
        return (this.options.appearance.fullScreenMode || this.options.appearance.scrollbarsMode) && this.reportParams.viewMode === 'Continuous';
    };
    Object.defineProperty(ModelService.prototype, "imagesForBookmark", {
        get: function () {
            var _this = this;
            if (!this._imagesForBookmark) {
                var names = ['root', 'folder', 'folderOpen', 'node', 'empty', 'line', 'join', 'joinBottom', 'plus', 'plusBottom', 'minus', 'minusBottom'];
                var imagesForBookmarks_1 = {};
                names.forEach(function (name) { return imagesForBookmarks_1[name] = _this.images['Bookmarks' + name + '.png']; });
                this._imagesForBookmark = imagesForBookmarks_1;
            }
            return this._imagesForBookmark;
        },
        enumerable: true,
        configurable: true
    });
    ModelService.prototype.isTouchDevice = function () {
        return ('ontouchstart' in document.documentElement);
    };
    ModelService.prototype.isMobileDevice = function () {
        return /iPhone|iPad|iPod|Macintosh|Android|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
    };
    ModelService.prototype.initializeMobile = function () {
        var e_1, _a;
        var isViewPortExist = false;
        var metas = this.controls.head.getElementsByTagName('meta');
        try {
            for (var metas_1 = __values(metas), metas_1_1 = metas_1.next(); !metas_1_1.done; metas_1_1 = metas_1.next()) {
                var meta = metas_1_1.value;
                if (meta.name && meta.name.toLowerCase() === 'viewport') {
                    isViewPortExist = true;
                    break;
                }
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (metas_1_1 && !metas_1_1.done && (_a = metas_1.return)) _a.call(metas_1);
            }
            finally { if (e_1) throw e_1.error; }
        }
        if (!isViewPortExist) {
            var viewPortTag = document.createElement('meta');
            viewPortTag.id = 'viewport';
            viewPortTag.name = 'viewport';
            viewPortTag.content = 'initial-scale=1.0,width=device-width,user-scalable=0';
            this.controls.head.appendChild(viewPortTag);
        }
        this.options.appearance.fullScreenMode = true;
        this.options.appearance.scrollbarsMode = true;
        this.options.appearance.parametersPanelPosition = 'Left';
        this.options.appearance.parametersPanelColumnsCount = 1;
        this.options.toolbar.displayMode = 'Separated';
        this.options.toolbar.viewMode = 'SinglePage';
        this.options.toolbar.showZoomButton = false;
        var defaultZoom = this.options.toolbar.zoom === -2 ? -2 : -1; // PageWidth or PageHeight
        this.options.toolbar.zoom = this.reportParams.zoom = defaultZoom;
        this.options.toolbar.showButtonCaptions = false;
        this.options.toolbar.showOpenButton = false;
        this.options.toolbar.showFindButton = false;
        this.options.toolbar.showEditorButton = false;
        this.options.toolbar.showFullScreenButton = false;
        this.options.toolbar.showAboutButton = false;
        this.options.toolbar.showViewModeButton = false;
    };
    ModelService.prototype.getImagesScalingFactor = function () {
        var wnd = window;
        var devicePixelRatio = window.devicePixelRatio || (wnd.deviceXDPI && wnd.logicalXDPI ? wnd.deviceXDPI / wnd.logicalXDPI : 1);
        if (!devicePixelRatio || devicePixelRatio <= 1) {
            return '1';
        }
        else {
            return devicePixelRatio.toString();
        }
    };
    ModelService.prototype.getHTMLColor = function (color) {
        if (color.indexOf(',') > 0 && color.indexOf('rgb') < 0) {
            return "rgb(" + color + ")";
        }
        return color;
    };
    ModelService.prototype.getResourceUrl = function (resourceParameter) {
        var url = this.getActionRequestUrl(this.options.requestResourcesUrl, this.options.actions.viewerEvent);
        url += url.indexOf('?') > 0 ? '&' : '?';
        url += 'stiweb_component=Viewer&stiweb_action=Resource&stiweb_data=' + resourceParameter + '&stiweb_theme=' + this.options.theme;
        url += '&stiweb_cachemode=' + (this.options.server.useCacheForResources
            ? this.options.server.cacheMode === 'ObjectSession' || this.options.server.cacheMode === 'StringSession'
                ? 'session'
                : 'cache'
            : 'none');
        url += '&stiweb_version=' + this.options.shortProductVersion;
        return url;
    };
    ModelService.prototype.getActionRequestUrl = function (requestUrl, action) {
        if (!action) {
            return requestUrl;
        }
        if (action.indexOf('?') < 0) {
            return requestUrl.replace('{action}', action);
        }
        var query = action.substring(action.indexOf('?') + 1);
        action = action.substring(0, action.indexOf('?'));
        return requestUrl.replace('{action}', action) + (requestUrl.indexOf('?') > 0 ? '&' : '?') + query;
    };
    ModelService = __decorate([
        Injectable()
    ], ModelService);
    return ModelService;
}());
export { ModelService };
