// stimulsoft-viewer-angular/esm5/lib/services/mouse.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
var MouseService = /** @class */ (function () {
    function MouseService() {
        var _this = this;
        this.documetMouseDown = new Subject();
        this.documetMouseMove = new Subject();
        this.documetMouseUp = new Subject();
        document.addEventListener('mouseup', function (event) {
            _this.documetMouseUp.next(event);
        });
        document.addEventListener('mousemove', function (event) {
            _this.documetMouseMove.next(event);
        });
        document.addEventListener('mousedown', function (event) {
            _this.documetMouseDown.next(event);
        });
    }
    MouseService.prototype.getDocumentMouseUp = function () {
        return this.documetMouseUp.asObservable();
    };
    MouseService.prototype.getDocumentMouseMove = function () {
        return this.documetMouseMove.asObservable();
    };
    MouseService.prototype.getDocumentMouseDown = function () {
        return this.documetMouseDown.asObservable();
    };
    MouseService = __decorate([
        Injectable()
    ], MouseService);
    return MouseService;
}());
export { MouseService };
