// stimulsoft-viewer-angular/esm5/lib/services/objects.js
import { Subject } from 'rxjs';
var ComponentDescription = /** @class */ (function () {
    function ComponentDescription(type, caption, img, action, tooltip, arrow, menuItems, margin, selected) {
        this.type = type;
        this.caption = caption;
        this.img = img;
        this.action = action;
        this.tooltip = tooltip;
        this.arrow = arrow;
        this.menuItems = menuItems;
        this.margin = margin;
        this.selected = selected;
        this.selected = false;
    }
    return ComponentDescription;
}());
export { ComponentDescription };
var ViewerEvent = /** @class */ (function () {
    function ViewerEvent(name, value, bookmarkPage, bookmarkAnchor, componentGuid) {
        this.name = name;
        this.value = value;
        this.bookmarkPage = bookmarkPage;
        this.bookmarkAnchor = bookmarkAnchor;
        this.componentGuid = componentGuid;
    }
    return ViewerEvent;
}());
export { ViewerEvent };
var ControlClass = /** @class */ (function () {
    function ControlClass() {
        this.toolbar = new ControlProps();
        this.reportPanel = new ControlProps();
        this.drillDownPanel = new ControlProps();
        this.findPanel = new ControlProps();
        this.resourcesPanel = new ControlProps();
        this.bookmarksPanel = new ControlProps();
        this.navigatePanel = new ControlProps();
        this.parametersPanel = new ControlProps();
        this.dashboardsPanel = new ControlProps();
        this.viewer = new ControlProps();
        this.tooltip = new ControlProps();
        this.aboutPanel = new ControlProps();
    }
    Object.defineProperty(ControlClass.prototype, "head", {
        get: function () {
            return document.getElementsByTagName('head')[0];
        },
        enumerable: true,
        configurable: true
    });
    return ControlClass;
}());
export { ControlClass };
var ControlProps = /** @class */ (function () {
    function ControlProps(pageService, el) {
        this.pageService = pageService;
        this.el = el;
        this._visible = false;
        this.subject = new Subject();
        this.enabled = true;
        this.layout = new Rectangle();
    }
    Object.defineProperty(ControlProps.prototype, "offsetHeight", {
        get: function () {
            var _a;
            return ((_a = this.el) === null || _a === void 0 ? void 0 : _a.nativeElement.offsetHeight) || 0;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ControlProps.prototype, "offsetWidth", {
        get: function () {
            var _a;
            return ((_a = this.el) === null || _a === void 0 ? void 0 : _a.nativeElement.offsetWidth) || 0;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ControlProps.prototype, "visible", {
        get: function () {
            return this._visible;
        },
        set: function (value) {
            var _a;
            this._visible = value;
            this.subject.next(value);
            (_a = this.pageService) === null || _a === void 0 ? void 0 : _a.calculateLayout();
        },
        enumerable: true,
        configurable: true
    });
    ControlProps.prototype.getVisibility = function () {
        return this.subject.asObservable();
    };
    Object.defineProperty(ControlProps.prototype, "exists", {
        get: function () {
            return this.el != null;
        },
        enumerable: true,
        configurable: true
    });
    return ControlProps;
}());
export { ControlProps };
var Rectangle = /** @class */ (function () {
    function Rectangle(width, height, left, bottom, right) {
        if (width === void 0) { width = 0; }
        if (height === void 0) { height = 0; }
        if (left === void 0) { left = 0; }
        if (bottom === void 0) { bottom = 0; }
        if (right === void 0) { right = 0; }
        this.width = width;
        this.height = height;
        this.left = left;
        this.bottom = bottom;
        this.right = right;
        this._top = 0;
    }
    Object.defineProperty(Rectangle.prototype, "top", {
        get: function () {
            return this._top;
        },
        set: function (value) {
            this._top = value;
        },
        enumerable: true,
        configurable: true
    });
    return Rectangle;
}());
export { Rectangle };
var BookmarkNode = /** @class */ (function () {
    function BookmarkNode(name, url, page, compunentGuid, nodes, open, selected) {
        this.name = name;
        this.url = url;
        this.page = page;
        this.compunentGuid = compunentGuid;
        this.nodes = nodes;
        this.open = open;
        this.selected = selected;
    }
    return BookmarkNode;
}());
export { BookmarkNode };
var InteractionParams = /** @class */ (function () {
    function InteractionParams(action, drillDownParameters, drillDownGuid, dashboardDrillDownGuid, variables, sortingParameters, collapsingParameters) {
        this.action = action;
        this.drillDownParameters = drillDownParameters;
        this.drillDownGuid = drillDownGuid;
        this.dashboardDrillDownGuid = dashboardDrillDownGuid;
        this.variables = variables;
        this.sortingParameters = sortingParameters;
        this.collapsingParameters = collapsingParameters;
    }
    return InteractionParams;
}());
export { InteractionParams };
var InteractionObject = /** @class */ (function () {
    function InteractionObject(paramsVariables, countColumns, countInColumn) {
        this.paramsVariables = paramsVariables;
        this.countColumns = countColumns;
        this.countInColumn = countInColumn;
    }
    return InteractionObject;
}());
export { InteractionObject };
var Variable = /** @class */ (function () {
    function Variable(name, alias, description, basicType, type, allowUserValues, dateTimeType, items, key, value, keyTo, isChecked, focusOnCreate, visible, binding, isNull) {
        if (isChecked === void 0) { isChecked = true; }
        if (visible === void 0) { visible = true; }
        this.name = name;
        this.alias = alias;
        this.description = description;
        this.basicType = basicType;
        this.type = type;
        this.allowUserValues = allowUserValues;
        this.dateTimeType = dateTimeType;
        this.items = items;
        this.key = key;
        this.value = value;
        this.keyTo = keyTo;
        this.isChecked = isChecked;
        this.focusOnCreate = focusOnCreate;
        this.visible = visible;
        this.binding = binding;
        this.isNull = isNull;
    }
    return Variable;
}());
export { Variable };
var DateTimeObject = /** @class */ (function () {
    function DateTimeObject(year, month, day, hours, minutes, seconds) {
        this.year = year;
        this.month = month;
        this.day = day;
        this.hours = hours;
        this.minutes = minutes;
        this.seconds = seconds;
    }
    return DateTimeObject;
}());
export { DateTimeObject };
var Item = /** @class */ (function () {
    function Item(name, caption, imageName, key, haveSubMenu, imageSizes, selected, type) {
        this.name = name;
        this.caption = caption;
        this.imageName = imageName;
        this.key = key;
        this.haveSubMenu = haveSubMenu;
        this.imageSizes = imageSizes;
        this.selected = selected;
        this.type = type;
    }
    return Item;
}());
export { Item };
var Message = /** @class */ (function () {
    function Message(action, data, subAction) {
        this.action = action;
        this.data = data;
        this.subAction = subAction;
    }
    return Message;
}());
export { Message };
var Form = /** @class */ (function () {
    function Form(name, left, top, isMooving, level, formData) {
        if (isMooving === void 0) { isMooving = false; }
        this.name = name;
        this.left = left;
        this.top = top;
        this.isMooving = isMooving;
        this.level = level;
        this.formData = formData;
    }
    return Form;
}());
export { Form };
var ExportFormSettings = /** @class */ (function () {
    function ExportFormSettings(components, cSettings, openAfterExport, groups, update) {
        this.components = components;
        this.cSettings = cSettings;
        this.openAfterExport = openAfterExport;
        this.groups = groups;
        this.update = update;
    }
    return ExportFormSettings;
}());
export { ExportFormSettings };
var ExportGroup = /** @class */ (function () {
    function ExportGroup(opened) {
        this.opened = opened;
    }
    return ExportGroup;
}());
export { ExportGroup };
var ExportComponent = /** @class */ (function () {
    function ExportComponent(name, type, margin, label, tooltip, caption, checked, width, disabled, items, key) {
        this.name = name;
        this.type = type;
        this.margin = margin;
        this.label = label;
        this.tooltip = tooltip;
        this.caption = caption;
        this.checked = checked;
        this.width = width;
        this.disabled = disabled;
        this.items = items;
        this.key = key;
    }
    return ExportComponent;
}());
export { ExportComponent };
var DrillDown = /** @class */ (function () {
    function DrillDown(caption, selected, reportParams, visible) {
        if (selected === void 0) { selected = false; }
        this.caption = caption;
        this.selected = selected;
        this.reportParams = reportParams;
        this.visible = visible;
    }
    return DrillDown;
}());
export { DrillDown };
var ErrorMessage = /** @class */ (function () {
    function ErrorMessage(error, type) {
        this.error = error;
        this.type = type;
    }
    return ErrorMessage;
}());
export { ErrorMessage };
var Resource = /** @class */ (function () {
    function Resource(name, type, alias, id) {
        this.name = name;
        this.type = type;
        this.alias = alias;
        this.id = id;
    }
    return Resource;
}());
export { Resource };
var NotificationFormOptions = /** @class */ (function () {
    function NotificationFormOptions(image, message, description, buttonCaption, cancelAction, action) {
        this.image = image;
        this.message = message;
        this.description = description;
        this.buttonCaption = buttonCaption;
        this.cancelAction = cancelAction;
        this.action = action;
    }
    return NotificationFormOptions;
}());
export { NotificationFormOptions };
