// stimulsoft-viewer-angular/esm5/lib/services/print.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { ModelService } from './model.service';
import { ExportService } from './export.service';
import { HelperService } from './helper.service';
import { StiHttpClientService } from './http-client.service';
import { ControllerService } from './controller.service';
var PrintService = /** @class */ (function () {
    function PrintService(model, exportService, helper, httpClient, controller) {
        var _this = this;
        this.model = model;
        this.exportService = exportService;
        this.helper = helper;
        this.httpClient = httpClient;
        this.controller = controller;
        controller.getActionMessage().subscribe(function (message) {
            if (message.action === 'Print') {
                _this.postPrint(message.data.format);
            }
        });
        controller.getMessage().subscribe(function (message) {
            if (message.action === 'PrintReport') {
                switch (message.subAction) {
                    case 'PrintWithPreview':
                        _this.printAsPopup(message.data);
                        break;
                    case 'PrintWithoutPreview':
                        _this.printAsHtml(message.data);
                        break;
                }
            }
        });
    }
    PrintService.prototype.postPrint = function (printAction) {
        if (printAction === 'PrintPdf' && /iPhone|iPad|iPod|Macintosh/i.test(navigator.userAgent) && this.helper.isTouchDevice()) {
            var settings = this.exportService.getDefaultExportSettings('Pdf');
            settings.OpenAfterExport = true;
            this.postExport('Pdf', settings);
            return;
        }
        var data = {
            action: 'PrintReport',
            printAction: printAction,
            bookmarksPrint: this.model.options.appearance.bookmarksPrint
        };
        var url = ''; // this.model.options.requestUrl.replace('{action}', this.model.options.actions.printReport);
        switch (printAction) {
            case 'PrintPdf':
                if (this.model.options.appearance.printToPdfMode === 'Popup' || this.helper.getNavigatorName() == 'Safari' || this.helper.getNavigatorName() == 'iPad') {
                    this.printAsPdfPopup(data);
                }
                else {
                    this.printAsPdf(url, data);
                }
                break;
            default:
                this.controller.post('PrintReport', this.model.options.actions.printReport, data, 'text', printAction);
                break;
        }
    };
    PrintService.prototype.postExport = function (format, settings, elementName, isDashboardExport) {
        var data = {
            action: isDashboardExport ? 'ExportDashboard' : 'ExportReport',
            exportFormat: format,
            exportSettings: settings,
            elementName: elementName
        };
        var doc = settings && settings.OpenAfterExport && this.model.options.appearance.openExportedReportWindow === '_blank' ? this.helper.openNewWindow('about:blank', '_blank').document : null;
        var url = this.model.requestUrl;
        this.httpClient.postForm(url.replace('{action}', this.model.options.actions.exportReport), data, doc);
    };
    PrintService.prototype.printAsPdfPopup = function (data) {
        var url = this.model.requestUrl.replace('{action}', this.model.options.actions.printReport);
        var win = this.helper.openNewWindow('about:blank', '_blank');
        if (win != null) {
            this.httpClient.postForm(url, data, win.document);
        }
    };
    PrintService.prototype.printAsPdf = function (url, data) {
        data.responseType = 'blob';
        var viewer = this.model.controls.viewer.el.nativeElement;
        var printFrameId = this.model.options.viewerId + '_PdfPrintFrame';
        var printFrame = document.getElementById(printFrameId);
        if (printFrame) {
            viewer.removeChild(printFrame);
        }
        printFrame = document.createElement('iframe');
        printFrame.id = printFrameId;
        printFrame.name = printFrameId;
        printFrame.width = '0';
        printFrame.height = '0';
        printFrame.style.position = 'absolute';
        printFrame.style.border = 'none';
        // Firefox does not load the invisible content of the iframe
        if (this.helper.getNavigatorName() === 'Mozilla') {
            printFrame.width = '100px';
            printFrame.height = '100px';
            printFrame.style.visibility = 'hidden';
            printFrame.style.zIndex = '-100';
            printFrame.style.pointerEvents = 'none';
        }
        viewer.insertBefore(printFrame, viewer.firstChild);
        // Manual printing in browsers that do not support automatic PDF printing
        if (this.helper.getNavigatorName() !== 'Mozilla') {
            printFrame.onload = function () {
                printFrame.contentWindow.focus();
                printFrame.contentWindow.print();
            };
        }
        var form = document.createElement('FORM');
        form.setAttribute('id', 'printForm');
        form.setAttribute('method', 'POST');
        form.setAttribute('action', this.model.requestUrl.replace('{action}', this.model.options.actions.printReport));
        form.setAttribute('target', this.model.options.viewerId + '_PdfPrintFrame');
        var params = this.model.createPostParameters(data, true);
        Object.keys(params).forEach(function (key) {
            var paramsField = document.createElement('INPUT');
            paramsField.setAttribute('type', 'hidden');
            paramsField.setAttribute('name', key);
            paramsField.setAttribute('value', params[key]);
            form.appendChild(paramsField);
        });
        document.body.appendChild(form);
        form.submit();
        document.body.removeChild(form);
    };
    PrintService.prototype.printAsPopup = function (text) {
        var width = this.model.reportParams.pagesWidth || 790;
        var win = this.helper.openNewWindow('about:blank', 'PrintReport', 'height=900,width=' + width + ',toolbar=no,menubar=yes,scrollbars=yes,resizable=yes,location=no,directories=no,status=no');
        if (win != null) {
            win.document.open();
            win.document.write(text);
            win.document.close();
        }
    };
    PrintService.prototype.printAsHtml = function (text) {
        if (this.controller.showError(text)) {
            return;
        }
        var viewer = this.model.controls.viewer.el.nativeElement;
        // Remove '_PdfPrintFrame', this should fix IE strange error
        var printFrameId = this.model.options.viewerId + '_PdfPrintFrame';
        var printFrame = document.getElementById(printFrameId);
        if (printFrame) {
            viewer.removeChild(printFrame);
        }
        printFrameId = this.model.options.viewerId + '_HtmlPrintFrame';
        printFrame = document.getElementById(printFrameId);
        if (printFrame) {
            viewer.removeChild(printFrame);
        }
        printFrame = document.createElement('iframe');
        printFrame.id = printFrameId;
        printFrame.name = printFrameId;
        printFrame.width = '0';
        printFrame.height = '0';
        printFrame.style.position = 'absolute';
        printFrame.style.border = 'none';
        viewer.insertBefore(printFrame, viewer.firstChild);
        printFrame.contentWindow.document.open();
        printFrame.contentWindow.document.write(text);
        printFrame.contentWindow.document.close();
        setTimeout(function () {
            printFrame.contentWindow.focus();
            printFrame.contentWindow.print();
        });
    };
    PrintService.ctorParameters = function () { return [
        { type: ModelService },
        { type: ExportService },
        { type: HelperService },
        { type: StiHttpClientService },
        { type: ControllerService }
    ]; };
    PrintService = __decorate([
        Injectable()
    ], PrintService);
    return PrintService;
}());
export { PrintService };
