// stimulsoft-viewer-angular/esm5/lib/services/styles.service.js
import { __decorate, __values } from "tslib";
import { Injectable } from '@angular/core';
import { ModelService } from './model.service';
var StylesService = /** @class */ (function () {
    function StylesService(model) {
        this.model = model;
        this.pagesCss = {};
    }
    StylesService.prototype.setupStyle = function (style, id) {
        var _a;
        if (this.pagesCss[id] == null) {
            this.pagesCss[id] = document.createElement('STYLE');
            this.pagesCss[id].id = ((_a = this.model.options) === null || _a === void 0 ? void 0 : _a.viewerId) + 'Styles';
            this.pagesCss[id].setAttribute('type', 'text/css');
            this.model.controls.head.appendChild(this.pagesCss[id]);
        }
        if (this.pagesCss[id].styleSheet) {
            this.pagesCss[id].styleSheet.cssText = style;
        }
        else {
            this.pagesCss[id].innerHTML = style;
        }
    };
    StylesService.prototype.addCustomFontStyles = function (customFonts) {
        if (!customFonts) {
            return;
        }
        var existsStyles = [];
        try {
            existsStyles = document.getElementsByTagName('head')[0].getElementsByTagName('style');
        }
        catch (e) { }
        customFonts.forEach(function (customFont) {
            var e_1, _a;
            if (customFont.contentForCss && customFont.originalFontFamily) {
                var style = document.createElement('style');
                style.innerHTML = "@font-face {\r\nfont-family: '" + customFont.originalFontFamily + "';\r\n src: url(" + customFont.contentForCss + ");\r\n }";
                var existsThisStyle = false;
                try {
                    for (var existsStyles_1 = __values(existsStyles), existsStyles_1_1 = existsStyles_1.next(); !existsStyles_1_1.done; existsStyles_1_1 = existsStyles_1.next()) {
                        var estyle = existsStyles_1_1.value;
                        if (estyle.innerHTML.indexOf("font-family: '" + customFont.originalFontFamily + "'") > 0) {
                            existsThisStyle = true;
                            break;
                        }
                    }
                }
                catch (e_1_1) { e_1 = { error: e_1_1 }; }
                finally {
                    try {
                        if (existsStyles_1_1 && !existsStyles_1_1.done && (_a = existsStyles_1.return)) _a.call(existsStyles_1);
                    }
                    finally { if (e_1) throw e_1.error; }
                }
                if (!existsThisStyle) {
                    document.getElementsByTagName('head')[0].appendChild(style);
                }
            }
        });
    };
    StylesService.prototype.addChartScript = function (script) {
        var currChartScripts = document.getElementById(this.model.options.viewerId + 'chartScriptJsViewer');
        if (currChartScripts) {
            this.model.controls.head.removeChild(currChartScripts);
        }
        if (script) {
            var chartScripts = document.createElement('Script');
            chartScripts.setAttribute('type', 'text/javascript');
            chartScripts.id = this.model.options.viewerId + 'chartScriptJsViewer';
            chartScripts.textContent = script;
            this.model.controls.head.appendChild(chartScripts);
        }
    };
    StylesService.ctorParameters = function () { return [
        { type: ModelService }
    ]; };
    StylesService = __decorate([
        Injectable()
    ], StylesService);
    return StylesService;
}());
export { StylesService };
