// stimulsoft-viewer-angular/esm5/lib/services/tooltip.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { ModelService } from './model.service';
import { MenuService } from '../menu/menu.service';
var TooltipService = /** @class */ (function () {
    function TooltipService(model, menuService) {
        this.model = model;
        this.menuService = menuService;
        this.state = 'initial';
        this.caption = '';
        this.helpLink = '';
        this.top = 0;
        this.left = 0;
        this.innerState = 'initial';
    }
    TooltipService.prototype.show = function (left, top, caption, below, helpLink) {
        var _this = this;
        if (below === void 0) { below = true; }
        if (this.menuService.isMenuVisible() || !this.model.options.appearance.showTooltips) {
            this.hideImmediately();
            return;
        }
        if (this.caption === caption && this.helpLink === helpLink && this.state !== 'initial') {
            this.showInternal();
            return;
        }
        this.caption = caption;
        this.helpLink = helpLink;
        this.state = this.innerState = this.state !== 'expanded' ? 'preInitial' : 'preInitial2';
        setTimeout(function () {
            var tooltip = _this.model.controls.tooltip.el.nativeElement;
            var browserWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
            if (!below) {
                top -= tooltip.offsetHeight;
            }
            if (left + tooltip.offsetWidth > browserWidth) {
                left = browserWidth - tooltip.offsetWidth;
            }
            _this.left = left;
            _this.top = top;
            if (_this.innerState !== 'initial') {
                _this.showInternal();
            }
        });
    };
    TooltipService.prototype.showInternal = function () {
        this.state = 'expanded';
        this.innerState = 'expanded';
    };
    TooltipService.prototype.showFromTooltip = function () {
        if (this.state !== 'preInitial') {
            this.state = 'expanded';
            this.innerState = 'expanded';
        }
    };
    TooltipService.prototype.hideImmediately = function () {
        this.innerState = 'initial';
        this.state = 'initial';
    };
    TooltipService.prototype.hide = function () {
        var _this = this;
        this.innerState = 'initial';
        if (this.timeout) {
            clearTimeout(this.timeout);
        }
        this.timeout = setTimeout(function () {
            if (_this.innerState === 'initial') {
                _this.state = 'initial';
            }
            _this.timeout = null;
        }, 1000);
    };
    TooltipService.ctorParameters = function () { return [
        { type: ModelService },
        { type: MenuService }
    ]; };
    TooltipService = __decorate([
        Injectable()
    ], TooltipService);
    return TooltipService;
}());
export { TooltipService };
