// stimulsoft-viewer-angular/esm5/lib/components/about-panel.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ControllerService } from '../services/controller.service';
import { HelperService } from '../services/helper.service';
import { FormService } from '../forms/form.service';
var AboutPanelComponent = /** @class */ (function () {
    function AboutPanelComponent(model, controller, helper, formService) {
        var _this = this;
        this.model = model;
        this.controller = controller;
        this.helper = helper;
        this.formService = formService;
        this.year = new Date().getFullYear();
        this.top = 0;
        this.left = 0;
        this.opacity = 0;
        controller.getActionMessage().subscribe(function (message) {
            if (message.action === 'About') {
                _this.formService.form = { name: 'about', left: 0, top: 0, isMooving: false, level: 2 };
                _this.opacity = 0;
                setTimeout(function () {
                    _this.top = _this.model.viewerSize.height / 2 - _this.element.nativeElement.offsetHeight / 2;
                    _this.left = _this.model.viewerSize.width / 2 - _this.element.nativeElement.offsetWidth / 2;
                    _this.opacity = 1;
                });
            }
        });
    }
    AboutPanelComponent.prototype.ngOnInit = function () { };
    AboutPanelComponent.prototype.close = function () {
        this.formService.closeForm('about');
    };
    AboutPanelComponent.prototype.click = function (event) {
        if (event) {
            event.stopPropagation();
            event.preventDefault();
        }
        this.helper.openNewWindow('https://www.stimulsoft.com');
    };
    AboutPanelComponent.prototype.isLicensed = function () {
        return !this.model.options.cloudMode && !this.model.options.serverMode && !this.model.options.standaloneJsMode &&
            !this.model.options.reportDesignerMode && !this.model.options.licenseIsValid;
    };
    Object.defineProperty(AboutPanelComponent.prototype, "userLabel", {
        get: function () {
            var userName = this.model.options.licenseUserName || '';
            if (this.isLicensed()) {
                if (userName) {
                    userName += ', ';
                }
                return userName + this.helper.getBackText(true) + ' Version';
            }
            else {
                return userName;
            }
        },
        enumerable: true,
        configurable: true
    });
    AboutPanelComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: HelperService },
        { type: FormService }
    ]; };
    __decorate([
        ViewChild('element')
    ], AboutPanelComponent.prototype, "element", void 0);
    AboutPanelComponent = __decorate([
        Component({
            selector: 'sti-about-panel',
            template: "\n    <div #element class=\"stiJsViewerAboutPanel\" style=\"background-size: contain;\"\n      [style.display]=\"formService.form?.name == 'about' ? '' : 'none'\"\n      [style.left.px]=\"left\"\n      [style.top.px]=\"top\"\n      [style.opacity]=\"opacity\"\n      (click)=\"close()\" >\n      <div class=\"stiJsViewerAboutPanelHeader\">\n        Stimulsoft Reports\n      </div>\n      <img src=\"{{model.img('About.png')}}\" style=\"margin-top: 30px; height: 94px; width: 94px;\">\n      <div class=\"stiJsViewerAboutPanelCopyright\">\n        {{'Copyright 2003-' + year + ' Stimulsoft'}}\n      </div>\n      <div class=\"stiJsViewerAboutPanelVersion\">\n        {{model.options.productVersion?.trim()}}, {{model.options?.frameworkType}}, Angular\n      </div>\n      <div class=\"stiJsViewerAboutPanelVersion\">\n        All rights reserved\n      </div>\n      <div class=\"stiJsViewerAboutPanelVersion\" style=\"margin-top: 20px;font-weight: bold; \"\n        [style.color]=\"isLicensed() ? 'red' : '#444444'\"\n        [style.display]=\"isLicensed() || userLabel ? '' : 'none'\">\n        {{userLabel}}\n      </div>\n      <div class=\"stiJsViewerFormSeparator\" style=\"margin-top: 20px;\">\n      </div>\n      <div class=\"stiJsViewerAboutPanelStiLink\" (click)='click($event)'>\n        www.stimulsoft.com\n      </div>\n    </div>\n  "
        })
    ], AboutPanelComponent);
    return AboutPanelComponent;
}());
export { AboutPanelComponent };
