// stimulsoft-viewer-angular/esm5/lib/components/bookmarks-panel.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { HelperService } from '../services/helper.service';
import { ControllerService } from '../services/controller.service';
import { trigger, state, transition, animate, style, keyframes } from '@angular/animations';
var BookmarksPanelComponent = /** @class */ (function () {
    function BookmarksPanelComponent(model, helper, controller) {
        var _this = this;
        this.model = model;
        this.helper = helper;
        this.controller = controller;
        controller.getMessage().subscribe(function (message) {
            if (message.action === 'GetReport' || message.action === 'OpenReport') {
                setTimeout(function () {
                    if (_this.model.reportParams.bookmarksContent != null) {
                        _this.create();
                    }
                    _this.model.controls.bookmarksPanel.visible = _this.model.reportParams.bookmarksContent !== null && !_this.model.options.isMobileDevice;
                });
            }
            else {
                // Go to the bookmark, if it present
                setTimeout(function () {
                    if (_this.model.options.bookmarkAnchor != null) {
                        _this.helper.scrollToAnchor(_this.model.options.bookmarkAnchor, _this.model.options.componentGuid);
                        _this.model.options.bookmarkAnchor = null;
                        _this.model.options.componentGuid = null;
                    }
                });
            }
        });
        this.model.controls.bookmarksPanel.getVisibility().subscribe(function (value) {
            if (!value) {
                _this.helper.removeBookmarksLabel();
                _this.clearSelected();
            }
        });
    }
    BookmarksPanelComponent.prototype.ngAfterViewInit = function () {
        this.model.controls.bookmarksPanel.el = this.element;
    };
    BookmarksPanelComponent.prototype.ngOnInit = function () { };
    BookmarksPanelComponent.prototype.getImg1 = function (node, i) {
        var _a;
        if (((_a = node.nodes) === null || _a === void 0 ? void 0 : _a.length) === 0) {
            return i !== this.model.nodes.length - 1 ? this.model.imagesForBookmark['join'] : this.model.imagesForBookmark['joinBottom'];
        }
        return node.open ? (i === this.model.nodes.length - 1 ? this.model.imagesForBookmark['minusBottom'] : this.model.imagesForBookmark['minus']) :
            (i === this.model.nodes.length - 1 ? this.model.imagesForBookmark['plusBottom'] : this.model.imagesForBookmark['plus']);
    };
    BookmarksPanelComponent.prototype.postAction = function (node) {
        this.clearSelected();
        this.controller.postBookmarkNodeAction(node);
    };
    BookmarksPanelComponent.prototype.clearSelected = function () {
        var _a;
        (_a = this.model.nodes) === null || _a === void 0 ? void 0 : _a.forEach(function (n) {
            n.selected = false;
            n.nodes.forEach(function (element) { return element.selected = false; });
        });
    };
    BookmarksPanelComponent.prototype.create = function () {
        var bookmarks = this.model.reportParams.bookmarksContent.split('bookmarks.add(');
        var root = bookmarks[1].replace('0,-1,\'', '');
        this.rootName = root.substr(0, root.indexOf('\''));
        var nodes = [];
        this.parseNodes(bookmarks.splice(2), 0, nodes);
        this.model.nodes = nodes;
    };
    BookmarksPanelComponent.prototype.parseNodes = function (bookmarks, index, nodes) {
        var _this = this;
        var folder;
        bookmarks.forEach(function (bookmark) {
            var str = bookmark.substr(bookmark.indexOf(',') + 1);
            var nodeType = parseInt(str.substr(0, str.indexOf(',')), 10);
            var node = _this.parseNode(str);
            if (nodeType === 0) {
                nodes.push(node);
                folder = node;
            }
            else {
                folder.nodes.push(node);
            }
        });
    };
    BookmarksPanelComponent.prototype.parseNode = function (str) {
        str = str.substr(str.indexOf(',') + 2);
        var name = this.unescape(str.substr(0, str.indexOf('\',')));
        str = str.substr(str.indexOf('\',') + 3);
        var url = str.substr(0, str.indexOf('\','));
        str = str.substr(str.indexOf('\',') + 3);
        var pageTitle = str.substr(0, str.indexOf('\','));
        str = str.substr(str.indexOf('\',') + 3);
        var componentGuid = str.substr(0, str.length - 3);
        return { name: name, url: url, page: parseInt(pageTitle.substr(5), 10) - 1, compunentGuid: componentGuid, nodes: [], open: false, selected: false };
    };
    BookmarksPanelComponent.prototype.unescape = function (str) {
        return str.replace(/\\&apos;/g, '\'')
            .replace(/\\&quot;/g, '"')
            .replace(/\\&gt;/g, '>')
            .replace(/\\&lt;/g, '<')
            .replace(/\\&amp;/g, '&');
    };
    Object.defineProperty(BookmarksPanelComponent.prototype, "bottom", {
        get: function () {
            if (this.model.options.isMobileDevice) {
                return this.model.options.toolbar.autoHide ? '0' : '0.5in';
            }
            else {
                return this.model.options.toolbar.displayMode === 'Separated' && this.model.options.toolbar.visible ? '35px' : '0';
            }
        },
        enumerable: true,
        configurable: true
    });
    BookmarksPanelComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService },
        { type: ControllerService }
    ]; };
    __decorate([
        ViewChild('element')
    ], BookmarksPanelComponent.prototype, "element", void 0);
    __decorate([
        ViewChild('bookmarksPanel')
    ], BookmarksPanelComponent.prototype, "bookmarksPanel", void 0);
    BookmarksPanelComponent = __decorate([
        Component({
            selector: 'sti-bookmarks-panel',
            template: "\n    <div #element [style]=\"{fontFamily: model.options.toolbar.fontFamily, fontColor: helper.val(model.options.toolbar.fontColor)}\"\n      [class]=\"'stiJsViewerBookmarksPanel' + (model.options.toolbar.displayMode == 'Separated' ? ' stiJsViewerBookmarksPanelSeparated' : '')\"\n      [style.width.px]=\"model.options.appearance.bookmarksTreeWidth - (model.options.toolbar.displayMode == 'Simple' ? 0 : 1)\"\n      [style.bottom]=\"bottom\"\n      [style.top.px]=\"model.controls.bookmarksPanel.layout.top\"\n      [style.transition]=\"model.options.isMobileDevice ? 'opacity 300ms ease' : null\"\n      [style.display]=\"!this.model.options.isMobileDevice ? (model.controls.bookmarksPanel.visible ? '' : 'none') : null\"\n      [@visibility]=\"!this.model.options.isMobileDevice ? null : (model.controls.bookmarksPanel.visible ? 'visible' : 'hidden')\">\n      <div #bookmarksPanel [class]=\"'stiJsViewerBookmarksContainer' + (model.options.toolbar.displayMode == 'Simple' ? ' stiJsViewerBookmarksContainerSimple' : '')\"\n           [style.background]=\"helper.val(model.options.toolbar.backgroundColor)\"\n           [style.border]=\"helper.val(model.options.toolbar.borderColor) != '' ? '1px solid ' + helper.val(model.options.toolbar.borderColor): ''\">\n           <div class=\"stiTree\">\n              <div class=\"stiTreeNode\">\n                <img style=\"width: 16px; height: 16px;\" [src]=\"model.imagesForBookmark['root']\"/>\n                <a class=\"node\">{{rootName}}</a>\n              </div>\n              <div class=\"clip\" style=\"display:block;\">\n                <ng-container *ngFor=\"let node of model.nodes; index as i\">\n                    <div class=\"stiTreeNode\">\n                        <a (click)=\"node.open = !node.open\">\n                            <img style=\"width: 18px; height: 18px\"\n                             [src]=\"getImg1(node, i)\"/>\n                        </a>\n                        <img style=\"width: 16px; height: 16px;\" [src]=\"node.nodes?.length == 0 ? model.imagesForBookmark['node'] :(node.open ? model.imagesForBookmark['folderOpen'] : model.imagesForBookmark['folder'])\"/>\n                        <a [class]=\"node.selected ? 'nodeSel' : 'node'\" (click)=\"postAction(node)\">{{node.name}}</a>\n                    </div>\n                    <div class=\"clip\" [style.display]=\"node.open ? 'block' : 'none'\">\n                      <div *ngFor=\"let subNode of node.nodes; index as k\" class=\"stiTreeNode\">\n                        <img style=\"width: 18px; height: 18px;\" [src]=\"i != model.nodes.length - 1 ? model.imagesForBookmark['line'] : model.imagesForBookmark['empty']\"/>\n                        <img style=\"width: 18px; height: 18px;\" [src]=\"k == node.nodes.length - 1 ? model.imagesForBookmark['joinBottom'] : model.imagesForBookmark['join']\"/>\n                        <img style=\"width: 16px; height: 16px;\" [src]=\"model.imagesForBookmark['node']\" />\n                        <a [class]=\"subNode.selected ? 'nodeSel' : 'node'\" (click)=\"postAction(subNode)\">{{subNode.name}}</a>\n                      </div>\n                    </div>\n                </ng-container>\n              </div>\n          </div>\n      </div>\n    </div>\n  ",
            animations: [
                trigger('visibility', [
                    state('visible', style({ opacity: 1, display: 'block' })),
                    state('hidden', style({ opacity: 0, display: 'none' })),
                    transition('hidden => visible', [
                        animate('300ms ease-in-out', keyframes([
                            style({ display: 'block', opacity: 0, offset: 0 }),
                            style({ display: 'block', opacity: 1, offset: 1 }),
                        ]))
                    ]),
                    transition('visible => hidden', [
                        animate('300ms ease-in-out', keyframes([
                            style({ display: 'block', opacity: 1, offset: 0 }),
                            style({ display: 'none', opacity: 0, offset: 1 }),
                        ]))
                    ])
                ])
            ]
        })
    ], BookmarksPanelComponent);
    return BookmarksPanelComponent;
}());
export { BookmarksPanelComponent };
