// stimulsoft-viewer-angular/esm5/lib/components/center-text.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ControllerService } from '../services/controller.service';
import { HelperService } from '../services/helper.service';
var CenterTextComponent = /** @class */ (function () {
    function CenterTextComponent(model, controller, helper) {
        var _this = this;
        this.model = model;
        this.controller = controller;
        this.helper = helper;
        this.text = '';
        this.opacity = 0;
        this.display = 'none';
        controller.getActionMessage().subscribe(function (message) {
            switch (message.action) {
                case 'centerText':
                    _this.show(message.data);
                    break;
                case 'hideCenterText':
                    _this.hide();
                    break;
            }
        });
    }
    CenterTextComponent.prototype.ngOnInit = function () { };
    CenterTextComponent.prototype.show = function (text) {
        var _this = this;
        this.display = '';
        this.opacity = 0;
        this.text = text;
        setTimeout(function () {
            _this.helper.setObjectToCenter(_this.element.nativeElement);
            _this.opacity = 1;
        });
        if (this.hideTimer) {
            clearTimeout(this.hideTimer);
        }
        this.hideTimer = setTimeout(function () {
            _this.hide();
        }, 2000);
    };
    CenterTextComponent.prototype.hide = function () {
        var _this = this;
        this.opacity = 0;
        if (this.hideTimer) {
            clearTimeout(this.hideTimer);
        }
        this.hideTimer = setTimeout(function () {
            _this.display = 'none';
        }, 300);
    };
    CenterTextComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: HelperService }
    ]; };
    __decorate([
        ViewChild('element')
    ], CenterTextComponent.prototype, "element", void 0);
    CenterTextComponent = __decorate([
        Component({
            selector: 'sti-center-text',
            template: "\n    <div #element\n      style=\"position: absolute; z-index: 1000; transition-property: opacity; transition-duration: 300ms; text-shadow: rgb(0, 0, 0) -1px -1px 0px, rgb(0, 0, 0) 1px -1px 0px, rgb(0, 0, 0) -1px 1px 0px, rgb(0, 0, 0) 1px 1px 0px; font-size: 100px;\"\n      [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n      [style.color]=\"model.options.toolbar.fontColor\"\n      [style.opacity]=\"opacity\"\n      [style.display]=\"display\">\n      <div>\n        {{text}}\n      </div>\n    </div>\n  "
        })
    ], CenterTextComponent);
    return CenterTextComponent;
}());
export { CenterTextComponent };
