// stimulsoft-viewer-angular/esm5/lib/components/find-panel.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { FindService } from '../services/find.service';
import { ControllerService } from '../services/controller.service';
var FindPanelComponent = /** @class */ (function () {
    function FindPanelComponent(model, findService, controller) {
        var _this = this;
        this.model = model;
        this.findService = findService;
        this.controller = controller;
        this.changeFind = false;
        this.initItems();
        controller.getMessage().subscribe(function (message) {
            switch (message.action) {
                case 'GetReport':
                case 'GetPages':
                case 'OpenReport':
                case 'Variables':
                    if (model.controls.findPanel.visible) {
                        setTimeout(function () {
                            _this.findService.hideFindLabels();
                            _this.find('Next');
                        });
                    }
                    break;
            }
        });
    }
    FindPanelComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        this.model.controls.findPanel.el = this.element;
        this.model.controls.findPanel.getVisibility().subscribe(function () {
            if (_this.model.controls.findPanel.visible) {
                _this.findService.text = '';
                setTimeout(function () {
                    _this.textBox.element.nativeElement.focus();
                });
            }
        });
    };
    FindPanelComponent.prototype.ngOnInit = function () { };
    FindPanelComponent.prototype.selected = function (item) {
        switch (item.action) {
            case 'MatchCase':
                return this.findService.matchCase;
            case 'MatchWholeWord':
                return this.findService.matchWholeWord;
        }
        return false;
    };
    FindPanelComponent.prototype.textBoxAction = function (target) {
        this.findService.text = target.value;
        this.find('Next');
    };
    FindPanelComponent.prototype.find = function (direction) {
        if (this.findService.text === '') {
            this.findService.hideFindLabels();
            return;
        }
        if (this.findService.lastFindText !== this.findService.text || this.changeFind) {
            this.changeFind = false;
            this.findService.showFindLabels();
        }
        else {
            this.findService.selectFindLabel(direction);
        }
    };
    FindPanelComponent.prototype.initItems = function () {
        this.items = [
            { type: 'button', action: 'close', img: 'CloseForm.png', margin: '2px' },
            { type: 'textBlock', action: 'text', caption: this.model.loc('FindWhat').replace(":", ""), margin: '2px' },
            { type: 'textBox', action: 'findTextBox', margin: '2px' },
            { type: 'button', action: 'FindPreviows', caption: this.model.loc('FindPrevious'), img: 'Arrows.ArrowUpBlue.png', margin: '2px' },
            { type: 'button', action: 'FindNext', caption: this.model.loc('FindNext'), img: 'Arrows.ArrowDownBlue.png', margin: '2px' },
            { type: 'button', action: 'MatchCase', caption: this.model.loc('MatchCase').replace('&', ''), margin: '2px' },
            { type: 'button', action: 'MatchWholeWord', caption: this.model.loc('MatchWholeWord').replace('&', ''), margin: '2px' }
        ];
    };
    FindPanelComponent.prototype.action = function (action) {
        switch (action) {
            case 'close':
                this.model.controls.findPanel.visible = false;
                break;
            case 'MatchCase':
                this.findService.matchCase = !this.findService.matchCase;
                this.changeFind = true;
                break;
            case 'MatchWholeWord':
                this.findService.matchWholeWord = !this.findService.matchWholeWord;
                this.changeFind = true;
                break;
            case 'FindPreviows':
                this.findService.text = this.textBox.element.nativeElement.value;
                this.find('Previous');
                break;
            case 'FindNext':
                this.findService.text = this.textBox.element.nativeElement.value;
                this.find('Next');
                break;
        }
    };
    FindPanelComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: FindService },
        { type: ControllerService }
    ]; };
    __decorate([
        ViewChild('element')
    ], FindPanelComponent.prototype, "element", void 0);
    __decorate([
        ViewChild('textBox')
    ], FindPanelComponent.prototype, "textBox", void 0);
    FindPanelComponent = __decorate([
        Component({
            selector: 'sti-find-panel',
            template: "\n    <div #element [style.display]=\"model.controls.findPanel.visible ? '' : 'none'\"\n    [class]=\"'stiJsViewerToolBar' + (model.options.toolbar.displayMode == 'Separated' ? ' stiJsViewerToolBarSeparated' : '')\">\n      <div [style.paddingTop]=\"model.options.toolbar.displayMode == 'Simple' ? '2px' : ''\">\n        <div class=\"stiJsViewerToolBarTable\" [style.border]=\"model.options.toolbar.displayMode == 'Separated' ? '0px' : ''\" style=\"box-sizing: border-box\">\n          <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"margin: 0\">\n            <tbody>\n                <tr class=\"stiJsViewerClearAllStyles\">\n                   <td *ngFor=\"let item of items\"  class=\"stiJsViewerClearAllStyles\">\n                      <sti-button *ngIf=\"item.type=='button'\"\n                        [actionName]=\"item.action\"\n                        [imageName]=\"item.img\"\n                        [caption]=\"item.caption\"\n                        [margin]=\"item.margin\"\n                        [height]=\"model.options.toolbar.displayMode == 'Separated' ? '28px' : ''\"\n                        (action)=\"action(item.action)\"\n                        [selected]=\"selected(item)\"\n                        ></sti-button>\n                      <sti-text-box #textBox *ngIf=\"item.type=='textBox'\" [width]=\"170\"\n                        [margin]=\"item.margin\"\n                        [focusOnCreate]=\"true\"\n                        (action)=\"textBoxAction($event)\"\n                        [value]=\"findService.text\"></sti-text-box>\n                      <sti-text-block *ngIf=\"item.type=='textBlock'\" [margin]=\"item.margin\" [text]=\"item.caption\"></sti-text-block>\n                  </td>\n                </tr>\n            </tbody>\n          </table>\n        </div>\n      </div>\n    </div>\n  "
        })
    ], FindPanelComponent);
    return FindPanelComponent;
}());
export { FindPanelComponent };
