// stimulsoft-viewer-angular/esm5/lib/components/interactions-panel.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { HelperService } from '../services/helper.service';
import { InteractionsService } from '../services/interactions.service';
import { MenuService } from '../menu/menu.service';
import { ControllerService } from '../services/controller.service';
import { trigger, state, transition, style, keyframes, animate } from '@angular/animations';
var InteractionsPanelComponent = /** @class */ (function () {
    function InteractionsPanelComponent(model, helper, interactionService, menuService, controller) {
        this.model = model;
        this.helper = helper;
        this.interactionService = interactionService;
        this.menuService = menuService;
        this.controller = controller;
        this.currentOpeningParameter = null;
        this.dropDownButtonWasClicked = false;
        this.dateTimeButtonWasClicked = false;
    }
    InteractionsPanelComponent.prototype.ngAfterViewInit = function () {
        this.model.controls.parametersPanel.el = this.element;
    };
    InteractionsPanelComponent.prototype.ngOnInit = function () { };
    InteractionsPanelComponent.prototype.getNameAlign = function (indexRow, indexColumn) {
        var index = this.index(indexRow, indexColumn);
        if (index < this.length() &&
            this.model.interactions.paramsVariables[index].basicType === 'Range' &&
            this.model.options.appearance.parametersPanelPosition === 'Left') {
            return 'top';
        }
        return '';
    };
    InteractionsPanelComponent.prototype.getNamePadding = function (indexRow, indexColumn) {
        var index = this.index(indexRow, indexColumn);
        if (index < this.length() &&
            this.model.interactions.paramsVariables[index].basicType === 'Range' &&
            this.model.options.appearance.parametersPanelPosition === 'Left') {
            return this.model.options.isTouchDevice ? '11px' : '9px';
        }
        return '';
    };
    InteractionsPanelComponent.prototype.getCaption = function (indexRow, indexColumn, checkLeft) {
        if (checkLeft === void 0) { checkLeft = true; }
        var index = this.index(indexRow, indexColumn);
        if (checkLeft && this.ifLeftRange(indexRow, indexColumn)) {
            return '';
        }
        return index < this.length() ? this.model.interactions.paramsVariables[index].alias : '';
    };
    InteractionsPanelComponent.prototype.ifLeftRange = function (indexRow, indexColumn) {
        var index = this.index(indexRow, indexColumn);
        return this.model.interactions.paramsVariables[index].basicType === 'Range' && this.model.options.appearance.parametersPanelPosition === 'Left';
    };
    InteractionsPanelComponent.prototype.getTitle = function (indexRow, indexColumn) {
        var index = this.index(indexRow, indexColumn);
        return index < this.length() ? this.model.interactions.paramsVariables[index].description : '';
    };
    InteractionsPanelComponent.prototype.index = function (indexRow, indexColumn) {
        return indexColumn * this.model.interactions.countInColumn.length + indexRow;
    };
    InteractionsPanelComponent.prototype.length = function () {
        var _a, _b;
        return ((_a = this.model.interactions) === null || _a === void 0 ? void 0 : _a.paramsVariables) != null ? Object.keys((_b = this.model.interactions) === null || _b === void 0 ? void 0 : _b.paramsVariables).length : 0;
    };
    Object.defineProperty(InteractionsPanelComponent.prototype, "className", {
        get: function () {
            var className = 'stiJsViewerParametersPanel';
            if (this.model.options.appearance.parametersPanelPosition === 'Top') {
                className += ' stiJsViewerParametersPanelTop';
                if (this.model.options.toolbar.displayMode === 'Separated') {
                    className += ' stiJsViewerParametersPanelSeparatedTop';
                }
            }
            return className;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(InteractionsPanelComponent.prototype, "top", {
        get: function () {
            var styleTop = this.model.options.toolbar.visible ? this.model.controls.toolbar.offsetHeight : 0;
            if (this.model.options.isMobileDevice && this.model.options.toolbar.autoHide) {
                styleTop = 0;
            }
            styleTop += this.model.controls.drillDownPanel.exists ? this.model.controls.drillDownPanel.offsetHeight : 0;
            styleTop += this.model.controls.findPanel.exists ? this.model.controls.findPanel.offsetHeight : 0;
            styleTop += this.model.controls.resourcesPanel.exists ? this.model.controls.resourcesPanel.offsetHeight : 0;
            return styleTop;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(InteractionsPanelComponent.prototype, "bottom", {
        get: function () {
            if (this.model.options.appearance.parametersPanelPosition === 'Left') {
                if (this.model.options.isMobileDevice) {
                    return this.model.options.toolbar.autoHide ? '0' : '0.5in';
                }
                else {
                    return this.model.options.toolbar.displayMode === 'Separated' && this.model.options.toolbar.visible ? '35px' : '0';
                }
            }
            return '';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(InteractionsPanelComponent.prototype, "innerClassName", {
        get: function () {
            var className = this.model.options.toolbar.displayMode === 'Simple' ? 'stiJsViewerInnerParametersPanelSimple' : '';
            if (this.model.options.appearance.parametersPanelPosition === 'Left') {
                className += ' stiJsViewerInnerParametersPanelLeft';
                if (this.model.options.toolbar.displayMode === 'Separated') {
                    className += ' stiJsViewerInnerParametersPanelSeparatedLeft';
                }
            }
            return className;
        },
        enumerable: true,
        configurable: true
    });
    InteractionsPanelComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService },
        { type: InteractionsService },
        { type: MenuService },
        { type: ControllerService }
    ]; };
    __decorate([
        ViewChild('element')
    ], InteractionsPanelComponent.prototype, "element", void 0);
    InteractionsPanelComponent = __decorate([
        Component({
            selector: 'sti-interactions-panel',
            template: "\n    <div #element [class]=\"className\"\n       [style.display]=\"!this.model.options.isMobileDevice ? (model.controls.parametersPanel.visible ? '' : 'none') : null\"\n       [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n       [style.color]=\"model.options.toolbar.fontColor\"\n       [style.top.px]=\"top\"\n       [style.left.px]=\"model.controls.parametersPanel.layout.left\"\n       [style.bottom]=\"bottom\"\n       [style.transition]=\"model.options.isMobileDevice ? 'opacity 300ms ease' : null\"\n       [@visibility]=\"!this.model.options.isMobileDevice ? null : (model.controls.parametersPanel.visible ? 'visible' : 'hidden')\">\n       <div [class]=\"innerClassName\" [style.marginTop]=\"model.options.toolbar.displayMode == 'Simple' ? '2px' : ''\" (scroll)=\"menuService.closeAllMenus()\">\n         <div class=\"stiJsViewerInnerContainerParametersPanel\"\n              [style.background]=\"helper.val(model.options.toolbar.backgroundColor, '')\"\n              [style.border]=\"helper.val(model.options.toolbar.backgroundColor, '') != '' ? '1px solid ' + helper.val(model.options.toolbar.backgroundColor, '') : ''\"\n              [style.maxHeight]=\"model.options.appearance.parametersPanelPosition == 'Top' ? model.options.appearance.parametersPanelMaxHeight + 'px' : ''\"\n              (scroll)=\"menuService.closeAllMenus()\">\n            <table *ngIf=\"model.interactions\" class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"border: 0px;\">\n              <tbody>\n                <tr *ngFor=\"let item of model.interactions.countInColumn; let indexRow=index\" >\n                  <ng-container *ngFor=\"let item2 of model.interactions.countColumns; let indexColumn=index\">\n                    <ng-container *ngIf=\"index(indexRow, indexColumn) <= length()\">\n                        <td [style.padding]=\"'0 10px 0 ' + (indexColumn > 0 ? '30px' : '0')\"\n                            [style.whiteSpace]=\"'nowrap'\"\n                            [style.verticalAlign]='getNameAlign(indexRow, indexColumn)'\n                            [style.paddingTop]='getNamePadding(indexRow, indexColumn)'\n                            [attr.title]='getTitle(indexRow, indexColumn)'>\n                            {{getCaption(indexRow, indexColumn)}}\n                              <table *ngIf=\"ifLeftRange(indexRow, indexColumn)\" class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"height: 60px;\">\n                                <tbody>\n                                  <tr class=\"stiJsViewerClearAllStyles\">\n                                    <td class=\"stiJsViewerClearAllStyles\" rowspan=\"2\" style=\"vertical-align: top; padding-top: 9px;\">\n                                      {{getCaption(indexRow, indexColumn, false)}}\n                                    </td>\n                                    <td class=\"stiJsViewerClearAllStyles\" style=\"padding-left: 12px;\">\n                                      {{model.loc('RangeFrom')}}\n                                    </td>\n                                  </tr>\n                                  <tr class=\"stiJsViewerClearAllStyles\">\n                                    <td class=\"stiJsViewerClearAllStyles\" style=\"padding-left: 12px;\">\n                                      {{model.loc('RangeTo')}}\n                                    </td>\n                                  </tr>\n                                </tbody>\n                              </table>\n                            </td>\n                        <td [style.padding]=\"'0px'\">\n                          <sti-parameter *ngIf=\"index(indexRow, indexColumn) != length()\"  [params]=\"model.interactions.paramsVariables[index(indexRow, indexColumn)]\"></sti-parameter>\n\n                          <table *ngIf=\"index(indexRow, indexColumn) == length()\" class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\"\n                            align=\"right\" style=\"margin: 5px 2px 10px 0px;\">\n                            <tr class=\"stiJsViewerClearAllStyles\">\n                              <td class=\"stiJsViewerClearAllStyles\">\n                                <sti-form-button [caption]=\"model.loc('Reset')\" (action)=\"controller.action({name: 'Reset'})\"></sti-form-button>\n                              </td>\n                              <td class=\"stiJsViewerClearAllStyles\" style=\"padding-left: 10px;\">\n                                <sti-form-button [caption]=\"model.loc('Submit')\" (action)=\"controller.action({name: 'Submit'})\"></sti-form-button>\n                              </td>\n                            </tr>\n                          </table>\n                        </td>\n                    </ng-container>\n                  </ng-container>\n                </tr>\n                <tr *ngIf=\"length() == model.interactions.countInColumn.length * model.interactions.countColumns.length\">\n                  <td></td>\n                  <td *ngIf=\"model.interactions.countColumns.length > 1\"></td>\n                  <td *ngIf=\"model.interactions.countColumns.length > 1\"></td>\n                  <td [style.padding]=\"'0px'\">\n                      <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\"\n                         align=\"right\" style=\"margin: 5px 2px 10px 0px;\">\n                         <tr class=\"stiJsViewerClearAllStyles\">\n                           <td class=\"stiJsViewerClearAllStyles\">\n                             <sti-form-button [caption]=\"model.loc('Reset')\" (action)=\"controller.action({name: 'Reset'})\"></sti-form-button>\n                           </td>\n                           <td class=\"stiJsViewerClearAllStyles\" style=\"padding-left: 10px;\">\n                              <sti-form-button [caption]=\"model.loc('Submit')\" (action)=\"controller.action({name: 'Submit'})\"></sti-form-button>\n                           </td>\n                         </tr>\n                       </table>\n                  </td>\n                </tr>\n              </tbody>\n            </table>\n          </div>\n      </div>\n    </div>\n  ",
            animations: [
                trigger('visibility', [
                    state('visible', style({ opacity: 1, display: 'block' })),
                    state('hidden', style({ opacity: 0, display: 'none' })),
                    transition('hidden => visible', [
                        animate('300ms ease-in-out', keyframes([
                            style({ display: 'block', opacity: 0, offset: 0 }),
                            style({ display: 'block', opacity: 1, offset: 1 }),
                        ]))
                    ]),
                    transition('visible => hidden', [
                        animate('300ms ease-in-out', keyframes([
                            style({ display: 'block', opacity: 1, offset: 0 }),
                            style({ display: 'none', opacity: 0, offset: 1 }),
                        ]))
                    ])
                ])
            ]
        })
    ], InteractionsPanelComponent);
    return InteractionsPanelComponent;
}());
export { InteractionsPanelComponent };
