// stimulsoft-viewer-angular/esm5/lib/components/navigate-panel.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ControllerService } from '../services/controller.service';
import { HelperService } from '../services/helper.service';
import { PageService } from '../services/page.service';
var NavigatePanelComponent = /** @class */ (function () {
    function NavigatePanelComponent(model, controller, helper, pageService) {
        this.model = model;
        this.controller = controller;
        this.helper = helper;
        this.pageService = pageService;
        this.comps = [];
    }
    NavigatePanelComponent.prototype.ngOnInit = function () {
        var _this = this;
        if (this.model.options.toolbar.displayMode === 'Separated') {
            this.initButtons();
        }
        this.controller.getMessage().subscribe(function (message) {
            if (message.action !== 'viewer_loaded') {
                _this.updateButtons();
            }
        });
        this.model.controls.navigatePanel.visible = this.model.options.toolbar.displayMode === 'Separated';
    };
    NavigatePanelComponent.prototype.ngAfterViewInit = function () {
        this.model.controls.navigatePanel.el = this.element;
    };
    NavigatePanelComponent.prototype.updateButtons = function () {
        var _this = this;
        this.comps.filter(function (i) { return i.action === 'ViewMode'; }).forEach(function (m) { return m.caption = _this.model.loc(_this.model.reportParams.viewMode); });
        this.comps.filter(function (i) { return i.action === 'Zoom'; }).forEach(function (m) {
            m.menuItems.forEach(function (n) { return n.selected = n.name === ('Zoom' + _this.model.reportParams.zoom); });
            m.caption = Math.round(_this.model.reportParams.zoom) + '%';
        });
        this.comps.filter(function (i) { return i.action === 'ShowFind'; }).forEach(function (m) {
            m.selected = _this.model.controls.findPanel.visible;
        });
    };
    NavigatePanelComponent.prototype.initButtons = function () {
        var comps = [];
        if (this.model.options.toolbar.showFirstPageButton) {
            comps.push({ type: 'button', action: 'FirstPage', img: 'PageFirst20.png', tooltip: true });
        }
        if (this.model.options.toolbar.showPreviousPageButton) {
            comps.push({ type: 'button', action: 'PrevPage', img: 'PagePrevious20.png', tooltip: true });
        }
        if (this.model.options.toolbar.showCurrentPageControl) {
            comps.push({ type: 'separator' });
            comps.push({ type: 'pageControl' });
            comps.push({ type: 'separator' });
        }
        if (this.model.options.toolbar.showNextPageButton) {
            comps.push({ type: 'button', action: 'NextPage', img: 'PageNext20.png', tooltip: true });
        }
        if (this.model.options.toolbar.showLastPageButton) {
            comps.push({ type: 'button', action: 'LastPage', img: 'PageLast20.png', tooltip: true });
        }
        comps.push({ type: 'space' });
        comps.push({ type: 'button', action: 'ZoomPageWidth', img: 'ZoomPageWidth20.png', tooltip: true });
        comps.push({ type: 'button', action: 'ZoomOnePage', img: 'ZoomOnePage20.png', tooltip: true });
        if (this.model.options.toolbar.showZoomButton) {
            comps.push({ type: 'separator' });
            comps.push({
                type: 'button', action: 'Zoom', caption: '100%', tooltip: true, arrow: 'Arrows.SmallArrowUpWhite.png',
                menuItems: this.helper.getZoomMenuItems()
            });
        }
        this.comps = comps;
    };
    NavigatePanelComponent.prototype.enabled = function (desc) {
        var disableNaviButtons = this.disableNaviButtons();
        switch (desc.action) {
            case 'FirstPage':
            case 'PrevPage':
                return this.model.reportParams.pageNumber > 0 && !disableNaviButtons;
            case 'NextPage':
            case 'LastPage':
                return this.model.reportParams.pageNumber < this.model.reportParams.pagesCount - 1 && !disableNaviButtons;
        }
        return true;
    };
    NavigatePanelComponent.prototype.selected = function (desc) {
        switch (desc.action) {
            case 'ZoomPageWidth':
                return Math.round(this.model.reportParams.zoom) === Math.round(this.pageService.getZoomByPageWidth());
            case 'ZoomOnePage':
                return Math.round(this.model.reportParams.zoom) === Math.round(this.pageService.getZoomByPageHeight());
        }
        return false;
    };
    NavigatePanelComponent.prototype.disableNaviButtons = function () {
        return this.model.reportParams.viewMode === 'MultiplePages' || this.model.reportParams.viewMode === 'WholeReport' ||
            (this.model.reportParams.viewMode === 'Continuous' && !this.model.options.appearance.scrollbarsMode && !this.model.options.appearance.fullScreenMode);
    };
    NavigatePanelComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: HelperService },
        { type: PageService }
    ]; };
    __decorate([
        ViewChild('element')
    ], NavigatePanelComponent.prototype, "element", void 0);
    NavigatePanelComponent = __decorate([
        Component({
            selector: 'sti-navigate-panel',
            template: "\n  <div #element class=\"stiJsViewerNavigatePanel\" style=\"display: block;\"\n    [style.display]=\"!model.options.isMobileDevice ? (this.model.controls.navigatePanel.visible ? '' : 'none') : ''\"\n    [style.transition]=\"model.options.isMobileDevice ? 'margin 300ms ease, opacity 300ms ease' : null\"\n    [style.zIndex]=\"model.options.isMobileDevice ? (model.options.toolbar.autoHide ? 5 : 2) : null\"\n    [style.opacity]=\"model.options.isMobileDevice ? (model.controls.navigatePanel.visible ? (model.options.toolbar.autoHide ? 0.9 : 1) : 0) : 1\"\n    [style.marginBottom]=\"model.options.isMobileDevice && !model.controls.navigatePanel.visible ? '-0.55in' : 0\">\n    <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n      <tbody>\n        <tr class=\"stiJsViewerClearAllStyles\">\n          <td *ngFor=\"let comp of comps\" class=\"stiJsViewerClearAllStyles\" [style.width]=\"comp.type == 'space' ? '100%' : null\">\n            <sti-button *ngIf=\"comp.type=='button'\"\n              [caption]=\"comp.caption\"\n              [imageName]=\"comp.img\"\n              [arrow]=\"comp.arrow\"\n              [menuItems]=\"comp.menuItems\"\n              [actionName]=\"comp.action\"\n              [tooltip]=\"comp.tooltip\"\n              [styleName]=\"'stiJsViewerNavigateButton'\"\n              [height]=\"model.options.isMobileDevice ? '0.5in' : '35px'\"\n              [width]=\"model.options.isMobileDevice ? '0.4in' : '35px'\"\n              [boxSizing]=\"'border-box'\"\n              [arrowMarginTop]=\"'1px'\"\n              [enabled]=\"enabled(comp)\"\n              [imageCellTextAlign]=\"'center'\"\n              [innerTableWidth]=\"'100%'\"\n              [navagationPanelTooltip]=\"true\"\n              [margin]=\"comp.action == 'FirstPage' ? '0 1px 0 3px' : (comp.action == 'Zoom' ? '0 3px 0 1px' : '0px 1px 0 1px')\"\n              [selected]=\"selected(comp)\"\n              (action)=\"controller.action({ name: comp.action })\">\n            </sti-button>\n\n            <div *ngIf=\"comp.type=='separator'\"\n              [style.height]=\"model.options.isMobileDevice ? '0.5in' : '35px'\"\n              [style.margin]=\"'0px 1px 0 1px'\"\n              class=\"stiJsViewerNavigatePanelSeparator\"></div>\n\n            <sti-page-control *ngIf=\"comp.type=='pageControl'\"\n              [enabled]=\"!(model.reportParams.pagesCount <= 1 || disableNaviButtons())\"\n              [count]=\"model.reportParams.pagesCount\"\n              [margin]=\"'0px 1px 0 1px'\"\n              [textBoxBorder]=\"'0px'\"></sti-page-control>\n          </td>\n        </tr>\n      </tbody>\n    </table>\n    <div class=\"stiJsViewerNavigatePanelDisabledPanel\" [style.display]=\"this.model.controls.navigatePanel.enabled ? 'none' : 'block'\"></div>\n  </div>\n  "
        })
    ], NavigatePanelComponent);
    return NavigatePanelComponent;
}());
export { NavigatePanelComponent };
