// stimulsoft-viewer-angular/esm5/lib/components/open-dialog.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ControllerService } from '../services/controller.service';
var OpenDialogComponent = /** @class */ (function () {
    function OpenDialogComponent(model, controller) {
        this.model = model;
        this.controller = controller;
    }
    OpenDialogComponent.prototype.ngOnChanges = function (changes) {
        var _this = this;
        if (this.fileMask != null && this.model.openDialogFileMask != null) {
            setTimeout(function () {
                _this.element.nativeElement.focus();
                _this.element.nativeElement.click();
            });
        }
    };
    OpenDialogComponent.prototype.ngOnInit = function () { };
    OpenDialogComponent.prototype.onchange = function (event) {
        var _this = this;
        var files = event.target.files;
        var fileName = files[0] ? files[0].name : 'Report';
        var filePath = event.target.value;
        var reader = new FileReader();
        reader.onload = function (e) {
            _this.controller.loadFile(fileName, e.target.result);
        };
        this.model.openDialogFileMask = null;
        reader.readAsDataURL(files[0]);
    };
    OpenDialogComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService }
    ]; };
    __decorate([
        ViewChild('element')
    ], OpenDialogComponent.prototype, "element", void 0);
    __decorate([
        Input()
    ], OpenDialogComponent.prototype, "fileMask", void 0);
    OpenDialogComponent = __decorate([
        Component({
            selector: 'sti-open-dialog',
            template: "\n    <input #element *ngIf=\"model.openDialogFileMask != null\"  type=\"file\" name=\"files[]\" multiple=\"\" style=\"display: none;\"\n      [attr.accept]=\"model.openDialogFileMask\"\n      (change)=\"onchange($event)\">\n  "
        })
    ], OpenDialogComponent);
    return OpenDialogComponent;
}());
export { OpenDialogComponent };
