// stimulsoft-viewer-angular/esm5/lib/components/report-panel.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ControllerService } from '../services/controller.service';
import { ModelService } from '../services/model.service';
import { StylesService } from '../services/styles.service';
import { InteractionsService } from '../services/interactions.service';
var ReportPanelComponent = /** @class */ (function () {
    function ReportPanelComponent(controller, model, stylesService, interactionService) {
        var _this = this;
        this.controller = controller;
        this.model = model;
        this.stylesService = stylesService;
        this.interactionService = interactionService;
        this.touchZoomFirstDistance = 0;
        this.touchZoomSecondDistance = 0;
        this.touchZoomValue = 0;
        this.maxHeights = {};
        controller.getMessage().subscribe(function (message) {
            if (message.action !== 'viewer_loaded' && _this.model.reportParams.pagesArray) {
                if (_this.model.reportParams.repaintOnlyDashboardContent) {
                    // jsObject.controls.reportPanel.repaintDashboardContent(parameters);
                }
                else {
                    _this.loadPages();
                }
            }
            if (message.action === 'GetReport' || message.action === 'OpenReport') {
                setTimeout(function () {
                    if (!_this.model.options.isParametersReceived && ((_this.model.reportParams.type === 'Report' && _this.model.options.toolbar.showParametersButton) || _this.model.reportParams.type === 'Dashboard')) {
                        interactionService.postInteraction({ action: 'InitVars' });
                    }
                    if (_this.model.reportParams.autoZoom) {
                        _this.controller.action({ name: _this.model.reportParams.autoZoom === -1 ? 'ZoomPageWidth' : 'ZoomOnePage' });
                        _this.model.reportParams.autoZoom = null;
                    }
                });
            }
        });
    }
    ReportPanelComponent.prototype.ngOnInit = function () { };
    ReportPanelComponent.prototype.ngAfterViewInit = function () {
        this.model.controls.reportPanel.el = this.element;
    };
    ReportPanelComponent.prototype.loadPages = function () {
        var _this = this;
        this.maxHeights = {};
        var pagesArray = this.model.reportParams.pagesArray;
        this.stylesService.setupStyle(pagesArray[pagesArray.length - 2], 'page');
        var chartScript = pagesArray[pagesArray.length - 1];
        this.model.pages = pagesArray.slice(0, pagesArray.length - 2);
        setTimeout(function () {
            _this.model.pages.forEach(function (page) { return _this.interactionService.initializeInteractions(page.page); });
            _this.interactionService.updateAllHyperLinks();
            _this.stylesService.addChartScript(chartScript);
        });
    };
    Object.defineProperty(ReportPanelComponent.prototype, "position", {
        get: function () {
            return this.model.options.heightType !== 'Percentage' || this.model.options.appearance.scrollbarsMode ? 'absolute' : 'relative';
        },
        enumerable: true,
        configurable: true
    });
    ReportPanelComponent.prototype.scroll = function () {
        var _this = this;
        if (this.model.pagesNavigationIsActive()) {
            clearTimeout(this.scrollTimeout);
            // update current page number
            this.scrollTimeout = setTimeout(function () {
                var commonPagesHeight = 0;
                var index = 0;
                for (index = 0; index < _this.model.pages.length; index++) {
                    commonPagesHeight += _this.model.pages[index].page.offsetHeight + 10;
                    if (commonPagesHeight > _this.element.nativeElement.scrollTop) {
                        break;
                    }
                }
                if (index < _this.model.reportParams.pagesCount && index >= 0 && index !== _this.model.reportParams.pageNumber) {
                    _this.model.reportParams.pageNumber = index;
                }
            }, 300);
        }
    };
    ReportPanelComponent.prototype.eventTouchStart = function (e) {
        this.touchesLength++;
        this.touchStartX = parseInt(e.changedTouches[0].clientX, 10);
        if (this.model.options.appearance.allowTouchZoom && this.touchesLength === 1) {
            this.touchZoomFirstDistance = 0;
            this.touchZoomSecondDistance = 0;
            this.touchZoomValue = 0;
        }
    };
    ReportPanelComponent.prototype.eventTouchMove = function (e) {
        if (this.model.options.appearance.allowTouchZoom && e.touches.length > 1) {
            if ('preventDefault' in e) {
                e.preventDefault();
            }
            this.touchZoomSecondDistance = Math.sqrt(Math.pow(e.touches[0].pageX - e.touches[1].pageX, 2) + Math.pow(e.touches[0].pageY - e.touches[1].pageY, 2));
            if (this.touchZoomFirstDistance === 0) {
                this.touchZoomFirstDistance = Math.sqrt(Math.pow(e.touches[0].pageX - e.touches[1].pageX, 2) + Math.pow(e.touches[0].pageY - e.touches[1].pageY, 2));
            }
            var touchZoomOffset = Math.trunc((this.touchZoomSecondDistance - this.touchZoomFirstDistance) / 2.5);
            if (Math.abs(touchZoomOffset) >= 5) {
                this.touchZoomValue = Math.trunc((this.model.reportParams.zoom + touchZoomOffset) / 5) * 5;
                this.touchZoomValue = Math.min(Math.max(this.touchZoomValue, 20), 200);
                this.controller.actionSubject.next({ action: 'centerText', data: this.touchZoomValue.toString() });
            }
        }
    };
    ReportPanelComponent.prototype.eventTouchEnd = function (e) {
        if (this.touchesLength > 0) {
            this.touchesLength--;
        }
        if (this.model.options.isMobileDevice && this.model.options.toolbar.autoHide) {
            if (Math.trunc(this.touchStartX - e.changedTouches[0].clientX) !== 0) {
                this.controller.keepToolbar();
            }
            else {
                if (!this.model.controls.toolbar.visible) {
                    this.controller.showToolbar();
                }
                else {
                    this.controller.hideToolbar();
                }
            }
        }
        if (this.model.options.appearance.allowTouchZoom && this.touchZoomValue !== 0 && this.touchesLength === 0) {
            this.controller.actionSubject.next({ action: 'hideCenterText' });
            this.model.reportParams.zoom = this.touchZoomValue;
            this.controller.post('GetPages');
        }
    };
    ReportPanelComponent.ctorParameters = function () { return [
        { type: ControllerService },
        { type: ModelService },
        { type: StylesService },
        { type: InteractionsService }
    ]; };
    __decorate([
        ViewChild('element')
    ], ReportPanelComponent.prototype, "element", void 0);
    ReportPanelComponent = __decorate([
        Component({
            selector: 'sti-report-panel',
            template: "\n    <div #element\n         [style.textAlign]=\"model.options.appearance.pageAlignment == 'default' ? 'center' : model.options.appearance.pageAlignment\"\n         [style.top]=\"0\"\n         [style.bottom]=\"0\"\n         [style.position]=\"position\"\n         [style.height]=\"model.options.heightType != 'Percentage' || model.options.appearance.scrollbarsMode ? 'auto' : 'calc(100% - 35px)'\"\n         [style.overflow] = \"model.reportParams.type == 'Report' && model.options.appearance.scrollbarsMode ? 'auto' : 'hidden'\"\n         [style.marginTop.px]=\"model.controls.reportPanel.layout.top\"\n         [style.marginLeft.px]=\"model.controls.reportPanel.layout.left\"\n         [style.marginBottom.px]=\"position == 'absolute' ? model.controls.reportPanel.layout.bottom : 0\"\n         [style.paddingBottom.px]=\"position == 'relative' ? model.controls.reportPanel.layout.bottom : 0\"\n         [style.transition]=\"model.options.isMobileDevice ? 'margin 200ms ease' : null\"\n         (scroll)=\"scroll()\"\n         class=\"stiJsViewerReportPanel\"\n         (touchstart)=\"eventTouchStart($event)\"\n         (touchmove)=\"eventTouchMove($event)\"\n         (touchend)=\"eventTouchEnd($event)\">\n      <sti-page *ngFor=\"let page of model.pages\" [pageAttributes]=\"page\" [reportPanel]=\"this\"></sti-page>\n    </div>\n  "
        })
    ], ReportPanelComponent);
    return ReportPanelComponent;
}());
export { ReportPanelComponent };
