// stimulsoft-viewer-angular/esm5/lib/components/resources-panel.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ControllerService } from '../services/controller.service';
import { HelperService } from '../services/helper.service';
var ResourcesPanelComponent = /** @class */ (function () {
    function ResourcesPanelComponent(model, controller, helper) {
        var _this = this;
        this.model = model;
        this.controller = controller;
        this.helper = helper;
        controller.getMessage().subscribe(function (message) {
            var _a;
            if (message.action === 'GetReport' || message.action === 'OpenReport') {
                _this.model.controls.resourcesPanel.visible = ((_a = _this.model.reportParams.resources) === null || _a === void 0 ? void 0 : _a.length) > 0;
            }
        });
    }
    ResourcesPanelComponent.prototype.ngOnInit = function () { };
    ResourcesPanelComponent.prototype.ngAfterViewInit = function () {
        this.model.controls.resourcesPanel.el = this.element;
    };
    ResourcesPanelComponent.prototype.action = function (resource) {
        var resTypesAllowedViewInBrowser = ['Image', 'Pdf', 'Txt'];
        var viewType = resTypesAllowedViewInBrowser.indexOf(resource.type) >= 0 ? 'View' : 'SaveFile';
        this.controller.postReportResource(resource.name, this.model.options.jsMode ? 'SaveFile' : viewType);
    };
    ResourcesPanelComponent.prototype.getResourceImage = function (resourceType) {
        if (this.model.img('BigResource' + resourceType + '.png') !== '') {
            return 'BigResource' + resourceType + '.png';
        }
        else {
            return 'BigResource.png';
        }
    };
    ResourcesPanelComponent.prototype.getMenuItems = function (resource) {
        var items = [];
        items.push({ name: 'Resource' + 'View', caption: this.model.loc('ButtonView'), value: resource });
        items.push({ name: 'Resource' + 'SaveFile', caption: this.model.loc('SaveFile'), value: resource });
        return items;
    };
    ResourcesPanelComponent.prototype.getActionName = function (resource) {
        if (!resource.id) {
            resource.id = Math.random().toString();
        }
        return 'resource' + resource.name + resource.id;
    };
    ResourcesPanelComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: HelperService }
    ]; };
    __decorate([
        ViewChild('element')
    ], ResourcesPanelComponent.prototype, "element", void 0);
    ResourcesPanelComponent = __decorate([
        Component({
            selector: 'sti-resources-panel',
            template: "\n    <div #element style=\"z-index: 3\"\n      [style.display]=\"this.model.controls.resourcesPanel.visible ? '' : 'none'\"\n      [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n      [style.fontColor]=\"model.options.toolbar.fontColor\"\n      [class]=\"'stiJsViewerToolBar' + (model.options.toolbar.displayMode == 'Separated' ? ' stiJsViewerToolBarSeparated' : '')\">\n      <div [style.paddingTop]=\"model.options.toolbar.displayMode == 'Simple' ? '2px' : ''\">\n        <div class=\"stiJsViewerToolBarTable\" style=\"box-sizing: border-box; display: table;\"\n          [style.border]=\"model.options.toolbar.displayMode == 'Separated' ? '0px' : ''\">\n          <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n            <tbody>\n              <tr class=\"stiJsViewerClearAllStyles\">\n                <td *ngFor=\"let resource of model.reportParams.resources\" class=\"stiJsViewerClearAllStyles\">\n                  <sti-button [caption]=\"resource.name\"\n                    [caption2]=\"helper.getHumanFileSize(resource.size, 1)\"\n                    [imageName]=\"getResourceImage(resource.type)\"\n                    [styleName]=\"'stiJsViewerFormButton'\"\n                    [height]=\"'auto'\"\n                    [margin]=\"'3px 0 3px 3px'\"\n                    [innerTableWidth]=\"'100%'\"\n                    [minWidth]=\"'80px'\"\n                    [captionAlign]=\"'left'\"\n                    [captionPaddingLeft]=\"'3px'\"\n                    [captionMaxWidth]=\"'150px'\"\n                    [captionLineHeight]=\"'14px'\"\n                    [captionWhiteSpace]=\"'nowrap'\"\n                    [captionOverflow]=\"'hidden'\"\n                    [captionTextOverflow]=\"'ellipsis'\"\n                    [imageCellWidth]=\"'1px'\"\n                    [imageCellPadding]=\"'4px 8px 4px 4px'\"\n                    [menuItems]=\"getMenuItems(resource)\"\n                    [resourceButton]=\"true\"\n                    [imageSizesWidth]=\"32\"\n                    [imageSizesHeight]=\"32\"\n                    [actionName]=\"getActionName(resource)\"\n                    (action)=\"action(resource)\">\n                  </sti-button>\n                </td>\n              </tr>\n            </tbody>\n          </table>\n        </div>\n      </div>\n    </div>\n  "
        })
    ], ResourcesPanelComponent);
    return ResourcesPanelComponent;
}());
export { ResourcesPanelComponent };
