// stimulsoft-viewer-angular/esm5/lib/components/toolbar-dop.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ToolbarService } from '../services/toolbar.service';
import { ControllerService } from '../services/controller.service';
var ToolbarDopComponent = /** @class */ (function () {
    function ToolbarDopComponent(model, controller, toolbarService) {
        this.model = model;
        this.controller = controller;
        this.toolbarService = toolbarService;
        this.marginLeft = '1px';
        this.viewInit = false;
    }
    ToolbarDopComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        this.viewInit = true;
        setTimeout(function () {
            var _a, _b;
            var width = (_b = (_a = _this.mainTable) === null || _a === void 0 ? void 0 : _a.nativeElement) === null || _b === void 0 ? void 0 : _b.offsetWidth;
            _this.marginLeft = _this.viewInit && _this.model.options.toolbar.alignment === 'Center' && width ? "calc(50% - " + Math.trunc(width / 2) + "px)" : '1px';
        });
    };
    ToolbarDopComponent.prototype.ngOnInit = function () { };
    ToolbarDopComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: ToolbarService }
    ]; };
    __decorate([
        ViewChild('mainTable')
    ], ToolbarDopComponent.prototype, "mainTable", void 0);
    __decorate([
        Input()
    ], ToolbarDopComponent.prototype, "comps", void 0);
    __decorate([
        Input()
    ], ToolbarDopComponent.prototype, "align", void 0);
    ToolbarDopComponent = __decorate([
        Component({
            selector: 'sti-toolbar-dop',
            template: "\n    <table #mainTable\n      class=\"stiJsViewerClearAllStyles\"\n      cellpadding=\"0\"\n      cellspacing=\"0\"\n      style=\"margin: 1px;\"\n      [attr.align]=\"align\"\n      [style.marginLeft]=\"marginLeft\">\n      <tbody>\n        <tr class=\"stiJsViewerClearAllStyles\">\n          <td *ngFor=\"let comp of comps\" class=\"stiJsViewerClearAllStyles\">\n            <sti-button *ngIf=\"comp.type=='button'\"\n              [caption]=\"model.options.toolbar.showButtonCaptions ? comp.caption : null\"\n              [imageName]=\"comp.img\"\n              [arrow]=\"!model.options.isMobileDevice ? comp.arrow : null\"\n              [menuItems]=\"comp.menuItems\"\n              [actionName]=\"comp.action\"\n              [tooltip]=\"comp.tooltip\"\n              [enabled]=\"toolbarService.enabled(comp)\"\n              [display]=\"toolbarService.display(comp)\"\n              [innerTableWidth]=\"toolbarService.getInnerTableWidth(comp)\"\n              [width]=\"toolbarService.getButonWidth(comp)\"\n              [height]=\"model.options.toolbar.displayMode == 'Separated' ? (model.options.isMobileDevice ? '0.5in' : '28px') : null\"\n              (action)=\"controller.action({ name: comp.action });\"\n              [selected]=\"toolbarService.selected(comp)\"></sti-button>\n            <sti-toolbar-separator *ngIf=\"comp.type=='separator'\"></sti-toolbar-separator>\n            <sti-page-control *ngIf=\"comp.type=='pageControl'\" [enabled]=\"!(model.reportParams.pagesCount <= 1 || toolbarService.disableNaviButtons())\" [count]=\"model.reportParams.pagesCount\" ></sti-page-control>\n        </td>\n      </tr>\n    </tbody>\n  </table>\n  "
        })
    ], ToolbarDopComponent);
    return ToolbarDopComponent;
}());
export { ToolbarDopComponent };
