// stimulsoft-viewer-angular/esm2015/lib/controls/parameters/parameter-text-box.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild } from '@angular/core';
import { HelperService } from '../../services/helper.service';
import { ModelService } from '../../services/model.service';
let ParameterTextBoxComponent = class ParameterTextBoxComponent {
    constructor(helper, model) {
        this.helper = helper;
        this.model = model;
        this.focusOnCreate = false;
        this.isMenu = false;
        this.value = '';
        this.secondTextBox = false;
    }
    ngOnInit() { }
    onchange(target) {
        if (this.element.oldValue === target.value) {
            return;
        }
        const value = this.getValue(target);
        if (this.isMenu) {
            if (!this.secondTextBox) {
                this.item.key = value;
            }
            else {
                this.item.keyTo = value;
            }
        }
        else {
            if (!this.secondTextBox) {
                if (this.variable.basicType === 'Range' || this.variable.type === 'DateTime') {
                    this.variable.key = value;
                }
                else {
                    this.variable.value = value;
                }
            }
            else {
                this.variable.keyTo = value;
            }
        }
    }
    getValue(target) {
        if (this.variable.type === 'DateTime') {
            if (this.element.oldValue === target.value) {
                return;
            }
            try {
                const timeString = new Date().toLocaleTimeString();
                const isAmericanFormat = timeString.toLowerCase().indexOf('am') >= 0 || timeString.toLowerCase().indexOf('pm') >= 0;
                const formatDate = isAmericanFormat ? 'MM/dd/yyyy' : 'dd.MM.yyyy';
                let format = formatDate + (isAmericanFormat ? ' h:mm:ss tt' : ' hh:mm:ss');
                if (this.variable.dateTimeType === 'Date') {
                    format = formatDate;
                }
                if (this.variable.dateTimeType === 'Time') {
                    format = 'hh:mm:ss';
                }
                const date = this.helper.getDateTimeFromString(target.value, this.model.options.appearance.parametersPanelDateFormat || format);
                return this.helper.getDateTimeObject(date);
            }
            catch (e) {
                alert(e);
            }
        }
        else {
            return target.value;
        }
    }
    get width() {
        let width = 210;
        if (this.variable.basicType === 'Range') {
            width = 140;
            if (this.variable.type === 'Guid' || this.variable.type === 'String') {
                width = 190;
            }
            if (this.variable.type === 'DateTime') {
                width = 235;
            }
            if (this.variable.type === 'Char') {
                width = 60;
            }
        }
        else {
            if (this.variable.type === 'Guid') {
                width = 265;
            }
            else {
                width = 210;
            }
        }
        return width;
    }
    get color() {
        return !this.variable.isNull
            ? (this.model.options.toolbar.fontColor && this.model.options.toolbar.fontColor !== 'Empty' ? this.model.options.toolbar.fontColor : '#444444')
            : this.variable.type === 'DateTime' ? 'transparent' : '#c6c6c6';
    }
};
ParameterTextBoxComponent.ctorParameters = () => [
    { type: HelperService },
    { type: ModelService }
];
__decorate([
    ViewChild('element')
], ParameterTextBoxComponent.prototype, "element", void 0);
__decorate([
    Input()
], ParameterTextBoxComponent.prototype, "item", void 0);
__decorate([
    Input()
], ParameterTextBoxComponent.prototype, "variable", void 0);
__decorate([
    Input()
], ParameterTextBoxComponent.prototype, "readOnly", void 0);
__decorate([
    Input()
], ParameterTextBoxComponent.prototype, "focusOnCreate", void 0);
__decorate([
    Input()
], ParameterTextBoxComponent.prototype, "isMenu", void 0);
__decorate([
    Input()
], ParameterTextBoxComponent.prototype, "value", void 0);
__decorate([
    Input()
], ParameterTextBoxComponent.prototype, "secondTextBox", void 0);
ParameterTextBoxComponent = __decorate([
    Component({
        selector: 'sti-parameter-text-box',
        template: `
    <sti-text-box #element
      [maxLength]="variable.type === 'Char' ? 1 : ''"
      [value]="isMenu ? helper.getStringKey(item.key, variable): value"
      [width]="width"
      [readOnly]="readOnly"
      [focusOnCreate]="focusOnCreate"
      [enabled]="!variable?.isNull"
      [color]="color"
      (onchange)="onchange($event)">
    </sti-text-box>
  `
    })
], ParameterTextBoxComponent);
export { ParameterTextBoxComponent };
