// stimulsoft-viewer-angular/esm2015/lib/controls/parameters/parameter-checkbox.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
let ParameterCheckboxComponent = class ParameterCheckboxComponent {
    constructor() {
        this.isEnabled = true;
        this.isMenuParameter = false;
        this.nullable = false;
        this.action = new EventEmitter();
    }
    ngOnInit() { }
    onAction(checked) {
        this.action.emit(checked);
    }
    get checked() {
        if (this.nullable) {
            return this.params.isNull;
        }
        else {
            return this.isMenuParameter ? this.params.isChecked : (typeof (this.params.value) === 'boolean' && this.params.value) || this.params.value === 'true' || this.params.value === 'True';
        }
    }
};
__decorate([
    Input()
], ParameterCheckboxComponent.prototype, "params", void 0);
__decorate([
    Input()
], ParameterCheckboxComponent.prototype, "captionText", void 0);
__decorate([
    Input()
], ParameterCheckboxComponent.prototype, "margin", void 0);
__decorate([
    Input()
], ParameterCheckboxComponent.prototype, "width", void 0);
__decorate([
    Input()
], ParameterCheckboxComponent.prototype, "imageBlockParentWidth", void 0);
__decorate([
    Input()
], ParameterCheckboxComponent.prototype, "isEnabled", void 0);
__decorate([
    Input()
], ParameterCheckboxComponent.prototype, "isMenuParameter", void 0);
__decorate([
    Input()
], ParameterCheckboxComponent.prototype, "nullable", void 0);
__decorate([
    Output()
], ParameterCheckboxComponent.prototype, "action", void 0);
ParameterCheckboxComponent = __decorate([
    Component({
        selector: 'sti-parameter-checkbox',
        template: `
    <sti-check-box [isEnabled]="isEnabled"
                   [isChecked]="checked"
                   [captionText]="captionText"
                   [margin]="margin"
                   [width]="width"
                   [imageBlockParentWidth]="imageBlockParentWidth"
                   (action)="onAction($event)">
    </sti-check-box>
    `
    })
], ParameterCheckboxComponent);
export { ParameterCheckboxComponent };
