// stimulsoft-viewer-angular/esm2015/lib/controls/parameters/parameter-button.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../../services/model.service';
let ParameterButtonComponent = class ParameterButtonComponent {
    constructor(model) {
        this.model = model;
        this.action = new EventEmitter();
    }
    ngOnInit() { }
    onaction() {
        this.action.emit(this.params);
    }
};
ParameterButtonComponent.ctorParameters = () => [
    { type: ModelService }
];
__decorate([
    Input()
], ParameterButtonComponent.prototype, "params", void 0);
__decorate([
    Input()
], ParameterButtonComponent.prototype, "buttonType", void 0);
__decorate([
    Output()
], ParameterButtonComponent.prototype, "action", void 0);
ParameterButtonComponent = __decorate([
    Component({
        selector: 'sti-parameter-button',
        template: `
    <sti-button [imageName]="buttonType + '.png'"
        [styleName]="'stiJsViewerFormButton'"
        [height]="model.options.isTouchDevice ? '26px' : '21px'"
        [innerTableWidth]="'100%'"
        [enabled]="!params?.isNull"
        (action)="onaction()">
    </sti-button>
  `
    })
], ParameterButtonComponent);
export { ParameterButtonComponent };
