// stimulsoft-viewer-angular/esm5/lib/forms/base-form.component.js
import { __decorate } from "tslib";
import { Component, Input, EventEmitter, Output, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { HelperService } from '../services/helper.service';
import { trigger, state, transition, style, animate } from '@angular/animations';
import { FormService } from './form.service';
var BaseFormComponent = /** @class */ (function () {
    function BaseFormComponent(model, helper, formService) {
        this.model = model;
        this.helper = helper;
        this.formService = formService;
        this.changeVisibility = new EventEmitter();
        this.action = new EventEmitter();
        this.level = 1;
        this.showCancel = true;
        this.showButtons = true;
        this.showSeparator = true;
        this.display = 'none';
    }
    BaseFormComponent.prototype.ngOnInit = function () { };
    BaseFormComponent.prototype.startAnimation = function (event) {
        if (event.toState === 'visible') {
            this.display = '';
            if (this.defaultTop != null) {
                this.formService.centerForm(this, this.defaultTop);
            }
        }
    };
    BaseFormComponent.prototype.doneAnimation = function (event) {
        if (event.toState === 'hidden') {
            this.display = 'none';
        }
        this.changeVisibility.emit(event.toState);
    };
    BaseFormComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService },
        { type: FormService }
    ]; };
    __decorate([
        ViewChild('element')
    ], BaseFormComponent.prototype, "element", void 0);
    __decorate([
        Output()
    ], BaseFormComponent.prototype, "changeVisibility", void 0);
    __decorate([
        Output()
    ], BaseFormComponent.prototype, "action", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "level", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "caption", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "helpUrl", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "fontFamily", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "color", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "fontSize", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "containerPadding", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "name", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "defaultTop", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "showCancel", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "showButtons", void 0);
    __decorate([
        Input()
    ], BaseFormComponent.prototype, "showSeparator", void 0);
    BaseFormComponent = __decorate([
        Component({
            selector: 'sti-base-form',
            template: "\n    <div #element class=\"stiJsViewerForm\"\n      [style.zIndex]=\"level * 10 + 1\"\n      [@.disabled]=\"!model.options.toolbar.menuAnimation\"\n      [style.fontFamily]=\"fontFamily\"\n      [style.color]=\"color\"\n      [style.fontSize]=\"fontSize\"\n      [style.left.px]=\"formService.form?.left\"\n      [style.top.px]=\"formService.form?.top\"\n      [style.display]=\"display\"\n      [@showForm]=\"formService.form?.name == name ? 'visible' : 'hidden'\"\n      (@showForm.start)=\"startAnimation($event)\"\n      (@showForm.done)=\"doneAnimation($event)\">\n      <div class=\"stiJsViewerFormHeader\"\n        (mousedown)=\"formService.startMove(name, $event)\"\n        (touchstart)=\"formService.startMove(name, null, $event)\"\n        (touchmove)=\"formService.move($event)\"\n        (touchend)=\"formService.stopMove()\">\n        <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"width: 100%;\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\" style=\"text-align: left; padding: 5px 10px 8px 15px;\">\n                {{caption}}\n              </td>\n              <td *ngIf=\"helpUrl != null && model.options.appearance.showDialogsHelp\" class=\"stiJsViewerClearAllStyles\" style=\"width: 20px; text-align: right; padding: 2px 0px 1px 0px; vertical-align: top;\">\n                <sti-button [imageMargin]=\"'0 2px 0 2px'\" [display]=\"'inline-block'\" (action)=\"helper.showHelpWindow(helpUrl)\">\n                </sti-button>\n              </td>\n              <td class=\"stiJsViewerClearAllStyles\" style=\"vertical-align: top; width: 30px; text-align: right; padding: 2px 1px 1px;\">\n                <sti-button [imageMargin]=\"'0 2px 0 2px'\" [display]=\"'inline-block'\" [imageName]=\"'CloseForm.png'\" (action)=\"formService.closeForm(name)\">\n                </sti-button>\n              </td>\n            </tr>\n          </tbody>\n        </table>\n      </div>\n\n      <div class=\"stiJsViewerFormContainer\" [style.padding]=\"containerPadding\">\n        <ng-content>\n        </ng-content>\n      </div>\n\n      <div *ngIf=\"showSeparator\" class=\"stiJsViewerFormSeparator\"></div>\n\n      <div *ngIf=\"showButtons\" class=\"stiJsViewerFormButtonsPanel\">\n        <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\" style=\"padding: 8px;\">\n                <sti-form-button [caption]=\"model.loc('ButtonOk')\" (action)=\"action.emit()\">\n                </sti-form-button>\n              </td>\n              <td *ngIf=\"showCancel\" class=\"stiJsViewerClearAllStyles\" style=\"padding: 8px 8px 8px 0px;\">\n                <sti-form-button [caption]=\"model.loc('ButtonCancel')\" (action)=\"formService.closeForm(name)\">\n                </sti-form-button>\n              </td>\n            </tr>\n          </tbody>\n        </table>\n      </div>\n\n    </div>\n  ",
            animations: [
                trigger('showForm', [
                    state('hidden', style({ opacity: '0' })),
                    state('visible', style({ opacity: '1' })),
                    transition('hidden => visible', [
                        animate('200ms ease-in-out')
                    ])
                ])
            ]
        })
    ], BaseFormComponent);
    return BaseFormComponent;
}());
export { BaseFormComponent };
