// stimulsoft-viewer-angular/esm5/lib/forms/error-message-form.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { ModelService } from '../services/model.service';
import { FormService } from './form.service';
var ErrorMessageFormComponent = /** @class */ (function () {
    function ErrorMessageFormComponent(model, formService) {
        this.model = model;
        this.formService = formService;
    }
    ErrorMessageFormComponent.prototype.ngOnInit = function () { };
    ErrorMessageFormComponent.prototype.getImage = function () {
        var _a;
        var messageType = (_a = this.model.errorMessage) === null || _a === void 0 ? void 0 : _a.type;
        if (messageType === 'Warning') {
            return this.model.img('MsgFormWarning.png');
        }
        else if (messageType === true || messageType === 'Info') {
            return this.model.img('MsgFormInfo.png');
        }
        else {
            return this.model.img('MsgFormError.png');
        }
    };
    ErrorMessageFormComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: FormService }
    ]; };
    ErrorMessageFormComponent = __decorate([
        Component({
            selector: 'sti-error-message-form',
            template: "\n    <sti-base-form [fontFamily]=\"model.options.toolbar.fontFamily\"\n      [color]=\"model.options.toolbar.fontColor\"\n      [name]=\"'errorMessageForm'\"\n      [caption]=\"model.loc(model.errorMessage?.type == 'Error' ? 'Error' : 'FormViewerTitle')\"\n      [level]=\"4\"\n      [defaultTop]=\"150\"\n      [showCancel]=\"false\"\n      (action)=\"formService.closeForm('errorMessageForm')\">\n      <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n        <tbody>\n          <tr class=\"stiJsViewerClearAllStyles\">\n            <td class=\"stiJsViewerClearAllStyles\">\n              <img [attr.src]=\"getImage()\" style=\"padding: 15px; height: 32px; width: 32px;\">\n            </td>\n            <td class=\"stiJsViewerMessagesFormDescription\" style=\"max-width: 600px;\"\n              [style.color]=\"model.options.toolbar.fontColor\" [innerHTML]=\"model.errorMessage?.error\">\n            </td>\n          </tr>\n        </tbody>\n      </table>\n    </sti-base-form>\n  "
        })
    ], ErrorMessageFormComponent);
    return ErrorMessageFormComponent;
}());
export { ErrorMessageFormComponent };
