// stimulsoft-viewer-angular/esm5/lib/forms/form.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { MouseService } from '../services/mouse.service';
import { MenuService } from '../menu/menu.service';
import { HelperService } from '../services/helper.service';
import { ModelService } from '../services/model.service';
var FormService = /** @class */ (function () {
    function FormService(mouseService, menuService, helper, model) {
        var _this = this;
        this.mouseService = mouseService;
        this.menuService = menuService;
        this.helper = helper;
        this.model = model;
        this.mouseX = 0;
        this.mouseY = 0;
        this.formX = 0;
        this.formY = 0;
        mouseService.getDocumentMouseMove().subscribe(function (event) {
            _this.move(event);
        });
        mouseService.getDocumentMouseUp().subscribe(function (event) {
            _this.stopMove();
        });
    }
    Object.defineProperty(FormService.prototype, "form", {
        get: function () {
            return this.model.form;
        },
        set: function (form) {
            this.model.form = form;
        },
        enumerable: true,
        configurable: true
    });
    FormService.prototype.showForm = function (name, formData) {
        this.form = { name: name, left: 0, top: 0, isMooving: false, formData: formData };
    };
    FormService.prototype.closeForm = function (name) {
        var _a, _b;
        if (((_a = this.form) === null || _a === void 0 ? void 0 : _a.name) === 'notificationForm' && ((_b = this.model.notificationFormOptions) === null || _b === void 0 ? void 0 : _b.action)) {
            this.model.notificationFormOptions.action();
        }
        this.form = null;
    };
    FormService.prototype.centerForm = function (baseForm, defaultTop) {
        var _this = this;
        if (this.form != null) {
            setTimeout(function () {
                var position = _this.helper.setObjectToCenter(baseForm.element.nativeElement, defaultTop);
                _this.form.left = position.left;
                _this.form.top = position.top;
                _this.form.level = baseForm.level;
            });
        }
    };
    FormService.prototype.startMove = function (name, event, touchEvent) {
        var _a, _b;
        if (event || touchEvent && touchEvent.changedTouches.length > 0) {
            this.formX = this.form.left;
            this.formY = this.form.top;
            this.mouseX = ((_a = event) === null || _a === void 0 ? void 0 : _a.screenX) || touchEvent.changedTouches[0].screenX;
            this.mouseY = ((_b = event) === null || _b === void 0 ? void 0 : _b.screenY) || touchEvent.changedTouches[0].screenY;
            this.form.isMooving = true;
        }
    };
    FormService.prototype.move = function (event) {
        var _a;
        if ((_a = this.form) === null || _a === void 0 ? void 0 : _a.isMooving) {
            var screenX_1 = event.screenX || (event.changedTouches.length > 0 ? event.changedTouches[0].screenX : -1);
            var screenY_1 = event.screenY || (event.changedTouches.length > 0 ? event.changedTouches[0].screenY : -1);
            this.form.left = this.formX + (screenX_1 - this.mouseX);
            this.form.top = this.formY + (screenY_1 - this.mouseY);
            this.menuService.closeAllMenus();
        }
    };
    FormService.prototype.stopMove = function () {
        var _a;
        if ((_a = this.form) === null || _a === void 0 ? void 0 : _a.isMooving) {
            this.form.isMooving = false;
        }
    };
    FormService.ctorParameters = function () { return [
        { type: MouseService },
        { type: MenuService },
        { type: HelperService },
        { type: ModelService }
    ]; };
    FormService = __decorate([
        Injectable()
    ], FormService);
    return FormService;
}());
export { FormService };
