// stimulsoft-viewer-angular/esm5/lib/forms/group-panel.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
import { trigger, state, transition, animate, style } from '@angular/animations';
var GroupPanelComponent = /** @class */ (function () {
    function GroupPanelComponent(model) {
        this.model = model;
        this.opened = false;
        this.action = new EventEmitter();
        this.display = 'none';
    }
    GroupPanelComponent.prototype.ngOnInit = function () { };
    GroupPanelComponent.prototype.startAnimation = function (event) {
        if (event.toState === 'opened') {
            this.display = '';
        }
    };
    GroupPanelComponent.prototype.doneAnimation = function (event) {
        if (event.toState === 'closed') {
            this.display = 'none';
        }
    };
    GroupPanelComponent.ctorParameters = function () { return [
        { type: ModelService }
    ]; };
    __decorate([
        Input()
    ], GroupPanelComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], GroupPanelComponent.prototype, "caption", void 0);
    __decorate([
        Input()
    ], GroupPanelComponent.prototype, "innerPadding", void 0);
    __decorate([
        Input()
    ], GroupPanelComponent.prototype, "opened", void 0);
    __decorate([
        Input()
    ], GroupPanelComponent.prototype, "margin", void 0);
    __decorate([
        Output()
    ], GroupPanelComponent.prototype, "action", void 0);
    GroupPanelComponent = __decorate([
        Component({
            selector: 'sti-group-panel',
            template: "\n  <div #groupPanel [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n       [style.color]=\"model.options.toolbarFontColor\"\n       [style.minWidth.px]=\"width\"\n       [style.margin]=\"margin\"\n       style=\"overflow: hidden\">\n      <sti-form-button [caption]=\"caption\"\n          [imageName]=\"opened ? 'Arrows.BigArrowDown.png' : 'Arrows.BigArrowRight.png'\"\n          [imageCellWidth]=\"'1px'\"\n          [captionPadding]=\"'0 15px 0 5px'\"\n          [margin]=\"'0px'\"\n          [captionAlign]=\"'left'\"\n          (action)=\"opened = !opened; action.emit(opened)\">\n      </sti-form-button>\n\n      <div class=\"stiJsViewerGroupPanelContainer\"\n           [style.padding]=\"innerPadding\"\n           [style.display]=\"display\"\n           [@state]=\"opened ? 'opened' : 'closed'\"\n           (@state.start)=\"startAnimation($event)\"\n           (@state.done)=\"doneAnimation($event)\">\n          <ng-content>\n\n          </ng-content>\n      </div>\n\n\n  </div>\n  ",
            animations: [
                trigger('state', [
                    state('opened', style({ height: '*' })),
                    state('closed', style({ height: '0px' })),
                    transition('* => *', [
                        animate('150ms ease-in-out')
                    ])
                ])
            ]
        })
    ], GroupPanelComponent);
    return GroupPanelComponent;
}());
export { GroupPanelComponent };
