// stimulsoft-viewer-angular/esm5/lib/forms/password-form.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ControllerService } from '../services/controller.service';
import { FormService } from './form.service';
var PasswordFormComponent = /** @class */ (function () {
    function PasswordFormComponent(model, controller, formService) {
        this.model = model;
        this.controller = controller;
        this.formService = formService;
    }
    PasswordFormComponent.prototype.ngOnInit = function () { };
    PasswordFormComponent.prototype.changeVisibility = function (state) {
        if (state === 'visible') {
            this.password = '';
            this.textBox.element.nativeElement.focus();
        }
    };
    PasswordFormComponent.prototype.action = function () {
        var data = this.formService.form.formData;
        data.openingFilePassword = this.password;
        this.formService.closeForm('passwordForm');
        this.controller.postOpen(data);
    };
    PasswordFormComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: FormService }
    ]; };
    __decorate([
        ViewChild('textBox')
    ], PasswordFormComponent.prototype, "textBox", void 0);
    PasswordFormComponent = __decorate([
        Component({
            selector: 'sti-password-form',
            template: "\n    <sti-base-form [fontFamily]=\"model.options.toolbar.fontFamily\"\n      [color]=\"model.options.toolbar.fontColor\"\n      [name]=\"'passwordForm'\"\n      [caption]=\"model.loc('PasswordSaveReport').replace(':', '')\"\n      [level]=\"2\"\n      [defaultTop]=\"150\"\n      (changeVisibility)=\"changeVisibility($event)\"\n      (action)=\"action()\">\n      <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"margin: 5px;\">\n        <tbody>\n          <tr class=\"stiJsViewerClearAllStyles\">\n            <td class=\"stiJsViewerCaptionControls\">\n              {{model.loc('PasswordEnter')}}\n            </td>\n            <td class=\"stiJsViewerCaptionControls\">\n              <sti-text-box #textBox [type]=\"'password'\" [width]=\"200\" [focusOnCreate]=\"true\" [value]=\"password\" (onchange)=\"password = $event.value\">\n              </sti-text-box>\n            </td>\n          </tr>\n        </tbody>\n      </table>\n    </sti-base-form>\n  "
        })
    ], PasswordFormComponent);
    return PasswordFormComponent;
}());
export { PasswordFormComponent };
