// stimulsoft-viewer-angular/esm5/lib/controls/button.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild, Output, EventEmitter } from '@angular/core';
import { ControllerService } from '../services/controller.service';
import { ModelService } from '../services/model.service';
import { MenuService, Menu } from '../menu/menu.service';
import { TooltipService } from '../services/tooltip.service';
import { HelperService } from '../services/helper.service';
var ButtonComponent = /** @class */ (function () {
    function ButtonComponent(model, controller, menuService, tooltipService, helper) {
        this.model = model;
        this.controller = controller;
        this.menuService = menuService;
        this.tooltipService = tooltipService;
        this.helper = helper;
        this.margin = '1px';
        this.height = '23px';
        this.selected = false;
        this.imageCellTextAlign = 'center';
        this.imageSizesWidth = 16;
        this.imageSizesHeight = 16;
        this.closeButton = false;
        this.resourceButton = false;
        this.navagationPanelTooltip = false;
        this.helpLink = 'user-manual/index.html?viewer_reports.htm';
        this.action = new EventEmitter();
        this.closeButtonAction = new EventEmitter();
        this.over = false;
        this.showMenu = false;
        this.closeButtonPressed = false;
        this.resourceButtonPressed = false;
        this._enabled = true;
        this.canShowTooltip = true;
    }
    ButtonComponent.prototype.ngOnInit = function () {
        var _this = this;
        if (this.menuItems) {
            setTimeout(function () {
                _this.menuObj = { type: 'buttonMenu', name: _this.actionName, items: _this.menuItems, parent: !_this.resourceButton ? _this.button : _this.resButtonEl };
                _this.menuService.addMenu(_this.menuObj);
            }, 500);
        }
    };
    Object.defineProperty(ButtonComponent.prototype, "styleName", {
        get: function () {
            return this._styleName || 'stiJsViewerStandartSmallButton';
        },
        set: function (value) {
            this._styleName = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ButtonComponent.prototype, "enabled", {
        get: function () {
            return this._enabled;
        },
        set: function (value) {
            if (!value) {
                this.over = false;
            }
            this._enabled = value;
        },
        enumerable: true,
        configurable: true
    });
    ButtonComponent.prototype.mouseover = function () {
        var _this = this;
        if (!this.enabled) {
            return;
        }
        this.over = true;
        if (this.model.options.toolbar.showMenuMode === 'Hover' && ['Print', 'Save', 'SendEmail', 'Zoom', 'ViewMode'].some(function (a) { return a === _this.actionName; })) {
            this.tooltipService.hideImmediately();
            this.menuService.showMenu(this.actionName);
        }
        else if (this.tooltip && !this.menuService.menus.some(function (e) { return e.state === 'expanded'; }) && this.canShowTooltip) {
            var top_1 = !this.navagationPanelTooltip ? this.model.controls.toolbar.offsetHeight + this.model.controls.dashboardsPanel.offsetHeight :
                this.helper.findPosY(this.model.controls.navigatePanel.el.nativeElement, 'stiJsViewerMainPanel');
            var tooltip = this.tooltip === true ? (this.model.localization[this.actionName + 'ToolTip'] != null ? this.model.loc(this.actionName + 'ToolTip') : this.model.loc(this.actionName)) : this.tooltip;
            this.tooltipService.show(this.helper.findPosX(this.button.nativeElement, 'stiJsViewerMainPanel'), top_1, tooltip, !this.navagationPanelTooltip, this.helpLink);
        }
    };
    ButtonComponent.prototype.mouseout = function () {
        var _this = this;
        if (!this.enabled) {
            return;
        }
        this.over = false;
        this.tooltipService.hide();
        setTimeout(function () {
            _this.canShowTooltip = true;
        }, 1000);
    };
    ButtonComponent.prototype.click = function () {
        if (this.enabled && !this.closeButtonPressed) {
            this.tooltipService.hideImmediately();
            if (this.menuItems == null || (this.resourceButton && !this.resourceButtonPressed)) {
                this.action.emit();
            }
            else {
                this.menuService.showMenu(this.actionName);
            }
        }
        this.canShowTooltip = false;
        this.closeButtonPressed = false;
        this.resourceButtonPressed = false;
    };
    Object.defineProperty(ButtonComponent.prototype, "className", {
        get: function () {
            var _a;
            return this.styleName ? this.styleName + ' ' + (this.enabled ? (this.styleName + ((this.selected || ((_a = this.menuObj) === null || _a === void 0 ? void 0 : _a.state) === 'expanded') ? 'Selected' : (this.over ? 'Over' : 'Default'))) : this.styleName + 'Disabled') : '';
        },
        enumerable: true,
        configurable: true
    });
    ButtonComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: ControllerService },
        { type: MenuService },
        { type: TooltipService },
        { type: HelperService }
    ]; };
    __decorate([
        Input()
    ], ButtonComponent.prototype, "caption", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "caption2", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionAlign", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionPadding", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionPaddingLeft", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionWhiteSpace", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionOverflow", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionTextOverflow", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionWidth", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionMaxWidth", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "captionLineHeight", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "imageName", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "arrow", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "arrowMarginTop", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "margin", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "height", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "selected", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "minWidth", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "innerTableWidth", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "menuItems", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "actionName", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "tooltip", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "imageCellTextAlign", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "imageCellWidth", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "imageCellPadding", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "imageSizesWidth", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "imageSizesHeight", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "imageMargin", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "display", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "closeButton", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "resourceButton", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "styleColors", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "boxSizing", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "navagationPanelTooltip", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "cursor", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "fontSize", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "helpLink", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "borderColor", void 0);
    __decorate([
        Output()
    ], ButtonComponent.prototype, "action", void 0);
    __decorate([
        Output()
    ], ButtonComponent.prototype, "closeButtonAction", void 0);
    __decorate([
        ViewChild('button')
    ], ButtonComponent.prototype, "button", void 0);
    __decorate([
        ViewChild('resButtonEl')
    ], ButtonComponent.prototype, "resButtonEl", void 0);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "styleName", null);
    __decorate([
        Input()
    ], ButtonComponent.prototype, "enabled", null);
    ButtonComponent = __decorate([
        Component({
            selector: 'sti-button',
            template: "\n  <div #button\n    [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n    [style.cursor]=\"cursor\"\n    [style.borderColor]=\"borderColor\"\n    [class]=\"className\"\n    [style]=\"{height: helper.val(height, '23px'), boxSizing: helper.val(boxSizing, 'content-box'), margin: helper.val(margin, '1px'), minWidth: minWidth, width: width, display: display }\"\n    [style.fontSize]=\"fontSize\"\n    (mouseover)=\"mouseover()\"\n    (mouseout)=\"mouseout()\"\n    (click)=\"click()\" >\n    <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"height: 100%;width:100%\" [style.width]=\"innerTableWidth\">\n      <tbody>\n        <tr class=\"stiJsViewerClearAllStyles\">\n          <td *ngIf=\"imageName != null\" [style]=\"{lineHeight:'0', padding:helper.val(imageCellPadding, '0 3px'), textAlign: imageCellTextAlign, width: imageCellWidth}\" class=\"stiJsViewerClearAllStyles\">\n            <img src=\"{{model.img(imageName)}}\" [style.opacity]=\"enabled ? '1' : '0.5'\"\n                [style.width.px]=\"imageSizesWidth\"\n                [style.height.px]=\"imageSizesHeight\"\n                [style.margin]=\"imageMargin\">\n          </td>\n          <td *ngIf=\"caption != null\"\n             [style]=\"{whiteSpace:'nowrap', textAlign: captionAlign || 'left', padding:captionPadding || ((arrow ? '1px 0' : '1px 5px') + (imageName ? ' 0 0' : ' 0 5px')), width: captionWidth}\"\n             [style.maxWidth]=\"captionMaxWidth\"\n             [style.lineHeight]=\"captionLineHeight\"\n             [style.paddingLeft]=\"captionPaddingLeft\"\n             [style.whiteSpace]=\"captionWhiteSpace\"\n             [style.overflow]=\"captionOverflow\"\n             [style.textOverflow]=\"captionTextOverflow\"\n             class=\"stiJsViewerClearAllStyles\">\n            {{caption}}\n            <div *ngIf=\"caption2 != null\">\n            {{caption2}}\n            </div>\n          </td>\n          <td *ngIf=\"arrow\" class=\"stiJsViewerClearAllStyles\">\n            <img src=\"{{arrow == 'Up' || arrow == 'Down' ? model.img('Arrows.SmallArrow' + arrow + (arrow == 'Down' && styleColors && styleColors.isDarkStyle ? 'White.png' : '.png')) : model.img(arrow)}}\"\n            [style]=\"{lineHeight:'0', width:'8px', height:'8px', verticalAlign:'baseline', padding:caption ? '0 5px 0 5px' : '0 5px 0 2px', marginTop: arrowMarginTop}\"\n            [style.opacity]=\"enabled ? '1' : '0.5'\"/>\n          </td>\n\n          <td *ngIf=\"closeButton\" class=\"stiJsViewerClearAllStyles\">\n            <sti-button [imageName]=\"'CloseForm.png'\"\n              [imageMargin]=\"'1px 0 0 -1px'\"\n              [margin]=\"'0 2px 0 0'\"\n              [imageCellPadding]=\"'0'\"\n              [width]=\"model.options.isTouchDevice ? '22px' : '17px'\"\n              [height]=\"model.options.isTouchDevice ? '22px' : '17px'\"\n              (action)=\"closeButtonPressed = true; closeButtonAction.emit()\">\n            </sti-button>\n          </td>\n\n          <td #resButtonEl *ngIf=\"resourceButton\" style=\"width: 1px;\">\n            <sti-button [imageName]=\"'Arrows.SmallArrowDown.png'\"\n              [styleName]=\"'stiJsViewerResourceDropDownButton'\"\n              [innerTableWidth]=\"'100%'\"\n              [margin]=\"'0 7px 0 3px'\"\n              [imageCellTextAlign]=\"'center'\"\n              [imageSizesWidth]=\"8\"\n              [imageSizesHeight]=\"8\"\n              [height]=\"model.options.isTouchDevice ? '23px' : '17px'\"\n              [width]=\"model.options.isTouchDevice ? '23px' : '17px'\"\n              (action)=\"resourceButtonPressed = true;\">>\n            </sti-button>\n          </td>\n\n        </tr>\n      </tbody>\n    </table>\n  </div>\n  "
        })
    ], ButtonComponent);
    return ButtonComponent;
}());
export { ButtonComponent };
