// stimulsoft-viewer-angular/esm5/lib/controls/checkbox.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
var CheckboxComponent = /** @class */ (function () {
    function CheckboxComponent(model) {
        this.model = model;
        this.isEnabled = true;
        this.isChecked = false;
        this.indeterminate = false;
        this.action = new EventEmitter();
        this.over = false;
    }
    CheckboxComponent.prototype.ngOnInit = function () { };
    CheckboxComponent.prototype.mouseover = function () {
        this.over = true;
    };
    CheckboxComponent.prototype.mouseleave = function () {
        this.over = false;
    };
    CheckboxComponent.prototype.click = function () {
        if (this.isEnabled) {
            this.isChecked = !this.isChecked;
            this.action.emit(this.isChecked);
        }
    };
    CheckboxComponent.ctorParameters = function () { return [
        { type: ModelService }
    ]; };
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "isEnabled", void 0);
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "isChecked", void 0);
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "indeterminate", void 0);
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "captionText", void 0);
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "tooltip", void 0);
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "margin", void 0);
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "styleColors", void 0);
    __decorate([
        Input()
    ], CheckboxComponent.prototype, "imageBlockParentWidth", void 0);
    __decorate([
        Output()
    ], CheckboxComponent.prototype, "action", void 0);
    CheckboxComponent = __decorate([
        Component({
            selector: 'sti-check-box',
            template: "\n    <table cellpadding=\"0\" cellspacing=\"0\"\n      [class]=\"isEnabled ? 'stiJsViewerCheckBox' : 'stiJsViewerCheckBoxDisabled'\"\n      [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n      [style.color]=\"model.options.toolbar.fontColor\"\n      [attr.title]=\"tooltip\"\n      [style.margin]=\"margin\"\n      [style.width]=\"width\"\n      style=\"box-sizing:content-box\"\n      (mouseover)=\"mouseover()\"\n      (mouseleave)=\"mouseleave()\"\n      (click)=\"click()\">\n       <tbody>\n          <tr class=\"stiJsViewerClearAllStyles\">\n             <td style=\"line-height: 0\"\n                 [style.padding]=\"model.options.isTouchDevice ? '1px 3px 1px 1px' : ''\"\n                 [style.width]=\"imageBlockParentWidth\"\n                 class=\"stiJsViewerClearAllStyles\">\n                <div [style.width.px]=\"model.options.isTouchDevice ? 16 : 13\"\n                     [style.height.px]=\"model.options.isTouchDevice ? 16 : 13\"\n                     [class]=\"isEnabled ? (over ? 'stiJsViewerCheckBoxImageBlockOver' : 'stiJsViewerCheckBoxImageBlock') : 'stiJsViewerCheckBoxImageBlockDisabled'\"\n                     style=\"box-sizing: content-box\">\n                     <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"width: 100%; height: 100%;\">\n                        <tbody>\n                          <tr class=\"stiJsViewerClearAllStyles\">\n                            <td [style.textAlign]=\"model.options.isTouchDevice ? 'center' : 'right'\"\n                                [style.verticalAlign]=\"model.options.isTouchDevice ? 'middle' : 'top'\">\n                                <img [src]=\"model.img((indeterminate ? 'CheckBoxIndeterminate' : 'CheckBox') + (styleColors && styleColors.isDarkStyle ? 'White.png' : '.png'))\"\n                                    [style.visibility]=\"isChecked ? 'visible' : 'hidden'\"\n                                    [style.width.px]=\"indeterminate ? 13 : 12\"\n                                    [style.height.px]=\"indeterminate ? 13 : 12\"\n                                    [style.opacity]=\"isEnabled ? 1 : 0.5\"\n                                    style=\"vertical-align: baseline\"/>\n                            </td>\n                          </tr>\n                        </tbody>\n                      </table>\n                </div>\n             </td>\n             <td *ngIf=\"captionText != null\"\n                 [style.padding]=\"!model.options.isTouchDevice ? '1px 0 0 4px' : ''\"\n                 style=\"white-space: nowrap\">\n                 {{captionText}}\n             </td>\n          </tr>\n       </tbody>\n    </table>\n  "
        })
    ], CheckboxComponent);
    return CheckboxComponent;
}());
export { CheckboxComponent };
