// stimulsoft-viewer-angular/esm5/lib/controls/dashboard-panel-button.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { ModelService } from '../services/model.service';
var DashboardButtonComponent = /** @class */ (function () {
    function DashboardButtonComponent(model) {
        this.model = model;
        this.showCloseButton = false;
    }
    DashboardButtonComponent.prototype.ngOnInit = function () { };
    DashboardButtonComponent.ctorParameters = function () { return [
        { type: ModelService }
    ]; };
    __decorate([
        Input()
    ], DashboardButtonComponent.prototype, "showCloseButton", void 0);
    __decorate([
        Input()
    ], DashboardButtonComponent.prototype, "info", void 0);
    __decorate([
        Input()
    ], DashboardButtonComponent.prototype, "dbsMode", void 0);
    __decorate([
        Input()
    ], DashboardButtonComponent.prototype, "display", void 0);
    DashboardButtonComponent = __decorate([
        Component({
            selector: 'sti-dashboard-button',
            template: "\n    <sti-button\n      [margin]=\"'2px 1px 2px 2px'\"\n      [display]=\"display\"\n      [borderColor]=\"!dbsMode ? 'transparent' : ''\"\n      [height]=\"model.options.toolbar.displayMode == 'Separated' ? (dbsMode ? (model.options.isTouchDevice ? '28px' : '23px') : '28px') : null\">\n    </sti-button>\n  "
        })
    ], DashboardButtonComponent);
    return DashboardButtonComponent;
}());
export { DashboardButtonComponent };
