// stimulsoft-viewer-angular/esm5/lib/controls/date-picker-day-button.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { ModelService } from '../services/model.service';
import { HelperService } from '../services/helper.service';
import { MenuService } from '../menu/menu.service';
var DatePickerDayButtonComponent = /** @class */ (function () {
    function DatePickerDayButtonComponent(model, helper, menuService) {
        this.model = model;
        this.helper = helper;
        this.menuService = menuService;
        this.caption = '';
        this.enabled = false;
        this.selected = false;
        this.closeOnAction = true;
    }
    DatePickerDayButtonComponent.prototype.action = function () {
        this.date.day = this.row * 7 + this.col - this.getFirstDay() + 1;
        if (this.closeOnAction && this.menuService.getMenu('datePickerMenu') != null) {
            this.menuService.getMenu('datePickerMenu').state = 'initialUp';
        }
    };
    DatePickerDayButtonComponent.prototype.getFirstDay = function () {
        var firstDay = new Date(this.date.year, this.date.month - 1, 1).getDay();
        if (firstDay === 0) {
            firstDay = 7;
        }
        if (this.model.options.appearance.datePickerFirstDayOfWeek !== 'Sunday') {
            firstDay--;
        }
        return firstDay;
    };
    DatePickerDayButtonComponent.prototype.ngOnInit = function () { };
    DatePickerDayButtonComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService },
        { type: MenuService }
    ]; };
    __decorate([
        Input()
    ], DatePickerDayButtonComponent.prototype, "col", void 0);
    __decorate([
        Input()
    ], DatePickerDayButtonComponent.prototype, "row", void 0);
    __decorate([
        Input()
    ], DatePickerDayButtonComponent.prototype, "caption", void 0);
    __decorate([
        Input()
    ], DatePickerDayButtonComponent.prototype, "enabled", void 0);
    __decorate([
        Input()
    ], DatePickerDayButtonComponent.prototype, "selected", void 0);
    __decorate([
        Input()
    ], DatePickerDayButtonComponent.prototype, "date", void 0);
    __decorate([
        Input()
    ], DatePickerDayButtonComponent.prototype, "closeOnAction", void 0);
    DatePickerDayButtonComponent = __decorate([
        Component({
            selector: 'sti-date-picker-day-button',
            template: "\n    <sti-button [styleName]=\"'stiJsViewerDatePickerDayButton'\"\n                [width]=\"model.options.isTouchDevice ? '25px' : '23px'\"\n                [height]=\"model.options.isTouchDevice ? '25px' : '23px'\"\n                [captionAlign]=\"'center'\"\n                [innerTableWidth]=\"'100%'\"\n                [captionPadding]=\"'0px'\"\n                [margin]=\"'1px'\"\n                [caption]=\"caption\"\n                [enabled]=\"enabled\"\n                [selected]=\"selected\"\n                (action)=\"action()\">\n    </sti-button>\n  "
        })
    ], DatePickerDayButtonComponent);
    return DatePickerDayButtonComponent;
}());
export { DatePickerDayButtonComponent };
