// stimulsoft-viewer-angular/esm5/lib/controls/drop-down-list.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
import { MouseService } from '../services/mouse.service';
import { MenuService } from '../menu/menu.service';
var DropDownListComponent = /** @class */ (function () {
    function DropDownListComponent(model, mouseService, menuService) {
        var _this = this;
        this.model = model;
        this.mouseService = mouseService;
        this.menuService = menuService;
        this.showImage = false;
        this.action = new EventEmitter();
        this.over = false;
        this._readOnly = false;
        this._enabled = true;
        this.mouseService.getDocumentMouseUp().subscribe(function () {
            _this.hideListMenu();
        });
    }
    DropDownListComponent.prototype.ngOnInit = function () { };
    Object.defineProperty(DropDownListComponent.prototype, "readOnly", {
        get: function () {
            return this._readOnly;
        },
        set: function (value) {
            var _this = this;
            this._readOnly = value;
            setTimeout(function () {
                try {
                    _this.input.nativeElement.setAttribute('unselectable', value ? 'on' : 'off');
                    _this.input.nativeElement.setAttribute('onselectstart', value ? 'return false' : '');
                    _this.input.nativeElement.readOnly = value;
                }
                catch (e) { }
            });
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DropDownListComponent.prototype, "enabled", {
        get: function () {
            return this._enabled;
        },
        set: function (value) {
            if (!value) {
                this.over = false;
            }
            this._enabled = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DropDownListComponent.prototype, "key", {
        get: function () {
            return this._key;
        },
        set: function (key) {
            if (this.items != null) {
                this.items.forEach(function (i) { return i.selected = i.key === key; });
            }
            this._key = key;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DropDownListComponent.prototype, "inputValue", {
        get: function () {
            var _this = this;
            var _a;
            if (this.items != null) {
                return ((_a = this.items.find(function (i) { return i.key === _this._key; })) === null || _a === void 0 ? void 0 : _a.caption) || '';
            }
            return '';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DropDownListComponent.prototype, "selected", {
        get: function () {
            var _a, _b;
            var vm = this.menuService.getVerticalMenu();
            return ((_a = vm) === null || _a === void 0 ? void 0 : _a.state) === 'expanded' && ((_b = vm) === null || _b === void 0 ? void 0 : _b.parent) === this.element;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DropDownListComponent.prototype, "className", {
        get: function () {
            return this.selected ? 'stiJsViewerDropDownListOver' : (this.enabled ? (this.over ? 'stiJsViewerDropDownListOver' : 'stiJsViewerDropDownList') : 'stiJsViewerDropDownListDisabled');
        },
        enumerable: true,
        configurable: true
    });
    DropDownListComponent.prototype.mouseover = function () {
        if (!this.enabled) {
            return;
        }
        this.over = true;
    };
    DropDownListComponent.prototype.mouseout = function () {
        if (!this.enabled) {
            return;
        }
        this.over = false;
    };
    DropDownListComponent.prototype.inputClick = function () {
        if (this.readOnly) {
            this.showListMenu();
        }
    };
    DropDownListComponent.prototype.showListMenu = function () {
        var _this = this;
        this.menuService.addMenu({
            type: this.menuService.VERTICAL_MENU_NAME, name: this.menuService.VERTICAL_MENU_NAME, items: [], parent: this.element,
            verticalItems: this.items, itemStyleName: 'stiJsViewerMenuStandartItem', menuStyleName: 'stiJsViewerDropdownMenu',
            action: this.action, width: this.element.nativeElement.offsetWidth
        });
        setTimeout(function () {
            _this.menuService.showMenu(_this.menuService.VERTICAL_MENU_NAME);
        });
    };
    DropDownListComponent.prototype.hideListMenu = function () {
        if (this.menuService.getVerticalMenu()) {
            this.menuService.getVerticalMenu().state = 'initialDown';
        }
    };
    DropDownListComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: MouseService },
        { type: MenuService }
    ]; };
    __decorate([
        ViewChild('input')
    ], DropDownListComponent.prototype, "input", void 0);
    __decorate([
        ViewChild('element')
    ], DropDownListComponent.prototype, "element", void 0);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "toolTip", void 0);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "showImage", void 0);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "margin", void 0);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "items", void 0);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "styleDisplay", void 0);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "verticalAlign", void 0);
    __decorate([
        Output()
    ], DropDownListComponent.prototype, "action", void 0);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "readOnly", null);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "enabled", null);
    __decorate([
        Input()
    ], DropDownListComponent.prototype, "key", null);
    DropDownListComponent = __decorate([
        Component({
            selector: 'sti-drop-down-list',
            template: "\n        <table #element [class]=\"className\" cellpadding=\"0\" cellspacing=\"0\"\n              [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n              [style.color]=\"model.options.toolbar.fontColor\"\n              [style.margin]=\"margin\"\n              [style.verticalAlign]=\"verticalAlign\"\n              [style.display]=\"styleDisplay\"\n              [attr.title]=\"toolTip\"\n              (mouseover)=\"mouseover()\"\n              (mouseout)=\"mouseout()\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\">\n                <input #input\n                [style.width.px]=\"width - (model.options.isTouchDevice ? 23 : 15) - (showImage ? 38 : 0)\"\n                [style.border]=\"0\"\n                [style.cursor]=\"readOnly ? 'default' : 'text'\"\n                [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n                [style.color]=\"model.options.toolbar.fontColor\"\n                [style.height]=\"model.options.isTouchDevice ? '23px' : '18px'\"\n                [style.lineHeight]=\"model.options.isTouchDevice ? '23px' : '18px'\"\n                [style.visibility]=\"enabled ? 'visible' : 'hidden'\"\n                [value]=\"inputValue\"\n                (click)=\"inputClick()\"\n                class=\"stiJsViewerDropDownList_TextBox\">\n              </td>\n\n              <td class=\"stiJsViewerClearAllStyles\">\n                  <sti-button [imageName]=\"'Arrows.' + (model.options.isTouchDevice ? 'Big' : 'Small') + 'ArrowDown.png'\"\n                     [styleName]=\"'stiJsViewerDropDownListButton'\"\n                     [imageSizesWidth]=\"model.options.isTouchDevice ? 16 : 8\"\n                     [imageSizesHeight]=\"model.options.isTouchDevice ? 16 : 8\"\n                     [height]=\"model.options.isTouchDevice ? '26px' : '21px'\"\n                     [enabled]=\"enabled\"\n                     [selected]=\"selected\"\n                     [margin]=\"'0'\"\n                     (action)=\"showListMenu()\">\n                  </sti-button>\n              </td>\n            </tr>\n          </tbody>\n        </table>\n\n  "
        })
    ], DropDownListComponent);
    return DropDownListComponent;
}());
export { DropDownListComponent };
