// stimulsoft-viewer-angular/esm5/lib/controls/drop-down-list-for-export-form.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
var DropDownListForExportFormComponent = /** @class */ (function () {
    function DropDownListForExportFormComponent() {
        this.enabled = true;
        this.action = new EventEmitter();
    }
    DropDownListForExportFormComponent.prototype.ngOnInit = function () { };
    __decorate([
        Input()
    ], DropDownListForExportFormComponent.prototype, "items", void 0);
    __decorate([
        Input()
    ], DropDownListForExportFormComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], DropDownListForExportFormComponent.prototype, "key", void 0);
    __decorate([
        Input()
    ], DropDownListForExportFormComponent.prototype, "margin", void 0);
    __decorate([
        Input()
    ], DropDownListForExportFormComponent.prototype, "enabled", void 0);
    __decorate([
        Input()
    ], DropDownListForExportFormComponent.prototype, "styleDisplay", void 0);
    __decorate([
        Input()
    ], DropDownListForExportFormComponent.prototype, "verticalAlign", void 0);
    __decorate([
        Output()
    ], DropDownListForExportFormComponent.prototype, "action", void 0);
    DropDownListForExportFormComponent = __decorate([
        Component({
            selector: 'sti-drop-down-list-for-export-form',
            template: "\n    <sti-drop-down-list [items]=\"items\"\n      [width]=\"width\"\n      [readOnly]=\"true\"\n      [key]=\"key\"\n      [margin]=\"margin\"\n      [enabled]=\"enabled\"\n      [styleDisplay]=\"styleDisplay\"\n      [verticalAlign]=\"verticalAlign\"\n      (action)=\"key = $event.key; action.emit($event)\">\n    </sti-drop-down-list>\n  "
        })
    ], DropDownListForExportFormComponent);
    return DropDownListForExportFormComponent;
}());
export { DropDownListForExportFormComponent };
