// stimulsoft-viewer-angular/esm5/lib/controls/form-button.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
var FormButtonComponent = /** @class */ (function () {
    function FormButtonComponent() {
        this.captionAlign = 'center';
        this.action = new EventEmitter();
    }
    FormButtonComponent.prototype.ngOnInit = function () { };
    FormButtonComponent.prototype.onaction = function (event) {
        this.action.emit(event);
    };
    __decorate([
        Input()
    ], FormButtonComponent.prototype, "caption", void 0);
    __decorate([
        Input()
    ], FormButtonComponent.prototype, "actionName", void 0);
    __decorate([
        Input()
    ], FormButtonComponent.prototype, "imageName", void 0);
    __decorate([
        Input()
    ], FormButtonComponent.prototype, "imageCellWidth", void 0);
    __decorate([
        Input()
    ], FormButtonComponent.prototype, "captionPadding", void 0);
    __decorate([
        Input()
    ], FormButtonComponent.prototype, "margin", void 0);
    __decorate([
        Input()
    ], FormButtonComponent.prototype, "captionAlign", void 0);
    __decorate([
        Output()
    ], FormButtonComponent.prototype, "action", void 0);
    FormButtonComponent = __decorate([
        Component({
            selector: 'sti-form-button',
            template: "\n      <sti-button [styleName]=\"'stiJsViewerFormButton'\"\n                  [caption]=\"caption\"\n                  [actionName]=\"actionName\"\n                  [captionAlign]=\"captionAlign\"\n                  [innerTableWidth]=\"'100%'\"\n                  [minWidth]=\"'80px'\"\n                  [imageName]=\"imageName\"\n                  [imageCellWidth]=\"imageCellWidth\"\n                  [captionPadding]=\"captionPadding\"\n                  [margin]=\"margin\"\n                  (action)=\"onaction($event)\">\n      </sti-button>\n  "
        })
    ], FormButtonComponent);
    return FormButtonComponent;
}());
export { FormButtonComponent };
