// stimulsoft-viewer-angular/esm5/lib/controls/radio-button.components.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
import { RadioButtonService } from '../services/radio-button.service';
var RadioButtonComponent = /** @class */ (function () {
    function RadioButtonComponent(model, radioService) {
        this.model = model;
        this.radioService = radioService;
        this.action = new EventEmitter();
        this.enabled = true;
        this.checked = false;
        this.over = false;
    }
    RadioButtonComponent.prototype.ngAfterViewInit = function () {
        this.radioService.addButton(this);
    };
    RadioButtonComponent.prototype.ngOnDestroy = function () {
        this.radioService.removeButton(this);
    };
    RadioButtonComponent.prototype.ngOnInit = function () { };
    RadioButtonComponent.prototype.mouseover = function () {
        if (!this.enabled) {
            return;
        }
        this.over = true;
    };
    RadioButtonComponent.prototype.mouseout = function () {
        if (!this.enabled) {
            return;
        }
        this.over = false;
    };
    RadioButtonComponent.prototype.click = function () {
        if (this.enabled) {
            this.checked = true;
            this.radioService.check(this);
            this.action.emit(true);
        }
    };
    RadioButtonComponent.prototype.uncheck = function () {
        this.checked = false;
        this.action.emit(false);
    };
    Object.defineProperty(RadioButtonComponent.prototype, "outClass", {
        get: function () {
            return !this.enabled ? 'stiJsViewerRadioButtonOutCircleDisabled' : (this.over ? 'stiJsViewerRadioButtonOutCircleOver' : 'stiJsViewerRadioButtonOutCircle');
        },
        enumerable: true,
        configurable: true
    });
    RadioButtonComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: RadioButtonService }
    ]; };
    __decorate([
        Output()
    ], RadioButtonComponent.prototype, "action", void 0);
    __decorate([
        Input()
    ], RadioButtonComponent.prototype, "groupName", void 0);
    __decorate([
        Input()
    ], RadioButtonComponent.prototype, "name", void 0);
    __decorate([
        Input()
    ], RadioButtonComponent.prototype, "enabled", void 0);
    __decorate([
        Input()
    ], RadioButtonComponent.prototype, "checked", void 0);
    __decorate([
        Input()
    ], RadioButtonComponent.prototype, "tooltip", void 0);
    __decorate([
        Input()
    ], RadioButtonComponent.prototype, "caption", void 0);
    __decorate([
        Input()
    ], RadioButtonComponent.prototype, "margin", void 0);
    __decorate([
        Input()
    ], RadioButtonComponent.prototype, "paddingLeftLastCell", void 0);
    RadioButtonComponent = __decorate([
        Component({
            selector: 'sti-radio-button',
            template: "\n    <table [class]=\"enabled ? 'stiJsViewerRadioButton' : 'stiJsViewerRadioButtonDisabled'\" cellpadding=\"0\" cellspacing=\"0\"\n      [style.fontFamily]=\"model.options.toolbar.fontFamily\"\n      [style.margin]=\"margin\"\n      [attr.title]=\"tooltip\"\n      (mouseover)=\"mouseover()\" (mouseout)=\"mouseout()\" (click)=\"click()\">\n      <tbody>\n        <tr class=\"stiJsViewerClearAllStyles\">\n          <td class=\"stiJsViewerClearAllStyles\">\n            <div class=\"stiJsViewerRadioButtonOutCircle\" [class]=\"outClass\">\n              <div class=\"stiJsViewerRadioButtonInnerCircle\"\n                  [style.margin]=\"model.options.isTouchDevice ? '4px' : '3px'\"\n                  [style.width]=\"model.options.isTouchDevice ? '9px' : '7px'\"\n                  [style.height]=\"model.options.isTouchDevice ? '9px' : '7px'\"\n                  [style.visibility]=\"checked ? 'visible' : 'hidden'\"\n                  [style.opacity]=\"enabled ? 1 : 0.5\">\n              </div>\n            </div>\n          </td>\n          <td *ngIf=\"caption != null\" class=\"stiJsViewerClearAllStyles\" style=\"padding-left: 4px; white-space: nowrap;\">\n            {{caption}}\n          </td>\n          <td class=\"stiJsViewerClearAllStyles\" [style.paddingLeft]=\"paddingLeftLastCell\">\n            <ng-content></ng-content>\n          </td>\n        </tr>\n      </tbody>\n    </table>\n  "
        })
    ], RadioButtonComponent);
    return RadioButtonComponent;
}());
export { RadioButtonComponent };
