// stimulsoft-viewer-angular/esm5/lib/controls/text-area.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
var TextAreaComponent = /** @class */ (function () {
    function TextAreaComponent() {
        this.enabled = true;
        this.action = new EventEmitter();
        this.onchange = new EventEmitter();
        this.onblur = new EventEmitter();
        this.selected = false;
        this.focused = false;
        this.over = false;
        this._readOnly = false;
        this.styleName = 'stiJsViewerTextBox';
    }
    TextAreaComponent.prototype.ngOnInit = function () { };
    TextAreaComponent.prototype.keypress = function (event) {
        if (!this.enabled) {
            return false;
        }
        if (event.keyCode === 13) {
            this.action.emit(event.target);
        }
    };
    TextAreaComponent.prototype.onchanged = function (event) {
        this.onchange.emit(event.target);
    };
    Object.defineProperty(TextAreaComponent.prototype, "className", {
        get: function () {
            if (this._readOnly) {
                return this.styleName + ' ' + this.styleName + 'Default';
            }
            return this.styleName + ' ' + this.styleName + (this.selected ? 'Over' : (this.enabled ? (this.over ? 'Over' : 'Default') : 'Disabled'));
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TextAreaComponent.prototype, "readOnly", {
        get: function () {
            return this._readOnly;
        },
        enumerable: true,
        configurable: true
    });
    __decorate([
        Input()
    ], TextAreaComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], TextAreaComponent.prototype, "height", void 0);
    __decorate([
        Input()
    ], TextAreaComponent.prototype, "enabled", void 0);
    __decorate([
        Input()
    ], TextAreaComponent.prototype, "value", void 0);
    __decorate([
        Input()
    ], TextAreaComponent.prototype, "padding", void 0);
    __decorate([
        Input()
    ], TextAreaComponent.prototype, "margin", void 0);
    __decorate([
        Output()
    ], TextAreaComponent.prototype, "action", void 0);
    __decorate([
        Output()
    ], TextAreaComponent.prototype, "onchange", void 0);
    __decorate([
        Output()
    ], TextAreaComponent.prototype, "onblur", void 0);
    TextAreaComponent = __decorate([
        Component({
            selector: 'sti-text-area',
            template: "\n    <textarea\n      [style.width.px]=\"width\"\n      [style.minWidth.px]=\"width\"\n      [style.height.px]=\"height\"\n      [style.minHeight.px]=\"height\"\n      [class]=\"className\"\n      [value]=\"value || ''\"\n      [style.margin]=\"margin\"\n      [style.cursor]=\"readOnly ? 'default' : ''\"\n      [style.padding]=\"padding\"\n      style=\"padding-top: 3px, font-family: Arial\"\n      (mouseover)=\"over=true\"\n      (mouseleave)=\"over=false\"\n      (focused)=\"focused=true; selected=true\"\n      (blur)=\"focused=false; selected=false; onblur.emit($event.target)\"\n      (keypress)=\"keypress($event)\"\n      (keyup)=\"onchanged($event)\">\n    </textarea>\n  "
        })
    ], TextAreaComponent);
    return TextAreaComponent;
}());
export { TextAreaComponent };
