// stimulsoft-viewer-angular/esm5/lib/controls/text-box.componet.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
var TextBoxComponent = /** @class */ (function () {
    function TextBoxComponent(model) {
        this.model = model;
        this.enabled = true;
        this.focusOnCreate = false;
        this.action = new EventEmitter();
        this.onchange = new EventEmitter();
        this.onblur = new EventEmitter();
        this.selected = false;
        this.focused = false;
        this.over = false;
        this._readOnly = false;
        this.styleName = 'stiJsViewerTextBox';
    }
    TextBoxComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        if (this.focusOnCreate) {
            setTimeout(function () {
                _this.element.nativeElement.focus();
            });
        }
    };
    TextBoxComponent.prototype.keypress = function (event) {
        if (!this.enabled) {
            return false;
        }
        if (event.keyCode === 13) {
            this.action.emit(event.target);
        }
    };
    TextBoxComponent.prototype.onchanged = function (event) {
        this.onchange.emit(event.target);
    };
    TextBoxComponent.prototype.focus = function () {
        this.oldValue = this.element.nativeElement.value;
    };
    Object.defineProperty(TextBoxComponent.prototype, "height", {
        get: function () {
            return this.model.options.isTouchDevice ? '26px' : '21px';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TextBoxComponent.prototype, "className", {
        get: function () {
            if (this._readOnly) {
                return this.styleName + ' ' + this.styleName + 'Default';
            }
            return this.styleName + ' ' + this.styleName + (this.selected ? 'Over' : (this.enabled ? (this.over ? 'Over' : 'Default') : 'Disabled'));
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TextBoxComponent.prototype, "readOnly", {
        get: function () {
            return this._readOnly;
        },
        set: function (value) {
            var _this = this;
            this._readOnly = value;
            setTimeout(function () {
                try {
                    _this.element.nativeElement.setAttribute('unselectable', value ? 'on' : 'off');
                    _this.element.nativeElement.setAttribute('onselectstart', value ? 'return false' : '');
                    _this.element.nativeElement.readOnly = value;
                }
                catch (e) { }
            });
        },
        enumerable: true,
        configurable: true
    });
    TextBoxComponent.prototype.ngOnInit = function () { };
    TextBoxComponent.ctorParameters = function () { return [
        { type: ModelService }
    ]; };
    __decorate([
        ViewChild('element')
    ], TextBoxComponent.prototype, "element", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "actionLostFocus", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "tooltip", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "enabled", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "value", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "margin", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "focusOnCreate", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "maxLength", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "color", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "type", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "padding", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "border", void 0);
    __decorate([
        Output()
    ], TextBoxComponent.prototype, "action", void 0);
    __decorate([
        Output()
    ], TextBoxComponent.prototype, "onchange", void 0);
    __decorate([
        Output()
    ], TextBoxComponent.prototype, "onblur", void 0);
    __decorate([
        Input()
    ], TextBoxComponent.prototype, "readOnly", null);
    TextBoxComponent = __decorate([
        Component({
            selector: 'sti-text-box',
            template: "\n    <input #element\n    [style]=\"{fontFamily:model.options.toolbar.fontFamily, height: height, lineHeight: height, boxSizing:'content-box'}\"\n    [style.color]=\"color ? color : (model.options.toolbar.fontColor != '' ? model.options.toolbar.fontColor : '')\"\n    [style.width]=\"width != null ? width + 'px' : ''\"\n    [style.margin]=\"margin\"\n    [style.cursor]=\"readOnly ? 'default' : ''\"\n    [style.padding]=\"padding\"\n    [style.border]=\"border\"\n    [title]=\"tooltip || ''\"\n    [class]=\"className\"\n    [value]=\"value || ''\"\n    [attr.maxLength]=\"maxLength\"\n    [attr.type]=\"type\"\n    (mouseover)=\"over=true\"\n    (mouseleave)=\"over=false\"\n    (focused)=\"focused=true; selected=true\"\n    (blur)=\"focused=false; selected=false; onblur.emit($event.target)\"\n    (keypress)=\"keypress($event)\"\n    (keyup)=\"onchanged($event)\"\n    (focus)=\"focus()\"/>\n  "
        })
    ], TextBoxComponent);
    return TextBoxComponent;
}());
export { TextBoxComponent };
