// stimulsoft-viewer-angular/esm5/lib/controls/tooltip.component.js
import { __decorate } from "tslib";
import { Component, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { trigger, state, transition, animate, style, keyframes } from '@angular/animations';
import { TooltipService } from '../services/tooltip.service';
import { HelperService } from '../services/helper.service';
var TooltipComponent = /** @class */ (function () {
    function TooltipComponent(model, tooltipService, helper) {
        this.model = model;
        this.tooltipService = tooltipService;
        this.helper = helper;
    }
    TooltipComponent.prototype.ngAfterViewInit = function () {
        this.model.controls.tooltip.el = this.element;
    };
    TooltipComponent.prototype.ngOnInit = function () { };
    TooltipComponent.prototype.action = function () {
        this.helper.showHelpWindow(this.tooltipService.helpLink);
    };
    TooltipComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: TooltipService },
        { type: HelperService }
    ]; };
    __decorate([
        ViewChild('element')
    ], TooltipComponent.prototype, "element", void 0);
    TooltipComponent = __decorate([
        Component({
            selector: 'sti-tooltip',
            template: "\n      <div #element class='stiJsViewerToolTip' [@expandDown]=\"tooltipService.state\"\n        [style.left.px]='tooltipService.left' [style.top.px]='tooltipService.top'\n        (mouseover)=\"tooltipService.showFromTooltip()\" (mouseout)=\"tooltipService.hide()\">\n        <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" style=\"height: 100%;\"\n          [style.border]=\"model.options.appearance.showTooltipsHelp ? '' : 0\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerToolTipTextCell\">\n                {{tooltipService.caption}}\n              </td>\n            </tr>\n            <tr *ngIf=\"model.options.appearance.showTooltipsHelp\" class=\"stiJsViewerClearAllStyles\">\n              <td  class=\"stiJsViewerClearAllStyles\">\n                <sti-button [caption]=\"model.loc('TellMeMore')\"\n                    [imageName]=\"'HelpIcon.png'\"\n                    [margin]=\"'4px 8px 4px 8px'\"\n                    (action)='action()'\n                    [styleName]=\"'stiJsViewerHyperlinkButton'\">\n                </sti-button>\n              </td>\n            </tr>\n          </tbody>\n        </table>\n    </div>\n  ",
            animations: [
                trigger('expandDown', [
                    state('initial', style({ opacity: 0, display: 'none' })),
                    state('preInitial', style({ opacity: 0, display: 'block' })),
                    state('preInitial2', style({ opacity: 0, display: 'block' })),
                    state('expanded', style({ opacity: 1, display: 'block' })),
                    transition('preInitial => expanded', [
                        animate('300ms 300ms ease-in-out', keyframes([
                            style({ display: 'block', opacity: 0, offset: 0 }),
                            style({ display: 'block', opacity: 1, offset: 1 }),
                        ]))
                    ])
                ])
            ]
        })
    ], TooltipComponent);
    return TooltipComponent;
}());
export { TooltipComponent };
