// stimulsoft-viewer-angular/esm2015/lib/controls/button.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild, Output, EventEmitter } from '@angular/core';
import { ControllerService } from '../services/controller.service';
import { ModelService } from '../services/model.service';
import { MenuService, Menu } from '../menu/menu.service';
import { TooltipService } from '../services/tooltip.service';
import { HelperService } from '../services/helper.service';
let ButtonComponent = class ButtonComponent {
    constructor(model, controller, menuService, tooltipService, helper) {
        this.model = model;
        this.controller = controller;
        this.menuService = menuService;
        this.tooltipService = tooltipService;
        this.helper = helper;
        this.margin = '1px';
        this.height = '23px';
        this.selected = false;
        this.imageCellTextAlign = 'center';
        this.imageSizesWidth = 16;
        this.imageSizesHeight = 16;
        this.closeButton = false;
        this.resourceButton = false;
        this.navagationPanelTooltip = false;
        this.helpLink = 'user-manual/index.html?viewer_reports.htm';
        this.action = new EventEmitter();
        this.closeButtonAction = new EventEmitter();
        this.over = false;
        this.showMenu = false;
        this.closeButtonPressed = false;
        this.resourceButtonPressed = false;
        this._enabled = true;
        this.canShowTooltip = true;
    }
    ngOnInit() {
        if (this.menuItems) {
            setTimeout(() => {
                this.menuObj = { type: 'buttonMenu', name: this.actionName, items: this.menuItems, parent: !this.resourceButton ? this.button : this.resButtonEl };
                this.menuService.addMenu(this.menuObj);
            }, 500);
        }
    }
    get styleName() {
        return this._styleName || 'stiJsViewerStandartSmallButton';
    }
    set styleName(value) {
        this._styleName = value;
    }
    get enabled() {
        return this._enabled;
    }
    set enabled(value) {
        if (!value) {
            this.over = false;
        }
        this._enabled = value;
    }
    mouseover() {
        if (!this.enabled) {
            return;
        }
        this.over = true;
        if (this.model.options.toolbar.showMenuMode === 'Hover' && ['Print', 'Save', 'SendEmail', 'Zoom', 'ViewMode'].some(a => a === this.actionName)) {
            this.tooltipService.hideImmediately();
            this.menuService.showMenu(this.actionName);
        }
        else if (this.tooltip && !this.menuService.menus.some(e => e.state === 'expanded') && this.canShowTooltip) {
            const top = !this.navagationPanelTooltip ? this.model.controls.toolbar.offsetHeight + this.model.controls.dashboardsPanel.offsetHeight :
                this.helper.findPosY(this.model.controls.navigatePanel.el.nativeElement, 'stiJsViewerMainPanel');
            const tooltip = this.tooltip === true ? (this.model.localization[this.actionName + 'ToolTip'] != null ? this.model.loc(this.actionName + 'ToolTip') : this.model.loc(this.actionName)) : this.tooltip;
            this.tooltipService.show(this.helper.findPosX(this.button.nativeElement, 'stiJsViewerMainPanel'), top, tooltip, !this.navagationPanelTooltip, this.helpLink);
        }
    }
    mouseout() {
        if (!this.enabled) {
            return;
        }
        this.over = false;
        this.tooltipService.hide();
        setTimeout(() => {
            this.canShowTooltip = true;
        }, 1000);
    }
    click() {
        if (this.enabled && !this.closeButtonPressed) {
            this.tooltipService.hideImmediately();
            if (this.menuItems == null || (this.resourceButton && !this.resourceButtonPressed)) {
                this.action.emit();
            }
            else {
                this.menuService.showMenu(this.actionName);
            }
        }
        this.canShowTooltip = false;
        this.closeButtonPressed = false;
        this.resourceButtonPressed = false;
    }
    get className() {
        var _a;
        return this.styleName ? this.styleName + ' ' + (this.enabled ? (this.styleName + ((this.selected || ((_a = this.menuObj) === null || _a === void 0 ? void 0 : _a.state) === 'expanded') ? 'Selected' : (this.over ? 'Over' : 'Default'))) : this.styleName + 'Disabled') : '';
    }
};
ButtonComponent.ctorParameters = () => [
    { type: ModelService },
    { type: ControllerService },
    { type: MenuService },
    { type: TooltipService },
    { type: HelperService }
];
__decorate([
    Input()
], ButtonComponent.prototype, "caption", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "caption2", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionAlign", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionPadding", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionPaddingLeft", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionWhiteSpace", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionOverflow", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionTextOverflow", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionWidth", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionMaxWidth", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "captionLineHeight", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "imageName", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "arrow", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "arrowMarginTop", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "margin", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "height", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "selected", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "minWidth", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "innerTableWidth", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "menuItems", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "actionName", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "tooltip", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "imageCellTextAlign", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "imageCellWidth", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "imageCellPadding", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "imageSizesWidth", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "imageSizesHeight", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "imageMargin", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "width", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "display", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "closeButton", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "resourceButton", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "styleColors", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "boxSizing", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "navagationPanelTooltip", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "cursor", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "fontSize", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "helpLink", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "borderColor", void 0);
__decorate([
    Output()
], ButtonComponent.prototype, "action", void 0);
__decorate([
    Output()
], ButtonComponent.prototype, "closeButtonAction", void 0);
__decorate([
    ViewChild('button')
], ButtonComponent.prototype, "button", void 0);
__decorate([
    ViewChild('resButtonEl')
], ButtonComponent.prototype, "resButtonEl", void 0);
__decorate([
    Input()
], ButtonComponent.prototype, "styleName", null);
__decorate([
    Input()
], ButtonComponent.prototype, "enabled", null);
ButtonComponent = __decorate([
    Component({
        selector: 'sti-button',
        template: `
  <div #button
    [style.fontFamily]="model.options.toolbar.fontFamily"
    [style.cursor]="cursor"
    [style.borderColor]="borderColor"
    [class]="className"
    [style]="{height: helper.val(height, '23px'), boxSizing: helper.val(boxSizing, 'content-box'), margin: helper.val(margin, '1px'), minWidth: minWidth, width: width, display: display }"
    [style.fontSize]="fontSize"
    (mouseover)="mouseover()"
    (mouseout)="mouseout()"
    (click)="click()" >
    <table class="stiJsViewerClearAllStyles" cellpadding="0" cellspacing="0" style="height: 100%;width:100%" [style.width]="innerTableWidth">
      <tbody>
        <tr class="stiJsViewerClearAllStyles">
          <td *ngIf="imageName != null" [style]="{lineHeight:'0', padding:helper.val(imageCellPadding, '0 3px'), textAlign: imageCellTextAlign, width: imageCellWidth}" class="stiJsViewerClearAllStyles">
            <img src="{{model.img(imageName)}}" [style.opacity]="enabled ? '1' : '0.5'"
                [style.width.px]="imageSizesWidth"
                [style.height.px]="imageSizesHeight"
                [style.margin]="imageMargin">
          </td>
          <td *ngIf="caption != null"
             [style]="{whiteSpace:'nowrap', textAlign: captionAlign || 'left', padding:captionPadding || ((arrow ? '1px 0' : '1px 5px') + (imageName ? ' 0 0' : ' 0 5px')), width: captionWidth}"
             [style.maxWidth]="captionMaxWidth"
             [style.lineHeight]="captionLineHeight"
             [style.paddingLeft]="captionPaddingLeft"
             [style.whiteSpace]="captionWhiteSpace"
             [style.overflow]="captionOverflow"
             [style.textOverflow]="captionTextOverflow"
             class="stiJsViewerClearAllStyles">
            {{caption}}
            <div *ngIf="caption2 != null">
            {{caption2}}
            </div>
          </td>
          <td *ngIf="arrow" class="stiJsViewerClearAllStyles">
            <img src="{{arrow == 'Up' || arrow == 'Down' ? model.img('Arrows.SmallArrow' + arrow + (arrow == 'Down' && styleColors && styleColors.isDarkStyle ? 'White.png' : '.png')) : model.img(arrow)}}"
            [style]="{lineHeight:'0', width:'8px', height:'8px', verticalAlign:'baseline', padding:caption ? '0 5px 0 5px' : '0 5px 0 2px', marginTop: arrowMarginTop}"
            [style.opacity]="enabled ? '1' : '0.5'"/>
          </td>

          <td *ngIf="closeButton" class="stiJsViewerClearAllStyles">
            <sti-button [imageName]="'CloseForm.png'"
              [imageMargin]="'1px 0 0 -1px'"
              [margin]="'0 2px 0 0'"
              [imageCellPadding]="'0'"
              [width]="model.options.isTouchDevice ? '22px' : '17px'"
              [height]="model.options.isTouchDevice ? '22px' : '17px'"
              (action)="closeButtonPressed = true; closeButtonAction.emit()">
            </sti-button>
          </td>

          <td #resButtonEl *ngIf="resourceButton" style="width: 1px;">
            <sti-button [imageName]="'Arrows.SmallArrowDown.png'"
              [styleName]="'stiJsViewerResourceDropDownButton'"
              [innerTableWidth]="'100%'"
              [margin]="'0 7px 0 3px'"
              [imageCellTextAlign]="'center'"
              [imageSizesWidth]="8"
              [imageSizesHeight]="8"
              [height]="model.options.isTouchDevice ? '23px' : '17px'"
              [width]="model.options.isTouchDevice ? '23px' : '17px'"
              (action)="resourceButtonPressed = true;">>
            </sti-button>
          </td>

        </tr>
      </tbody>
    </table>
  </div>
  `
    })
], ButtonComponent);
export { ButtonComponent };
