// stimulsoft-viewer-angular/esm2015/lib/controls/date-picker-day-button.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { ModelService } from '../services/model.service';
import { HelperService } from '../services/helper.service';
import { MenuService } from '../menu/menu.service';
let DatePickerDayButtonComponent = class DatePickerDayButtonComponent {
    constructor(model, helper, menuService) {
        this.model = model;
        this.helper = helper;
        this.menuService = menuService;
        this.caption = '';
        this.enabled = false;
        this.selected = false;
        this.closeOnAction = true;
    }
    action() {
        this.date.day = this.row * 7 + this.col - this.getFirstDay() + 1;
        if (this.closeOnAction && this.menuService.getMenu('datePickerMenu') != null) {
            this.menuService.getMenu('datePickerMenu').state = 'initialUp';
        }
    }
    getFirstDay() {
        let firstDay = new Date(this.date.year, this.date.month - 1, 1).getDay();
        if (firstDay === 0) {
            firstDay = 7;
        }
        if (this.model.options.appearance.datePickerFirstDayOfWeek !== 'Sunday') {
            firstDay--;
        }
        return firstDay;
    }
    ngOnInit() { }
};
DatePickerDayButtonComponent.ctorParameters = () => [
    { type: ModelService },
    { type: HelperService },
    { type: MenuService }
];
__decorate([
    Input()
], DatePickerDayButtonComponent.prototype, "col", void 0);
__decorate([
    Input()
], DatePickerDayButtonComponent.prototype, "row", void 0);
__decorate([
    Input()
], DatePickerDayButtonComponent.prototype, "caption", void 0);
__decorate([
    Input()
], DatePickerDayButtonComponent.prototype, "enabled", void 0);
__decorate([
    Input()
], DatePickerDayButtonComponent.prototype, "selected", void 0);
__decorate([
    Input()
], DatePickerDayButtonComponent.prototype, "date", void 0);
__decorate([
    Input()
], DatePickerDayButtonComponent.prototype, "closeOnAction", void 0);
DatePickerDayButtonComponent = __decorate([
    Component({
        selector: 'sti-date-picker-day-button',
        template: `
    <sti-button [styleName]="'stiJsViewerDatePickerDayButton'"
                [width]="model.options.isTouchDevice ? '25px' : '23px'"
                [height]="model.options.isTouchDevice ? '25px' : '23px'"
                [captionAlign]="'center'"
                [innerTableWidth]="'100%'"
                [captionPadding]="'0px'"
                [margin]="'1px'"
                [caption]="caption"
                [enabled]="enabled"
                [selected]="selected"
                (action)="action()">
    </sti-button>
  `
    })
], DatePickerDayButtonComponent);
export { DatePickerDayButtonComponent };
