// stimulsoft-viewer-angular/esm2015/lib/controls/drop-down-list.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
import { MouseService } from '../services/mouse.service';
import { MenuService } from '../menu/menu.service';
let DropDownListComponent = class DropDownListComponent {
    constructor(model, mouseService, menuService) {
        this.model = model;
        this.mouseService = mouseService;
        this.menuService = menuService;
        this.showImage = false;
        this.action = new EventEmitter();
        this.over = false;
        this._readOnly = false;
        this._enabled = true;
        this.mouseService.getDocumentMouseUp().subscribe(() => {
            this.hideListMenu();
        });
    }
    ngOnInit() { }
    set readOnly(value) {
        this._readOnly = value;
        setTimeout(() => {
            try {
                this.input.nativeElement.setAttribute('unselectable', value ? 'on' : 'off');
                this.input.nativeElement.setAttribute('onselectstart', value ? 'return false' : '');
                this.input.nativeElement.readOnly = value;
            }
            catch (e) { }
        });
    }
    get readOnly() {
        return this._readOnly;
    }
    get enabled() {
        return this._enabled;
    }
    set enabled(value) {
        if (!value) {
            this.over = false;
        }
        this._enabled = value;
    }
    get key() {
        return this._key;
    }
    set key(key) {
        if (this.items != null) {
            this.items.forEach(i => i.selected = i.key === key);
        }
        this._key = key;
    }
    get inputValue() {
        var _a;
        if (this.items != null) {
            return ((_a = this.items.find(i => i.key === this._key)) === null || _a === void 0 ? void 0 : _a.caption) || '';
        }
        return '';
    }
    get selected() {
        var _a, _b;
        const vm = this.menuService.getVerticalMenu();
        return ((_a = vm) === null || _a === void 0 ? void 0 : _a.state) === 'expanded' && ((_b = vm) === null || _b === void 0 ? void 0 : _b.parent) === this.element;
    }
    get className() {
        return this.selected ? 'stiJsViewerDropDownListOver' : (this.enabled ? (this.over ? 'stiJsViewerDropDownListOver' : 'stiJsViewerDropDownList') : 'stiJsViewerDropDownListDisabled');
    }
    mouseover() {
        if (!this.enabled) {
            return;
        }
        this.over = true;
    }
    mouseout() {
        if (!this.enabled) {
            return;
        }
        this.over = false;
    }
    inputClick() {
        if (this.readOnly) {
            this.showListMenu();
        }
    }
    showListMenu() {
        this.menuService.addMenu({
            type: this.menuService.VERTICAL_MENU_NAME, name: this.menuService.VERTICAL_MENU_NAME, items: [], parent: this.element,
            verticalItems: this.items, itemStyleName: 'stiJsViewerMenuStandartItem', menuStyleName: 'stiJsViewerDropdownMenu',
            action: this.action, width: this.element.nativeElement.offsetWidth
        });
        setTimeout(() => {
            this.menuService.showMenu(this.menuService.VERTICAL_MENU_NAME);
        });
    }
    hideListMenu() {
        if (this.menuService.getVerticalMenu()) {
            this.menuService.getVerticalMenu().state = 'initialDown';
        }
    }
};
DropDownListComponent.ctorParameters = () => [
    { type: ModelService },
    { type: MouseService },
    { type: MenuService }
];
__decorate([
    ViewChild('input')
], DropDownListComponent.prototype, "input", void 0);
__decorate([
    ViewChild('element')
], DropDownListComponent.prototype, "element", void 0);
__decorate([
    Input()
], DropDownListComponent.prototype, "toolTip", void 0);
__decorate([
    Input()
], DropDownListComponent.prototype, "showImage", void 0);
__decorate([
    Input()
], DropDownListComponent.prototype, "width", void 0);
__decorate([
    Input()
], DropDownListComponent.prototype, "margin", void 0);
__decorate([
    Input()
], DropDownListComponent.prototype, "items", void 0);
__decorate([
    Input()
], DropDownListComponent.prototype, "styleDisplay", void 0);
__decorate([
    Input()
], DropDownListComponent.prototype, "verticalAlign", void 0);
__decorate([
    Output()
], DropDownListComponent.prototype, "action", void 0);
__decorate([
    Input()
], DropDownListComponent.prototype, "readOnly", null);
__decorate([
    Input()
], DropDownListComponent.prototype, "enabled", null);
__decorate([
    Input()
], DropDownListComponent.prototype, "key", null);
DropDownListComponent = __decorate([
    Component({
        selector: 'sti-drop-down-list',
        template: `
        <table #element [class]="className" cellpadding="0" cellspacing="0"
              [style.fontFamily]="model.options.toolbar.fontFamily"
              [style.color]="model.options.toolbar.fontColor"
              [style.margin]="margin"
              [style.verticalAlign]="verticalAlign"
              [style.display]="styleDisplay"
              [attr.title]="toolTip"
              (mouseover)="mouseover()"
              (mouseout)="mouseout()">
          <tbody>
            <tr class="stiJsViewerClearAllStyles">
              <td class="stiJsViewerClearAllStyles">
                <input #input
                [style.width.px]="width - (model.options.isTouchDevice ? 23 : 15) - (showImage ? 38 : 0)"
                [style.border]="0"
                [style.cursor]="readOnly ? 'default' : 'text'"
                [style.fontFamily]="model.options.toolbar.fontFamily"
                [style.color]="model.options.toolbar.fontColor"
                [style.height]="model.options.isTouchDevice ? '23px' : '18px'"
                [style.lineHeight]="model.options.isTouchDevice ? '23px' : '18px'"
                [style.visibility]="enabled ? 'visible' : 'hidden'"
                [value]="inputValue"
                (click)="inputClick()"
                class="stiJsViewerDropDownList_TextBox">
              </td>

              <td class="stiJsViewerClearAllStyles">
                  <sti-button [imageName]="'Arrows.' + (model.options.isTouchDevice ? 'Big' : 'Small') + 'ArrowDown.png'"
                     [styleName]="'stiJsViewerDropDownListButton'"
                     [imageSizesWidth]="model.options.isTouchDevice ? 16 : 8"
                     [imageSizesHeight]="model.options.isTouchDevice ? 16 : 8"
                     [height]="model.options.isTouchDevice ? '26px' : '21px'"
                     [enabled]="enabled"
                     [selected]="selected"
                     [margin]="'0'"
                     (action)="showListMenu()">
                  </sti-button>
              </td>
            </tr>
          </tbody>
        </table>

  `
    })
], DropDownListComponent);
export { DropDownListComponent };
