// stimulsoft-viewer-angular/esm2015/lib/controls/form-button.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
let FormButtonComponent = class FormButtonComponent {
    constructor() {
        this.captionAlign = 'center';
        this.action = new EventEmitter();
    }
    ngOnInit() { }
    onaction(event) {
        this.action.emit(event);
    }
};
__decorate([
    Input()
], FormButtonComponent.prototype, "caption", void 0);
__decorate([
    Input()
], FormButtonComponent.prototype, "actionName", void 0);
__decorate([
    Input()
], FormButtonComponent.prototype, "imageName", void 0);
__decorate([
    Input()
], FormButtonComponent.prototype, "imageCellWidth", void 0);
__decorate([
    Input()
], FormButtonComponent.prototype, "captionPadding", void 0);
__decorate([
    Input()
], FormButtonComponent.prototype, "margin", void 0);
__decorate([
    Input()
], FormButtonComponent.prototype, "captionAlign", void 0);
__decorate([
    Output()
], FormButtonComponent.prototype, "action", void 0);
FormButtonComponent = __decorate([
    Component({
        selector: 'sti-form-button',
        template: `
      <sti-button [styleName]="'stiJsViewerFormButton'"
                  [caption]="caption"
                  [actionName]="actionName"
                  [captionAlign]="captionAlign"
                  [innerTableWidth]="'100%'"
                  [minWidth]="'80px'"
                  [imageName]="imageName"
                  [imageCellWidth]="imageCellWidth"
                  [captionPadding]="captionPadding"
                  [margin]="margin"
                  (action)="onaction($event)">
      </sti-button>
  `
    })
], FormButtonComponent);
export { FormButtonComponent };
