// stimulsoft-viewer-angular/esm2015/lib/controls/radio-button.components.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
import { RadioButtonService } from '../services/radio-button.service';
let RadioButtonComponent = class RadioButtonComponent {
    constructor(model, radioService) {
        this.model = model;
        this.radioService = radioService;
        this.action = new EventEmitter();
        this.enabled = true;
        this.checked = false;
        this.over = false;
    }
    ngAfterViewInit() {
        this.radioService.addButton(this);
    }
    ngOnDestroy() {
        this.radioService.removeButton(this);
    }
    ngOnInit() { }
    mouseover() {
        if (!this.enabled) {
            return;
        }
        this.over = true;
    }
    mouseout() {
        if (!this.enabled) {
            return;
        }
        this.over = false;
    }
    click() {
        if (this.enabled) {
            this.checked = true;
            this.radioService.check(this);
            this.action.emit(true);
        }
    }
    uncheck() {
        this.checked = false;
        this.action.emit(false);
    }
    get outClass() {
        return !this.enabled ? 'stiJsViewerRadioButtonOutCircleDisabled' : (this.over ? 'stiJsViewerRadioButtonOutCircleOver' : 'stiJsViewerRadioButtonOutCircle');
    }
};
RadioButtonComponent.ctorParameters = () => [
    { type: ModelService },
    { type: RadioButtonService }
];
__decorate([
    Output()
], RadioButtonComponent.prototype, "action", void 0);
__decorate([
    Input()
], RadioButtonComponent.prototype, "groupName", void 0);
__decorate([
    Input()
], RadioButtonComponent.prototype, "name", void 0);
__decorate([
    Input()
], RadioButtonComponent.prototype, "enabled", void 0);
__decorate([
    Input()
], RadioButtonComponent.prototype, "checked", void 0);
__decorate([
    Input()
], RadioButtonComponent.prototype, "tooltip", void 0);
__decorate([
    Input()
], RadioButtonComponent.prototype, "caption", void 0);
__decorate([
    Input()
], RadioButtonComponent.prototype, "margin", void 0);
__decorate([
    Input()
], RadioButtonComponent.prototype, "paddingLeftLastCell", void 0);
RadioButtonComponent = __decorate([
    Component({
        selector: 'sti-radio-button',
        template: `
    <table [class]="enabled ? 'stiJsViewerRadioButton' : 'stiJsViewerRadioButtonDisabled'" cellpadding="0" cellspacing="0"
      [style.fontFamily]="model.options.toolbar.fontFamily"
      [style.margin]="margin"
      [attr.title]="tooltip"
      (mouseover)="mouseover()" (mouseout)="mouseout()" (click)="click()">
      <tbody>
        <tr class="stiJsViewerClearAllStyles">
          <td class="stiJsViewerClearAllStyles">
            <div class="stiJsViewerRadioButtonOutCircle" [class]="outClass">
              <div class="stiJsViewerRadioButtonInnerCircle"
                  [style.margin]="model.options.isTouchDevice ? '4px' : '3px'"
                  [style.width]="model.options.isTouchDevice ? '9px' : '7px'"
                  [style.height]="model.options.isTouchDevice ? '9px' : '7px'"
                  [style.visibility]="checked ? 'visible' : 'hidden'"
                  [style.opacity]="enabled ? 1 : 0.5">
              </div>
            </div>
          </td>
          <td *ngIf="caption != null" class="stiJsViewerClearAllStyles" style="padding-left: 4px; white-space: nowrap;">
            {{caption}}
          </td>
          <td class="stiJsViewerClearAllStyles" [style.paddingLeft]="paddingLeftLastCell">
            <ng-content></ng-content>
          </td>
        </tr>
      </tbody>
    </table>
  `
    })
], RadioButtonComponent);
export { RadioButtonComponent };
