// stimulsoft-viewer-angular/esm2015/lib/controls/text-box.componet.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
let TextBoxComponent = class TextBoxComponent {
    constructor(model) {
        this.model = model;
        this.enabled = true;
        this.focusOnCreate = false;
        this.action = new EventEmitter();
        this.onchange = new EventEmitter();
        this.onblur = new EventEmitter();
        this.selected = false;
        this.focused = false;
        this.over = false;
        this._readOnly = false;
        this.styleName = 'stiJsViewerTextBox';
    }
    ngAfterViewInit() {
        if (this.focusOnCreate) {
            setTimeout(() => {
                this.element.nativeElement.focus();
            });
        }
    }
    keypress(event) {
        if (!this.enabled) {
            return false;
        }
        if (event.keyCode === 13) {
            this.action.emit(event.target);
        }
    }
    onchanged(event) {
        this.onchange.emit(event.target);
    }
    focus() {
        this.oldValue = this.element.nativeElement.value;
    }
    get height() {
        return this.model.options.isTouchDevice ? '26px' : '21px';
    }
    get className() {
        if (this._readOnly) {
            return this.styleName + ' ' + this.styleName + 'Default';
        }
        return this.styleName + ' ' + this.styleName + (this.selected ? 'Over' : (this.enabled ? (this.over ? 'Over' : 'Default') : 'Disabled'));
    }
    set readOnly(value) {
        this._readOnly = value;
        setTimeout(() => {
            try {
                this.element.nativeElement.setAttribute('unselectable', value ? 'on' : 'off');
                this.element.nativeElement.setAttribute('onselectstart', value ? 'return false' : '');
                this.element.nativeElement.readOnly = value;
            }
            catch (e) { }
        });
    }
    get readOnly() {
        return this._readOnly;
    }
    ngOnInit() { }
};
TextBoxComponent.ctorParameters = () => [
    { type: ModelService }
];
__decorate([
    ViewChild('element')
], TextBoxComponent.prototype, "element", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "width", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "actionLostFocus", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "tooltip", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "enabled", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "value", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "margin", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "focusOnCreate", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "maxLength", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "color", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "type", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "padding", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "border", void 0);
__decorate([
    Output()
], TextBoxComponent.prototype, "action", void 0);
__decorate([
    Output()
], TextBoxComponent.prototype, "onchange", void 0);
__decorate([
    Output()
], TextBoxComponent.prototype, "onblur", void 0);
__decorate([
    Input()
], TextBoxComponent.prototype, "readOnly", null);
TextBoxComponent = __decorate([
    Component({
        selector: 'sti-text-box',
        template: `
    <input #element
    [style]="{fontFamily:model.options.toolbar.fontFamily, height: height, lineHeight: height, boxSizing:'content-box'}"
    [style.color]="color ? color : (model.options.toolbar.fontColor != '' ? model.options.toolbar.fontColor : '')"
    [style.width]="width != null ? width + 'px' : ''"
    [style.margin]="margin"
    [style.cursor]="readOnly ? 'default' : ''"
    [style.padding]="padding"
    [style.border]="border"
    [title]="tooltip || ''"
    [class]="className"
    [value]="value || ''"
    [attr.maxLength]="maxLength"
    [attr.type]="type"
    (mouseover)="over=true"
    (mouseleave)="over=false"
    (focused)="focused=true; selected=true"
    (blur)="focused=false; selected=false; onblur.emit($event.target)"
    (keypress)="keypress($event)"
    (keyup)="onchanged($event)"
    (focus)="focus()"/>
  `
    })
], TextBoxComponent);
export { TextBoxComponent };
