// stimulsoft-viewer-angular/esm2015/lib/controls/text-area.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
let TextAreaComponent = class TextAreaComponent {
    constructor() {
        this.enabled = true;
        this.action = new EventEmitter();
        this.onchange = new EventEmitter();
        this.onblur = new EventEmitter();
        this.selected = false;
        this.focused = false;
        this.over = false;
        this._readOnly = false;
        this.styleName = 'stiJsViewerTextBox';
    }
    ngOnInit() { }
    keypress(event) {
        if (!this.enabled) {
            return false;
        }
        if (event.keyCode === 13) {
            this.action.emit(event.target);
        }
    }
    onchanged(event) {
        this.onchange.emit(event.target);
    }
    get className() {
        if (this._readOnly) {
            return this.styleName + ' ' + this.styleName + 'Default';
        }
        return this.styleName + ' ' + this.styleName + (this.selected ? 'Over' : (this.enabled ? (this.over ? 'Over' : 'Default') : 'Disabled'));
    }
    get readOnly() {
        return this._readOnly;
    }
};
__decorate([
    Input()
], TextAreaComponent.prototype, "width", void 0);
__decorate([
    Input()
], TextAreaComponent.prototype, "height", void 0);
__decorate([
    Input()
], TextAreaComponent.prototype, "enabled", void 0);
__decorate([
    Input()
], TextAreaComponent.prototype, "value", void 0);
__decorate([
    Input()
], TextAreaComponent.prototype, "padding", void 0);
__decorate([
    Input()
], TextAreaComponent.prototype, "margin", void 0);
__decorate([
    Output()
], TextAreaComponent.prototype, "action", void 0);
__decorate([
    Output()
], TextAreaComponent.prototype, "onchange", void 0);
__decorate([
    Output()
], TextAreaComponent.prototype, "onblur", void 0);
TextAreaComponent = __decorate([
    Component({
        selector: 'sti-text-area',
        template: `
    <textarea
      [style.width.px]="width"
      [style.minWidth.px]="width"
      [style.height.px]="height"
      [style.minHeight.px]="height"
      [class]="className"
      [value]="value || ''"
      [style.margin]="margin"
      [style.cursor]="readOnly ? 'default' : ''"
      [style.padding]="padding"
      style="padding-top: 3px, font-family: Arial"
      (mouseover)="over=true"
      (mouseleave)="over=false"
      (focused)="focused=true; selected=true"
      (blur)="focused=false; selected=false; onblur.emit($event.target)"
      (keypress)="keypress($event)"
      (keyup)="onchanged($event)">
    </textarea>
  `
    })
], TextAreaComponent);
export { TextAreaComponent };
