// stimulsoft-viewer-angular/esm2015/lib/forms/error-message-form.component.js
import { __decorate } from "tslib";
import { Component } from '@angular/core';
import { ModelService } from '../services/model.service';
import { FormService } from './form.service';
let ErrorMessageFormComponent = class ErrorMessageFormComponent {
    constructor(model, formService) {
        this.model = model;
        this.formService = formService;
    }
    ngOnInit() { }
    getImage() {
        var _a;
        const messageType = (_a = this.model.errorMessage) === null || _a === void 0 ? void 0 : _a.type;
        if (messageType === 'Warning') {
            return this.model.img('MsgFormWarning.png');
        }
        else if (messageType === true || messageType === 'Info') {
            return this.model.img('MsgFormInfo.png');
        }
        else {
            return this.model.img('MsgFormError.png');
        }
    }
};
ErrorMessageFormComponent.ctorParameters = () => [
    { type: ModelService },
    { type: FormService }
];
ErrorMessageFormComponent = __decorate([
    Component({
        selector: 'sti-error-message-form',
        template: `
    <sti-base-form [fontFamily]="model.options.toolbar.fontFamily"
      [color]="model.options.toolbar.fontColor"
      [name]="'errorMessageForm'"
      [caption]="model.loc(model.errorMessage?.type == 'Error' ? 'Error' : 'FormViewerTitle')"
      [level]="4"
      [defaultTop]="150"
      [showCancel]="false"
      (action)="formService.closeForm('errorMessageForm')">
      <table class="stiJsViewerClearAllStyles" cellpadding="0" cellspacing="0">
        <tbody>
          <tr class="stiJsViewerClearAllStyles">
            <td class="stiJsViewerClearAllStyles">
              <img [attr.src]="getImage()" style="padding: 15px; height: 32px; width: 32px;">
            </td>
            <td class="stiJsViewerMessagesFormDescription" style="max-width: 600px;"
              [style.color]="model.options.toolbar.fontColor" [innerHTML]="model.errorMessage?.error">
            </td>
          </tr>
        </tbody>
      </table>
    </sti-base-form>
  `
    })
], ErrorMessageFormComponent);
export { ErrorMessageFormComponent };
