// stimulsoft-viewer-angular/esm2015/lib/forms/form.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { MouseService } from '../services/mouse.service';
import { MenuService } from '../menu/menu.service';
import { HelperService } from '../services/helper.service';
import { ModelService } from '../services/model.service';
let FormService = class FormService {
    constructor(mouseService, menuService, helper, model) {
        this.mouseService = mouseService;
        this.menuService = menuService;
        this.helper = helper;
        this.model = model;
        this.mouseX = 0;
        this.mouseY = 0;
        this.formX = 0;
        this.formY = 0;
        mouseService.getDocumentMouseMove().subscribe((event) => {
            this.move(event);
        });
        mouseService.getDocumentMouseUp().subscribe((event) => {
            this.stopMove();
        });
    }
    get form() {
        return this.model.form;
    }
    set form(form) {
        this.model.form = form;
    }
    showForm(name, formData) {
        this.form = { name, left: 0, top: 0, isMooving: false, formData };
    }
    closeForm(name) {
        var _a, _b;
        if (((_a = this.form) === null || _a === void 0 ? void 0 : _a.name) === 'notificationForm' && ((_b = this.model.notificationFormOptions) === null || _b === void 0 ? void 0 : _b.action)) {
            this.model.notificationFormOptions.action();
        }
        this.form = null;
    }
    centerForm(baseForm, defaultTop) {
        if (this.form != null) {
            setTimeout(() => {
                const position = this.helper.setObjectToCenter(baseForm.element.nativeElement, defaultTop);
                this.form.left = position.left;
                this.form.top = position.top;
                this.form.level = baseForm.level;
            });
        }
    }
    startMove(name, event, touchEvent) {
        var _a, _b;
        if (event || touchEvent && touchEvent.changedTouches.length > 0) {
            this.formX = this.form.left;
            this.formY = this.form.top;
            this.mouseX = ((_a = event) === null || _a === void 0 ? void 0 : _a.screenX) || touchEvent.changedTouches[0].screenX;
            this.mouseY = ((_b = event) === null || _b === void 0 ? void 0 : _b.screenY) || touchEvent.changedTouches[0].screenY;
            this.form.isMooving = true;
        }
    }
    move(event) {
        var _a;
        if ((_a = this.form) === null || _a === void 0 ? void 0 : _a.isMooving) {
            const screenX = event.screenX || (event.changedTouches.length > 0 ? event.changedTouches[0].screenX : -1);
            const screenY = event.screenY || (event.changedTouches.length > 0 ? event.changedTouches[0].screenY : -1);
            this.form.left = this.formX + (screenX - this.mouseX);
            this.form.top = this.formY + (screenY - this.mouseY);
            this.menuService.closeAllMenus();
        }
    }
    stopMove() {
        var _a;
        if ((_a = this.form) === null || _a === void 0 ? void 0 : _a.isMooving) {
            this.form.isMooving = false;
        }
    }
};
FormService.ctorParameters = () => [
    { type: MouseService },
    { type: MenuService },
    { type: HelperService },
    { type: ModelService }
];
FormService = __decorate([
    Injectable()
], FormService);
export { FormService };
