// stimulsoft-viewer-angular/esm2015/lib/forms/group-panel.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../services/model.service';
import { trigger, state, transition, animate, style } from '@angular/animations';
let GroupPanelComponent = class GroupPanelComponent {
    constructor(model) {
        this.model = model;
        this.opened = false;
        this.action = new EventEmitter();
        this.display = 'none';
    }
    ngOnInit() { }
    startAnimation(event) {
        if (event.toState === 'opened') {
            this.display = '';
        }
    }
    doneAnimation(event) {
        if (event.toState === 'closed') {
            this.display = 'none';
        }
    }
};
GroupPanelComponent.ctorParameters = () => [
    { type: ModelService }
];
__decorate([
    Input()
], GroupPanelComponent.prototype, "width", void 0);
__decorate([
    Input()
], GroupPanelComponent.prototype, "caption", void 0);
__decorate([
    Input()
], GroupPanelComponent.prototype, "innerPadding", void 0);
__decorate([
    Input()
], GroupPanelComponent.prototype, "opened", void 0);
__decorate([
    Input()
], GroupPanelComponent.prototype, "margin", void 0);
__decorate([
    Output()
], GroupPanelComponent.prototype, "action", void 0);
GroupPanelComponent = __decorate([
    Component({
        selector: 'sti-group-panel',
        template: `
  <div #groupPanel [style.fontFamily]="model.options.toolbar.fontFamily"
       [style.color]="model.options.toolbarFontColor"
       [style.minWidth.px]="width"
       [style.margin]="margin"
       style="overflow: hidden">
      <sti-form-button [caption]="caption"
          [imageName]="opened ? 'Arrows.BigArrowDown.png' : 'Arrows.BigArrowRight.png'"
          [imageCellWidth]="'1px'"
          [captionPadding]="'0 15px 0 5px'"
          [margin]="'0px'"
          [captionAlign]="'left'"
          (action)="opened = !opened; action.emit(opened)">
      </sti-form-button>

      <div class="stiJsViewerGroupPanelContainer"
           [style.padding]="innerPadding"
           [style.display]="display"
           [@state]="opened ? 'opened' : 'closed'"
           (@state.start)="startAnimation($event)"
           (@state.done)="doneAnimation($event)">
          <ng-content>

          </ng-content>
      </div>


  </div>
  `,
        animations: [
            trigger('state', [
                state('opened', style({ height: '*' })),
                state('closed', style({ height: '0px' })),
                transition('* => *', [
                    animate('150ms ease-in-out')
                ])
            ])
        ]
    })
], GroupPanelComponent);
export { GroupPanelComponent };
