// stimulsoft-viewer-angular/esm2015/lib/services/mouse.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
let MouseService = class MouseService {
    constructor() {
        this.documetMouseDown = new Subject();
        this.documetMouseMove = new Subject();
        this.documetMouseUp = new Subject();
        document.addEventListener('mouseup', (event) => {
            this.documetMouseUp.next(event);
        });
        document.addEventListener('mousemove', (event) => {
            this.documetMouseMove.next(event);
        });
        document.addEventListener('mousedown', (event) => {
            this.documetMouseDown.next(event);
        });
    }
    getDocumentMouseUp() {
        return this.documetMouseUp.asObservable();
    }
    getDocumentMouseMove() {
        return this.documetMouseMove.asObservable();
    }
    getDocumentMouseDown() {
        return this.documetMouseDown.asObservable();
    }
};
MouseService = __decorate([
    Injectable()
], MouseService);
export { MouseService };
