// stimulsoft-viewer-angular/esm2015/lib/components/open-dialog.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ControllerService } from '../services/controller.service';
let OpenDialogComponent = class OpenDialogComponent {
    constructor(model, controller) {
        this.model = model;
        this.controller = controller;
    }
    ngOnChanges(changes) {
        if (this.fileMask != null && this.model.openDialogFileMask != null) {
            setTimeout(() => {
                this.element.nativeElement.focus();
                this.element.nativeElement.click();
            });
        }
    }
    ngOnInit() { }
    onchange(event) {
        const files = event.target.files;
        const fileName = files[0] ? files[0].name : 'Report';
        const filePath = event.target.value;
        const reader = new FileReader();
        reader.onload = (e) => {
            this.controller.loadFile(fileName, e.target.result);
        };
        this.model.openDialogFileMask = null;
        reader.readAsDataURL(files[0]);
    }
};
OpenDialogComponent.ctorParameters = () => [
    { type: ModelService },
    { type: ControllerService }
];
__decorate([
    ViewChild('element')
], OpenDialogComponent.prototype, "element", void 0);
__decorate([
    Input()
], OpenDialogComponent.prototype, "fileMask", void 0);
OpenDialogComponent = __decorate([
    Component({
        selector: 'sti-open-dialog',
        template: `
    <input #element *ngIf="model.openDialogFileMask != null"  type="file" name="files[]" multiple="" style="display: none;"
      [attr.accept]="model.openDialogFileMask"
      (change)="onchange($event)">
  `
    })
], OpenDialogComponent);
export { OpenDialogComponent };
