// stimulsoft-viewer-angular/esm2015/lib/components/toolbar-dop.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild } from '@angular/core';
import { ModelService } from '../services/model.service';
import { ToolbarService } from '../services/toolbar.service';
import { ControllerService } from '../services/controller.service';
let ToolbarDopComponent = class ToolbarDopComponent {
    constructor(model, controller, toolbarService) {
        this.model = model;
        this.controller = controller;
        this.toolbarService = toolbarService;
        this.marginLeft = '1px';
        this.viewInit = false;
    }
    ngAfterViewInit() {
        this.viewInit = true;
        setTimeout(() => {
            var _a, _b;
            const width = (_b = (_a = this.mainTable) === null || _a === void 0 ? void 0 : _a.nativeElement) === null || _b === void 0 ? void 0 : _b.offsetWidth;
            this.marginLeft = this.viewInit && this.model.options.toolbar.alignment === 'Center' && width ? `calc(50% - ${Math.trunc(width / 2)}px)` : '1px';
        });
    }
    ngOnInit() { }
};
ToolbarDopComponent.ctorParameters = () => [
    { type: ModelService },
    { type: ControllerService },
    { type: ToolbarService }
];
__decorate([
    ViewChild('mainTable')
], ToolbarDopComponent.prototype, "mainTable", void 0);
__decorate([
    Input()
], ToolbarDopComponent.prototype, "comps", void 0);
__decorate([
    Input()
], ToolbarDopComponent.prototype, "align", void 0);
ToolbarDopComponent = __decorate([
    Component({
        selector: 'sti-toolbar-dop',
        template: `
    <table #mainTable
      class="stiJsViewerClearAllStyles"
      cellpadding="0"
      cellspacing="0"
      style="margin: 1px;"
      [attr.align]="align"
      [style.marginLeft]="marginLeft">
      <tbody>
        <tr class="stiJsViewerClearAllStyles">
          <td *ngFor="let comp of comps" class="stiJsViewerClearAllStyles">
            <sti-button *ngIf="comp.type=='button'"
              [caption]="model.options.toolbar.showButtonCaptions ? comp.caption : null"
              [imageName]="comp.img"
              [arrow]="!model.options.isMobileDevice ? comp.arrow : null"
              [menuItems]="comp.menuItems"
              [actionName]="comp.action"
              [tooltip]="comp.tooltip"
              [enabled]="toolbarService.enabled(comp)"
              [display]="toolbarService.display(comp)"
              [innerTableWidth]="toolbarService.getInnerTableWidth(comp)"
              [width]="toolbarService.getButonWidth(comp)"
              [height]="model.options.toolbar.displayMode == 'Separated' ? (model.options.isMobileDevice ? '0.5in' : '28px') : null"
              (action)="controller.action({ name: comp.action });"
              [selected]="toolbarService.selected(comp)"></sti-button>
            <sti-toolbar-separator *ngIf="comp.type=='separator'"></sti-toolbar-separator>
            <sti-page-control *ngIf="comp.type=='pageControl'" [enabled]="!(model.reportParams.pagesCount <= 1 || toolbarService.disableNaviButtons())" [count]="model.reportParams.pagesCount" ></sti-page-control>
        </td>
      </tr>
    </tbody>
  </table>
  `
    })
], ToolbarDopComponent);
export { ToolbarDopComponent };
