// @coreui/angular/fesm2022/coreui-angular.mjs
import * as i0 from '@angular/core';
import { DOCUMENT, Inject, Injectable, HostListener, Input, Directive, NgModule, HostBinding, Component, EventEmitter, Output, Pipe, forwardRef } from '@angular/core';
import { NgClass, AsyncPipe } from '@angular/common';
import * as i1 from '@angular/router';
import { NavigationEnd, RouterLink, RouterModule } from '@angular/router';
import { BehaviorSubject } from 'rxjs';
import { filter } from 'rxjs/operators';

const sidebarCssClasses = [
    'sidebar-show',
    'sidebar-sm-show',
    'sidebar-md-show',
    'sidebar-lg-show',
    'sidebar-xl-show'
];
const asideMenuCssClasses = [
    'aside-menu-show',
    'aside-menu-sm-show',
    'aside-menu-md-show',
    'aside-menu-lg-show',
    'aside-menu-xl-show'
];

const RemoveClasses = (NewClassNames) => {
    const MatchClasses = NewClassNames.map((Class) => document.body.classList.contains(Class));
    return MatchClasses.indexOf(true) !== -1;
};
const ToggleClasses = (Toggle, ClassNames) => {
    const Level = ClassNames.indexOf(Toggle);
    const NewClassNames = ClassNames.slice(0, Level + 1);
    if (RemoveClasses(NewClassNames)) {
        NewClassNames.map((Class) => document.body.classList.remove(Class));
    }
    else {
        document.body.classList.add(Toggle);
    }
};
class ClassToggler {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
    }
    removeClasses(NewClassNames) {
        const MatchClasses = NewClassNames.map((Class) => this.document.body.classList.contains(Class));
        return MatchClasses.indexOf(true) !== -1;
    }
    toggleClasses(Toggle, ClassNames) {
        const Level = ClassNames.indexOf(Toggle);
        const NewClassNames = ClassNames.slice(0, Level + 1);
        if (this.removeClasses(NewClassNames)) {
            NewClassNames.map((Class) => this.renderer.removeClass(this.document.body, Class));
        }
        else {
            this.renderer.addClass(this.document.body, Toggle);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: ClassToggler, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: ClassToggler }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: ClassToggler, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }] });

/**
 * Allows the sidebar to be toggled via click.
 */
class SidebarToggleDirective {
    constructor(classToggler) {
        this.classToggler = classToggler;
        this.breakpoint = false;
    }
    ngOnInit() {
        this.bp = this.breakpoint;
    }
    toggleOpen($event) {
        $event.preventDefault();
        const cssClass = this.bp ? `sidebar-${this.bp}-show` : sidebarCssClasses[0];
        this.classToggler.toggleClasses(cssClass, sidebarCssClasses);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarToggleDirective, deps: [{ token: ClassToggler }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: SidebarToggleDirective, isStandalone: true, selector: "[appSidebarToggler]", inputs: { breakpoint: ["appSidebarToggler", "breakpoint"] }, host: { listeners: { "click": "toggleOpen($event)" } }, providers: [ClassToggler], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarToggleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appSidebarToggler]',
                    providers: [ClassToggler],
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: ClassToggler }], propDecorators: { breakpoint: [{
                type: Input,
                args: ['appSidebarToggler']
            }], toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
class SidebarMinimizeDirective {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
    }
    toggleOpen($event) {
        $event.preventDefault();
        const body = this.document.body;
        body.classList.contains('sidebar-minimized') ?
            this.renderer.removeClass(body, 'sidebar-minimized') :
            this.renderer.addClass(body, 'sidebar-minimized');
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarMinimizeDirective, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: SidebarMinimizeDirective, isStandalone: true, selector: "[appSidebarMinimizer]", host: { listeners: { "click": "toggleOpen($event)" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarMinimizeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appSidebarMinimizer]',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }], propDecorators: { toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
class MobileSidebarToggleDirective {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
    }
    toggleOpen($event) {
        $event.preventDefault();
        const body = this.document.body;
        body.classList.contains('sidebar-show') ?
            this.renderer.removeClass(body, 'sidebar-show') :
            this.renderer.addClass(body, 'sidebar-show');
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: MobileSidebarToggleDirective, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: MobileSidebarToggleDirective, isStandalone: true, selector: "[appMobileSidebarToggler]", host: { listeners: { "click": "toggleOpen($event)" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: MobileSidebarToggleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appMobileSidebarToggler]',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }], propDecorators: { toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
/**
 * Allows the off-canvas sidebar to be closed via click.
 */
class SidebarOffCanvasCloseDirective {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
    }
    toggleOpen($event) {
        $event.preventDefault();
        const body = this.document.body;
        if (body.classList.contains('sidebar-off-canvas')) {
            body.classList.contains('sidebar-show') ?
                this.renderer.removeClass(body, 'sidebar-show') :
                this.renderer.addClass(body, 'sidebar-show');
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarOffCanvasCloseDirective, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: SidebarOffCanvasCloseDirective, isStandalone: true, selector: "[appSidebarClose]", host: { listeners: { "click": "toggleOpen($event)" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarOffCanvasCloseDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appSidebarClose]',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }], propDecorators: { toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
class BrandMinimizeDirective {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
    }
    toggleOpen($event) {
        $event.preventDefault();
        const body = this.document.body;
        body.classList.contains('brand-minimized') ?
            this.renderer.removeClass(body, 'brand-minimized') :
            this.renderer.addClass(body, 'brand-minimized');
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: BrandMinimizeDirective, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: BrandMinimizeDirective, isStandalone: true, selector: "[appBrandMinimizer]", host: { listeners: { "click": "toggleOpen($event)" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: BrandMinimizeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appBrandMinimizer]',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }], propDecorators: { toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
/**
 * Allows the aside to be toggled via click.
 */
class AsideToggleDirective {
    constructor(classToggler) {
        this.classToggler = classToggler;
        this.breakpoint = false;
    }
    ngOnInit() {
        this.bp = this.breakpoint;
    }
    toggleOpen($event) {
        $event.preventDefault();
        const cssClass = this.bp ? `aside-menu-${this.bp}-show` : asideMenuCssClasses[0];
        this.classToggler.toggleClasses(cssClass, asideMenuCssClasses);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AsideToggleDirective, deps: [{ token: ClassToggler }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: AsideToggleDirective, isStandalone: true, selector: "[appAsideMenuToggler]", inputs: { breakpoint: ["appAsideMenuToggler", "breakpoint"] }, host: { listeners: { "click": "toggleOpen($event)" } }, providers: [ClassToggler], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AsideToggleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appAsideMenuToggler]',
                    providers: [ClassToggler],
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: ClassToggler }], propDecorators: { breakpoint: [{
                type: Input,
                args: ['appAsideMenuToggler']
            }], toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
class HtmlAttributesDirective {
    constructor(renderer, el) {
        this.renderer = renderer;
        this.el = el;
        this.appHtmlAttr = {};
    }
    ngOnInit() {
        const attribs = this.appHtmlAttr;
        for (const attr in attribs) {
            if (attr === 'style' && typeof (attribs[attr]) === 'object') {
                this.setStyle(attribs[attr]);
            }
            else if (attr === 'class') {
                this.addClass(attribs[attr]);
            }
            else {
                this.setAttrib(attr, attribs[attr]);
            }
        }
    }
    setStyle(styles) {
        for (const style in styles) {
            if (style) {
                this.renderer.setStyle(this.el.nativeElement, style, styles[style]);
            }
        }
    }
    addClass(classes) {
        const classArray = (Array.isArray(classes) ? classes : classes.split(' '));
        classArray.filter((element) => element.length > 0).forEach(element => {
            this.renderer.addClass(this.el.nativeElement, element);
        });
    }
    setAttrib(key, value) {
        value !== null ?
            this.renderer.setAttribute(this.el.nativeElement, key, value) :
            this.renderer.removeAttribute(this.el.nativeElement, key);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: HtmlAttributesDirective, deps: [{ token: i0.Renderer2 }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: HtmlAttributesDirective, isStandalone: true, selector: "[appHtmlAttr]", inputs: { appHtmlAttr: "appHtmlAttr" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: HtmlAttributesDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appHtmlAttr]',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: i0.Renderer2 }, { type: i0.ElementRef }], propDecorators: { appHtmlAttr: [{
                type: Input
            }] } });

class LayoutModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: LayoutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: LayoutModule, imports: [AsideToggleDirective,
            BrandMinimizeDirective,
            SidebarToggleDirective,
            SidebarMinimizeDirective,
            SidebarOffCanvasCloseDirective,
            MobileSidebarToggleDirective,
            HtmlAttributesDirective], exports: [AsideToggleDirective,
            BrandMinimizeDirective,
            MobileSidebarToggleDirective,
            SidebarToggleDirective,
            SidebarMinimizeDirective,
            SidebarOffCanvasCloseDirective,
            HtmlAttributesDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: LayoutModule, providers: [
            ClassToggler
        ] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: LayoutModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AsideToggleDirective,
                        BrandMinimizeDirective,
                        SidebarToggleDirective,
                        SidebarMinimizeDirective,
                        SidebarOffCanvasCloseDirective,
                        MobileSidebarToggleDirective,
                        HtmlAttributesDirective
                    ],
                    exports: [
                        AsideToggleDirective,
                        BrandMinimizeDirective,
                        MobileSidebarToggleDirective,
                        SidebarToggleDirective,
                        SidebarMinimizeDirective,
                        SidebarOffCanvasCloseDirective,
                        HtmlAttributesDirective
                    ],
                    providers: [
                        ClassToggler
                    ]
                }]
        }] });

function Replace(el) {
    const nativeElement = el.nativeElement;
    // @ts-ignore
    const parentElement = nativeElement.parentElement;
    // move all children out of the element
    while (nativeElement.firstChild) {
        parentElement.insertBefore(nativeElement.firstChild, nativeElement);
    }
    // remove the empty element(the host)
    parentElement.removeChild(nativeElement);
}

class AppAsideComponent {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
        this.fixedClass = 'aside-menu-fixed';
        this.asideMenuClass = true;
    }
    ngOnInit() {
        this.isFixed(this.fixed);
        this.isOffCanvas(this.offCanvas);
        this.displayBreakpoint(this.display);
    }
    ngOnDestroy() {
        this.renderer.removeClass(this.document.body, this.fixedClass);
    }
    isFixed(fixed = this.fixed) {
        if (fixed) {
            this.renderer.addClass(this.document.body, this.fixedClass);
        }
    }
    isOffCanvas(offCanvas = this.offCanvas) {
        if (offCanvas) {
            this.renderer.addClass(this.document.body, 'aside-menu-off-canvas');
        }
    }
    displayBreakpoint(display = this.display) {
        if (display !== false) {
            const cssClass = this.display ? `aside-menu-${this.display}-show` : asideMenuCssClasses[0];
            this.renderer.addClass(this.document.body, cssClass);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppAsideComponent, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppAsideComponent, isStandalone: true, selector: "app-aside, cui-aside", inputs: { display: "display", fixed: "fixed", offCanvas: "offCanvas" }, host: { properties: { "class.aside-menu": "this.asideMenuClass" } }, ngImport: i0, template: `<ng-content />`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppAsideComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-aside, cui-aside',
                    template: `<ng-content />`,
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }], propDecorators: { display: [{
                type: Input
            }], fixed: [{
                type: Input
            }], offCanvas: [{
                type: Input
            }], asideMenuClass: [{
                type: HostBinding,
                args: ['class.aside-menu']
            }] } });

class AppAsideModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppAsideModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppAsideModule, imports: [AppAsideComponent], exports: [AppAsideComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppAsideModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppAsideModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AppAsideComponent
                    ],
                    exports: [
                        AppAsideComponent
                    ],
                }]
        }] });

class AppBreadcrumbService {
    constructor(router, route) {
        this.router = router;
        this.route = route;
        this.breadcrumbSubject = new BehaviorSubject(new Array());
        this.breadcrumbs = this.breadcrumbSubject.asObservable();
        this.router.events.pipe(filter(event => event instanceof NavigationEnd)).subscribe((event) => {
            const breadcrumbs = [];
            let currentRoute = this.route.root;
            let url = '';
            do {
                const childrenRoutes = currentRoute.children;
                currentRoute = null;
                // tslint:disable-next-line:no-shadowed-variable
                childrenRoutes.forEach(route => {
                    if (route.outlet === 'primary') {
                        const routeSnapshot = route.snapshot;
                        url += '/' + routeSnapshot.url.map(segment => segment.path).join('/');
                        breadcrumbs.push({
                            label: route.snapshot.data,
                            url
                        });
                        currentRoute = route;
                    }
                });
            } while (currentRoute);
            this.breadcrumbSubject.next(Object.assign([], breadcrumbs));
            return breadcrumbs;
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbService, deps: [{ token: i1.Router }, { token: i1.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.Router }, { type: i1.ActivatedRoute }] });

class AppBreadcrumbComponent {
    constructor(document, renderer, service, el) {
        this.document = document;
        this.renderer = renderer;
        this.service = service;
        this.el = el;
        this.fixedClass = 'breadcrumb-fixed';
    }
    ngOnInit() {
        Replace(this.el);
        this.isFixed(this.fixed);
        this.breadcrumbs = this.service.breadcrumbs;
    }
    ngOnDestroy() {
        this.renderer.removeClass(this.document.body, this.fixedClass);
    }
    isFixed(fixed = this.fixed) {
        if (fixed) {
            this.renderer.addClass(this.document.body, this.fixedClass);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbComponent, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }, { token: AppBreadcrumbService }, { token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.2.2", type: AppBreadcrumbComponent, isStandalone: true, selector: "app-breadcrumb", inputs: { fixed: "fixed" }, ngImport: i0, template: `
    @for (breadcrumb of breadcrumbs | async; track breadcrumb; let last = $last) {
      @if (breadcrumb.label.title && (breadcrumb.url.slice(-1) == '/' || last)) {
        <li class="breadcrumb-item"
          [ngClass]="{active: last}">
          @if (!last) {
            <a [routerLink]="breadcrumb.url">{{breadcrumb.label.title}}</a>
          }
          @if (last) {
            <span [routerLink]="breadcrumb.url">{{breadcrumb.label.title}}</span>
          }
        </li>
      }
    }
    `, isInline: true, dependencies: [{ kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "info", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "pipe", type: AsyncPipe, name: "async" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-breadcrumb',
                    template: `
    @for (breadcrumb of breadcrumbs | async; track breadcrumb; let last = $last) {
      @if (breadcrumb.label.title && (breadcrumb.url.slice(-1) == '/' || last)) {
        <li class="breadcrumb-item"
          [ngClass]="{active: last}">
          @if (!last) {
            <a [routerLink]="breadcrumb.url">{{breadcrumb.label.title}}</a>
          }
          @if (last) {
            <span [routerLink]="breadcrumb.url">{{breadcrumb.label.title}}</span>
          }
        </li>
      }
    }
    `,
                    imports: [NgClass, RouterLink, AsyncPipe]
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }, { type: AppBreadcrumbService }, { type: i0.ElementRef }], propDecorators: { fixed: [{
                type: Input
            }] } });

class CuiBreadcrumbComponent {
    constructor(document, renderer, service) {
        this.document = document;
        this.renderer = renderer;
        this.service = service;
        this.fixedClass = 'breadcrumb-fixed';
    }
    ngOnInit() {
        this.isFixed(this.fixed);
        this.breadcrumbs = this.service.breadcrumbs;
    }
    ngOnDestroy() {
        this.renderer.removeClass(this.document.body, this.fixedClass);
    }
    isFixed(fixed = this.fixed) {
        if (fixed) {
            this.renderer.addClass(this.document.body, this.fixedClass);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: CuiBreadcrumbComponent, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }, { token: AppBreadcrumbService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.2.2", type: CuiBreadcrumbComponent, isStandalone: true, selector: "cui-breadcrumb", inputs: { fixed: "fixed" }, ngImport: i0, template: "<ol class=\"breadcrumb\">\n  @for (breadcrumb of breadcrumbs | async; track breadcrumb; let last = $last) {\n    @if (breadcrumb.label.title && (breadcrumb.url.slice(-1) == '/' || last)) {\n      <li class=\"breadcrumb-item\"\n        [ngClass]=\"{active: last}\">\n        @if (!last) {\n          <a [routerLink]=\"breadcrumb.url\">{{breadcrumb.label.title}}</a>\n        }\n        @if (last) {\n          <span [routerLink]=\"breadcrumb.url\">{{breadcrumb.label.title}}</span>\n        }\n      </li>\n    }\n  }\n  <ng-content />\n</ol>\n", dependencies: [{ kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "info", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "pipe", type: AsyncPipe, name: "async" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: CuiBreadcrumbComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cui-breadcrumb', imports: [NgClass, RouterLink, AsyncPipe], template: "<ol class=\"breadcrumb\">\n  @for (breadcrumb of breadcrumbs | async; track breadcrumb; let last = $last) {\n    @if (breadcrumb.label.title && (breadcrumb.url.slice(-1) == '/' || last)) {\n      <li class=\"breadcrumb-item\"\n        [ngClass]=\"{active: last}\">\n        @if (!last) {\n          <a [routerLink]=\"breadcrumb.url\">{{breadcrumb.label.title}}</a>\n        }\n        @if (last) {\n          <span [routerLink]=\"breadcrumb.url\">{{breadcrumb.label.title}}</span>\n        }\n      </li>\n    }\n  }\n  <ng-content />\n</ol>\n" }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }, { type: AppBreadcrumbService }], propDecorators: { fixed: [{
                type: Input
            }] } });

// @dynamic
class AppBreadcrumbModule {
    static forRoot(config) {
        return {
            ngModule: AppBreadcrumbModule,
            providers: [
                AppBreadcrumbService
            ]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbModule, imports: [AppBreadcrumbComponent, CuiBreadcrumbComponent], exports: [AppBreadcrumbComponent, CuiBreadcrumbComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppBreadcrumbModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [AppBreadcrumbComponent, CuiBreadcrumbComponent],
                    exports: [AppBreadcrumbComponent, CuiBreadcrumbComponent],
                }]
        }] });

class AppFooterComponent {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
        this.fixedClass = 'footer-fixed';
        this.appFooterClass = true;
    }
    ngOnInit() {
        this.isFixed(this.fixed);
    }
    ngOnDestroy() {
        this.renderer.removeClass(this.document.body, this.fixedClass);
    }
    isFixed(fixed = this.fixed) {
        if (fixed) {
            this.renderer.addClass(this.document.body, this.fixedClass);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppFooterComponent, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppFooterComponent, isStandalone: true, selector: "app-footer, cui-footer", inputs: { fixed: "fixed" }, host: { properties: { "class.app-footer": "this.appFooterClass" } }, ngImport: i0, template: `<ng-content />`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppFooterComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-footer, cui-footer',
                    template: `<ng-content />`,
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }], propDecorators: { fixed: [{
                type: Input
            }], appFooterClass: [{
                type: HostBinding,
                args: ['class.app-footer']
            }] } });

class AppFooterModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppFooterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppFooterModule, imports: [AppFooterComponent], exports: [AppFooterComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppFooterModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppFooterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [AppFooterComponent],
                    exports: [AppFooterComponent],
                }]
        }] });

class AppHeaderComponent {
    constructor(document, renderer) {
        this.document = document;
        this.renderer = renderer;
        this.navbarBrandText = { icon: '🅲', text: '🅲 CoreUI' };
        this.navbarBrandHref = ''; // deprecated, use navbarBrandRouterLink instead
        this.navbarBrandRouterLink = '';
        this.sidebarToggler = 'lg';
        this.mobileSidebarToggler = true;
        this.asideMenuToggler = 'lg';
        this.mobileAsideMenuToggler = true;
        this.fixedClass = 'header-fixed';
        this.appHeaderClass = true;
        this.navbarClass = true;
        this.breakpoints = ['xl', 'lg', 'md', 'sm', 'xs'];
        this.sidebarTogglerClass = 'd-none d-md-block';
        this.sidebarTogglerMobileClass = 'd-lg-none';
        this.asideTogglerClass = 'd-none d-md-block';
        this.asideTogglerMobileClass = 'd-lg-none';
    }
    ngOnInit() {
        this.isFixed(this.fixed);
        this.navbarBrandImg = Boolean(this.navbarBrand || this.navbarBrandFull || this.navbarBrandMinimized);
        this.navbarBrandRouterLink = this.navbarBrandRouterLink[0] ? this.navbarBrandRouterLink : this.navbarBrandHref;
        this.sidebarTogglerClass = this.setTogglerBreakpointClass(this.sidebarToggler);
        this.sidebarTogglerMobileClass = this.setTogglerMobileBreakpointClass(this.sidebarToggler);
        this.asideTogglerClass = this.setTogglerBreakpointClass(this.asideMenuToggler);
        this.asideTogglerMobileClass = this.setTogglerMobileBreakpointClass(this.asideMenuToggler);
    }
    ngOnDestroy() {
        this.renderer.removeClass(this.document.body, this.fixedClass);
    }
    isFixed(fixed = this.fixed) {
        if (fixed) {
            this.renderer.addClass(this.document.body, this.fixedClass);
        }
    }
    setTogglerBreakpointClass(breakpoint = 'md') {
        let togglerClass = 'd-none d-md-block';
        if (this.breakpoints.includes(breakpoint)) {
            const breakpointIndex = this.breakpoints.indexOf(breakpoint);
            togglerClass = `d-none d-${breakpoint}-block`;
        }
        return togglerClass;
    }
    setTogglerMobileBreakpointClass(breakpoint = 'lg') {
        let togglerClass = 'd-lg-none';
        if (this.breakpoints.includes(breakpoint)) {
            togglerClass = `d-${breakpoint}-none`;
        }
        return togglerClass;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppHeaderComponent, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.2.2", type: AppHeaderComponent, isStandalone: true, selector: "app-header, cui-header", inputs: { fixed: "fixed", navbarBrand: "navbarBrand", navbarBrandFull: "navbarBrandFull", navbarBrandMinimized: "navbarBrandMinimized", navbarBrandText: "navbarBrandText", navbarBrandHref: "navbarBrandHref", navbarBrandRouterLink: "navbarBrandRouterLink", sidebarToggler: "sidebarToggler", mobileSidebarToggler: "mobileSidebarToggler", asideMenuToggler: "asideMenuToggler", mobileAsideMenuToggler: "mobileAsideMenuToggler" }, host: { properties: { "class.app-header": "this.appHeaderClass", "class.navbar": "this.navbarClass" } }, ngImport: i0, template: "@if (mobileSidebarToggler != false) {\n  <button class=\"navbar-toggler {{sidebarTogglerMobileClass}}\" type=\"button\" appSidebarToggler>\n    <span class=\"navbar-toggler-icon\"></span>\n  </button>\n}\n<a class=\"navbar-brand\" [routerLink]=\"navbarBrandRouterLink\">\n  @if (navbarBrandImg) {\n    @if (navbarBrand) {\n      <img\n        [appHtmlAttr]=\"navbarBrand\"\n        [ngClass]=\"'navbar-brand'\">\n    }\n    @if (navbarBrandFull) {\n      <img\n        [appHtmlAttr]=\"navbarBrandFull\"\n        [ngClass]=\"'navbar-brand-full'\">\n    }\n    @if (navbarBrandMinimized) {\n      <img\n        [appHtmlAttr]=\"navbarBrandMinimized\"\n        [ngClass]=\"'navbar-brand-minimized'\">\n    }\n  }\n  @if (!navbarBrandImg) {\n    <div class=\"navbar-brand-full\" [innerHTML]=\"navbarBrandText.text\"></div>\n    <div class=\"navbar-brand-minimized\" [innerHTML]=\"navbarBrandText.icon\"></div>\n  }\n</a>\n@if (sidebarToggler != false) {\n  <button class=\"navbar-toggler {{sidebarTogglerClass}}\" type=\"button\" [appSidebarToggler]=\"sidebarToggler\">\n    <span class=\"navbar-toggler-icon\"></span>\n  </button>\n}\n<ng-content />\n@if (asideMenuToggler != false) {\n  <button class=\"navbar-toggler {{asideTogglerClass}}\" type=\"button\" [appAsideMenuToggler]=\"asideMenuToggler\">\n    <span class=\"navbar-toggler-icon\"></span>\n  </button>\n}\n@if (mobileAsideMenuToggler != false) {\n  <button class=\"navbar-toggler {{asideTogglerMobileClass}}\" type=\"button\" appAsideMenuToggler>\n    <span class=\"navbar-toggler-icon\"></span>\n  </button>\n}\n", dependencies: [{ kind: "directive", type: RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "info", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: SidebarToggleDirective, selector: "[appSidebarToggler]", inputs: ["appSidebarToggler"] }, { kind: "directive", type: HtmlAttributesDirective, selector: "[appHtmlAttr]", inputs: ["appHtmlAttr"] }, { kind: "directive", type: AsideToggleDirective, selector: "[appAsideMenuToggler]", inputs: ["appAsideMenuToggler"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppHeaderComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-header, cui-header', standalone: true, imports: [RouterLink, NgClass, SidebarToggleDirective, HtmlAttributesDirective, AsideToggleDirective], template: "@if (mobileSidebarToggler != false) {\n  <button class=\"navbar-toggler {{sidebarTogglerMobileClass}}\" type=\"button\" appSidebarToggler>\n    <span class=\"navbar-toggler-icon\"></span>\n  </button>\n}\n<a class=\"navbar-brand\" [routerLink]=\"navbarBrandRouterLink\">\n  @if (navbarBrandImg) {\n    @if (navbarBrand) {\n      <img\n        [appHtmlAttr]=\"navbarBrand\"\n        [ngClass]=\"'navbar-brand'\">\n    }\n    @if (navbarBrandFull) {\n      <img\n        [appHtmlAttr]=\"navbarBrandFull\"\n        [ngClass]=\"'navbar-brand-full'\">\n    }\n    @if (navbarBrandMinimized) {\n      <img\n        [appHtmlAttr]=\"navbarBrandMinimized\"\n        [ngClass]=\"'navbar-brand-minimized'\">\n    }\n  }\n  @if (!navbarBrandImg) {\n    <div class=\"navbar-brand-full\" [innerHTML]=\"navbarBrandText.text\"></div>\n    <div class=\"navbar-brand-minimized\" [innerHTML]=\"navbarBrandText.icon\"></div>\n  }\n</a>\n@if (sidebarToggler != false) {\n  <button class=\"navbar-toggler {{sidebarTogglerClass}}\" type=\"button\" [appSidebarToggler]=\"sidebarToggler\">\n    <span class=\"navbar-toggler-icon\"></span>\n  </button>\n}\n<ng-content />\n@if (asideMenuToggler != false) {\n  <button class=\"navbar-toggler {{asideTogglerClass}}\" type=\"button\" [appAsideMenuToggler]=\"asideMenuToggler\">\n    <span class=\"navbar-toggler-icon\"></span>\n  </button>\n}\n@if (mobileAsideMenuToggler != false) {\n  <button class=\"navbar-toggler {{asideTogglerMobileClass}}\" type=\"button\" appAsideMenuToggler>\n    <span class=\"navbar-toggler-icon\"></span>\n  </button>\n}\n" }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }], propDecorators: { fixed: [{
                type: Input
            }], navbarBrand: [{
                type: Input
            }], navbarBrandFull: [{
                type: Input
            }], navbarBrandMinimized: [{
                type: Input
            }], navbarBrandText: [{
                type: Input
            }], navbarBrandHref: [{
                type: Input
            }], navbarBrandRouterLink: [{
                type: Input
            }], sidebarToggler: [{
                type: Input
            }], mobileSidebarToggler: [{
                type: Input
            }], asideMenuToggler: [{
                type: Input
            }], mobileAsideMenuToggler: [{
                type: Input
            }], appHeaderClass: [{
                type: HostBinding,
                args: ['class.app-header']
            }], navbarClass: [{
                type: HostBinding,
                args: ['class.navbar']
            }] } });

class AppHeaderModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppHeaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppHeaderModule, imports: [AppHeaderComponent], exports: [AppHeaderComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppHeaderModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppHeaderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        AppHeaderComponent
                    ],
                    exports: [
                        AppHeaderComponent,
                    ],
                }]
        }] });

class AppSidebarService {
    constructor() {
        this.events = new BehaviorSubject({});
        this.events$ = this.events.asObservable();
    }
    toggle(action) {
        this.events.next(action);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [] });

class AppSidebarComponent {
    get minimized() {
        return this._minimized;
    }
    set minimized(value) {
        // only update / emit events when the value changes
        if (this._minimized !== value) {
            this._minimized = value;
            this._updateMinimized(value);
            this.minimizedChange.emit(value);
            this.sidebarService.toggle({ minimize: value });
        }
    }
    constructor(document, renderer, sidebarService) {
        this.document = document;
        this.renderer = renderer;
        this.sidebarService = sidebarService;
        this._minimized = false;
        /**
         * Emits whenever the minimized state of the sidebar changes.
         * Primarily used to facilitate two-way binding.
         */
        this.minimizedChange = new EventEmitter();
        this.sidebarClass = true;
    }
    ngOnInit() {
        this.displayBreakpoint(this.display);
        this.isCompact(this.compact);
        this.isFixed(this.fixed);
        this.isOffCanvas(this.offCanvas);
        this.sidebarService.toggle({ minimize: this.minimized });
        this.subscriptionEvents = this.sidebarService.events$.subscribe(action => {
            if (action.minimize !== undefined) {
                action.minimize === 'toggle' ? this.toggleMinimized() : this.minimized = !!action.minimize;
            }
        });
    }
    ngOnDestroy() {
        this.subscriptionEvents.unsubscribe();
        this.minimizedChange.complete();
        this.renderer.removeClass(this.document.body, 'sidebar-fixed');
        this._updateMinimized(false);
    }
    isCompact(compact = this.compact) {
        if (compact) {
            this.renderer.addClass(this.document.body, 'sidebar-compact');
        }
    }
    isFixed(fixed = this.fixed) {
        if (fixed) {
            this.renderer.addClass(this.document.body, 'sidebar-fixed');
        }
    }
    toggleMinimized() {
        this.minimized = !this._minimized;
    }
    isOffCanvas(offCanvas = this.offCanvas) {
        if (offCanvas) {
            this.renderer.addClass(this.document.body, 'sidebar-off-canvas');
        }
    }
    displayBreakpoint(display = this.display) {
        if (display !== false) {
            const cssClass = display ? `sidebar-${display}-show` : sidebarCssClasses[0];
            this.renderer.addClass(this.document.body, cssClass);
        }
    }
    _updateMinimized(minimized) {
        const body = this.document.body;
        if (minimized) {
            this.renderer.addClass(body, 'sidebar-minimized');
            this.renderer.addClass(body, 'brand-minimized');
        }
        else {
            this.renderer.removeClass(body, 'sidebar-minimized');
            this.renderer.removeClass(body, 'brand-minimized');
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarComponent, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }, { token: AppSidebarService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppSidebarComponent, isStandalone: true, selector: "app-sidebar, cui-sidebar", inputs: { compact: "compact", display: "display", fixed: "fixed", offCanvas: "offCanvas", minimized: "minimized" }, outputs: { minimizedChange: "minimizedChange" }, host: { properties: { "class.sidebar": "this.sidebarClass" } }, ngImport: i0, template: `<ng-content />`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar, cui-sidebar',
                    template: `<ng-content />`,
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }, { type: AppSidebarService }], propDecorators: { compact: [{
                type: Input
            }], display: [{
                type: Input
            }], fixed: [{
                type: Input
            }], offCanvas: [{
                type: Input
            }], minimized: [{
                type: Input
            }], minimizedChange: [{
                type: Output
            }], sidebarClass: [{
                type: HostBinding,
                args: ['class.sidebar']
            }] } });

class SidebarNavService {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarNavService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarNavService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarNavService, decorators: [{
            type: Injectable
        }] });
class SidebarNavHelper {
    constructor() {
        this.hasBadge = (item) => Boolean(item.badge);
        this.hasIcon = (item) => Boolean(item.icon);
    }
    itemType(item) {
        if (item.divider) {
            return 'divider';
        }
        else if (item.title) {
            return 'title';
        }
        else if (item.children) {
            return 'dropdown';
        }
        else if (item.label) {
            return 'label';
        }
        else if (!Object.keys(item).length) {
            return 'empty';
        }
        return 'link';
    }
    isActive(router, item) {
        return router.isActive(item.url, false);
    }
    getIconClass(item) {
        const classes = {
            'nav-icon': true
        };
        if (this.hasIcon(item)) {
            const icon = item.icon;
            // @ts-ignore
            classes[icon] = this.hasIcon(item);
        }
        return classes;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarNavHelper, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarNavHelper }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: SidebarNavHelper, decorators: [{
            type: Injectable
        }] });

class AppSidebarNavItemClassPipe {
    constructor(helper) {
        this.helper = helper;
    }
    transform(item, ...args) {
        const itemType = this.helper.itemType(item);
        let itemClass;
        if (['divider', 'title'].includes(itemType)) {
            itemClass = `nav-${itemType}`;
        }
        else if (itemType === 'dropdown') {
            itemClass = 'nav-item nav-dropdown';
        }
        else {
            itemClass = 'nav-item';
        }
        return item.class ? `${itemClass} ${item.class}` : itemClass;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavItemClassPipe, deps: [{ token: SidebarNavHelper }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavItemClassPipe, isStandalone: true, name: "appSidebarNavItemClass" }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavItemClassPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'appSidebarNavItemClass',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: SidebarNavHelper }] });

class AppSidebarNavBadgePipe {
    transform(item, args) {
        const classes = {
            badge: true
        };
        const variant = `badge-${item.badge.variant}`;
        classes[variant] = !!item.badge.variant;
        classes[item.badge.class] = !!item.badge.class;
        return classes;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavBadgePipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavBadgePipe, isStandalone: true, name: "appSidebarNavBadge" }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavBadgePipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'appSidebarNavBadge',
                    standalone: true
                }]
        }] });

class AppSidebarNavIconPipe {
    transform(item, args) {
        const classes = {
            'nav-icon': true
        };
        const icon = item.icon;
        classes[icon] = !!item.icon;
        return classes;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavIconPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavIconPipe, isStandalone: true, name: "appSidebarNavIcon" }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavIconPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'appSidebarNavIcon',
                    standalone: true
                }]
        }] });

class AppSidebarNavLinkPipe {
    transform(item) {
        const classes = { 'nav-link': true };
        const disabled = item.attributes && item.attributes.disabled;
        classes['disabled'] = disabled;
        classes['btn-link'] = disabled;
        classes[`nav-link-${item.variant}`] = !!item.variant;
        return classes;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLinkPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLinkPipe, isStandalone: true, name: "appSidebarNavLink" }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLinkPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'appSidebarNavLink',
                    standalone: true
                }]
        }] });

class AppSidebarNavLinkContentComponent {
    constructor(helper) {
        this.helper = helper;
        this.item = {};
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLinkContentComponent, deps: [{ token: SidebarNavHelper }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.2.2", type: AppSidebarNavLinkContentComponent, isStandalone: true, selector: "app-sidebar-nav-link-content, cui-sidebar-nav-link-content", inputs: { item: "item" }, providers: [SidebarNavHelper], ngImport: i0, template: `
    @if (true) {
      @if (helper.hasIcon(item)) {
        <i [ngClass]="item | appSidebarNavIcon"></i>
      }
      <ng-container>{{item.name}}</ng-container>
      @if (helper.hasBadge(item)) {
        <span [ngClass]="item | appSidebarNavBadge">{{ item.badge?.text }}</span>
      }
    }
    `, isInline: true, dependencies: [{ kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "pipe", type: AppSidebarNavIconPipe, name: "appSidebarNavIcon" }, { kind: "pipe", type: AppSidebarNavBadgePipe, name: "appSidebarNavBadge" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLinkContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-nav-link-content, cui-sidebar-nav-link-content',
                    template: `
    @if (true) {
      @if (helper.hasIcon(item)) {
        <i [ngClass]="item | appSidebarNavIcon"></i>
      }
      <ng-container>{{item.name}}</ng-container>
      @if (helper.hasBadge(item)) {
        <span [ngClass]="item | appSidebarNavBadge">{{ item.badge?.text }}</span>
      }
    }
    `,
                    providers: [SidebarNavHelper],
                    standalone: true,
                    imports: [NgClass, AppSidebarNavIconPipe, AppSidebarNavBadgePipe]
                }]
        }], ctorParameters: () => [{ type: SidebarNavHelper }], propDecorators: { item: [{
                type: Input
            }] } });
class AppSidebarNavLinkComponent {
    set item(item) {
        this._Item = JSON.parse(JSON.stringify(item));
    }
    get item() {
        return this._Item;
    }
    constructor(router) {
        this.router = router;
        this.linkClick = new EventEmitter();
        this.navigationEndObservable = router.events.pipe(filter(event => {
            return event instanceof NavigationEnd;
        }));
    }
    ngOnInit() {
        // @ts-ignore
        this.url = typeof this.item.url === 'string' ? this.item.url : this.router.serializeUrl(this.router.createUrlTree(this.item.url));
        this.linkType = this.getLinkType();
        this.href = this.isDisabled() ? '' : (this.item.href || this.url);
        this.linkActive = this.router.url.split(/[?#(;]/)[0] === this.href.split(/[?#(;]/)[0];
        this.navSubscription = this.navigationEndObservable.subscribe(event => {
            const itemUrlArray = this.href.split(/[?#(;]/)[0].split('/');
            const urlArray = event.urlAfterRedirects.split(/[?#(;]/)[0].split('/');
            this.linkActive = itemUrlArray.every((value, index) => value === urlArray[index]);
        });
    }
    ngOnDestroy() {
        this.navSubscription.unsubscribe();
    }
    getLinkType() {
        return this.isDisabled() ? 'disabled' : this.isExternalLink() ? 'external' : 'link';
    }
    isDisabled() {
        return (this.item.attributes && this.item.attributes.disabled) ? true : null;
    }
    isExternalLink() {
        return !!this.item.href || this.url.substring(0, 4) === 'http';
    }
    linkClicked() {
        this.linkClick.emit();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLinkComponent, deps: [{ token: i1.Router }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.2.2", type: AppSidebarNavLinkComponent, isStandalone: true, selector: "app-sidebar-nav-link, cui-sidebar-nav-link", inputs: { item: "item" }, outputs: { linkClick: "linkClick" }, providers: [SidebarNavHelper], ngImport: i0, template: "\n@switch (linkType) {\n  @case ('disabled') {\n    <a\n      [ngClass]=\"item | appSidebarNavLink\"\n      [appHtmlAttr]=\"item.attributes\"\n      >\n      <app-sidebar-nav-link-content [item]=\"item\" />\n    </a>\n  }\n  @case ('external') {\n    <a\n      [ngClass]=\"item | appSidebarNavLink\"\n      [href]=\"href\"\n      [appHtmlAttr]=\"item.attributes\"\n      (click)=\"linkClicked()\"\n      >\n      <app-sidebar-nav-link-content [item]=\"item\" />\n    </a>\n  }\n  @default {\n    <a\n      [ngClass]=\"item | appSidebarNavLink\"\n      [appHtmlAttr]=\"item.attributes\"\n      [target]=\"item.attributes?.target\"\n      [queryParams]=\"item.linkProps?.queryParams\"\n      [fragment]=\"item.linkProps?.fragment\"\n      [queryParamsHandling]=\"item.linkProps?.queryParamsHandling\"\n      [preserveFragment]=\"!!item.linkProps?.preserveFragment\"\n      [skipLocationChange]=\"!!item.linkProps?.skipLocationChange\"\n      [replaceUrl]=\"!!item.linkProps?.replaceUrl\"\n      [state]=\"item.linkProps?.state\"\n      [routerLink]=\"item.url\"\n      [routerLinkActive]=\"item.linkProps?.routerLinkActive || 'active'\"\n      [routerLinkActiveOptions]=\"item.linkProps?.routerLinkActiveOptions ?? {exact: false}\"\n      [class.active]=\"linkActive && !item.linkProps?.routerLinkActiveOptions\"\n      (click)=\"linkClicked()\"\n      >\n      <app-sidebar-nav-link-content [item]=\"item\" />\n    </a>\n  }\n}\n", dependencies: [{ kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: HtmlAttributesDirective, selector: "[appHtmlAttr]", inputs: ["appHtmlAttr"] }, { kind: "component", type: AppSidebarNavLinkContentComponent, selector: "app-sidebar-nav-link-content, cui-sidebar-nav-link-content", inputs: ["item"] }, { kind: "ngmodule", type: RouterModule }, { kind: "directive", type: i1.RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "info", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "directive", type: i1.RouterLinkActive, selector: "[routerLinkActive]", inputs: ["routerLinkActiveOptions", "ariaCurrentWhenActive", "routerLinkActive"], outputs: ["isActiveChange"], exportAs: ["routerLinkActive"] }, { kind: "pipe", type: AppSidebarNavLinkPipe, name: "appSidebarNavLink" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLinkComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-sidebar-nav-link, cui-sidebar-nav-link', providers: [SidebarNavHelper], standalone: true, imports: [NgClass, AppSidebarNavLinkPipe, HtmlAttributesDirective, AppSidebarNavLinkContentComponent, RouterModule], template: "\n@switch (linkType) {\n  @case ('disabled') {\n    <a\n      [ngClass]=\"item | appSidebarNavLink\"\n      [appHtmlAttr]=\"item.attributes\"\n      >\n      <app-sidebar-nav-link-content [item]=\"item\" />\n    </a>\n  }\n  @case ('external') {\n    <a\n      [ngClass]=\"item | appSidebarNavLink\"\n      [href]=\"href\"\n      [appHtmlAttr]=\"item.attributes\"\n      (click)=\"linkClicked()\"\n      >\n      <app-sidebar-nav-link-content [item]=\"item\" />\n    </a>\n  }\n  @default {\n    <a\n      [ngClass]=\"item | appSidebarNavLink\"\n      [appHtmlAttr]=\"item.attributes\"\n      [target]=\"item.attributes?.target\"\n      [queryParams]=\"item.linkProps?.queryParams\"\n      [fragment]=\"item.linkProps?.fragment\"\n      [queryParamsHandling]=\"item.linkProps?.queryParamsHandling\"\n      [preserveFragment]=\"!!item.linkProps?.preserveFragment\"\n      [skipLocationChange]=\"!!item.linkProps?.skipLocationChange\"\n      [replaceUrl]=\"!!item.linkProps?.replaceUrl\"\n      [state]=\"item.linkProps?.state\"\n      [routerLink]=\"item.url\"\n      [routerLinkActive]=\"item.linkProps?.routerLinkActive || 'active'\"\n      [routerLinkActiveOptions]=\"item.linkProps?.routerLinkActiveOptions ?? {exact: false}\"\n      [class.active]=\"linkActive && !item.linkProps?.routerLinkActiveOptions\"\n      (click)=\"linkClicked()\"\n      >\n      <app-sidebar-nav-link-content [item]=\"item\" />\n    </a>\n  }\n}\n" }]
        }], ctorParameters: () => [{ type: i1.Router }], propDecorators: { item: [{
                type: Input
            }], linkClick: [{
                type: Output
            }] } });

class AppSidebarNavLabelComponent {
    constructor(helper) {
        this.helper = helper;
        this.item = {};
        this.classes = {
            'nav-label': true,
            active: true
        };
        this.iconClasses = {};
    }
    ngOnInit() {
        this.iconClasses = this.helper.getIconClass(this.item);
    }
    getItemClass() {
        const itemClass = this.item.class;
        // @ts-ignore
        this.classes[itemClass] = !!itemClass;
        return this.classes;
    }
    getLabelIconClass() {
        const variant = `text-${this.item.label?.variant}`;
        // @ts-ignore
        this.iconClasses[variant] = !!this.item.label?.variant;
        const labelClass = this.item.label?.class;
        // @ts-ignore
        this.iconClasses[labelClass] = !!labelClass;
        return this.iconClasses;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLabelComponent, deps: [{ token: SidebarNavHelper }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.2.2", type: AppSidebarNavLabelComponent, isStandalone: true, selector: "app-sidebar-nav-label, cui-sidebar-nav-label", inputs: { item: "item" }, ngImport: i0, template: "<a [ngClass]=\"getItemClass()\"\n  href=\"{{item.url}}\"\n  [appHtmlAttr]=\"item.attributes\">\n  @if (helper.hasIcon(item)) {\n    <i [ngClass]=\"getLabelIconClass()\"></i>\n  }\n  <ng-container>{{item.name}}</ng-container>\n  @if (helper.hasBadge(item)) {\n    <span [ngClass]=\"item | appSidebarNavBadge\">{{ item.badge?.text }}</span>\n  }\n</a>\n", dependencies: [{ kind: "directive", type: HtmlAttributesDirective, selector: "[appHtmlAttr]", inputs: ["appHtmlAttr"] }, { kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "pipe", type: AppSidebarNavBadgePipe, name: "appSidebarNavBadge" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavLabelComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-sidebar-nav-label, cui-sidebar-nav-label', standalone: true, imports: [AppSidebarNavBadgePipe, HtmlAttributesDirective, NgClass], template: "<a [ngClass]=\"getItemClass()\"\n  href=\"{{item.url}}\"\n  [appHtmlAttr]=\"item.attributes\">\n  @if (helper.hasIcon(item)) {\n    <i [ngClass]=\"getLabelIconClass()\"></i>\n  }\n  <ng-container>{{item.name}}</ng-container>\n  @if (helper.hasBadge(item)) {\n    <span [ngClass]=\"item | appSidebarNavBadge\">{{ item.badge?.text }}</span>\n  }\n</a>\n" }]
        }], ctorParameters: () => [{ type: SidebarNavHelper }], propDecorators: { item: [{
                type: Input
            }] } });

class AppSidebarNavTitleComponent {
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.item = {};
    }
    ngOnInit() {
        const nativeElement = this.el.nativeElement;
        const name = this.renderer.createText(this.item.name);
        if (this.item.class) {
            const classes = this.item.class;
            this.renderer.addClass(nativeElement, classes);
        }
        if (this.item.wrapper) {
            const wrapper = this.renderer.createElement(this.item.wrapper.element);
            this.addAttribs(this.item.wrapper.attributes, wrapper);
            this.renderer.appendChild(wrapper, name);
            this.renderer.appendChild(nativeElement, wrapper);
        }
        else {
            this.renderer.appendChild(nativeElement, name);
        }
    }
    addAttribs(attribs, element) {
        if (attribs) {
            for (const attr in attribs) {
                if (attr === 'style' && typeof (attribs[attr]) === 'object') {
                    this.setStyle(attribs[attr], element);
                }
                else if (attr === 'class') {
                    this.addClass(attribs[attr], element);
                }
                else {
                    this.setAttrib(attr, attribs[attr], element);
                }
            }
        }
    }
    setStyle(styles, el) {
        for (const style in styles) {
            this.renderer.setStyle(el, style, styles[style]);
        }
    }
    addClass(classes, el) {
        const classArray = (Array.isArray(classes) ? classes : classes.split(' '));
        classArray.filter((element) => element.length > 0).forEach(element => {
            this.renderer.addClass(el, element);
        });
    }
    setAttrib(key, value, el) {
        this.renderer.setAttribute(el, key, value);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavTitleComponent, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppSidebarNavTitleComponent, isStandalone: true, selector: "app-sidebar-nav-title, cui-sidebar-nav-title", inputs: { item: "item" }, ngImport: i0, template: '', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-nav-title, cui-sidebar-nav-title',
                    template: '',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: i0.ElementRef }, { type: i0.Renderer2 }], propDecorators: { item: [{
                type: Input
            }] } });

class AppSidebarNavDividerComponent {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavDividerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppSidebarNavDividerComponent, isStandalone: true, selector: "app-sidebar-nav-divider, cui-sidebar-nav-divider", inputs: { item: "item" }, ngImport: i0, template: ``, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavDividerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-nav-divider, cui-sidebar-nav-divider',
                    template: ``,
                    standalone: true
                }]
        }], propDecorators: { item: [{
                type: Input
            }] } });

class NavDropdownDirective {
    constructor(elementRef, renderer) {
        this.elementRef = elementRef;
        this.renderer = renderer;
    }
    toggle() {
        this.elementRef.nativeElement.classList.contains('open') ?
            this.renderer.removeClass(this.elementRef.nativeElement, 'open') :
            this.renderer.addClass(this.elementRef.nativeElement, 'open');
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: NavDropdownDirective, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: NavDropdownDirective, isStandalone: true, selector: "[appNavDropdown]", ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: NavDropdownDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appNavDropdown]',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: i0.ElementRef }, { type: i0.Renderer2 }] });
/**
 * Allows the dropdown to be toggled via click.
 */
class NavDropdownToggleDirective {
    constructor(dropdown) {
        this.dropdown = dropdown;
    }
    toggleOpen($event) {
        $event.preventDefault();
        this.dropdown.toggle();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: NavDropdownToggleDirective, deps: [{ token: NavDropdownDirective }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "20.2.2", type: NavDropdownToggleDirective, isStandalone: true, selector: "[appNavDropdownToggle]", host: { listeners: { "click": "toggleOpen($event)" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: NavDropdownToggleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appNavDropdownToggle]',
                    standalone: true,
                }]
        }], ctorParameters: () => [{ type: NavDropdownDirective }], propDecorators: { toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });

// lightweight injection token
class DropdownToken {
}
class AppSidebarNavItemsComponent {
    set items(items) {
        this._items = [...items];
    }
    get items() {
        return this._items;
    }
    constructor(document, renderer, router, helper) {
        this.document = document;
        this.renderer = renderer;
        this.router = router;
        this.helper = helper;
        this._items = [];
    }
    hideMobile() {
        if (this.document.body.classList.contains('sidebar-show')) {
            this.renderer.removeClass(this.document.body, 'sidebar-show');
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavItemsComponent, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }, { token: i1.Router }, { token: SidebarNavHelper }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.2.2", type: AppSidebarNavItemsComponent, isStandalone: true, selector: "app-sidebar-nav-items, cui-sidebar-nav-items", inputs: { items: "items" }, ngImport: i0, template: `
    @for (item of items; track item) {
      @switch (helper.itemType(item)) {
        @case ('dropdown') {
          <app-sidebar-nav-dropdown
            [item]="item"
            [class.open]="helper.isActive(router, item)"
            [ngClass]="item | appSidebarNavItemClass"
            appNavDropdown>
          </app-sidebar-nav-dropdown>
        }
        @case ('divider') {
          <app-sidebar-nav-divider
            [item]="item"
            [ngClass]="item | appSidebarNavItemClass"
            [appHtmlAttr]="item.attributes">
          </app-sidebar-nav-divider>
        }
        @case ('title') {
          <app-sidebar-nav-title
            [item]="item"
            [ngClass]="item | appSidebarNavItemClass"
            [appHtmlAttr]="item.attributes">
          </app-sidebar-nav-title>
        }
        @case ('label') {
          <app-sidebar-nav-label
            [item]="item"
            class="nav-item"
            [ngClass]="item | appSidebarNavItemClass">
          </app-sidebar-nav-label>
        }
        @case ('empty') {
        }
        @default {
          <app-sidebar-nav-link
            [item]="item"
            class="nav-item"
            [ngClass]="item | appSidebarNavItemClass"
            (linkClick)="hideMobile()"
            >
          </app-sidebar-nav-link>
        }
      }
    }
    `, isInline: true, dependencies: [{ kind: "directive", type: i0.forwardRef(() => NgClass), selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i0.forwardRef(() => HtmlAttributesDirective), selector: "[appHtmlAttr]", inputs: ["appHtmlAttr"] }, { kind: "component", type: i0.forwardRef(() => AppSidebarNavDropdownComponent), selector: "app-sidebar-nav-dropdown, cui-sidebar-nav-dropdown", inputs: ["item"] }, { kind: "component", type: i0.forwardRef(() => AppSidebarNavLinkComponent), selector: "app-sidebar-nav-link, cui-sidebar-nav-link", inputs: ["item"], outputs: ["linkClick"] }, { kind: "component", type: i0.forwardRef(() => AppSidebarNavLabelComponent), selector: "app-sidebar-nav-label, cui-sidebar-nav-label", inputs: ["item"] }, { kind: "component", type: i0.forwardRef(() => AppSidebarNavTitleComponent), selector: "app-sidebar-nav-title, cui-sidebar-nav-title", inputs: ["item"] }, { kind: "component", type: i0.forwardRef(() => AppSidebarNavDividerComponent), selector: "app-sidebar-nav-divider, cui-sidebar-nav-divider", inputs: ["item"] }, { kind: "directive", type: i0.forwardRef(() => NavDropdownDirective), selector: "[appNavDropdown]" }, { kind: "pipe", type: i0.forwardRef(() => AppSidebarNavItemClassPipe), name: "appSidebarNavItemClass" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavItemsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-nav-items, cui-sidebar-nav-items',
                    template: `
    @for (item of items; track item) {
      @switch (helper.itemType(item)) {
        @case ('dropdown') {
          <app-sidebar-nav-dropdown
            [item]="item"
            [class.open]="helper.isActive(router, item)"
            [ngClass]="item | appSidebarNavItemClass"
            appNavDropdown>
          </app-sidebar-nav-dropdown>
        }
        @case ('divider') {
          <app-sidebar-nav-divider
            [item]="item"
            [ngClass]="item | appSidebarNavItemClass"
            [appHtmlAttr]="item.attributes">
          </app-sidebar-nav-divider>
        }
        @case ('title') {
          <app-sidebar-nav-title
            [item]="item"
            [ngClass]="item | appSidebarNavItemClass"
            [appHtmlAttr]="item.attributes">
          </app-sidebar-nav-title>
        }
        @case ('label') {
          <app-sidebar-nav-label
            [item]="item"
            class="nav-item"
            [ngClass]="item | appSidebarNavItemClass">
          </app-sidebar-nav-label>
        }
        @case ('empty') {
        }
        @default {
          <app-sidebar-nav-link
            [item]="item"
            class="nav-item"
            [ngClass]="item | appSidebarNavItemClass"
            (linkClick)="hideMobile()"
            >
          </app-sidebar-nav-link>
        }
      }
    }
    `,
                    standalone: true,
                    // providers: [{provide: DropdownToken, useExisting: forwardRef(()=> AppSidebarNavDropdownComponent)}],
                    imports: [
                        NgClass,
                        HtmlAttributesDirective,
                        AppSidebarNavItemClassPipe,
                        forwardRef(() => AppSidebarNavDropdownComponent),
                        AppSidebarNavLinkComponent,
                        AppSidebarNavLabelComponent,
                        AppSidebarNavTitleComponent,
                        AppSidebarNavDividerComponent,
                        NavDropdownDirective
                    ]
                }]
        }], ctorParameters: () => [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }, { type: i1.Router }, { type: SidebarNavHelper }], propDecorators: { items: [{
                type: Input
            }] } });
class AppSidebarNavDropdownComponent {
    constructor(helper) {
        this.helper = helper;
        this.item = {};
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavDropdownComponent, deps: [{ token: SidebarNavHelper }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "20.2.2", type: AppSidebarNavDropdownComponent, isStandalone: true, selector: "app-sidebar-nav-dropdown, cui-sidebar-nav-dropdown", inputs: { item: "item" }, providers: [SidebarNavHelper], ngImport: i0, template: `
    <a class="nav-link nav-dropdown-toggle"
      appNavDropdownToggle
      [appHtmlAttr]="item.attributes">
      @if (helper.hasIcon(item)) {
        <i [ngClass]="item | appSidebarNavIcon"></i>
      }
      <ng-container>{{item.name}}</ng-container>
      @if (helper.hasBadge(item)) {
        <span [ngClass]="item | appSidebarNavBadge">{{ item.badge?.text }}</span>
      }
    </a>
    <app-sidebar-nav-items
      class="nav-dropdown-items"
      [items]="item.children ?? []">
    </app-sidebar-nav-items>
    `, isInline: true, styles: [".nav-dropdown-toggle{cursor:pointer}\n", ".nav-dropdown-items{display:block}\n"], dependencies: [{ kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: HtmlAttributesDirective, selector: "[appHtmlAttr]", inputs: ["appHtmlAttr"] }, { kind: "component", type: AppSidebarNavItemsComponent, selector: "app-sidebar-nav-items, cui-sidebar-nav-items", inputs: ["items"] }, { kind: "directive", type: NavDropdownToggleDirective, selector: "[appNavDropdownToggle]" }, { kind: "pipe", type: AppSidebarNavBadgePipe, name: "appSidebarNavBadge" }, { kind: "pipe", type: AppSidebarNavIconPipe, name: "appSidebarNavIcon" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavDropdownComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-sidebar-nav-dropdown, cui-sidebar-nav-dropdown', template: `
    <a class="nav-link nav-dropdown-toggle"
      appNavDropdownToggle
      [appHtmlAttr]="item.attributes">
      @if (helper.hasIcon(item)) {
        <i [ngClass]="item | appSidebarNavIcon"></i>
      }
      <ng-container>{{item.name}}</ng-container>
      @if (helper.hasBadge(item)) {
        <span [ngClass]="item | appSidebarNavBadge">{{ item.badge?.text }}</span>
      }
    </a>
    <app-sidebar-nav-items
      class="nav-dropdown-items"
      [items]="item.children ?? []">
    </app-sidebar-nav-items>
    `, providers: [SidebarNavHelper], standalone: true, imports: [NgClass, HtmlAttributesDirective, AppSidebarNavBadgePipe, AppSidebarNavIconPipe, AppSidebarNavItemsComponent, NavDropdownToggleDirective], styles: [".nav-dropdown-toggle{cursor:pointer}\n", ".nav-dropdown-items{display:block}\n"] }]
        }], ctorParameters: () => [{ type: SidebarNavHelper }], propDecorators: { item: [{
                type: Input
            }] } });

class AppSidebarNavComponent {
    constructor(router) {
        this.router = router;
        this.navItems = [];
        this.sidebarNavClass = true;
        this.role = 'nav';
        this.navItemsArray = [];
    }
    ngOnChanges(changes) {
        this.navItemsArray = Array.isArray(this.navItems) ? this.navItems.slice() : [];
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavComponent, deps: [{ token: i1.Router }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppSidebarNavComponent, isStandalone: true, selector: "app-sidebar-nav, cui-sidebar-nav", inputs: { navItems: "navItems", role: "role" }, host: { properties: { "class.sidebar-nav": "this.sidebarNavClass", "attr.role": "this.role" } }, usesOnChanges: true, ngImport: i0, template: "<app-sidebar-nav-items\n  class=\"nav\"\n  [items]=\"navItemsArray\">\n</app-sidebar-nav-items>\n", dependencies: [{ kind: "component", type: AppSidebarNavItemsComponent, selector: "app-sidebar-nav-items, cui-sidebar-nav-items", inputs: ["items"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarNavComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-sidebar-nav, cui-sidebar-nav', standalone: true, imports: [AppSidebarNavItemsComponent], template: "<app-sidebar-nav-items\n  class=\"nav\"\n  [items]=\"navItemsArray\">\n</app-sidebar-nav-items>\n" }]
        }], ctorParameters: () => [{ type: i1.Router }], propDecorators: { navItems: [{
                type: Input
            }], sidebarNavClass: [{
                type: HostBinding,
                args: ['class.sidebar-nav']
            }], role: [{
                type: HostBinding,
                args: ['attr.role']
            }, {
                type: Input
            }] } });

class AppSidebarMinimizerComponent {
    toggleOpen($event) {
        $event.preventDefault();
        this.sidebarService.toggle({ minimize: 'toggle' });
    }
    constructor(sidebarService) {
        this.sidebarService = sidebarService;
        this.role = 'button';
        this.sidebarMinimizerClass = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarMinimizerComponent, deps: [{ token: AppSidebarService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppSidebarMinimizerComponent, isStandalone: true, selector: "app-sidebar-minimizer, cui-sidebar-minimizer", inputs: { role: "role" }, host: { listeners: { "click": "toggleOpen($event)" }, properties: { "attr.role": "this.role", "class.sidebar-minimizer": "this.sidebarMinimizerClass" } }, ngImport: i0, template: ``, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarMinimizerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-minimizer, cui-sidebar-minimizer',
                    template: ``,
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: AppSidebarService }], propDecorators: { role: [{
                type: HostBinding,
                args: ['attr.role']
            }, {
                type: Input
            }], sidebarMinimizerClass: [{
                type: HostBinding,
                args: ['class.sidebar-minimizer']
            }], toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });

class AppSidebarFooterComponent {
    constructor() {
        this.sidebarFooterClass = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarFooterComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppSidebarFooterComponent, isStandalone: true, selector: "app-sidebar-footer, cui-sidebar-footer", host: { properties: { "class.sidebar-footer": "this.sidebarFooterClass" } }, ngImport: i0, template: `<ng-content />`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarFooterComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-footer, cui-sidebar-footer',
                    template: `<ng-content />`,
                    standalone: true
                }]
        }], ctorParameters: () => [], propDecorators: { sidebarFooterClass: [{
                type: HostBinding,
                args: ['class.sidebar-footer']
            }] } });

class AppSidebarFormComponent {
    constructor() {
        this.sidebarFormClass = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarFormComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppSidebarFormComponent, isStandalone: true, selector: "app-sidebar-form, cui-sidebar-form", host: { properties: { "class.sidebar-form": "this.sidebarFormClass" } }, ngImport: i0, template: `<ng-content />`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarFormComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-form, cui-sidebar-form',
                    template: `<ng-content />`,
                    standalone: true
                }]
        }], propDecorators: { sidebarFormClass: [{
                type: HostBinding,
                args: ['class.sidebar-form']
            }] } });

class AppSidebarHeaderComponent {
    constructor() {
        this.sidebarHeaderClass = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarHeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "20.2.2", type: AppSidebarHeaderComponent, isStandalone: true, selector: "app-sidebar-header, cui-sidebar-header", host: { properties: { "class.sidebar-header": "this.sidebarHeaderClass" } }, ngImport: i0, template: `<ng-content />`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-header, cui-sidebar-header',
                    template: `<ng-content />`,
                    standalone: true
                }]
        }], propDecorators: { sidebarHeaderClass: [{
                type: HostBinding,
                args: ['class.sidebar-header']
            }] } });

class AppSidebarModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarModule, imports: [RouterModule,
            AppSidebarComponent,
            AppSidebarFooterComponent,
            AppSidebarFormComponent,
            AppSidebarHeaderComponent,
            AppSidebarMinimizerComponent,
            AppSidebarMinimizerComponent,
            AppSidebarNavComponent,
            NavDropdownDirective,
            AppSidebarNavItemsComponent,
            AppSidebarNavDividerComponent,
            AppSidebarNavDropdownComponent,
            AppSidebarNavLinkContentComponent,
            AppSidebarNavTitleComponent,
            AppSidebarNavLabelComponent,
            AppSidebarNavIconPipe,
            AppSidebarNavBadgePipe,
            AppSidebarNavLinkPipe,
            AppSidebarNavItemClassPipe,
            AppSidebarNavLinkComponent,
            NavDropdownToggleDirective,
            LayoutModule], exports: [AppSidebarFooterComponent,
            AppSidebarFormComponent,
            AppSidebarHeaderComponent,
            AppSidebarMinimizerComponent,
            AppSidebarComponent,
            // AppSidebarNavItemsComponent,
            AppSidebarNavComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarModule, providers: [
            SidebarNavHelper,
            AppSidebarService
        ], imports: [RouterModule,
            AppSidebarNavComponent,
            AppSidebarNavItemsComponent,
            AppSidebarNavDropdownComponent,
            AppSidebarNavLinkComponent,
            LayoutModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "20.2.2", ngImport: i0, type: AppSidebarModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        RouterModule,
                        AppSidebarComponent,
                        AppSidebarFooterComponent,
                        AppSidebarFormComponent,
                        AppSidebarHeaderComponent,
                        AppSidebarMinimizerComponent,
                        AppSidebarMinimizerComponent,
                        AppSidebarNavComponent,
                        NavDropdownDirective,
                        AppSidebarNavItemsComponent,
                        AppSidebarNavDividerComponent,
                        AppSidebarNavDropdownComponent,
                        AppSidebarNavLinkContentComponent,
                        AppSidebarNavTitleComponent,
                        AppSidebarNavLabelComponent,
                        AppSidebarNavIconPipe,
                        AppSidebarNavBadgePipe,
                        AppSidebarNavLinkPipe,
                        AppSidebarNavItemClassPipe,
                        AppSidebarNavLinkComponent,
                        NavDropdownToggleDirective,
                        LayoutModule
                    ],
                    exports: [
                        AppSidebarFooterComponent,
                        AppSidebarFormComponent,
                        AppSidebarHeaderComponent,
                        AppSidebarMinimizerComponent,
                        AppSidebarComponent,
                        // AppSidebarNavItemsComponent,
                        AppSidebarNavComponent,
                        // AppSidebarNavDividerComponent,
                        // AppSidebarNavDropdownComponent,
                        // AppSidebarNavLabelComponent,
                        // AppSidebarNavLinkComponent,
                        // AppSidebarNavLinkContentComponent,
                        // AppSidebarNavTitleComponent,
                        // NavDropdownDirective,
                        // NavDropdownToggleDirective,
                    ],
                    providers: [
                        SidebarNavHelper,
                        AppSidebarService
                    ]
                }]
        }] });

// export { NavDropdownDirective, NavDropdownToggleDirective } from './app-sidebar-nav.directive';
// export { AppSidebarNavDividerComponent } from './app-sidebar-nav/app-sidebar-nav-divider.component';
// export { AppSidebarNavDropdownComponent } from './app-sidebar-nav/app-sidebar-nav-dropdown.component';
// export { AppSidebarNavItemsComponent } from './app-sidebar-nav/app-sidebar-nav-items.component';
// export { AppSidebarNavLinkComponent, AppSidebarNavLinkContentComponent } from './app-sidebar-nav/app-sidebar-nav-link.component';
// export { AppSidebarNavTitleComponent } from './app-sidebar-nav/app-sidebar-nav-title.component';
// export { AppSidebarNavLabelComponent } from './app-sidebar-nav/app-sidebar-nav-label.component';

/*
 * Public API Surface of @coreui/angular
 */
// export * from './lib/shared/index';
// export * from './lib/coreui.module';

/**
 * Generated bundle index. Do not edit.
 */

export { AppAsideComponent, AppAsideModule, AppBreadcrumbComponent, AppBreadcrumbModule, AppFooterComponent, AppFooterModule, AppHeaderComponent, AppHeaderModule, AppSidebarComponent, AppSidebarFooterComponent, AppSidebarFormComponent, AppSidebarHeaderComponent, AppSidebarMinimizerComponent, AppSidebarModule, AppSidebarNavComponent, CuiBreadcrumbComponent, SidebarNavHelper };
