// stimulsoft-viewer-angular/esm5/lib/controls/parameters/parameter-checkbox.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
var ParameterCheckboxComponent = /** @class */ (function () {
    function ParameterCheckboxComponent() {
        this.isEnabled = true;
        this.isMenuParameter = false;
        this.nullable = false;
        this.action = new EventEmitter();
    }
    ParameterCheckboxComponent.prototype.ngOnInit = function () { };
    ParameterCheckboxComponent.prototype.onAction = function (checked) {
        this.action.emit(checked);
    };
    Object.defineProperty(ParameterCheckboxComponent.prototype, "checked", {
        get: function () {
            if (this.nullable) {
                return this.params.isNull;
            }
            else {
                return this.isMenuParameter ? this.params.isChecked : (typeof (this.params.value) === 'boolean' && this.params.value) || this.params.value === 'true' || this.params.value === 'True';
            }
        },
        enumerable: true,
        configurable: true
    });
    __decorate([
        Input()
    ], ParameterCheckboxComponent.prototype, "params", void 0);
    __decorate([
        Input()
    ], ParameterCheckboxComponent.prototype, "captionText", void 0);
    __decorate([
        Input()
    ], ParameterCheckboxComponent.prototype, "margin", void 0);
    __decorate([
        Input()
    ], ParameterCheckboxComponent.prototype, "width", void 0);
    __decorate([
        Input()
    ], ParameterCheckboxComponent.prototype, "imageBlockParentWidth", void 0);
    __decorate([
        Input()
    ], ParameterCheckboxComponent.prototype, "isEnabled", void 0);
    __decorate([
        Input()
    ], ParameterCheckboxComponent.prototype, "isMenuParameter", void 0);
    __decorate([
        Input()
    ], ParameterCheckboxComponent.prototype, "nullable", void 0);
    __decorate([
        Output()
    ], ParameterCheckboxComponent.prototype, "action", void 0);
    ParameterCheckboxComponent = __decorate([
        Component({
            selector: 'sti-parameter-checkbox',
            template: "\n    <sti-check-box [isEnabled]=\"isEnabled\"\n                   [isChecked]=\"checked\"\n                   [captionText]=\"captionText\"\n                   [margin]=\"margin\"\n                   [width]=\"width\"\n                   [imageBlockParentWidth]=\"imageBlockParentWidth\"\n                   (action)=\"onAction($event)\">\n    </sti-check-box>\n    "
        })
    ], ParameterCheckboxComponent);
    return ParameterCheckboxComponent;
}());
export { ParameterCheckboxComponent };
