// stimulsoft-viewer-angular/esm5/lib/controls/parameters/parameter-find.component.js
import { __decorate } from "tslib";
import { Component, Input } from '@angular/core';
import { ModelService } from '../../services/model.service';
import { HelperService } from '../../services/helper.service';
var ParameterFindComponent = /** @class */ (function () {
    function ParameterFindComponent(model, helper) {
        this.model = model;
        this.helper = helper;
    }
    ParameterFindComponent.prototype.ngOnInit = function () { };
    ParameterFindComponent.prototype.onchange = function (event) {
        var _this = this;
        var text = event.value.toLowerCase();
        this.variable.items.forEach(function (item) {
            var itemText = _this.helper.val(item.value, _this.helper.getStringKey(item.key, _this.variable)).toLowerCase();
            item.visible = itemText.indexOf(text) >= 0;
        });
    };
    ParameterFindComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService }
    ]; };
    __decorate([
        Input()
    ], ParameterFindComponent.prototype, "variable", void 0);
    ParameterFindComponent = __decorate([
        Component({
            selector: 'sti-parameter-find',
            template: "\n      <table class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\">\n          <tbody>\n            <tr class=\"stiJsViewerClearAllStyles\">\n              <td class=\"stiJsViewerClearAllStyles\" style=\"padding-left: 8px;\">\n                <sti-text-block [text]=\"model.loc('FindWhat')\"></sti-text-block>\n              </td>\n\n              <td class=\"stiJsViewerClearAllStyles\" style=\"padding-left: 8px;\">\n                <sti-text-box [width]=\"80\"\n                  [margin]=\"'4px'\"\n                  [focusOnCreate]=\"true\"\n                  (onchange)=\"onchange($event)\">\n                </sti-text-box>\n              </td>\n            </tr>\n          </tbody>\n      </table>\n\n  "
        })
    ], ParameterFindComponent);
    return ParameterFindComponent;
}());
export { ParameterFindComponent };
