// stimulsoft-viewer-angular/esm5/lib/controls/parameters/parameter-button.component.js
import { __decorate } from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { ModelService } from '../../services/model.service';
var ParameterButtonComponent = /** @class */ (function () {
    function ParameterButtonComponent(model) {
        this.model = model;
        this.action = new EventEmitter();
    }
    ParameterButtonComponent.prototype.ngOnInit = function () { };
    ParameterButtonComponent.prototype.onaction = function () {
        this.action.emit(this.params);
    };
    ParameterButtonComponent.ctorParameters = function () { return [
        { type: ModelService }
    ]; };
    __decorate([
        Input()
    ], ParameterButtonComponent.prototype, "params", void 0);
    __decorate([
        Input()
    ], ParameterButtonComponent.prototype, "buttonType", void 0);
    __decorate([
        Output()
    ], ParameterButtonComponent.prototype, "action", void 0);
    ParameterButtonComponent = __decorate([
        Component({
            selector: 'sti-parameter-button',
            template: "\n    <sti-button [imageName]=\"buttonType + '.png'\"\n        [styleName]=\"'stiJsViewerFormButton'\"\n        [height]=\"model.options.isTouchDevice ? '26px' : '21px'\"\n        [innerTableWidth]=\"'100%'\"\n        [enabled]=\"!params?.isNull\"\n        (action)=\"onaction()\">\n    </sti-button>\n  "
        })
    ], ParameterButtonComponent);
    return ParameterButtonComponent;
}());
export { ParameterButtonComponent };
