// stimulsoft-viewer-angular/esm5/lib/controls/parameters/parameter.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewChild } from '@angular/core';
import { ModelService } from '../../services/model.service';
import { HelperService } from '../../services/helper.service';
import { MenuService } from '../../menu/menu.service';
var ParameterComponent = /** @class */ (function () {
    function ParameterComponent(model, helper, menuService) {
        this.model = model;
        this.helper = helper;
        this.menuService = menuService;
    }
    ParameterComponent.prototype.ngOnInit = function () { };
    Object.defineProperty(ParameterComponent.prototype, "params", {
        get: function () {
            return this._params;
        },
        set: function (params) {
            this._params = params;
            if (params.basicType === 'Range') {
                if (params.type === 'DateTime' && params.keyTo && params.keyTo.isNull) {
                    params.keyTo = this.helper.getDateTimeObject(new Date());
                }
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ParameterComponent.prototype, "showParameterInTwoRows", {
        get: function () {
            return this.params.basicType === 'Range' && this.model.options.appearance.parametersPanelPosition === 'Left';
        },
        enumerable: true,
        configurable: true
    });
    ParameterComponent.prototype.getReadOnlyCheckbox = function () {
        return this.params.basicType === 'List' || !this.params.allowUserValues;
    };
    ParameterComponent.prototype.getFirstTextBoxValue = function () {
        var _this = this;
        var value = '';
        if (this.params.basicType === 'Value' || this.params.basicType === 'NullableValue') {
            if (this.params.type === 'DateTime' && this.params.value === null) {
                this.params.value = new Date();
                this.params.key = this.helper.getDateTimeObject(this.params.value);
            }
            value = (this.params.type === 'DateTime') ? this.helper.getStringKey(this.params.key, this.params) : this.params.value;
        }
        // Range
        if (this.params.basicType === 'Range') {
            if (this.params.type === 'DateTime' && this.params.key && this.params.key.isNull) {
                this.params.key = this.helper.getDateTimeObject(new Date());
            }
            value = this.helper.getStringKey(this.params.key, this.params);
        }
        // List
        if (this.params.basicType === 'List' && this.params.items) {
            this.params.items.forEach(function (item) {
                if (item.isChecked) {
                    if (value !== '') {
                        value += (_this.model.options.listSeparator ? _this.model.options.listSeparator + " " : "; ");
                    }
                    if (_this.params.allowUserValues) {
                        value += _this.helper.getStringKey(item.key, _this.params);
                    }
                    else {
                        value += item.value !== '' ? item.value : _this.helper.getStringKey(item.key, _this.params);
                    }
                }
            });
        }
        return value != null ? value.toString() : value;
    };
    ParameterComponent.prototype.nullableAction = function (checked) {
        this.params.isNull = !this.params.isNull;
    };
    ParameterComponent.prototype.firstGuidAction = function () {
        if (this.params.basicType === 'Range') {
            this.params.key = this.helper.newGuid();
        }
        else {
            this.params.value = this.helper.newGuid();
        }
    };
    ParameterComponent.prototype.doubleDateTimeAction = function (event) {
        var _this = this;
        this.menuService.addMenu({
            type: 'doubleDatePickerMenu', name: 'doubleDatePickerMenu', items: [], parent: this.rangeFrom,
            params: this.params
        });
        setTimeout(function () {
            _this.menuService.showMenu('doubleDatePickerMenu');
        });
    };
    ParameterComponent.prototype.firstDateTimeAction = function (event) {
        var _this = this;
        this.menuService.addMenu({
            type: 'datePickerMenu', name: 'datePickerMenu', items: [], parent: this.firstDateTimeButton,
            params: this.params
        });
        setTimeout(function () {
            _this.menuService.showMenu('datePickerMenu');
        });
    };
    ParameterComponent.prototype.dropDownButtonAction = function (event) {
        var _this = this;
        var menuType = 'parameterMenuForValue';
        switch (this.params.basicType) {
            case 'Range':
                menuType = 'parameterMenuForRange';
                break;
            case 'List':
                menuType = this.params.allowUserValues ? 'parameterMenuForEditList' : 'parameterMenuForNotEditList';
                break;
        }
        if (this.params.items) {
            this.params.items.forEach(function (i) { return i.visible = true; });
        }
        this.menuService.addMenu({
            type: menuType, name: 'parameterMenu', items: [], parent: this.element,
            params: this.params
        });
        setTimeout(function () {
            _this.menuService.showMenu('parameterMenu');
        });
    };
    ParameterComponent.ctorParameters = function () { return [
        { type: ModelService },
        { type: HelperService },
        { type: MenuService }
    ]; };
    __decorate([
        ViewChild('element')
    ], ParameterComponent.prototype, "element", void 0);
    __decorate([
        ViewChild('firstDateTimeButton')
    ], ParameterComponent.prototype, "firstDateTimeButton", void 0);
    __decorate([
        ViewChild('doubleDateTimeButton')
    ], ParameterComponent.prototype, "doubleDateTimeButton", void 0);
    __decorate([
        ViewChild('rangeFrom')
    ], ParameterComponent.prototype, "rangeFrom", void 0);
    __decorate([
        Input()
    ], ParameterComponent.prototype, "params", null);
    ParameterComponent = __decorate([
        Component({
            selector: 'sti-parameter',
            template: "\n    <table #element class=\"stiJsViewerClearAllStyles\" cellpadding=\"0\" cellspacing=\"0\" #rangeFrom>\n       <tbody>\n          <tr class=\"stiJsViewerClearAllStyles\" >\n            <td *ngIf=\"params.type == 'Bool' && (params.basicType == 'Value' || params.basicType == 'NullableValue')\"\n                style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n                <sti-parameter-checkbox [params]=\"params\" [isEnabled]=\"params.allowUserValues && !params.isNull\" (action)=\"params.value = $event\">\n                </sti-parameter-checkbox>\n            </td>\n\n            <td *ngIf=\"params.basicType == 'Range' && model.options.appearance.parametersPanelPosition === 'Top'\"  style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n                {{model.loc('RangeFrom')}}\n            </td>\n\n            <td *ngIf=\"params.type != 'Bool' || params.basicType == 'List'\"\n                style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n                <sti-parameter-text-box [variable]=\"params\"\n                    [readOnly]=\"getReadOnlyCheckbox()\"\n                    [value]=\"getFirstTextBoxValue()\">\n                </sti-parameter-text-box>\n            </td>\n\n            <td *ngIf=\"params.type == 'DateTime' && params.allowUserValues && params.basicType != 'List' && params.basicType != 'Range'\"\n                style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\" #firstDateTimeButton>\n                <sti-parameter-button [buttonType]=\"'DateTimeButton'\" [params]=\"params\"  (action)=\"firstDateTimeAction($event)\">\n                </sti-parameter-button>\n            </td>\n\n            <td *ngIf=\"params.type == 'Guid' && params.allowUserValues && params.basicType != 'List'\"\n                style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n                <sti-parameter-button  [buttonType]=\"'GuidButton'\" [params]=\"params\" (action)=\"firstGuidAction()\">\n                </sti-parameter-button>\n            </td>\n\n            <td *ngIf=\"!showParameterInTwoRows && params.basicType == 'Range' && model.options.appearance.parametersPanelPosition === 'Top'\" style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              {{model.loc('RangeTo')}}\n            </td>\n\n            <!-- second -->\n            <td *ngIf=\"!showParameterInTwoRows && params.basicType == 'Range'\" style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              <sti-parameter-text-box [variable]=\"params\"\n                    [readOnly]=\"!params.allowUserValues\"\n                    [value]=\"helper.getStringKey(params.keyTo, params)\"\n                    [secondTextBox]=\"true\">\n              </sti-parameter-text-box>\n            </td>\n\n            <td *ngIf=\"!showParameterInTwoRows && params.basicType == 'Range' && params.type == 'DateTime' && params.allowUserValues\" #doubleDateTimeButton\n              style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              <sti-parameter-button  [buttonType]=\"'DateTimeButton'\" [params]=\"params\" (action)=\"doubleDateTimeAction($event)\">\n              </sti-parameter-button>\n            </td>\n\n            <td *ngIf=\"!showParameterInTwoRows && params.basicType == 'Range' && params.type == 'Guid' && params.allowUserValues\"\n              style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              <sti-parameter-button  [buttonType]=\"'GuidButton'\" [params]=\"params\" (action)=\"params.keyTo=helper.newGuid()\">\n              </sti-parameter-button>\n            </td>\n\n            <td *ngIf=\"!showParameterInTwoRows && params.items != null || (params.basicType == 'List' && params.allowUserValues)\"\n                style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              <sti-parameter-button [buttonType]=\"'DropDownButton'\" [params]=\"params\" (action)=\"dropDownButtonAction($event)\">\n              </sti-parameter-button>\n            </td>\n\n            <td *ngIf=\"!showParameterInTwoRows && params.basicType == 'NullableValue' && params.allowUserValues\"\n                style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n                <sti-parameter-checkbox [params]=\"params\" [isEnabled]=\"params.allowUserValues\" (action)=\"nullableAction($event)\" [nullable]=\"true\" >\n                </sti-parameter-checkbox>\n            </td>\n\n            <td *ngIf=\"!showParameterInTwoRows && params.basicType == 'NullableValue' && params.allowUserValues\"\n                style=\"padding: 0px\" [style.height.px]=\"model.options.parameterRowHeight\">\n                {{model.loc('Null')}}\n            </td>\n          </tr>\n\n          <!-- NEW LINE -->\n          <tr *ngIf=\"showParameterInTwoRows\">\n            <td *ngIf=\"params.basicType == 'Range' && model.options.appearance.parametersPanelPosition === 'Top'\" style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              {{model.loc('RangeTo')}}\n            </td>\n\n            <!-- second -->\n            <td *ngIf=\"params.basicType == 'Range'\" style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              <sti-parameter-text-box [variable]=\"params\"\n                    [readOnly]=\"!params.allowUserValues\"\n                    [value]=\"helper.getStringKey(params.keyTo, params)\"\n                    [secondTextBox]=\"true\">\n              </sti-parameter-text-box>\n            </td>\n\n            <td *ngIf=\"params.basicType == 'Range' && params.type == 'DateTime' && params.allowUserValues\" #doubleDateTimeButton\n              style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              <sti-parameter-button  [buttonType]=\"'DateTimeButton'\" [params]=\"params\" (action)=\"doubleDateTimeAction($event)\">\n              </sti-parameter-button>\n            </td>\n\n            <td *ngIf=\"params.basicType == 'Range' && params.type == 'Guid' && params.allowUserValues\"\n              style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              <sti-parameter-button  [buttonType]=\"'GuidButton'\" [params]=\"params\" (action)=\"params.keyTo=helper.newGuid()\">\n              </sti-parameter-button>\n            </td>\n\n            <td *ngIf=\"params.items != null || (params.basicType == 'List' && params.allowUserValues)\"\n                style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n              <sti-parameter-button [buttonType]=\"'DropDownButton'\" [params]=\"params\" (action)=\"dropDownButtonAction($event)\">\n              </sti-parameter-button>\n            </td>\n\n            <td *ngIf=\"params.basicType == 'NullableValue' && params.allowUserValues\"\n                style=\"padding: 0px 2px;\" [style.height.px]=\"model.options.parameterRowHeight\">\n                <sti-parameter-checkbox [params]=\"params\" [isEnabled]=\"params.allowUserValues\" (action)=\"nullableAction($event)\" [nullable]=\"true\" >\n                </sti-parameter-checkbox>\n            </td>\n\n            <td *ngIf=\"params.basicType == 'NullableValue' && params.allowUserValues\"\n                style=\"padding: 0px\" [style.height.px]=\"model.options.parameterRowHeight\">\n                {{model.loc('Null')}}\n            </td>\n\n          </tr>\n       </tbody>\n    </table>\n  "
        })
    ], ParameterComponent);
    return ParameterComponent;
}());
export { ParameterComponent };
