// stimulsoft-viewer-angular/esm2015/lib/menu/parameter-menu-item.component.js
import { __decorate } from "tslib";
import { Component, Output, EventEmitter, Input } from '@angular/core';
import { ModelService } from '../services/model.service';
let ParameterMenuItemComponent = class ParameterMenuItemComponent {
    constructor(model) {
        this.model = model;
        this.action = new EventEmitter();
        this.active = true;
        this.padding = '0px 5px';
        this.isOver = false;
        this.mouseDown = false;
    }
    ngOnInit() { }
    mouseover() {
        this.isOver = true;
    }
    mouseout() {
        this.isOver = false;
        this.mouseDown = false;
    }
    mousedown() {
        this.mouseDown = true;
    }
    mouseup() {
        this.mouseDown = false;
        this.action.emit();
    }
    get className() {
        return this.active ? this.mouseDown ? 'stiJsViewerParametersMenuItemPressed' : (this.isOver ? 'stiJsViewerParametersMenuItemOver' : 'stiJsViewerParametersMenuItem') : 'stiJsViewerParametersMenuItem';
    }
};
ParameterMenuItemComponent.ctorParameters = () => [
    { type: ModelService }
];
__decorate([
    Output()
], ParameterMenuItemComponent.prototype, "action", void 0);
__decorate([
    Input()
], ParameterMenuItemComponent.prototype, "active", void 0);
__decorate([
    Input()
], ParameterMenuItemComponent.prototype, "padding", void 0);
ParameterMenuItemComponent = __decorate([
    Component({
        selector: 'sti-parameter-menu-item',
        template: `
      <div [class]="className"
        [style.height.px]="model.options.isTouchDevice ? 30 : 24"
        (mouseover)="mouseover()"
        (mouseout)="mouseout()"
        (mousedown)="mousedown()"
        (mouseup)="mouseup()">
        <table class="stiJsViewerClearAllStyles stiJsViewerParametersMenuItemInnerTable" cellpadding="0" cellspacing="0">
          <tbody>
            <tr class="stiJsViewerClearAllStyles">
              <td class="stiJsViewerClearAllStyles" [style.padding]="padding">
                <ng-content>
                </ng-content>
              </td>
            </tr>
          </tbody>
        </table>

      </div>
  `
    })
], ParameterMenuItemComponent);
export { ParameterMenuItemComponent };
